{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "f3de1a83",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Merged file created successfully!\n",
      "Total rows: 1060588\n",
      "\n",
      "First few rows:\n",
      "   CHROM      POS            ID REF ALT        AF  GENE    AF_AFR    AF_AMR  \\\n",
      "0  chr19  1248349   rs112363642   G   A       NaN   NaN  0.019037  0.217347   \n",
      "1  chr19  1248372   rs145305690   C   T  0.000468  MIDN  0.000000  0.000000   \n",
      "2  chr19  1248399  rs1404444736   G   C  0.000312  MIDN  0.000000  0.000000   \n",
      "3  chr19  1248405   rs535078794   C   T  0.000468  MIDN  0.000000  0.000000   \n",
      "4  chr19  1248419   rs371906837   C   A  0.003748  MIDN  0.012878  0.001020   \n",
      "\n",
      "   AF_AMR.1  ...  P_EUR_AMR  P_EUR_AMR1  P_EUR_AMR2  P_EUR_BEB  P_EUR_EAS  \\\n",
      "0  0.241697  ...   0.973014    0.969612    0.897267    0.99913   0.603798   \n",
      "1  0.000000  ...   1.000000    1.000000    1.000000    1.00000   0.000000   \n",
      "2  0.000000  ...   0.000000    0.000000    0.000000    0.00000   0.000000   \n",
      "3  0.000000  ...   1.000000    1.000000    1.000000    1.00000   0.000000   \n",
      "4  0.000000  ...   0.000000    1.000000    0.000000    1.00000   1.000000   \n",
      "\n",
      "   P_EUR_GIH  P_EUR_ITU  P_EUR_PJL  P_EUR_SAS  P_EUR_STU  \n",
      "0   0.987329     0.8941   0.926662   0.967103   0.967103  \n",
      "1   1.000000     1.0000   1.000000   1.000000   1.000000  \n",
      "2   0.000000     0.0000   0.000000   0.000000   0.000000  \n",
      "3   1.000000     1.0000   1.000000   1.000000   1.000000  \n",
      "4   1.000000     1.0000   1.000000   1.000000   1.000000  \n",
      "\n",
      "[5 rows x 41 columns]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Read the two TSV files\n",
    "final_chi = pd.read_csv('Final_chi_sq_fst_result.tsv', sep='\\t')\n",
    "pop_af = pd.read_csv('pop_AF_with_gene.raw.tsv', sep=' ', comment='#')\n",
    "\n",
    "# The pop_af file has a comment line, so we need to set proper column names\n",
    "# Based on the comment line in the file\n",
    "pop_af.columns = ['CHROM', 'POS', 'ID', 'REF', 'ALT', 'AF', 'AF_EUR', 'AF_AFR', \n",
    "                  'AF_EAS', 'AF_AMR', 'AF_AMR1', 'AF_AMR2', 'AF_SAS', 'AF_GIH', \n",
    "                  'AF_ITU', 'AF_BEB', 'AF_PJL', 'AF_STU', 'GENE']\n",
    "\n",
    "# Extract only AF and GENE columns along with merge keys\n",
    "pop_af_subset = pop_af[['CHROM', 'POS', 'ID', 'AF', 'GENE']]\n",
    "\n",
    "# Merge the dataframes on CHROM, POS, and ID\n",
    "merged = pd.merge(final_chi, pop_af_subset, on=['CHROM', 'POS', 'ID'], how='left')\n",
    "\n",
    "# Reorder columns to put AF and GENE after the basic variant info (CHROM, POS, ID, REF, ALT)\n",
    "cols = merged.columns.tolist()\n",
    "# Move AF and GENE to position 5 (right after ALT)\n",
    "basic_cols = ['CHROM', 'POS', 'ID', 'REF', 'ALT', 'AF', 'GENE']\n",
    "other_cols = [col for col in cols if col not in basic_cols]\n",
    "final_cols = basic_cols + other_cols\n",
    "\n",
    "merged = merged[final_cols]\n",
    "\n",
    "# Save the result\n",
    "merged.to_csv('Final_chi_sq_fst_result_with_AF_GENE.tsv', sep='\\t', index=False)\n",
    "\n",
    "print(f\"Merged file created successfully!\")\n",
    "print(f\"Total rows: {len(merged)}\")\n",
    "print(f\"\\nFirst few rows:\")\n",
    "print(merged.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "ee5d55e0",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"Final_chi_sq_fst_result_with_AF_GENE.tsv\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "590fbf94",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "CHROM",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "POS",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "ID",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "REF",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "ALT",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "AF",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "GENE",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "AF_AFR",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "AF_AMR",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "AF_AMR.1",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "AF_AMR.2",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "AF_EAS",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "AF_EUR",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "AF_SAS",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "AF_BEB",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "AF_GIH",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "AF_ITU",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "AF_PJL",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "AF_STU",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "FST_EUR_VS_AFR",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "FST_EUR_VS_AMR",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "FST_EUR_VS_AMR1",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "FST_EUR_VS_AMR2",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "FST_EUR_VS_BEB",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "FST_EUR_VS_EAS",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "FST_EUR_VS_GIH",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "FST_EUR_VS_ITU",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "FST_EUR_VS_PJL",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "FST_EUR_VS_SAS",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "FST_EUR_VS_STU",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "P_EUR_AFR",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "P_EUR_AMR",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "P_EUR_AMR1",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "P_EUR_AMR2",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "P_EUR_BEB",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "P_EUR_EAS",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "P_EUR_GIH",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "P_EUR_ITU",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "P_EUR_PJL",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "P_EUR_SAS",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "P_EUR_STU",
         "rawType": "float64",
         "type": "float"
        }
       ],
       "ref": "75fe48e8-a618-4b6e-980a-cbde824bcff8",
       "rows": [
        [
         "0",
         "chr19",
         "1248349",
         "rs112363642",
         "G",
         "A",
         null,
         null,
         "0.019037",
         "0.217347",
         "0.241697",
         "0.187215",
         "0.42906",
         "0.228639",
         "0.214912",
         "0.229008",
         "0.223301",
         "0.275701",
         "0.19863",
         "0.214912",
         "0.206717",
         "0.0",
         "0.0",
         "0.00348294",
         "0.0",
         "0.0867858",
         "0.0",
         "0.00336793",
         "0.000420678",
         "0.0",
         "0.0",
         "0.113852149",
         "0.97301411",
         "0.969611797",
         "0.897266583",
         "0.999129575",
         "0.603798001",
         "0.987328726",
         "0.894099823",
         "0.926661539",
         "0.967102804",
         "0.967102804"
        ],
        [
         "1",
         "chr19",
         "1248372",
         "rs145305690",
         "C",
         "T",
         "0.000468457",
         "MIDN",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0025641",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.00184687",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0",
         "0.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0"
        ],
        [
         "2",
         "chr19",
         "1248399",
         "rs1404444736",
         "G",
         "C",
         "0.000312305",
         "MIDN",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.00158228",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.00123435",
         "0.000500099",
         "0.0",
         "0.0",
         "0.0",
         "0.00070182",
         "0.0",
         "0.0",
         "0.0",
         "0.000733212",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0"
        ],
        [
         "3",
         "chr19",
         "1248405",
         "rs535078794",
         "C",
         "T",
         "0.000468457",
         "MIDN",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0025641",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.00184687",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0",
         "0.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0"
        ],
        [
         "4",
         "chr19",
         "1248419",
         "rs371906837",
         "C",
         "A",
         "0.00374766",
         "MIDN",
         "0.0128779",
         "0.00102041",
         "0.0",
         "0.0022831",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0102978",
         "0.000262689",
         "0.0",
         "0.002897",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "1.0",
         "0.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0"
        ]
       ],
       "shape": {
        "columns": 41,
        "rows": 5
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AF</th>\n",
       "      <th>GENE</th>\n",
       "      <th>AF_AFR</th>\n",
       "      <th>AF_AMR</th>\n",
       "      <th>AF_AMR.1</th>\n",
       "      <th>...</th>\n",
       "      <th>P_EUR_AMR</th>\n",
       "      <th>P_EUR_AMR1</th>\n",
       "      <th>P_EUR_AMR2</th>\n",
       "      <th>P_EUR_BEB</th>\n",
       "      <th>P_EUR_EAS</th>\n",
       "      <th>P_EUR_GIH</th>\n",
       "      <th>P_EUR_ITU</th>\n",
       "      <th>P_EUR_PJL</th>\n",
       "      <th>P_EUR_SAS</th>\n",
       "      <th>P_EUR_STU</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248349</td>\n",
       "      <td>rs112363642</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.019037</td>\n",
       "      <td>0.217347</td>\n",
       "      <td>0.241697</td>\n",
       "      <td>...</td>\n",
       "      <td>0.973014</td>\n",
       "      <td>0.969612</td>\n",
       "      <td>0.897267</td>\n",
       "      <td>0.99913</td>\n",
       "      <td>0.603798</td>\n",
       "      <td>0.987329</td>\n",
       "      <td>0.8941</td>\n",
       "      <td>0.926662</td>\n",
       "      <td>0.967103</td>\n",
       "      <td>0.967103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248372</td>\n",
       "      <td>rs145305690</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000468</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248399</td>\n",
       "      <td>rs1404444736</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000312</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248405</td>\n",
       "      <td>rs535078794</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000468</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248419</td>\n",
       "      <td>rs371906837</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.003748</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.012878</td>\n",
       "      <td>0.001020</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM      POS            ID REF ALT        AF  GENE    AF_AFR    AF_AMR  \\\n",
       "0  chr19  1248349   rs112363642   G   A       NaN   NaN  0.019037  0.217347   \n",
       "1  chr19  1248372   rs145305690   C   T  0.000468  MIDN  0.000000  0.000000   \n",
       "2  chr19  1248399  rs1404444736   G   C  0.000312  MIDN  0.000000  0.000000   \n",
       "3  chr19  1248405   rs535078794   C   T  0.000468  MIDN  0.000000  0.000000   \n",
       "4  chr19  1248419   rs371906837   C   A  0.003748  MIDN  0.012878  0.001020   \n",
       "\n",
       "   AF_AMR.1  ...  P_EUR_AMR  P_EUR_AMR1  P_EUR_AMR2  P_EUR_BEB  P_EUR_EAS  \\\n",
       "0  0.241697  ...   0.973014    0.969612    0.897267    0.99913   0.603798   \n",
       "1  0.000000  ...   1.000000    1.000000    1.000000    1.00000   0.000000   \n",
       "2  0.000000  ...   0.000000    0.000000    0.000000    0.00000   0.000000   \n",
       "3  0.000000  ...   1.000000    1.000000    1.000000    1.00000   0.000000   \n",
       "4  0.000000  ...   0.000000    1.000000    0.000000    1.00000   1.000000   \n",
       "\n",
       "   P_EUR_GIH  P_EUR_ITU  P_EUR_PJL  P_EUR_SAS  P_EUR_STU  \n",
       "0   0.987329     0.8941   0.926662   0.967103   0.967103  \n",
       "1   1.000000     1.0000   1.000000   1.000000   1.000000  \n",
       "2   0.000000     0.0000   0.000000   0.000000   0.000000  \n",
       "3   1.000000     1.0000   1.000000   1.000000   1.000000  \n",
       "4   1.000000     1.0000   1.000000   1.000000   1.000000  \n",
       "\n",
       "[5 rows x 41 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6fa694dd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "0",
         "rawType": "int64",
         "type": "integer"
        }
       ],
       "ref": "c3a1d824-ee8f-4933-a232-5fbcdd25a8b7",
       "rows": [
        [
         "CHROM",
         "0"
        ],
        [
         "POS",
         "0"
        ],
        [
         "ID",
         "0"
        ],
        [
         "REF",
         "0"
        ],
        [
         "ALT",
         "0"
        ],
        [
         "AF",
         "1"
        ],
        [
         "GENE",
         "1"
        ],
        [
         "AF_AFR",
         "0"
        ],
        [
         "AF_AMR",
         "0"
        ],
        [
         "AF_AMR.1",
         "0"
        ],
        [
         "AF_AMR.2",
         "0"
        ],
        [
         "AF_EAS",
         "0"
        ],
        [
         "AF_EUR",
         "0"
        ],
        [
         "AF_SAS",
         "0"
        ],
        [
         "AF_BEB",
         "0"
        ],
        [
         "AF_GIH",
         "0"
        ],
        [
         "AF_ITU",
         "0"
        ],
        [
         "AF_PJL",
         "0"
        ],
        [
         "AF_STU",
         "0"
        ],
        [
         "FST_EUR_VS_AFR",
         "0"
        ],
        [
         "FST_EUR_VS_AMR",
         "0"
        ],
        [
         "FST_EUR_VS_AMR1",
         "0"
        ],
        [
         "FST_EUR_VS_AMR2",
         "0"
        ],
        [
         "FST_EUR_VS_BEB",
         "0"
        ],
        [
         "FST_EUR_VS_EAS",
         "0"
        ],
        [
         "FST_EUR_VS_GIH",
         "0"
        ],
        [
         "FST_EUR_VS_ITU",
         "0"
        ],
        [
         "FST_EUR_VS_PJL",
         "0"
        ],
        [
         "FST_EUR_VS_SAS",
         "0"
        ],
        [
         "FST_EUR_VS_STU",
         "0"
        ],
        [
         "P_EUR_AFR",
         "0"
        ],
        [
         "P_EUR_AMR",
         "0"
        ],
        [
         "P_EUR_AMR1",
         "0"
        ],
        [
         "P_EUR_AMR2",
         "0"
        ],
        [
         "P_EUR_BEB",
         "0"
        ],
        [
         "P_EUR_EAS",
         "0"
        ],
        [
         "P_EUR_GIH",
         "0"
        ],
        [
         "P_EUR_ITU",
         "0"
        ],
        [
         "P_EUR_PJL",
         "0"
        ],
        [
         "P_EUR_SAS",
         "0"
        ],
        [
         "P_EUR_STU",
         "0"
        ]
       ],
       "shape": {
        "columns": 1,
        "rows": 41
       }
      },
      "text/plain": [
       "CHROM              0\n",
       "POS                0\n",
       "ID                 0\n",
       "REF                0\n",
       "ALT                0\n",
       "AF                 1\n",
       "GENE               1\n",
       "AF_AFR             0\n",
       "AF_AMR             0\n",
       "AF_AMR.1           0\n",
       "AF_AMR.2           0\n",
       "AF_EAS             0\n",
       "AF_EUR             0\n",
       "AF_SAS             0\n",
       "AF_BEB             0\n",
       "AF_GIH             0\n",
       "AF_ITU             0\n",
       "AF_PJL             0\n",
       "AF_STU             0\n",
       "FST_EUR_VS_AFR     0\n",
       "FST_EUR_VS_AMR     0\n",
       "FST_EUR_VS_AMR1    0\n",
       "FST_EUR_VS_AMR2    0\n",
       "FST_EUR_VS_BEB     0\n",
       "FST_EUR_VS_EAS     0\n",
       "FST_EUR_VS_GIH     0\n",
       "FST_EUR_VS_ITU     0\n",
       "FST_EUR_VS_PJL     0\n",
       "FST_EUR_VS_SAS     0\n",
       "FST_EUR_VS_STU     0\n",
       "P_EUR_AFR          0\n",
       "P_EUR_AMR          0\n",
       "P_EUR_AMR1         0\n",
       "P_EUR_AMR2         0\n",
       "P_EUR_BEB          0\n",
       "P_EUR_EAS          0\n",
       "P_EUR_GIH          0\n",
       "P_EUR_ITU          0\n",
       "P_EUR_PJL          0\n",
       "P_EUR_SAS          0\n",
       "P_EUR_STU          0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.isnull().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "1ca30276",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "2c3271b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_excel(\"Final_gene_list.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "c8d3c855",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Protein</th>\n",
       "      <th>Population</th>\n",
       "      <th>Gene Function</th>\n",
       "      <th>Primary Pathway</th>\n",
       "      <th>Cellular Location</th>\n",
       "      <th>Interacting Genes</th>\n",
       "      <th>SRNS/SSNS</th>\n",
       "      <th>Phenotypes</th>\n",
       "      <th>Inheritance</th>\n",
       "      <th>Pediatric (Y/N)</th>\n",
       "      <th>Adult (Y/N)</th>\n",
       "      <th>Both (Y/N)</th>\n",
       "      <th>Database</th>\n",
       "      <th>Database Link</th>\n",
       "      <th>Source Paper 1</th>\n",
       "      <th>Source Paper 2</th>\n",
       "      <th>Source Paper 3</th>\n",
       "      <th>Category</th>\n",
       "      <th>Variation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NPHS1</td>\n",
       "      <td>nephrin</td>\n",
       "      <td>Finnish type , highest in Finland but occurs i...</td>\n",
       "      <td>mutations in the gene encoSeems to play a role...</td>\n",
       "      <td>Cell-Cell communication , Nephrin family inter...</td>\n",
       "      <td>The gene is primarily expressed in renal tissu...</td>\n",
       "      <td>IQGAP1, NCK1, ARRB2 , CANX , FYN</td>\n",
       "      <td>SRNS</td>\n",
       "      <td>CNS, SRNS (NPHS1)</td>\n",
       "      <td>AR</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>Gwas</td>\n",
       "      <td>https://www.ebi.ac.uk/gwas/variants/rs2285450</td>\n",
       "      <td>https://pmc.ncbi.nlm.nih.gov/articles/PMC53836...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NPHS2</td>\n",
       "      <td>podocin</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Plays a role in the regulation of glomerular p...</td>\n",
       "      <td>Cell-Cell communication , Nephrin family inter...</td>\n",
       "      <td>Almost exclusively expressed in the podocytes ...</td>\n",
       "      <td>bcs1l , yme1l1 ,sgp7, afg3l2</td>\n",
       "      <td>SRNS</td>\n",
       "      <td>CNS, SRNS (NPHS2) &amp; Adult both</td>\n",
       "      <td>AR</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>Omim</td>\n",
       "      <td>https://www.omim.org/entry/604766?search=%22ne...</td>\n",
       "      <td>https://pmc.ncbi.nlm.nih.gov/articles/PMC53836...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PLCE1</td>\n",
       "      <td>phospholipase C, ε1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The production of the second messenger molecul...</td>\n",
       "      <td>GPR40 pathway, Nephrotic syndrome, Phosphoinos...</td>\n",
       "      <td>Expressed in podocytes</td>\n",
       "      <td>RHOA, HRAS, RAP1A, RAP2A, RAP2B, RRAS, AVIL , ...</td>\n",
       "      <td>SRNS</td>\n",
       "      <td>DMS, SRNS (NPHS3)</td>\n",
       "      <td>AR</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>Omim</td>\n",
       "      <td>https://www.omim.org/entry/610725?search=%22ne...</td>\n",
       "      <td>https://pmc.ncbi.nlm.nih.gov/articles/PMC53836...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CD2AP</td>\n",
       "      <td>CD2-associated protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seems to act as an adapter protein between mem...</td>\n",
       "      <td>GDNF signaling,Genes controlling nephrogenesis...</td>\n",
       "      <td>Widely expressed in fetal and adult tissues.</td>\n",
       "      <td>PKD2, NPHS1, NPHS2, WTIP, DDN, CBL BCAR1/p130...</td>\n",
       "      <td>SRNS</td>\n",
       "      <td>SRNS (FSGS3)</td>\n",
       "      <td>AD/AR</td>\n",
       "      <td>yes(recessive mutation)</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Omim</td>\n",
       "      <td>https://www.omim.org/entry/607832?search=%22ne...</td>\n",
       "      <td>https://pmc.ncbi.nlm.nih.gov/articles/PMC53836...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FAT1</td>\n",
       "      <td>FAT1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NS, Ciliopathy</td>\n",
       "      <td>AR</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>Omim</td>\n",
       "      <td>https://www.omim.org/entry/600976?search=%22ne...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene                 Protein  \\\n",
       "0  NPHS1                 nephrin   \n",
       "1  NPHS2                 podocin   \n",
       "2  PLCE1     phospholipase C, ε1   \n",
       "3  CD2AP  CD2-associated protein   \n",
       "4   FAT1                    FAT1   \n",
       "\n",
       "                                          Population  \\\n",
       "0  Finnish type , highest in Finland but occurs i...   \n",
       "1                                                NaN   \n",
       "2                                                NaN   \n",
       "3                                                NaN   \n",
       "4                                                NaN   \n",
       "\n",
       "                                       Gene Function  \\\n",
       "0  mutations in the gene encoSeems to play a role...   \n",
       "1  Plays a role in the regulation of glomerular p...   \n",
       "2  The production of the second messenger molecul...   \n",
       "3  Seems to act as an adapter protein between mem...   \n",
       "4                                                NaN   \n",
       "\n",
       "                                     Primary Pathway  \\\n",
       "0  Cell-Cell communication , Nephrin family inter...   \n",
       "1  Cell-Cell communication , Nephrin family inter...   \n",
       "2  GPR40 pathway, Nephrotic syndrome, Phosphoinos...   \n",
       "3  GDNF signaling,Genes controlling nephrogenesis...   \n",
       "4                                                NaN   \n",
       "\n",
       "                                   Cellular Location  \\\n",
       "0  The gene is primarily expressed in renal tissu...   \n",
       "1  Almost exclusively expressed in the podocytes ...   \n",
       "2                             Expressed in podocytes   \n",
       "3       Widely expressed in fetal and adult tissues.   \n",
       "4                                                NaN   \n",
       "\n",
       "                                   Interacting Genes SRNS/SSNS  \\\n",
       "0                   IQGAP1, NCK1, ARRB2 , CANX , FYN      SRNS   \n",
       "1                       bcs1l , yme1l1 ,sgp7, afg3l2      SRNS   \n",
       "2  RHOA, HRAS, RAP1A, RAP2A, RAP2B, RRAS, AVIL , ...      SRNS   \n",
       "3   PKD2, NPHS1, NPHS2, WTIP, DDN, CBL BCAR1/p130...      SRNS   \n",
       "4                                                NaN       NaN   \n",
       "\n",
       "                       Phenotypes Inheritance          Pediatric (Y/N)  \\\n",
       "0               CNS, SRNS (NPHS1)          AR                      yes   \n",
       "1  CNS, SRNS (NPHS2) & Adult both          AR                      yes   \n",
       "2               DMS, SRNS (NPHS3)          AR                      yes   \n",
       "3                    SRNS (FSGS3)       AD/AR  yes(recessive mutation)   \n",
       "4                  NS, Ciliopathy          AR                     yes    \n",
       "\n",
       "  Adult (Y/N) Both (Y/N) Database  \\\n",
       "0          no         no     Gwas   \n",
       "1          no        yes     Omim   \n",
       "2          no         no     Omim   \n",
       "3          no        NaN     Omim   \n",
       "4          no         no     Omim   \n",
       "\n",
       "                                       Database Link  \\\n",
       "0      https://www.ebi.ac.uk/gwas/variants/rs2285450   \n",
       "1  https://www.omim.org/entry/604766?search=%22ne...   \n",
       "2  https://www.omim.org/entry/610725?search=%22ne...   \n",
       "3  https://www.omim.org/entry/607832?search=%22ne...   \n",
       "4  https://www.omim.org/entry/600976?search=%22ne...   \n",
       "\n",
       "                                      Source Paper 1 Source Paper 2  \\\n",
       "0  https://pmc.ncbi.nlm.nih.gov/articles/PMC53836...            NaN   \n",
       "1  https://pmc.ncbi.nlm.nih.gov/articles/PMC53836...            NaN   \n",
       "2  https://pmc.ncbi.nlm.nih.gov/articles/PMC53836...            NaN   \n",
       "3  https://pmc.ncbi.nlm.nih.gov/articles/PMC53836...            NaN   \n",
       "4                                                NaN            NaN   \n",
       "\n",
       "   Source Paper 3  Category  Variation  \n",
       "0             NaN       NaN        NaN  \n",
       "1             NaN       NaN        NaN  \n",
       "2             NaN       NaN        NaN  \n",
       "3             NaN       NaN        NaN  \n",
       "4             NaN       NaN        NaN  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "4c0c9aed",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Gene', 'Protein', 'Population', 'Gene Function', 'Primary Pathway',\n",
       "       'Cellular Location', 'Interacting Genes', 'SRNS/SSNS', 'Phenotypes',\n",
       "       'Inheritance', ' Pediatric (Y/N)', 'Adult (Y/N)', 'Both (Y/N)',\n",
       "       'Database', 'Database Link', 'Source Paper 1', 'Source Paper 2',\n",
       "       'Source Paper 3', 'Category', 'Variation'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "b627a507",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 493 entries, 0 to 492\n",
      "Data columns (total 20 columns):\n",
      " #   Column             Non-Null Count  Dtype  \n",
      "---  ------             --------------  -----  \n",
      " 0   Gene               493 non-null    object \n",
      " 1   Protein            54 non-null     object \n",
      " 2   Population         6 non-null      object \n",
      " 3   Gene Function      24 non-null     object \n",
      " 4   Primary Pathway    18 non-null     object \n",
      " 5   Cellular Location  15 non-null     object \n",
      " 6   Interacting Genes  18 non-null     object \n",
      " 7   SRNS/SSNS          89 non-null     object \n",
      " 8   Phenotypes         166 non-null    object \n",
      " 9   Inheritance        82 non-null     object \n",
      " 10   Pediatric (Y/N)   152 non-null    object \n",
      " 11  Adult (Y/N)        154 non-null    object \n",
      " 12  Both (Y/N)         137 non-null    object \n",
      " 13  Database           468 non-null    object \n",
      " 14  Database Link      183 non-null    object \n",
      " 15  Source Paper 1     34 non-null     object \n",
      " 16  Source Paper 2     15 non-null     object \n",
      " 17  Source Paper 3     0 non-null      float64\n",
      " 18  Category           0 non-null      float64\n",
      " 19  Variation          0 non-null      float64\n",
      "dtypes: float64(3), object(17)\n",
      "memory usage: 77.2+ KB\n"
     ]
    }
   ],
   "source": [
    "df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "e14e9be5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(493, 20)"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "08bffa74",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"Final_chi_sq_fst_result_with_AF_GENE.tsv\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "37b6bb4b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AF</th>\n",
       "      <th>GENE</th>\n",
       "      <th>AF_AFR</th>\n",
       "      <th>AF_AMR</th>\n",
       "      <th>AF_AMR1</th>\n",
       "      <th>...</th>\n",
       "      <th>P_EUR_AMR</th>\n",
       "      <th>P_EUR_AMR1</th>\n",
       "      <th>P_EUR_AMR2</th>\n",
       "      <th>P_EUR_BEB</th>\n",
       "      <th>P_EUR_EAS</th>\n",
       "      <th>P_EUR_GIH</th>\n",
       "      <th>P_EUR_ITU</th>\n",
       "      <th>P_EUR_PJL</th>\n",
       "      <th>P_EUR_SAS</th>\n",
       "      <th>P_EUR_STU</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248349</td>\n",
       "      <td>rs112363642</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.204560</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.019037</td>\n",
       "      <td>0.217347</td>\n",
       "      <td>0.241697</td>\n",
       "      <td>...</td>\n",
       "      <td>0.973014</td>\n",
       "      <td>0.969612</td>\n",
       "      <td>0.897267</td>\n",
       "      <td>0.99913</td>\n",
       "      <td>0.603798</td>\n",
       "      <td>0.987329</td>\n",
       "      <td>0.8941</td>\n",
       "      <td>0.926662</td>\n",
       "      <td>0.967103</td>\n",
       "      <td>0.967103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248372</td>\n",
       "      <td>rs145305690</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000468</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248399</td>\n",
       "      <td>rs1404444736</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000312</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248405</td>\n",
       "      <td>rs535078794</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000468</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248419</td>\n",
       "      <td>rs371906837</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.003748</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.012878</td>\n",
       "      <td>0.001020</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM      POS            ID REF ALT        AF  GENE    AF_AFR    AF_AMR  \\\n",
       "0  chr19  1248349   rs112363642   G   A  0.204560  MIDN  0.019037  0.217347   \n",
       "1  chr19  1248372   rs145305690   C   T  0.000468  MIDN  0.000000  0.000000   \n",
       "2  chr19  1248399  rs1404444736   G   C  0.000312  MIDN  0.000000  0.000000   \n",
       "3  chr19  1248405   rs535078794   C   T  0.000468  MIDN  0.000000  0.000000   \n",
       "4  chr19  1248419   rs371906837   C   A  0.003748  MIDN  0.012878  0.001020   \n",
       "\n",
       "    AF_AMR1  ...  P_EUR_AMR  P_EUR_AMR1  P_EUR_AMR2  P_EUR_BEB  P_EUR_EAS  \\\n",
       "0  0.241697  ...   0.973014    0.969612    0.897267    0.99913   0.603798   \n",
       "1  0.000000  ...   1.000000    1.000000    1.000000    1.00000   0.000000   \n",
       "2  0.000000  ...   0.000000    0.000000    0.000000    0.00000   0.000000   \n",
       "3  0.000000  ...   1.000000    1.000000    1.000000    1.00000   0.000000   \n",
       "4  0.000000  ...   0.000000    1.000000    0.000000    1.00000   1.000000   \n",
       "\n",
       "   P_EUR_GIH  P_EUR_ITU  P_EUR_PJL  P_EUR_SAS  P_EUR_STU  \n",
       "0   0.987329     0.8941   0.926662   0.967103   0.967103  \n",
       "1   1.000000     1.0000   1.000000   1.000000   1.000000  \n",
       "2   0.000000     0.0000   0.000000   0.000000   0.000000  \n",
       "3   1.000000     1.0000   1.000000   1.000000   1.000000  \n",
       "4   1.000000     1.0000   1.000000   1.000000   1.000000  \n",
       "\n",
       "[5 rows x 41 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "55d0b061",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POS</th>\n",
       "      <th>AF</th>\n",
       "      <th>AF_AFR</th>\n",
       "      <th>AF_AMR</th>\n",
       "      <th>AF_AMR1</th>\n",
       "      <th>AF_AMR2</th>\n",
       "      <th>AF_EAS</th>\n",
       "      <th>AF_EUR</th>\n",
       "      <th>AF_SAS</th>\n",
       "      <th>AF_BEB</th>\n",
       "      <th>...</th>\n",
       "      <th>P_EUR_AMR</th>\n",
       "      <th>P_EUR_AMR1</th>\n",
       "      <th>P_EUR_AMR2</th>\n",
       "      <th>P_EUR_BEB</th>\n",
       "      <th>P_EUR_EAS</th>\n",
       "      <th>P_EUR_GIH</th>\n",
       "      <th>P_EUR_ITU</th>\n",
       "      <th>P_EUR_PJL</th>\n",
       "      <th>P_EUR_SAS</th>\n",
       "      <th>P_EUR_STU</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>...</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>7.477768e+07</td>\n",
       "      <td>5.146699e-02</td>\n",
       "      <td>5.649458e-02</td>\n",
       "      <td>4.984452e-02</td>\n",
       "      <td>4.969536e-02</td>\n",
       "      <td>5.002973e-02</td>\n",
       "      <td>5.040034e-02</td>\n",
       "      <td>4.831791e-02</td>\n",
       "      <td>4.991778e-02</td>\n",
       "      <td>4.983308e-02</td>\n",
       "      <td>...</td>\n",
       "      <td>6.961169e-01</td>\n",
       "      <td>7.321370e-01</td>\n",
       "      <td>7.289671e-01</td>\n",
       "      <td>7.194738e-01</td>\n",
       "      <td>6.229560e-01</td>\n",
       "      <td>7.395275e-01</td>\n",
       "      <td>7.295889e-01</td>\n",
       "      <td>7.366329e-01</td>\n",
       "      <td>7.245362e-01</td>\n",
       "      <td>7.245362e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>5.793876e+07</td>\n",
       "      <td>1.451645e-01</td>\n",
       "      <td>1.494555e-01</td>\n",
       "      <td>1.500193e-01</td>\n",
       "      <td>1.477009e-01</td>\n",
       "      <td>1.556383e-01</td>\n",
       "      <td>1.604505e-01</td>\n",
       "      <td>1.497605e-01</td>\n",
       "      <td>1.525278e-01</td>\n",
       "      <td>1.521766e-01</td>\n",
       "      <td>...</td>\n",
       "      <td>4.256450e-01</td>\n",
       "      <td>4.114855e-01</td>\n",
       "      <td>4.053948e-01</td>\n",
       "      <td>4.148443e-01</td>\n",
       "      <td>4.523035e-01</td>\n",
       "      <td>4.054333e-01</td>\n",
       "      <td>4.098429e-01</td>\n",
       "      <td>4.074041e-01</td>\n",
       "      <td>4.126985e-01</td>\n",
       "      <td>4.126985e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>1.160080e+05</td>\n",
       "      <td>2.080730e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>1.993037e+07</td>\n",
       "      <td>4.161460e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6.600250e-01</td>\n",
       "      <td>5.785506e-01</td>\n",
       "      <td>4.894919e-01</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6.324544e-01</td>\n",
       "      <td>5.688892e-01</td>\n",
       "      <td>6.369542e-01</td>\n",
       "      <td>5.226347e-01</td>\n",
       "      <td>5.226347e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>7.362441e+07</td>\n",
       "      <td>1.249220e-03</td>\n",
       "      <td>1.679730e-03</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>9.634765e-01</td>\n",
       "      <td>9.870043e-01</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>9.646361e-01</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>1.129737e+08</td>\n",
       "      <td>1.389760e-02</td>\n",
       "      <td>2.407610e-02</td>\n",
       "      <td>6.613760e-03</td>\n",
       "      <td>9.225090e-03</td>\n",
       "      <td>4.566210e-03</td>\n",
       "      <td>2.564100e-03</td>\n",
       "      <td>3.164560e-03</td>\n",
       "      <td>4.385960e-03</td>\n",
       "      <td>7.633590e-03</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>2.474497e+08</td>\n",
       "      <td>9.996880e-01</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows × 36 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                POS            AF        AF_AFR        AF_AMR       AF_AMR1  \\\n",
       "count  1.048575e+06  1.048575e+06  1.048575e+06  1.048575e+06  1.048575e+06   \n",
       "mean   7.477768e+07  5.146699e-02  5.649458e-02  4.984452e-02  4.969536e-02   \n",
       "std    5.793876e+07  1.451645e-01  1.494555e-01  1.500193e-01  1.477009e-01   \n",
       "min    1.160080e+05  2.080730e-04  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "25%    1.993037e+07  4.161460e-04  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "50%    7.362441e+07  1.249220e-03  1.679730e-03  0.000000e+00  0.000000e+00   \n",
       "75%    1.129737e+08  1.389760e-02  2.407610e-02  6.613760e-03  9.225090e-03   \n",
       "max    2.474497e+08  9.996880e-01  1.000000e+00  1.000000e+00  1.000000e+00   \n",
       "\n",
       "            AF_AMR2        AF_EAS        AF_EUR        AF_SAS        AF_BEB  \\\n",
       "count  1.048575e+06  1.048575e+06  1.048575e+06  1.048575e+06  1.048575e+06   \n",
       "mean   5.002973e-02  5.040034e-02  4.831791e-02  4.991778e-02  4.983308e-02   \n",
       "std    1.556383e-01  1.604505e-01  1.497605e-01  1.525278e-01  1.521766e-01   \n",
       "min    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "25%    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "50%    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "75%    4.566210e-03  2.564100e-03  3.164560e-03  4.385960e-03  7.633590e-03   \n",
       "max    1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00   \n",
       "\n",
       "       ...     P_EUR_AMR    P_EUR_AMR1    P_EUR_AMR2     P_EUR_BEB  \\\n",
       "count  ...  1.048575e+06  1.048575e+06  1.048575e+06  1.048575e+06   \n",
       "mean   ...  6.961169e-01  7.321370e-01  7.289671e-01  7.194738e-01   \n",
       "std    ...  4.256450e-01  4.114855e-01  4.053948e-01  4.148443e-01   \n",
       "min    ...  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "25%    ...  0.000000e+00  6.600250e-01  5.785506e-01  4.894919e-01   \n",
       "50%    ...  9.634765e-01  9.870043e-01  1.000000e+00  1.000000e+00   \n",
       "75%    ...  1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00   \n",
       "max    ...  1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00   \n",
       "\n",
       "          P_EUR_EAS     P_EUR_GIH     P_EUR_ITU     P_EUR_PJL     P_EUR_SAS  \\\n",
       "count  1.048575e+06  1.048575e+06  1.048575e+06  1.048575e+06  1.048575e+06   \n",
       "mean   6.229560e-01  7.395275e-01  7.295889e-01  7.366329e-01  7.245362e-01   \n",
       "std    4.523035e-01  4.054333e-01  4.098429e-01  4.074041e-01  4.126985e-01   \n",
       "min    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "25%    0.000000e+00  6.324544e-01  5.688892e-01  6.369542e-01  5.226347e-01   \n",
       "50%    9.646361e-01  1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00   \n",
       "75%    1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00   \n",
       "max    1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00   \n",
       "\n",
       "          P_EUR_STU  \n",
       "count  1.048575e+06  \n",
       "mean   7.245362e-01  \n",
       "std    4.126985e-01  \n",
       "min    0.000000e+00  \n",
       "25%    5.226347e-01  \n",
       "50%    1.000000e+00  \n",
       "75%    1.000000e+00  \n",
       "max    1.000000e+00  \n",
       "\n",
       "[8 rows x 36 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "fb439149",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 1048575 entries, 0 to 1048574\n",
      "Data columns (total 41 columns):\n",
      " #   Column           Non-Null Count    Dtype  \n",
      "---  ------           --------------    -----  \n",
      " 0   CHROM            1048575 non-null  object \n",
      " 1   POS              1048575 non-null  int64  \n",
      " 2   ID               1048575 non-null  object \n",
      " 3   REF              1048575 non-null  object \n",
      " 4   ALT              1048575 non-null  object \n",
      " 5   AF               1048575 non-null  float64\n",
      " 6   GENE             1048575 non-null  object \n",
      " 7   AF_AFR           1048575 non-null  float64\n",
      " 8   AF_AMR           1048575 non-null  float64\n",
      " 9   AF_AMR1          1048575 non-null  float64\n",
      " 10  AF_AMR2          1048575 non-null  float64\n",
      " 11  AF_EAS           1048575 non-null  float64\n",
      " 12  AF_EUR           1048575 non-null  float64\n",
      " 13  AF_SAS           1048575 non-null  float64\n",
      " 14  AF_BEB           1048575 non-null  float64\n",
      " 15  AF_GIH           1048575 non-null  float64\n",
      " 16  AF_ITU           1048575 non-null  float64\n",
      " 17  AF_PJL           1048575 non-null  float64\n",
      " 18  AF_STU           1048575 non-null  float64\n",
      " 19  FST_EUR_VS_AFR   1048575 non-null  float64\n",
      " 20  FST_EUR_VS_AMR   1048575 non-null  float64\n",
      " 21  FST_EUR_VS_AMR1  1048575 non-null  float64\n",
      " 22  FST_EUR_VS_AMR2  1048575 non-null  float64\n",
      " 23  FST_EUR_VS_BEB   1048575 non-null  float64\n",
      " 24  FST_EUR_VS_EAS   1048575 non-null  float64\n",
      " 25  FST_EUR_VS_GIH   1048575 non-null  float64\n",
      " 26  FST_EUR_VS_ITU   1048575 non-null  float64\n",
      " 27  FST_EUR_VS_PJL   1048575 non-null  float64\n",
      " 28  FST_EUR_VS_SAS   1048575 non-null  float64\n",
      " 29  FST_EUR_VS_STU   1048575 non-null  float64\n",
      " 30  P_EUR_AFR        1048575 non-null  float64\n",
      " 31  P_EUR_AMR        1048575 non-null  float64\n",
      " 32  P_EUR_AMR1       1048575 non-null  float64\n",
      " 33  P_EUR_AMR2       1048575 non-null  float64\n",
      " 34  P_EUR_BEB        1048575 non-null  float64\n",
      " 35  P_EUR_EAS        1048575 non-null  float64\n",
      " 36  P_EUR_GIH        1048575 non-null  float64\n",
      " 37  P_EUR_ITU        1048575 non-null  float64\n",
      " 38  P_EUR_PJL        1048575 non-null  float64\n",
      " 39  P_EUR_SAS        1048575 non-null  float64\n",
      " 40  P_EUR_STU        1048575 non-null  float64\n",
      "dtypes: float64(35), int64(1), object(5)\n",
      "memory usage: 328.0+ MB\n"
     ]
    }
   ],
   "source": [
    "df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "c73e4748",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1048575, 41)"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "f7fd1c0c",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"After_LD/Significant_Chisq_result/AFR_significant_variants.tsv\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "a105eed1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AF_EUR</th>\n",
       "      <th>AF_AFR</th>\n",
       "      <th>P_EUR_AFR</th>\n",
       "      <th>COUNT</th>\n",
       "      <th>PASS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248572</td>\n",
       "      <td>rs10425167</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001582</td>\n",
       "      <td>0.092945</td>\n",
       "      <td>0.020542</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1250398</td>\n",
       "      <td>rs3746107</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.013449</td>\n",
       "      <td>0.533595</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1251398</td>\n",
       "      <td>rs60724761</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.010285</td>\n",
       "      <td>0.210526</td>\n",
       "      <td>0.043161</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1251409</td>\n",
       "      <td>rs58898131</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.013449</td>\n",
       "      <td>0.530235</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1253350</td>\n",
       "      <td>rs113173533</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.155095</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM      POS           ID REF ALT    AF_EUR    AF_AFR  P_EUR_AFR  COUNT  \\\n",
       "0  chr19  1248572   rs10425167   G   C  0.001582  0.092945   0.020542      1   \n",
       "1  chr19  1250398    rs3746107   C   T  0.013449  0.533595   0.000006      1   \n",
       "2  chr19  1251398   rs60724761   C   T  0.010285  0.210526   0.043161      1   \n",
       "3  chr19  1251409   rs58898131   G   T  0.013449  0.530235   0.000006      1   \n",
       "4  chr19  1253350  rs113173533   C   T  0.000000  0.155095   0.000000      1   \n",
       "\n",
       "   PASS  \n",
       "0     1  \n",
       "1     1  \n",
       "2     1  \n",
       "3     1  \n",
       "4     1  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "a6f8019f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 40098 entries, 0 to 40097\n",
      "Data columns (total 10 columns):\n",
      " #   Column     Non-Null Count  Dtype  \n",
      "---  ------     --------------  -----  \n",
      " 0   CHROM      40098 non-null  object \n",
      " 1   POS        40098 non-null  int64  \n",
      " 2   ID         40098 non-null  object \n",
      " 3   REF        40098 non-null  object \n",
      " 4   ALT        40098 non-null  object \n",
      " 5   AF_EUR     40098 non-null  float64\n",
      " 6   AF_AFR     40098 non-null  float64\n",
      " 7   P_EUR_AFR  40098 non-null  float64\n",
      " 8   COUNT      40098 non-null  int64  \n",
      " 9   PASS       40098 non-null  int64  \n",
      "dtypes: float64(3), int64(3), object(4)\n",
      "memory usage: 3.1+ MB\n"
     ]
    }
   ],
   "source": [
    "df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "406fcbcd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(40098, 10)"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a525ba3e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df = pd.read_csv(\"Final_chi_sq_fst_result_with_AF_GENE.tsv\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "62ccd1e4",
   "metadata": {},
   "outputs": [],
   "source": [
    "df2 = pd.read_excel(\"Final_gene_list.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "0b3c9c79",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AF</th>\n",
       "      <th>GENE</th>\n",
       "      <th>AF_AFR</th>\n",
       "      <th>AF_AMR</th>\n",
       "      <th>AF_AMR1</th>\n",
       "      <th>...</th>\n",
       "      <th>P_EUR_AMR</th>\n",
       "      <th>P_EUR_AMR1</th>\n",
       "      <th>P_EUR_AMR2</th>\n",
       "      <th>P_EUR_BEB</th>\n",
       "      <th>P_EUR_EAS</th>\n",
       "      <th>P_EUR_GIH</th>\n",
       "      <th>P_EUR_ITU</th>\n",
       "      <th>P_EUR_PJL</th>\n",
       "      <th>P_EUR_SAS</th>\n",
       "      <th>P_EUR_STU</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248349</td>\n",
       "      <td>rs112363642</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.204560</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.019037</td>\n",
       "      <td>0.217347</td>\n",
       "      <td>0.241697</td>\n",
       "      <td>...</td>\n",
       "      <td>0.973014</td>\n",
       "      <td>0.969612</td>\n",
       "      <td>0.897267</td>\n",
       "      <td>0.99913</td>\n",
       "      <td>0.603798</td>\n",
       "      <td>0.987329</td>\n",
       "      <td>0.8941</td>\n",
       "      <td>0.926662</td>\n",
       "      <td>0.967103</td>\n",
       "      <td>0.967103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248372</td>\n",
       "      <td>rs145305690</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000468</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248399</td>\n",
       "      <td>rs1404444736</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000312</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248405</td>\n",
       "      <td>rs535078794</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000468</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248419</td>\n",
       "      <td>rs371906837</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.003748</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.012878</td>\n",
       "      <td>0.001020</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM      POS            ID REF ALT        AF  GENE    AF_AFR    AF_AMR  \\\n",
       "0  chr19  1248349   rs112363642   G   A  0.204560  MIDN  0.019037  0.217347   \n",
       "1  chr19  1248372   rs145305690   C   T  0.000468  MIDN  0.000000  0.000000   \n",
       "2  chr19  1248399  rs1404444736   G   C  0.000312  MIDN  0.000000  0.000000   \n",
       "3  chr19  1248405   rs535078794   C   T  0.000468  MIDN  0.000000  0.000000   \n",
       "4  chr19  1248419   rs371906837   C   A  0.003748  MIDN  0.012878  0.001020   \n",
       "\n",
       "    AF_AMR1  ...  P_EUR_AMR  P_EUR_AMR1  P_EUR_AMR2  P_EUR_BEB  P_EUR_EAS  \\\n",
       "0  0.241697  ...   0.973014    0.969612    0.897267    0.99913   0.603798   \n",
       "1  0.000000  ...   1.000000    1.000000    1.000000    1.00000   0.000000   \n",
       "2  0.000000  ...   0.000000    0.000000    0.000000    0.00000   0.000000   \n",
       "3  0.000000  ...   1.000000    1.000000    1.000000    1.00000   0.000000   \n",
       "4  0.000000  ...   0.000000    1.000000    0.000000    1.00000   1.000000   \n",
       "\n",
       "   P_EUR_GIH  P_EUR_ITU  P_EUR_PJL  P_EUR_SAS  P_EUR_STU  \n",
       "0   0.987329     0.8941   0.926662   0.967103   0.967103  \n",
       "1   1.000000     1.0000   1.000000   1.000000   1.000000  \n",
       "2   0.000000     0.0000   0.000000   0.000000   0.000000  \n",
       "3   1.000000     1.0000   1.000000   1.000000   1.000000  \n",
       "4   1.000000     1.0000   1.000000   1.000000   1.000000  \n",
       "\n",
       "[5 rows x 41 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "593764fd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Ensembl_ID</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>SRNS/SSNS</th>\n",
       "      <th>Phenotypes</th>\n",
       "      <th>Inheritance</th>\n",
       "      <th>Pediatric (Y/N)</th>\n",
       "      <th>Adult (Y/N)</th>\n",
       "      <th>...</th>\n",
       "      <th>GWAS_Link</th>\n",
       "      <th>ClinPGx_Link</th>\n",
       "      <th>MalaCards_Link</th>\n",
       "      <th>Function</th>\n",
       "      <th>Pathway</th>\n",
       "      <th>Source Paper 1</th>\n",
       "      <th>Source Paper 2</th>\n",
       "      <th>Source Paper 3</th>\n",
       "      <th>Category</th>\n",
       "      <th>Variation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NPHS1</td>\n",
       "      <td>ENSG00000161270</td>\n",
       "      <td>19</td>\n",
       "      <td>35825372.0</td>\n",
       "      <td>35869287.0</td>\n",
       "      <td>SRNS</td>\n",
       "      <td>CNS, SRNS (NPHS1)</td>\n",
       "      <td>AR</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.ebi.ac.uk/gwas/variants/rs2285450</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>PMC5383633</td>\n",
       "      <td>9543371,9660941 ,9915943 ,10577936 ,10652016 ,...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NPHS2</td>\n",
       "      <td>ENSG00000116218</td>\n",
       "      <td>1</td>\n",
       "      <td>179550494.0</td>\n",
       "      <td>179575952.0</td>\n",
       "      <td>SRNS</td>\n",
       "      <td>CNS, SRNS (NPHS2) &amp; Adult both</td>\n",
       "      <td>AR</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>PMC5383633</td>\n",
       "      <td>8589695,8606597 ,10742096 ,11729243 ,11733557 ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PLCE1</td>\n",
       "      <td>ENSG00000138193</td>\n",
       "      <td>10</td>\n",
       "      <td>93993894.0</td>\n",
       "      <td>94332823.0</td>\n",
       "      <td>SRNS</td>\n",
       "      <td>DMS, SRNS (NPHS3)</td>\n",
       "      <td>AR</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>PMC5383633</td>\n",
       "      <td>29127259</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CD2AP</td>\n",
       "      <td>ENSG00000198087</td>\n",
       "      <td>6</td>\n",
       "      <td>47477752.0</td>\n",
       "      <td>47627263.0</td>\n",
       "      <td>SRNS</td>\n",
       "      <td>SRNS (FSGS3)</td>\n",
       "      <td>AD/AR</td>\n",
       "      <td>yes(recessive mutation)</td>\n",
       "      <td>no</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>PMC5383633</td>\n",
       "      <td>19131354, 10997929, 12217865, 17459670, 146431...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FAT1</td>\n",
       "      <td>ENSG00000083857</td>\n",
       "      <td>4</td>\n",
       "      <td>186587791.0</td>\n",
       "      <td>186726722.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NS, Ciliopathy</td>\n",
       "      <td>AR</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>26905694</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene       Ensembl_ID chromosome        start          end SRNS/SSNS  \\\n",
       "0  NPHS1  ENSG00000161270         19   35825372.0   35869287.0      SRNS   \n",
       "1  NPHS2  ENSG00000116218          1  179550494.0  179575952.0      SRNS   \n",
       "2  PLCE1  ENSG00000138193         10   93993894.0   94332823.0      SRNS   \n",
       "3  CD2AP  ENSG00000198087          6   47477752.0   47627263.0      SRNS   \n",
       "4   FAT1  ENSG00000083857          4  186587791.0  186726722.0       NaN   \n",
       "\n",
       "                       Phenotypes Inheritance          Pediatric (Y/N)  \\\n",
       "0               CNS, SRNS (NPHS1)          AR                      yes   \n",
       "1  CNS, SRNS (NPHS2) & Adult both          AR                      yes   \n",
       "2               DMS, SRNS (NPHS3)          AR                      yes   \n",
       "3                    SRNS (FSGS3)       AD/AR  yes(recessive mutation)   \n",
       "4                  NS, Ciliopathy          AR                     yes    \n",
       "\n",
       "  Adult (Y/N)  ...                                      GWAS_Link  \\\n",
       "0          no  ...  https://www.ebi.ac.uk/gwas/variants/rs2285450   \n",
       "1          no  ...                                            NaN   \n",
       "2          no  ...                                            NaN   \n",
       "3          no  ...                                            NaN   \n",
       "4          no  ...                                            NaN   \n",
       "\n",
       "  ClinPGx_Link                                     MalaCards_Link  \\\n",
       "0          NaN  https://www.genecards.org/cgi-bin/carddisp.pl?...   \n",
       "1          NaN  https://www.genecards.org/cgi-bin/carddisp.pl?...   \n",
       "2          NaN  https://www.genecards.org/cgi-bin/carddisp.pl?...   \n",
       "3          NaN  https://www.genecards.org/cgi-bin/carddisp.pl?...   \n",
       "4          NaN  https://www.genecards.org/cgi-bin/carddisp.pl?...   \n",
       "\n",
       "                                            Function  \\\n",
       "0  https://www.genecards.org/cgi-bin/carddisp.pl?...   \n",
       "1  https://www.genecards.org/cgi-bin/carddisp.pl?...   \n",
       "2  https://www.genecards.org/cgi-bin/carddisp.pl?...   \n",
       "3  https://www.genecards.org/cgi-bin/carddisp.pl?...   \n",
       "4  https://www.genecards.org/cgi-bin/carddisp.pl?...   \n",
       "\n",
       "                                             Pathway Source Paper 1  \\\n",
       "0  https://www.genecards.org/cgi-bin/carddisp.pl?...     PMC5383633   \n",
       "1  https://www.genecards.org/cgi-bin/carddisp.pl?...     PMC5383633   \n",
       "2  https://www.genecards.org/cgi-bin/carddisp.pl?...     PMC5383633   \n",
       "3  https://www.genecards.org/cgi-bin/carddisp.pl?...     PMC5383633   \n",
       "4  https://www.genecards.org/cgi-bin/carddisp.pl?...       26905694   \n",
       "\n",
       "                                      Source Paper 2 Source Paper 3 Category  \\\n",
       "0  9543371,9660941 ,9915943 ,10577936 ,10652016 ,...            NaN      NaN   \n",
       "1  8589695,8606597 ,10742096 ,11729243 ,11733557 ...            NaN      NaN   \n",
       "2                                           29127259            NaN      NaN   \n",
       "3  19131354, 10997929, 12217865, 17459670, 146431...            NaN      NaN   \n",
       "4                                                NaN            NaN      NaN   \n",
       "\n",
       "  Variation  \n",
       "0       NaN  \n",
       "1       NaN  \n",
       "2       NaN  \n",
       "3       NaN  \n",
       "4       NaN  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "40a104e2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CHROM', 'POS', 'ID', 'REF', 'ALT', 'AF', 'GENE', 'AF_AFR', 'AF_AMR',\n",
       "       'AF_AMR1', 'AF_AMR2', 'AF_EAS', 'AF_EUR', 'AF_SAS', 'AF_BEB', 'AF_GIH',\n",
       "       'AF_ITU', 'AF_PJL', 'AF_STU', 'FST_EUR_VS_AFR', 'FST_EUR_VS_AMR',\n",
       "       'FST_EUR_VS_AMR1', 'FST_EUR_VS_AMR2', 'FST_EUR_VS_BEB',\n",
       "       'FST_EUR_VS_EAS', 'FST_EUR_VS_GIH', 'FST_EUR_VS_ITU', 'FST_EUR_VS_PJL',\n",
       "       'FST_EUR_VS_SAS', 'FST_EUR_VS_STU', 'P_EUR_AFR', 'P_EUR_AMR',\n",
       "       'P_EUR_AMR1', 'P_EUR_AMR2', 'P_EUR_BEB', 'P_EUR_EAS', 'P_EUR_GIH',\n",
       "       'P_EUR_ITU', 'P_EUR_PJL', 'P_EUR_SAS', 'P_EUR_STU'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "f89d7b5c",
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"db_SNP_link\"] = \"https://www.ncbi.nlm.nih.gov/snp/\" + df[\"ID\"].str.strip()\n",
    "df.to_csv(\"Final_with_dbsnp.tsv\", sep=\"\\t\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "afb777dd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CHROM', 'POS', 'ID', 'REF', 'ALT', 'AF', 'GENE', 'AF_AFR', 'AF_AMR',\n",
       "       'AF_AMR1', 'AF_AMR2', 'AF_EAS', 'AF_EUR', 'AF_SAS', 'AF_BEB', 'AF_GIH',\n",
       "       'AF_ITU', 'AF_PJL', 'AF_STU', 'FST_EUR_VS_AFR', 'FST_EUR_VS_AMR',\n",
       "       'FST_EUR_VS_AMR1', 'FST_EUR_VS_AMR2', 'FST_EUR_VS_BEB',\n",
       "       'FST_EUR_VS_EAS', 'FST_EUR_VS_GIH', 'FST_EUR_VS_ITU', 'FST_EUR_VS_PJL',\n",
       "       'FST_EUR_VS_SAS', 'FST_EUR_VS_STU', 'P_EUR_AFR', 'P_EUR_AMR',\n",
       "       'P_EUR_AMR1', 'P_EUR_AMR2', 'P_EUR_BEB', 'P_EUR_EAS', 'P_EUR_GIH',\n",
       "       'P_EUR_ITU', 'P_EUR_PJL', 'P_EUR_SAS', 'P_EUR_STU', 'db_SNP_link'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "defe8ea2",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_excel(\"Final_gene_list.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "628f6d17",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Gene', 'Ensembl_ID', 'chromosome', 'start', 'end', 'SRNS/SSNS',\n",
       "       'Phenotypes', 'Inheritance', ' Pediatric (Y/N)', 'Adult (Y/N)',\n",
       "       'Both (Y/N)', 'OMIM_Link', 'GWAS_Link', 'ClinPGx_Link',\n",
       "       'MalaCards_Link', 'Function', 'Pathway', 'Source Paper 1',\n",
       "       'Source Paper 2', 'Source Paper 3', 'Category', 'Variation'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "e7c2645f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "============================================================\n",
      "Nephrotic Syndrome Genetics Database Setup - UPDATED\n",
      "============================================================\n",
      "✓ Connected to database: nephrotic_syndrome.db\n",
      "✓ Database schema created successfully\n",
      "\n",
      "============================================================\n",
      "LOADING DATA\n",
      "============================================================\n",
      "\n",
      "📊 Loading gene data...\n",
      "✓ Loaded 494 genes\n",
      "\n",
      "📊 Loading variant data (this may take a few minutes)...\n",
      "  Processed chunk 1 (10000 variants)\n",
      "  Processed chunk 2 (10000 variants)\n",
      "  Processed chunk 3 (10000 variants)\n",
      "  Processed chunk 4 (10000 variants)\n",
      "  Processed chunk 5 (10000 variants)\n",
      "  Processed chunk 6 (10000 variants)\n",
      "  Processed chunk 7 (10000 variants)\n",
      "  Processed chunk 8 (10000 variants)\n",
      "  Processed chunk 9 (10000 variants)\n",
      "  Processed chunk 10 (10000 variants)\n",
      "  Processed chunk 11 (10000 variants)\n",
      "  Processed chunk 12 (10000 variants)\n",
      "  Processed chunk 13 (10000 variants)\n",
      "  Processed chunk 14 (10000 variants)\n",
      "  Processed chunk 15 (10000 variants)\n",
      "  Processed chunk 16 (10000 variants)\n",
      "  Processed chunk 17 (10000 variants)\n",
      "  Processed chunk 18 (10000 variants)\n",
      "  Processed chunk 19 (10000 variants)\n",
      "  Processed chunk 20 (10000 variants)\n",
      "  Processed chunk 21 (10000 variants)\n",
      "  Processed chunk 22 (10000 variants)\n",
      "  Processed chunk 23 (10000 variants)\n",
      "  Processed chunk 24 (10000 variants)\n",
      "  Processed chunk 25 (10000 variants)\n",
      "  Processed chunk 26 (10000 variants)\n",
      "  Processed chunk 27 (10000 variants)\n",
      "  Processed chunk 28 (10000 variants)\n",
      "  Processed chunk 29 (10000 variants)\n",
      "  Processed chunk 30 (10000 variants)\n",
      "  Processed chunk 31 (10000 variants)\n",
      "  Processed chunk 32 (10000 variants)\n",
      "  Processed chunk 33 (10000 variants)\n",
      "  Processed chunk 34 (10000 variants)\n",
      "  Processed chunk 35 (10000 variants)\n",
      "  Processed chunk 36 (10000 variants)\n",
      "  Processed chunk 37 (10000 variants)\n",
      "  Processed chunk 38 (10000 variants)\n",
      "  Processed chunk 39 (10000 variants)\n",
      "  Processed chunk 40 (10000 variants)\n",
      "  Processed chunk 41 (10000 variants)\n",
      "  Processed chunk 42 (10000 variants)\n",
      "  Processed chunk 43 (10000 variants)\n",
      "  Processed chunk 44 (10000 variants)\n",
      "  Processed chunk 45 (10000 variants)\n",
      "  Processed chunk 46 (10000 variants)\n",
      "  Processed chunk 47 (10000 variants)\n",
      "  Processed chunk 48 (10000 variants)\n",
      "  Processed chunk 49 (10000 variants)\n",
      "  Processed chunk 50 (10000 variants)\n",
      "  Processed chunk 51 (10000 variants)\n",
      "  Processed chunk 52 (10000 variants)\n",
      "  Processed chunk 53 (10000 variants)\n",
      "  Processed chunk 54 (10000 variants)\n",
      "  Processed chunk 55 (10000 variants)\n",
      "  Processed chunk 56 (10000 variants)\n",
      "  Processed chunk 57 (10000 variants)\n",
      "  Processed chunk 58 (10000 variants)\n",
      "  Processed chunk 59 (10000 variants)\n",
      "  Processed chunk 60 (10000 variants)\n",
      "  Processed chunk 61 (10000 variants)\n",
      "  Processed chunk 62 (10000 variants)\n",
      "  Processed chunk 63 (10000 variants)\n",
      "  Processed chunk 64 (10000 variants)\n",
      "  Processed chunk 65 (10000 variants)\n",
      "  Processed chunk 66 (10000 variants)\n",
      "  Processed chunk 67 (10000 variants)\n",
      "  Processed chunk 68 (10000 variants)\n",
      "  Processed chunk 69 (10000 variants)\n",
      "  Processed chunk 70 (10000 variants)\n",
      "  Processed chunk 71 (10000 variants)\n",
      "  Processed chunk 72 (10000 variants)\n",
      "  Processed chunk 73 (10000 variants)\n",
      "  Processed chunk 74 (10000 variants)\n",
      "  Processed chunk 75 (10000 variants)\n",
      "  Processed chunk 76 (10000 variants)\n",
      "  Processed chunk 77 (10000 variants)\n",
      "  Processed chunk 78 (10000 variants)\n",
      "  Processed chunk 79 (10000 variants)\n",
      "  Processed chunk 80 (10000 variants)\n",
      "  Processed chunk 81 (10000 variants)\n",
      "  Processed chunk 82 (10000 variants)\n",
      "  Processed chunk 83 (10000 variants)\n",
      "  Processed chunk 84 (10000 variants)\n",
      "  Processed chunk 85 (10000 variants)\n",
      "  Processed chunk 86 (10000 variants)\n",
      "  Processed chunk 87 (10000 variants)\n",
      "  Processed chunk 88 (10000 variants)\n",
      "  Processed chunk 89 (10000 variants)\n",
      "  Processed chunk 90 (10000 variants)\n",
      "  Processed chunk 91 (10000 variants)\n",
      "  Processed chunk 92 (10000 variants)\n",
      "  Processed chunk 93 (10000 variants)\n",
      "  Processed chunk 94 (10000 variants)\n",
      "  Processed chunk 95 (10000 variants)\n",
      "  Processed chunk 96 (10000 variants)\n",
      "  Processed chunk 97 (10000 variants)\n",
      "  Processed chunk 98 (10000 variants)\n",
      "  Processed chunk 99 (10000 variants)\n",
      "  Processed chunk 100 (10000 variants)\n",
      "  Processed chunk 101 (10000 variants)\n",
      "  Processed chunk 102 (10000 variants)\n",
      "  Processed chunk 103 (10000 variants)\n",
      "  Processed chunk 104 (10000 variants)\n",
      "  Processed chunk 105 (8575 variants)\n",
      "✓ Loaded all variants\n",
      "\n",
      "📊 Loading significant variants (before LD)...\n",
      "✓ Loaded significant variants (before LD)\n",
      "  Updated: 214167 variant-population combinations\n",
      "  Skipped: 1322 variants not found in main dataset\n",
      "\n",
      "📊 Loading significant variants (after LD)...\n",
      "✓ Loaded significant variants (after LD)\n",
      "  Updated: 169000 variant-population combinations\n",
      "  Skipped: 1329 variants not found in main dataset\n",
      "\n",
      "✓ Database connection closed\n",
      "\n",
      "============================================================\n",
      "DATABASE SETUP COMPLETE!\n",
      "============================================================\n",
      "\n",
      "Your database 'nephrotic_syndrome.db' is ready to use!\n",
      "\n",
      "Next steps:\n",
      "1. Test queries with the query examples script\n",
      "2. Set up your web dashboard using Flask/Django\n",
      "3. Deploy your application\n"
     ]
    }
   ],
   "source": [
    "\"\"\"\n",
    "Nephrotic Syndrome Genetic Diversity Database Setup - UPDATED VERSION\n",
    "This script creates and populates a SQLite database with enhanced links and annotations\n",
    "\"\"\"\n",
    "\n",
    "import sqlite3\n",
    "import pandas as pd\n",
    "import os\n",
    "from pathlib import Path\n",
    "\n",
    "# Database Schema Design - UPDATED\n",
    "DATABASE_SCHEMA = \"\"\"\n",
    "-- Table 1: Genes - Core gene information with enhanced annotations\n",
    "CREATE TABLE IF NOT EXISTS genes (\n",
    "    gene_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    gene_name TEXT UNIQUE NOT NULL,\n",
    "    ensembl_id TEXT,\n",
    "    chromosome TEXT,\n",
    "    start_position INTEGER,\n",
    "    end_position INTEGER,\n",
    "    srns_ssns TEXT,\n",
    "    phenotypes TEXT,\n",
    "    inheritance TEXT,\n",
    "    pediatric TEXT,\n",
    "    adult TEXT,\n",
    "    both TEXT,\n",
    "    gene_function TEXT,\n",
    "    pathway TEXT,\n",
    "    category TEXT,\n",
    "    variation TEXT,\n",
    "    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP\n",
    ");\n",
    "\n",
    "-- Table 2: Gene Links - All database and literature links\n",
    "CREATE TABLE IF NOT EXISTS gene_links (\n",
    "    link_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    gene_name TEXT NOT NULL,\n",
    "    omim_link TEXT,\n",
    "    gwas_link TEXT,\n",
    "    clinpgx_link TEXT,\n",
    "    malacards_link TEXT,\n",
    "    function_link TEXT,\n",
    "    pathway_link TEXT,\n",
    "    source_paper_1 TEXT,\n",
    "    source_paper_2 TEXT,\n",
    "    source_paper_3 TEXT,\n",
    "    FOREIGN KEY (gene_name) REFERENCES genes(gene_name)\n",
    ");\n",
    "\n",
    "-- Table 3: Variants - All variant information with dbSNP links\n",
    "CREATE TABLE IF NOT EXISTS variants (\n",
    "    variant_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    rs_id TEXT UNIQUE NOT NULL,\n",
    "    chrom TEXT NOT NULL,\n",
    "    position INTEGER NOT NULL,\n",
    "    ref_allele TEXT NOT NULL,\n",
    "    alt_allele TEXT NOT NULL,\n",
    "    gene_name TEXT NOT NULL,\n",
    "    global_af REAL,\n",
    "    dbsnp_link TEXT,\n",
    "    FOREIGN KEY (gene_name) REFERENCES genes(gene_name),\n",
    "    UNIQUE(chrom, position, ref_allele, alt_allele)\n",
    ");\n",
    "\n",
    "-- Table 4: Allele Frequencies - Population-specific frequencies\n",
    "CREATE TABLE IF NOT EXISTS allele_frequencies (\n",
    "    af_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    variant_id INTEGER NOT NULL,\n",
    "    population TEXT NOT NULL,\n",
    "    allele_frequency REAL NOT NULL,\n",
    "    FOREIGN KEY (variant_id) REFERENCES variants(variant_id),\n",
    "    UNIQUE(variant_id, population)\n",
    ");\n",
    "\n",
    "-- Table 5: Statistical Tests - FST and Chi-square results\n",
    "CREATE TABLE IF NOT EXISTS statistical_tests (\n",
    "    test_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    variant_id INTEGER NOT NULL,\n",
    "    population TEXT NOT NULL,\n",
    "    fst_value REAL,\n",
    "    chi_square_p_value REAL,\n",
    "    is_significant_chisq BOOLEAN DEFAULT 0,\n",
    "    is_significant_bonferroni BOOLEAN DEFAULT 0,\n",
    "    is_significant_fst BOOLEAN DEFAULT 0,\n",
    "    FOREIGN KEY (variant_id) REFERENCES variants(variant_id),\n",
    "    UNIQUE(variant_id, population)\n",
    ");\n",
    "\n",
    "-- Table 6: LD Status - Linkage Disequilibrium information\n",
    "CREATE TABLE IF NOT EXISTS ld_status (\n",
    "    ld_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    variant_id INTEGER NOT NULL,\n",
    "    population TEXT NOT NULL,\n",
    "    in_ld BOOLEAN DEFAULT 0,\n",
    "    ld_pruned BOOLEAN DEFAULT 0,\n",
    "    FOREIGN KEY (variant_id) REFERENCES variants(variant_id),\n",
    "    UNIQUE(variant_id, population)\n",
    ");\n",
    "\n",
    "-- Indexes for faster queries\n",
    "CREATE INDEX IF NOT EXISTS idx_variants_gene ON variants(gene_name);\n",
    "CREATE INDEX IF NOT EXISTS idx_variants_rs ON variants(rs_id);\n",
    "CREATE INDEX IF NOT EXISTS idx_variants_chrom_pos ON variants(chrom, position);\n",
    "CREATE INDEX IF NOT EXISTS idx_af_variant ON allele_frequencies(variant_id);\n",
    "CREATE INDEX IF NOT EXISTS idx_af_population ON allele_frequencies(population);\n",
    "CREATE INDEX IF NOT EXISTS idx_stats_variant ON statistical_tests(variant_id);\n",
    "CREATE INDEX IF NOT EXISTS idx_ld_variant ON ld_status(variant_id);\n",
    "CREATE INDEX IF NOT EXISTS idx_gene_links ON gene_links(gene_name);\n",
    "\"\"\"\n",
    "\n",
    "class NephroticSyndromeDB:\n",
    "    def __init__(self, db_path='nephrotic_syndrome.db'):\n",
    "        \"\"\"Initialize database connection\"\"\"\n",
    "        self.db_path = db_path\n",
    "        self.conn = None\n",
    "        self.cursor = None\n",
    "        \n",
    "    def connect(self):\n",
    "        \"\"\"Connect to SQLite database\"\"\"\n",
    "        self.conn = sqlite3.connect(self.db_path)\n",
    "        self.cursor = self.conn.cursor()\n",
    "        print(f\"✓ Connected to database: {self.db_path}\")\n",
    "        \n",
    "    def create_schema(self):\n",
    "        \"\"\"Create database schema\"\"\"\n",
    "        self.cursor.executescript(DATABASE_SCHEMA)\n",
    "        self.conn.commit()\n",
    "        print(\"✓ Database schema created successfully\")\n",
    "        \n",
    "    def load_gene_data(self, excel_path):\n",
    "        \"\"\"Load gene information from Excel file - UPDATED\"\"\"\n",
    "        print(\"\\n📊 Loading gene data...\")\n",
    "        df = pd.read_excel(excel_path)\n",
    "        \n",
    "        # Clean column names\n",
    "        df.columns = df.columns.str.strip()\n",
    "        \n",
    "        gene_count = 0\n",
    "        for idx, row in df.iterrows():\n",
    "            try:\n",
    "                # Insert gene information\n",
    "                self.cursor.execute(\"\"\"\n",
    "                    INSERT OR IGNORE INTO genes (\n",
    "                        gene_name, ensembl_id, chromosome, start_position, end_position,\n",
    "                        srns_ssns, phenotypes, inheritance, pediatric, adult, both,\n",
    "                        gene_function, pathway, category, variation\n",
    "                    ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)\n",
    "                \"\"\", (\n",
    "                    row.get('Gene'),\n",
    "                    row.get('Ensembl_ID'),\n",
    "                    row.get('chromosome'),\n",
    "                    row.get('start'),\n",
    "                    row.get('end'),\n",
    "                    row.get('SRNS/SSNS'),\n",
    "                    row.get('Phenotypes'),\n",
    "                    row.get('Inheritance'),\n",
    "                    row.get('Pediatric (Y/N)') or row.get(' Pediatric (Y/N)'),\n",
    "                    row.get('Adult (Y/N)'),\n",
    "                    row.get('Both (Y/N)'),\n",
    "                    row.get('Function'),\n",
    "                    row.get('Pathway'),\n",
    "                    row.get('Category'),\n",
    "                    row.get('Variation')\n",
    "                ))\n",
    "                \n",
    "                # Insert link information\n",
    "                self.cursor.execute(\"\"\"\n",
    "                    INSERT OR IGNORE INTO gene_links (\n",
    "                        gene_name, omim_link, gwas_link, clinpgx_link, malacards_link,\n",
    "                        function_link, pathway_link,\n",
    "                        source_paper_1, source_paper_2, source_paper_3\n",
    "                    ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)\n",
    "                \"\"\", (\n",
    "                    row.get('Gene'),\n",
    "                    row.get('OMIM_Link'),\n",
    "                    row.get('GWAS_Link'),\n",
    "                    row.get('ClinPGx_Link'),\n",
    "                    row.get('MalaCards_Link'),\n",
    "                    row.get('Function'),  # This is the function link\n",
    "                    row.get('Pathway'),   # This is the pathway link\n",
    "                    row.get('Source Paper 1'),\n",
    "                    row.get('Source Paper 2'),\n",
    "                    row.get('Source Paper 3')\n",
    "                ))\n",
    "                \n",
    "                gene_count += 1\n",
    "                \n",
    "            except Exception as e:\n",
    "                print(f\"Error inserting gene {row.get('Gene')}: {e}\")\n",
    "                \n",
    "        self.conn.commit()\n",
    "        print(f\"✓ Loaded {gene_count} genes\")\n",
    "        \n",
    "    def load_variant_data(self, tsv_path):\n",
    "        \"\"\"Load variant and statistical test data from main TSV file - UPDATED\"\"\"\n",
    "        print(\"\\n📊 Loading variant data (this may take a few minutes)...\")\n",
    "        \n",
    "        # Read in chunks to handle large file\n",
    "        chunk_size = 10000\n",
    "        chunks_processed = 0\n",
    "        \n",
    "        for chunk in pd.read_csv(tsv_path, sep='\\t', chunksize=chunk_size):\n",
    "            variant_count = 0\n",
    "            \n",
    "            for idx, row in chunk.iterrows():\n",
    "                try:\n",
    "                    # Insert variant with dbSNP link\n",
    "                    self.cursor.execute(\"\"\"\n",
    "                        INSERT OR IGNORE INTO variants (\n",
    "                            rs_id, chrom, position, ref_allele, alt_allele, \n",
    "                            gene_name, global_af, dbsnp_link\n",
    "                        ) VALUES (?, ?, ?, ?, ?, ?, ?, ?)\n",
    "                    \"\"\", (\n",
    "                        row['ID'],\n",
    "                        row['CHROM'],\n",
    "                        row['POS'],\n",
    "                        row['REF'],\n",
    "                        row['ALT'],\n",
    "                        row['GENE'],\n",
    "                        row['AF'],\n",
    "                        row.get('db_SNP_link')\n",
    "                    ))\n",
    "                    \n",
    "                    # Get variant_id\n",
    "                    variant_id = self.cursor.execute(\n",
    "                        \"SELECT variant_id FROM variants WHERE rs_id = ?\",\n",
    "                        (row['ID'],)\n",
    "                    ).fetchone()[0]\n",
    "                    \n",
    "                    # Insert allele frequencies for all populations\n",
    "                    populations = {\n",
    "                        'AFR': row['AF_AFR'], 'AMR': row['AF_AMR'], 'AMR1': row['AF_AMR1'],\n",
    "                        'AMR2': row['AF_AMR2'], 'EAS': row['AF_EAS'], 'EUR': row['AF_EUR'],\n",
    "                        'SAS': row['AF_SAS'], 'BEB': row['AF_BEB'], 'GIH': row['AF_GIH'],\n",
    "                        'ITU': row['AF_ITU'], 'PJL': row['AF_PJL'], 'STU': row['AF_STU']\n",
    "                    }\n",
    "                    \n",
    "                    for pop, freq in populations.items():\n",
    "                        self.cursor.execute(\"\"\"\n",
    "                            INSERT OR IGNORE INTO allele_frequencies (variant_id, population, allele_frequency)\n",
    "                            VALUES (?, ?, ?)\n",
    "                        \"\"\", (variant_id, pop, freq))\n",
    "                    \n",
    "                    # Insert statistical test results\n",
    "                    stat_populations = ['AFR', 'AMR', 'AMR1', 'AMR2', 'EAS', 'SAS', 'BEB', 'GIH', 'ITU', 'PJL', 'STU']\n",
    "                    \n",
    "                    for pop in stat_populations:\n",
    "                        fst_col = f'FST_EUR_VS_{pop}'\n",
    "                        p_col = f'P_EUR_{pop}'\n",
    "                        \n",
    "                        self.cursor.execute(\"\"\"\n",
    "                            INSERT OR IGNORE INTO statistical_tests (\n",
    "                                variant_id, population, fst_value, chi_square_p_value\n",
    "                            ) VALUES (?, ?, ?, ?)\n",
    "                        \"\"\", (variant_id, pop, row[fst_col], row[p_col]))\n",
    "                    \n",
    "                    variant_count += 1\n",
    "                    \n",
    "                except Exception as e:\n",
    "                    print(f\"Error processing variant {row.get('ID')}: {e}\")\n",
    "                    continue\n",
    "            \n",
    "            chunks_processed += 1\n",
    "            self.conn.commit()\n",
    "            print(f\"  Processed chunk {chunks_processed} ({variant_count} variants)\")\n",
    "            \n",
    "        print(f\"✓ Loaded all variants\")\n",
    "        \n",
    "    def load_significant_variants(self, folder_path, ld_status='before'):\n",
    "        \"\"\"Load significant variants and mark them in the database\"\"\"\n",
    "        print(f\"\\n📊 Loading significant variants ({ld_status} LD)...\")\n",
    "        \n",
    "        test_types = {\n",
    "            'Significant Chisq result': 'chisq',\n",
    "            'Significant_Chisq_result': 'chisq',\n",
    "            'Significant_Chisq_result_Bonferroni': 'bonferroni',\n",
    "            'Significant_Bonferroni_with_FST': 'bonferroni',\n",
    "            'Significant_Fst': 'fst'\n",
    "        }\n",
    "        \n",
    "        populations = ['AFR', 'AMR', 'AMR1', 'AMR2', 'EAS', 'SAS', 'BEB', 'GIH', 'ITU', 'PJL', 'STU']\n",
    "        \n",
    "        total_updated = 0\n",
    "        total_skipped = 0\n",
    "        \n",
    "        for test_folder, test_type in test_types.items():\n",
    "            test_path = os.path.join(folder_path, test_folder)\n",
    "            if not os.path.exists(test_path):\n",
    "                continue\n",
    "                \n",
    "            for pop in populations:\n",
    "                # Find matching file\n",
    "                files = list(Path(test_path).glob(f\"{pop}_*.tsv\"))\n",
    "                if not files:\n",
    "                    continue\n",
    "                    \n",
    "                file_path = files[0]\n",
    "                \n",
    "                try:\n",
    "                    df = pd.read_csv(file_path, sep='\\t')\n",
    "                    \n",
    "                    for _, row in df.iterrows():\n",
    "                        rs_id = row['ID']\n",
    "                        \n",
    "                        # Check if variant exists in database\n",
    "                        result = self.cursor.execute(\n",
    "                            \"SELECT variant_id FROM variants WHERE rs_id = ?\",\n",
    "                            (rs_id,)\n",
    "                        ).fetchone()\n",
    "                        \n",
    "                        if result is None:\n",
    "                            total_skipped += 1\n",
    "                            continue\n",
    "                        \n",
    "                        variant_id = result[0]\n",
    "                        \n",
    "                        # Update significance flags\n",
    "                        update_query = f\"\"\"\n",
    "                            UPDATE statistical_tests\n",
    "                            SET is_significant_{test_type} = 1\n",
    "                            WHERE variant_id = ? AND population = ?\n",
    "                        \"\"\"\n",
    "                        self.cursor.execute(update_query, (variant_id, pop))\n",
    "                        \n",
    "                        # Update LD status\n",
    "                        self.cursor.execute(\"\"\"\n",
    "                            INSERT OR REPLACE INTO ld_status (variant_id, population, in_ld, ld_pruned)\n",
    "                            VALUES (?, ?, ?, ?)\n",
    "                        \"\"\", (variant_id, pop, 1 if ld_status == 'after' else 0, ld_status == 'after'))\n",
    "                        \n",
    "                        total_updated += 1\n",
    "                        \n",
    "                except Exception as e:\n",
    "                    print(f\"  Warning: Error processing {file_path.name}: {e}\")\n",
    "                    \n",
    "        self.conn.commit()\n",
    "        print(f\"✓ Loaded significant variants ({ld_status} LD)\")\n",
    "        print(f\"  Updated: {total_updated} variant-population combinations\")\n",
    "        if total_skipped > 0:\n",
    "            print(f\"  Skipped: {total_skipped} variants not found in main dataset\")\n",
    "        \n",
    "    def close(self):\n",
    "        \"\"\"Close database connection\"\"\"\n",
    "        if self.conn:\n",
    "            self.conn.close()\n",
    "            print(\"\\n✓ Database connection closed\")\n",
    "\n",
    "\n",
    "# Main execution function\n",
    "def main():\n",
    "    print(\"=\" * 60)\n",
    "    print(\"Nephrotic Syndrome Genetics Database Setup - UPDATED\")\n",
    "    print(\"=\" * 60)\n",
    "    \n",
    "    # Initialize database\n",
    "    db = NephroticSyndromeDB('nephrotic_syndrome.db')\n",
    "    db.connect()\n",
    "    \n",
    "    # Create schema\n",
    "    db.create_schema()\n",
    "    \n",
    "    # Load data (Update these paths to your actual file locations)\n",
    "    print(\"\\n\" + \"=\" * 60)\n",
    "    print(\"LOADING DATA\")\n",
    "    print(\"=\" * 60)\n",
    "    \n",
    "    # 1. Load gene information\n",
    "    db.load_gene_data('Final_gene_list.xlsx')\n",
    "    \n",
    "    # 2. Load variant data\n",
    "    db.load_variant_data('Final_with_dbsnp.tsv')\n",
    "    \n",
    "    # 3. Load significant variants (Before LD)\n",
    "    db.load_significant_variants('Before_LD', ld_status='before')\n",
    "    \n",
    "    # 4. Load significant variants (After LD)\n",
    "    db.load_significant_variants('After_LD', ld_status='after')\n",
    "    \n",
    "    # Close connection\n",
    "    db.close()\n",
    "    \n",
    "    print(\"\\n\" + \"=\" * 60)\n",
    "    print(\"DATABASE SETUP COMPLETE!\")\n",
    "    print(\"=\" * 60)\n",
    "    print(f\"\\nYour database 'nephrotic_syndrome.db' is ready to use!\")\n",
    "    print(\"\\nNext steps:\")\n",
    "    print(\"1. Test queries with the query examples script\")\n",
    "    print(\"2. Set up your web dashboard using Flask/Django\")\n",
    "    print(\"3. Deploy your application\")\n",
    "\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b69bdca0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "======================================================================\n",
      "NEPHROTIC SYNDROME GENETICS DATABASE - CORRECTED VERSION\n",
      "======================================================================\n",
      "\n",
      "📏 Using Significance Criteria:\n",
      "  • Chi-square: p ≤ 0.05 AND AF ≥ 0.05 (in EITHER population)\n",
      "  • Bonferroni: p ≤ 0.01\n",
      "  • FST: FST ≥ 0.15\n",
      "✓ Connected to database: nephrotic_syndrome.db\n",
      "\n",
      "Creating database schema...\n",
      "✓ Database schema created successfully\n",
      "\n",
      "======================================================================\n",
      "LOADING GENE DATA\n",
      "======================================================================\n",
      "Reading from: Final_gene_list.xlsx\n",
      "Found 494 genes in Excel file\n",
      "\n",
      "Columns in Excel: ['Gene', 'Ensembl_ID', 'chromosome', 'start', 'end', 'SRNS/SSNS', 'Phenotypes', 'Inheritance', 'Pediatric (Y/N)', 'Adult (Y/N)', 'Both (Y/N)', 'OMIM_Link', 'GWAS_Link', 'ClinPGx_Link', 'MalaCards_Link', 'Function', 'Pathway', 'Source Paper 1', 'Source Paper 2', 'Source Paper 3', 'Category', 'Variation']\n",
      "✓ Loaded 493 genes\n",
      "✓ Loaded 493 gene link records (including function & pathway links)\n",
      "\n",
      "======================================================================\n",
      "LOADING VARIANT DATA\n",
      "======================================================================\n",
      "Reading from: Final_with_dbsnp.tsv\n",
      "\n",
      "This may take several minutes for large files...\n",
      "\n",
      "📊 Significance Criteria:\n",
      "  • Chi-square: p ≤ 0.05 AND AF ≥ 0.05 (in EITHER population)\n",
      "  • Bonferroni: p ≤ 0.01\n",
      "  • FST: FST ≥ 0.15\n",
      "\n",
      "  Processed 100,000 variants...\n",
      "  Processed 200,000 variants...\n",
      "  Processed 300,000 variants...\n",
      "  Processed 400,000 variants...\n",
      "  Processed 500,000 variants...\n",
      "  Processed 600,000 variants...\n",
      "  Processed 700,000 variants...\n",
      "  Processed 800,000 variants...\n",
      "  Processed 900,000 variants...\n",
      "  Processed 1,000,000 variants...\n",
      "\n",
      "✓ Loaded 1,048,575 variants with statistical tests\n",
      "\n",
      "======================================================================\n",
      "MARKING LD-PRUNED VARIANTS\n",
      "======================================================================\n",
      "Scanning directory: After_LD/\n",
      "\n",
      "Scanning all TSV files...\n",
      "  Processed 10 files, 5,112 unique variants found...\n",
      "  Processed 20 files, 49,012 unique variants found...\n",
      "  Processed 30 files, 49,085 unique variants found...\n",
      "  Processed 40 files, 59,964 unique variants found...\n",
      "\n",
      "📁 Files processed: 44\n",
      "🧬 Unique variants found (PASSED LD): 60,338\n",
      "\n",
      "Updating database...\n",
      "  Updated 5,000 / 60,338 variants...\n",
      "  Updated 10,000 / 60,338 variants...\n",
      "  Updated 15,000 / 60,338 variants...\n",
      "  Updated 20,000 / 60,338 variants...\n",
      "  Updated 25,000 / 60,338 variants...\n",
      "  Updated 30,000 / 60,338 variants...\n",
      "  Updated 35,000 / 60,338 variants...\n",
      "  Updated 40,000 / 60,338 variants...\n",
      "  Updated 45,000 / 60,338 variants...\n",
      "  Updated 50,000 / 60,338 variants...\n",
      "  Updated 55,000 / 60,338 variants...\n",
      "  Updated 60,000 / 60,338 variants...\n",
      "\n",
      "📊 LD Pruning Statistics:\n",
      "  • Variants KEPT (passed LD): 59,710\n",
      "  • Variants REMOVED (pruned): 816,365\n",
      "  • Total variants: 876,075\n",
      "  • Retention rate: 6.82%\n",
      "\n",
      "======================================================================\n",
      "UPDATING GENE POSITIONS\n",
      "======================================================================\n",
      "✓ Updated positions for 1 genes from variant data\n",
      "\n",
      "======================================================================\n",
      "DATABASE STATISTICS\n",
      "======================================================================\n",
      "\n",
      "📚 Total Genes: 491\n",
      "🧬 Total Variants: 876,075\n",
      "✨ Significant Variants (Bonferroni): 770,395 (87.94%)\n",
      "\n",
      "🔗 LD Pruning:\n",
      "  • Kept: 59,710\n",
      "  • Pruned: 816,365\n",
      "\n",
      "🌍 Top Populations with Significant Variants (Bonferroni):\n",
      "  • AFR: 440,710\n",
      "  • EAS: 286,195\n",
      "  • AMR: 239,757\n",
      "  • AMR1: 208,739\n",
      "  • BEB: 205,331\n",
      "  • STU: 199,009\n",
      "  • SAS: 199,009\n",
      "  • ITU: 194,745\n",
      "  • PJL: 193,755\n",
      "  • AMR2: 192,248\n",
      "\n",
      "======================================================================\n",
      "\n",
      "✓ Database connection closed\n",
      "\n",
      "======================================================================\n",
      "✅ DATABASE SETUP COMPLETE!\n",
      "======================================================================\n",
      "\n",
      "Database file: nephrotic_syndrome.db\n",
      "\n",
      "Next steps:\n",
      "1. Start Flask API: python app.py\n",
      "2. Open dashboard in browser\n",
      "3. Test queries and explore your data\n"
     ]
    }
   ],
   "source": [
    "\"\"\"\n",
    "Nephrotic Syndrome Genetic Diversity Database Setup - CORRECTED VERSION\n",
    "Fixed to match actual Excel column names - Function and Pathway are LINKS\n",
    "\"\"\"\n",
    "\n",
    "import sqlite3\n",
    "import pandas as pd\n",
    "import os\n",
    "from pathlib import Path\n",
    "\n",
    "# ==================== SIGNIFICANCE CRITERIA ====================\n",
    "CHISQ_P_THRESHOLD = 0.05        # Chi-square p-value threshold (p <= 0.05)\n",
    "CHISQ_AF_THRESHOLD = 0.05       # Minimum allele frequency (AF >= 0.05 in EITHER population)\n",
    "BONFERRONI_P_THRESHOLD = 0.01   # Bonferroni corrected p-value threshold (p <= 0.01)\n",
    "FST_THRESHOLD = 0.15            # FST threshold for significant differentiation (FST >= 0.15)\n",
    "\n",
    "# ==================== DATABASE SCHEMA ====================\n",
    "DATABASE_SCHEMA = \"\"\"\n",
    "-- Table 1: Genes - Core gene information\n",
    "CREATE TABLE IF NOT EXISTS genes (\n",
    "    gene_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    gene_name TEXT UNIQUE NOT NULL,\n",
    "    ensembl_id TEXT,\n",
    "    chromosome TEXT,\n",
    "    start_position INTEGER,\n",
    "    end_position INTEGER,\n",
    "    srns_ssns TEXT,\n",
    "    phenotypes TEXT,\n",
    "    inheritance TEXT,\n",
    "    pediatric TEXT,\n",
    "    adult TEXT,\n",
    "    both TEXT,\n",
    "    category TEXT,\n",
    "    variation TEXT,\n",
    "    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP\n",
    ");\n",
    "\n",
    "-- Table 2: Gene Links - All database and literature links (including Function and Pathway links)\n",
    "CREATE TABLE IF NOT EXISTS gene_links (\n",
    "    link_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    gene_name TEXT NOT NULL,\n",
    "    omim_link TEXT,\n",
    "    gwas_link TEXT,\n",
    "    clinpgx_link TEXT,\n",
    "    malacards_link TEXT,\n",
    "    function_link TEXT,\n",
    "    pathway_link TEXT,\n",
    "    source_paper_1 TEXT,\n",
    "    source_paper_2 TEXT,\n",
    "    source_paper_3 TEXT,\n",
    "    FOREIGN KEY (gene_name) REFERENCES genes(gene_name),\n",
    "    UNIQUE(gene_name)\n",
    ");\n",
    "\n",
    "-- Table 3: Variants - All variant information with dbSNP links\n",
    "CREATE TABLE IF NOT EXISTS variants (\n",
    "    variant_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    rs_id TEXT UNIQUE NOT NULL,\n",
    "    chrom TEXT NOT NULL,\n",
    "    position INTEGER NOT NULL,\n",
    "    ref_allele TEXT NOT NULL,\n",
    "    alt_allele TEXT NOT NULL,\n",
    "    gene_name TEXT NOT NULL,\n",
    "    global_af REAL,\n",
    "    dbsnp_link TEXT,\n",
    "    FOREIGN KEY (gene_name) REFERENCES genes(gene_name),\n",
    "    UNIQUE(chrom, position, ref_allele, alt_allele)\n",
    ");\n",
    "\n",
    "-- Table 4: Allele Frequencies - Population-specific frequencies\n",
    "CREATE TABLE IF NOT EXISTS allele_frequencies (\n",
    "    af_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    variant_id INTEGER NOT NULL,\n",
    "    population TEXT NOT NULL,\n",
    "    allele_frequency REAL NOT NULL,\n",
    "    FOREIGN KEY (variant_id) REFERENCES variants(variant_id),\n",
    "    UNIQUE(variant_id, population)\n",
    ");\n",
    "\n",
    "-- Table 5: Statistical Tests - FST and Chi-square results with significance flags\n",
    "CREATE TABLE IF NOT EXISTS statistical_tests (\n",
    "    test_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    variant_id INTEGER NOT NULL,\n",
    "    population TEXT NOT NULL,\n",
    "    fst_value REAL,\n",
    "    chi_square_p_value REAL,\n",
    "    is_significant_chisq BOOLEAN DEFAULT 0,\n",
    "    is_significant_bonferroni BOOLEAN DEFAULT 0,\n",
    "    is_significant_fst BOOLEAN DEFAULT 0,\n",
    "    FOREIGN KEY (variant_id) REFERENCES variants(variant_id),\n",
    "    UNIQUE(variant_id, population)\n",
    ");\n",
    "\n",
    "-- Table 6: LD Status - Linkage Disequilibrium information\n",
    "CREATE TABLE IF NOT EXISTS ld_status (\n",
    "    ld_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    variant_id INTEGER NOT NULL,\n",
    "    population TEXT DEFAULT 'ALL',\n",
    "    in_ld BOOLEAN DEFAULT 0,\n",
    "    ld_pruned BOOLEAN DEFAULT 0,\n",
    "    FOREIGN KEY (variant_id) REFERENCES variants(variant_id),\n",
    "    UNIQUE(variant_id, population)\n",
    ");\n",
    "\n",
    "-- Indexes for faster queries\n",
    "CREATE INDEX IF NOT EXISTS idx_variants_gene ON variants(gene_name);\n",
    "CREATE INDEX IF NOT EXISTS idx_variants_rs ON variants(rs_id);\n",
    "CREATE INDEX IF NOT EXISTS idx_variants_chrom_pos ON variants(chrom, position);\n",
    "CREATE INDEX IF NOT EXISTS idx_af_variant ON allele_frequencies(variant_id);\n",
    "CREATE INDEX IF NOT EXISTS idx_af_population ON allele_frequencies(population);\n",
    "CREATE INDEX IF NOT EXISTS idx_stats_variant ON statistical_tests(variant_id);\n",
    "CREATE INDEX IF NOT EXISTS idx_stats_significant ON statistical_tests(is_significant_bonferroni);\n",
    "CREATE INDEX IF NOT EXISTS idx_ld_variant ON ld_status(variant_id);\n",
    "CREATE INDEX IF NOT EXISTS idx_ld_pruned ON ld_status(ld_pruned);\n",
    "CREATE INDEX IF NOT EXISTS idx_gene_links ON gene_links(gene_name);\n",
    "CREATE INDEX IF NOT EXISTS idx_genes_chrom ON genes(chromosome);\n",
    "\"\"\"\n",
    "\n",
    "\n",
    "class NephroticSyndromeDB:\n",
    "    def __init__(self, db_path='nephrotic_syndrome.db'):\n",
    "        \"\"\"Initialize database connection\"\"\"\n",
    "        self.db_path = db_path\n",
    "        self.conn = None\n",
    "        self.cursor = None\n",
    "        \n",
    "    def connect(self):\n",
    "        \"\"\"Connect to SQLite database\"\"\"\n",
    "        self.conn = sqlite3.connect(self.db_path)\n",
    "        self.cursor = self.conn.cursor()\n",
    "        print(f\"✓ Connected to database: {self.db_path}\")\n",
    "        \n",
    "    def create_schema(self):\n",
    "        \"\"\"Create database schema\"\"\"\n",
    "        print(\"\\nCreating database schema...\")\n",
    "        self.cursor.executescript(DATABASE_SCHEMA)\n",
    "        self.conn.commit()\n",
    "        print(\"✓ Database schema created successfully\")\n",
    "        \n",
    "    def load_gene_data(self, excel_path):\n",
    "        \"\"\"Load gene information from Excel file\"\"\"\n",
    "        print(\"\\n\" + \"=\"*70)\n",
    "        print(\"LOADING GENE DATA\")\n",
    "        print(\"=\"*70)\n",
    "        print(f\"Reading from: {excel_path}\")\n",
    "        \n",
    "        df = pd.read_excel(excel_path)\n",
    "        \n",
    "        # Clean column names (remove extra spaces)\n",
    "        df.columns = df.columns.str.strip()\n",
    "        \n",
    "        print(f\"Found {len(df)} genes in Excel file\")\n",
    "        print(f\"\\nColumns in Excel: {list(df.columns)}\")\n",
    "        \n",
    "        gene_count = 0\n",
    "        link_count = 0\n",
    "        \n",
    "        for idx, row in df.iterrows():\n",
    "            try:\n",
    "                gene_name = row.get('Gene')\n",
    "                \n",
    "                if pd.isna(gene_name):\n",
    "                    continue\n",
    "                \n",
    "                # Insert gene information (removed gene_function and pathway as they are links)\n",
    "                self.cursor.execute(\"\"\"\n",
    "                    INSERT OR IGNORE INTO genes (\n",
    "                        gene_name, ensembl_id, chromosome, start_position, end_position,\n",
    "                        srns_ssns, phenotypes, inheritance,\n",
    "                        pediatric, adult, both, category, variation\n",
    "                    ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)\n",
    "                \"\"\", (\n",
    "                    gene_name,\n",
    "                    row.get('Ensembl_ID'),\n",
    "                    row.get('chromosome'),\n",
    "                    row.get('start'),\n",
    "                    row.get('end'),\n",
    "                    row.get('SRNS/SSNS'),\n",
    "                    row.get('Phenotypes'),\n",
    "                    row.get('Inheritance'),\n",
    "                    row.get('Pediatric (Y/N)'),\n",
    "                    row.get('Adult (Y/N)'),\n",
    "                    row.get('Both (Y/N)'),\n",
    "                    row.get('Category'),\n",
    "                    row.get('Variation')\n",
    "                ))\n",
    "                \n",
    "                gene_count += 1\n",
    "                \n",
    "                # Insert gene links - Function and Pathway are LINKS, not text\n",
    "                self.cursor.execute(\"\"\"\n",
    "                    INSERT OR IGNORE INTO gene_links (\n",
    "                        gene_name, omim_link, gwas_link, clinpgx_link, malacards_link,\n",
    "                        function_link, pathway_link,\n",
    "                        source_paper_1, source_paper_2, source_paper_3\n",
    "                    ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)\n",
    "                \"\"\", (\n",
    "                    gene_name,\n",
    "                    row.get('OMIM_Link'),\n",
    "                    row.get('GWAS_Link'),\n",
    "                    row.get('ClinPGx_Link'),\n",
    "                    row.get('MalaCards_Link'),\n",
    "                    row.get('Function'),      # This is a LINK\n",
    "                    row.get('Pathway'),       # This is a LINK\n",
    "                    row.get('Source Paper 1'),\n",
    "                    row.get('Source Paper 2'),\n",
    "                    row.get('Source Paper 3')\n",
    "                ))\n",
    "                \n",
    "                link_count += 1\n",
    "                \n",
    "            except Exception as e:\n",
    "                print(f\"  ⚠️  Error inserting gene {row.get('Gene')}: {e}\")\n",
    "                \n",
    "        self.conn.commit()\n",
    "        print(f\"✓ Loaded {gene_count} genes\")\n",
    "        print(f\"✓ Loaded {link_count} gene link records (including function & pathway links)\")\n",
    "        \n",
    "    def load_variant_data(self, tsv_path):\n",
    "        \"\"\"Load variant and statistical test data from main TSV file\"\"\"\n",
    "        print(\"\\n\" + \"=\"*70)\n",
    "        print(\"LOADING VARIANT DATA\")\n",
    "        print(\"=\"*70)\n",
    "        print(f\"Reading from: {tsv_path}\")\n",
    "        print(\"\\nThis may take several minutes for large files...\")\n",
    "        \n",
    "        # Define populations\n",
    "        populations = ['AFR', 'AMR', 'AMR1', 'AMR2', 'EAS', 'SAS', 'BEB', 'GIH', 'ITU', 'PJL', 'STU']\n",
    "        \n",
    "        # Read in chunks to handle large files\n",
    "        chunk_size = 10000\n",
    "        total_variants = 0\n",
    "        chunks_processed = 0\n",
    "        \n",
    "        print(f\"\\n📊 Significance Criteria:\")\n",
    "        print(f\"  • Chi-square: p ≤ {CHISQ_P_THRESHOLD} AND AF ≥ {CHISQ_AF_THRESHOLD} (in EITHER population)\")\n",
    "        print(f\"  • Bonferroni: p ≤ {BONFERRONI_P_THRESHOLD}\")\n",
    "        print(f\"  • FST: FST ≥ {FST_THRESHOLD}\\n\")\n",
    "        \n",
    "        for chunk in pd.read_csv(tsv_path, sep='\\t', chunksize=chunk_size, low_memory=False):\n",
    "            for idx, row in chunk.iterrows():\n",
    "                try:\n",
    "                    # Create dbSNP link\n",
    "                    dbsnp_link = row.get('db_SNP_link')\n",
    "                    if pd.isna(dbsnp_link) or not dbsnp_link:\n",
    "                        dbsnp_link = f\"https://www.ncbi.nlm.nih.gov/snp/{row['ID']}\"\n",
    "                    \n",
    "                    # Insert variant with dbSNP link\n",
    "                    self.cursor.execute(\"\"\"\n",
    "                        INSERT OR IGNORE INTO variants (\n",
    "                            rs_id, chrom, position, ref_allele, alt_allele, \n",
    "                            gene_name, global_af, dbsnp_link\n",
    "                        ) VALUES (?, ?, ?, ?, ?, ?, ?, ?)\n",
    "                    \"\"\", (\n",
    "                        row['ID'],\n",
    "                        row['CHROM'],\n",
    "                        row['POS'],\n",
    "                        row['REF'],\n",
    "                        row['ALT'],\n",
    "                        row['GENE'],\n",
    "                        row.get('AF', 0),\n",
    "                        dbsnp_link\n",
    "                    ))\n",
    "                    \n",
    "                    # Get variant_id\n",
    "                    result = self.cursor.execute(\n",
    "                        \"SELECT variant_id FROM variants WHERE rs_id = ?\",\n",
    "                        (row['ID'],)\n",
    "                    ).fetchone()\n",
    "                    \n",
    "                    if not result:\n",
    "                        continue\n",
    "                    \n",
    "                    variant_id = result[0]\n",
    "                    \n",
    "                    # Get EUR allele frequency for significance calculations\n",
    "                    af_eur = row.get('AF_EUR', 0)\n",
    "                    \n",
    "                    # Insert allele frequencies for EUR\n",
    "                    self.cursor.execute(\"\"\"\n",
    "                        INSERT OR IGNORE INTO allele_frequencies (variant_id, population, allele_frequency)\n",
    "                        VALUES (?, ?, ?)\n",
    "                    \"\"\", (variant_id, 'EUR', af_eur))\n",
    "                    \n",
    "                    # Insert allele frequencies for all other populations\n",
    "                    for pop in populations:\n",
    "                        af_col = f'AF_{pop}'\n",
    "                        if af_col in row.index:\n",
    "                            af_pop = row[af_col]\n",
    "                            if not pd.isna(af_pop):\n",
    "                                self.cursor.execute(\"\"\"\n",
    "                                    INSERT OR IGNORE INTO allele_frequencies (variant_id, population, allele_frequency)\n",
    "                                    VALUES (?, ?, ?)\n",
    "                                \"\"\", (variant_id, pop, af_pop))\n",
    "                    \n",
    "                    # Insert statistical tests with significance flags\n",
    "                    for pop in populations:\n",
    "                        fst_col = f'FST_EUR_VS_{pop}'\n",
    "                        p_col = f'P_EUR_{pop}'\n",
    "                        af_pop_col = f'AF_{pop}'\n",
    "                        \n",
    "                        if fst_col in row.index and p_col in row.index:\n",
    "                            fst_value = row[fst_col]\n",
    "                            p_value = row[p_col]\n",
    "                            af_pop = row.get(af_pop_col, 0)\n",
    "                            \n",
    "                            # Skip if values are NaN\n",
    "                            if pd.isna(fst_value) or pd.isna(p_value):\n",
    "                                continue\n",
    "                            \n",
    "                            # Calculate significance flags\n",
    "                            af_sufficient = (af_eur >= CHISQ_AF_THRESHOLD) or (af_pop >= CHISQ_AF_THRESHOLD)\n",
    "                            is_sig_chisq = (p_value <= CHISQ_P_THRESHOLD) and af_sufficient\n",
    "                            is_sig_bonf = p_value <= BONFERRONI_P_THRESHOLD\n",
    "                            is_sig_fst = fst_value >= FST_THRESHOLD\n",
    "                            \n",
    "                            self.cursor.execute(\"\"\"\n",
    "                                INSERT OR IGNORE INTO statistical_tests (\n",
    "                                    variant_id, population, fst_value, chi_square_p_value,\n",
    "                                    is_significant_chisq, is_significant_bonferroni, is_significant_fst\n",
    "                                ) VALUES (?, ?, ?, ?, ?, ?, ?)\n",
    "                            \"\"\", (\n",
    "                                variant_id, pop, fst_value, p_value,\n",
    "                                1 if is_sig_chisq else 0,\n",
    "                                1 if is_sig_bonf else 0,\n",
    "                                1 if is_sig_fst else 0\n",
    "                            ))\n",
    "                    \n",
    "                    # Initialize LD status as NOT pruned\n",
    "                    self.cursor.execute(\"\"\"\n",
    "                        INSERT OR IGNORE INTO ld_status (variant_id, population, in_ld, ld_pruned)\n",
    "                        VALUES (?, ?, ?, ?)\n",
    "                    \"\"\", (variant_id, 'ALL', 0, 0))\n",
    "                    \n",
    "                    total_variants += 1\n",
    "                    \n",
    "                except Exception as e:\n",
    "                    print(f\"  ⚠️  Error processing variant {row.get('ID')}: {e}\")\n",
    "                    continue\n",
    "            \n",
    "            chunks_processed += 1\n",
    "            \n",
    "            # Commit every chunk\n",
    "            self.conn.commit()\n",
    "            \n",
    "            if chunks_processed % 10 == 0:\n",
    "                print(f\"  Processed {total_variants:,} variants...\")\n",
    "        \n",
    "        print(f\"\\n✓ Loaded {total_variants:,} variants with statistical tests\")\n",
    "        \n",
    "    def load_ld_pruned_variants(self, after_ld_dir):\n",
    "        \"\"\"Mark variants that PASSED LD pruning from After_LD folder\"\"\"\n",
    "        print(\"\\n\" + \"=\"*70)\n",
    "        print(\"MARKING LD-PRUNED VARIANTS\")\n",
    "        print(\"=\"*70)\n",
    "        print(f\"Scanning directory: {after_ld_dir}\")\n",
    "        \n",
    "        after_path = Path(after_ld_dir)\n",
    "        \n",
    "        if not after_path.exists():\n",
    "            print(f\"⚠️  Warning: After_LD directory not found at {after_path}\")\n",
    "            print(\"All variants will be marked as NOT LD-pruned\")\n",
    "            return\n",
    "        \n",
    "        ld_passed_variants = set()\n",
    "        files_processed = 0\n",
    "        \n",
    "        print(\"\\nScanning all TSV files...\")\n",
    "        \n",
    "        for file in after_path.rglob('*.tsv'):\n",
    "            try:\n",
    "                df = pd.read_csv(file, sep='\\t', low_memory=False)\n",
    "                \n",
    "                if 'ID' in df.columns:\n",
    "                    variant_ids = df['ID'].unique()\n",
    "                    ld_passed_variants.update(variant_ids)\n",
    "                    files_processed += 1\n",
    "                    \n",
    "                    if files_processed % 10 == 0:\n",
    "                        print(f\"  Processed {files_processed} files, {len(ld_passed_variants):,} unique variants found...\")\n",
    "                        \n",
    "            except Exception as e:\n",
    "                print(f\"  ⚠️  Error reading {file.name}: {e}\")\n",
    "        \n",
    "        print(f\"\\n📁 Files processed: {files_processed}\")\n",
    "        print(f\"🧬 Unique variants found (PASSED LD): {len(ld_passed_variants):,}\")\n",
    "        \n",
    "        if len(ld_passed_variants) == 0:\n",
    "            print(\"⚠️  Warning: No variants found in After_LD files!\")\n",
    "            return\n",
    "        \n",
    "        print(\"\\nUpdating database...\")\n",
    "        \n",
    "        # First, ensure all variants have ld_pruned = 0\n",
    "        self.cursor.execute(\"UPDATE ld_status SET ld_pruned = 0, in_ld = 0\")\n",
    "        \n",
    "        # Mark the ones that passed as ld_pruned = 1\n",
    "        updated_count = 0\n",
    "        batch_size = 1000\n",
    "        \n",
    "        ld_variants_list = list(ld_passed_variants)\n",
    "        for i in range(0, len(ld_variants_list), batch_size):\n",
    "            batch = ld_variants_list[i:i+batch_size]\n",
    "            \n",
    "            placeholders = ','.join(['?'] * len(batch))\n",
    "            self.cursor.execute(f'''\n",
    "                UPDATE ld_status \n",
    "                SET ld_pruned = 1, in_ld = 1\n",
    "                WHERE variant_id IN (\n",
    "                    SELECT variant_id FROM variants WHERE rs_id IN ({placeholders})\n",
    "                )\n",
    "            ''', batch)\n",
    "            \n",
    "            updated_count += len(batch)\n",
    "            \n",
    "            if updated_count % 5000 == 0:\n",
    "                print(f\"  Updated {updated_count:,} / {len(ld_passed_variants):,} variants...\")\n",
    "                self.conn.commit()\n",
    "        \n",
    "        self.conn.commit()\n",
    "        \n",
    "        # Show statistics\n",
    "        self.cursor.execute('''\n",
    "            SELECT \n",
    "                SUM(CASE WHEN ld_pruned = 1 THEN 1 ELSE 0 END) as kept,\n",
    "                SUM(CASE WHEN ld_pruned = 0 THEN 1 ELSE 0 END) as removed,\n",
    "                COUNT(*) as total\n",
    "            FROM ld_status\n",
    "        ''')\n",
    "        stats = self.cursor.fetchone()\n",
    "        \n",
    "        print(f\"\\n📊 LD Pruning Statistics:\")\n",
    "        print(f\"  • Variants KEPT (passed LD): {stats[0]:,}\")\n",
    "        print(f\"  • Variants REMOVED (pruned): {stats[1]:,}\")\n",
    "        print(f\"  • Total variants: {stats[2]:,}\")\n",
    "        \n",
    "        if stats[2] > 0:\n",
    "            retention_rate = (stats[0]/stats[2]*100)\n",
    "            print(f\"  • Retention rate: {retention_rate:.2f}%\")\n",
    "    \n",
    "    def update_gene_positions(self):\n",
    "        \"\"\"Update gene chromosome and position info from variants (if not in Excel)\"\"\"\n",
    "        print(\"\\n\" + \"=\"*70)\n",
    "        print(\"UPDATING GENE POSITIONS\")\n",
    "        print(\"=\"*70)\n",
    "        \n",
    "        # Only update genes that don't have positions from Excel\n",
    "        self.cursor.execute('''\n",
    "            UPDATE genes\n",
    "            SET chromosome = (\n",
    "                SELECT DISTINCT chrom\n",
    "                FROM variants\n",
    "                WHERE variants.gene_name = genes.gene_name\n",
    "                LIMIT 1\n",
    "            ),\n",
    "            start_position = (\n",
    "                SELECT MIN(position)\n",
    "                FROM variants\n",
    "                WHERE variants.gene_name = genes.gene_name\n",
    "            ),\n",
    "            end_position = (\n",
    "                SELECT MAX(position)\n",
    "                FROM variants\n",
    "                WHERE variants.gene_name = genes.gene_name\n",
    "            )\n",
    "            WHERE (chromosome IS NULL OR start_position IS NULL OR end_position IS NULL)\n",
    "            AND EXISTS (\n",
    "                SELECT 1 FROM variants WHERE variants.gene_name = genes.gene_name\n",
    "            )\n",
    "        ''')\n",
    "        \n",
    "        updated = self.cursor.rowcount\n",
    "        self.conn.commit()\n",
    "        print(f\"✓ Updated positions for {updated} genes from variant data\")\n",
    "    \n",
    "    def show_statistics(self):\n",
    "        \"\"\"Display database statistics\"\"\"\n",
    "        print(\"\\n\" + \"=\"*70)\n",
    "        print(\"DATABASE STATISTICS\")\n",
    "        print(\"=\"*70)\n",
    "        \n",
    "        # Genes\n",
    "        self.cursor.execute(\"SELECT COUNT(*) FROM genes\")\n",
    "        total_genes = self.cursor.fetchone()[0]\n",
    "        print(f\"\\n📚 Total Genes: {total_genes:,}\")\n",
    "        \n",
    "        # Variants\n",
    "        self.cursor.execute(\"SELECT COUNT(*) FROM variants\")\n",
    "        total_variants = self.cursor.fetchone()[0]\n",
    "        print(f\"🧬 Total Variants: {total_variants:,}\")\n",
    "        \n",
    "        # Significant variants (Bonferroni)\n",
    "        self.cursor.execute('''\n",
    "            SELECT COUNT(DISTINCT variant_id)\n",
    "            FROM statistical_tests\n",
    "            WHERE is_significant_bonferroni = 1\n",
    "        ''')\n",
    "        sig_variants = self.cursor.fetchone()[0]\n",
    "        if total_variants > 0:\n",
    "            print(f\"✨ Significant Variants (Bonferroni): {sig_variants:,} ({sig_variants/total_variants*100:.2f}%)\")\n",
    "        \n",
    "        # LD statistics\n",
    "        self.cursor.execute('''\n",
    "            SELECT \n",
    "                SUM(CASE WHEN ld_pruned = 1 THEN 1 ELSE 0 END) as kept,\n",
    "                SUM(CASE WHEN ld_pruned = 0 THEN 1 ELSE 0 END) as pruned\n",
    "            FROM ld_status\n",
    "        ''')\n",
    "        ld_stats = self.cursor.fetchone()\n",
    "        if ld_stats:\n",
    "            print(f\"\\n🔗 LD Pruning:\")\n",
    "            print(f\"  • Kept: {ld_stats[0]:,}\")\n",
    "            print(f\"  • Pruned: {ld_stats[1]:,}\")\n",
    "        \n",
    "        # Significance by population\n",
    "        self.cursor.execute('''\n",
    "            SELECT population, COUNT(DISTINCT variant_id) as count\n",
    "            FROM statistical_tests\n",
    "            WHERE is_significant_bonferroni = 1\n",
    "            GROUP BY population\n",
    "            ORDER BY count DESC\n",
    "            LIMIT 10\n",
    "        ''')\n",
    "        \n",
    "        print(f\"\\n🌍 Top Populations with Significant Variants (Bonferroni):\")\n",
    "        for row in self.cursor.fetchall():\n",
    "            print(f\"  • {row[0]}: {row[1]:,}\")\n",
    "        \n",
    "        print(\"\\n\" + \"=\"*70)\n",
    "    \n",
    "    def close(self):\n",
    "        \"\"\"Close database connection\"\"\"\n",
    "        if self.conn:\n",
    "            self.conn.close()\n",
    "            print(\"\\n✓ Database connection closed\")\n",
    "\n",
    "\n",
    "def main():\n",
    "    print(\"=\"*70)\n",
    "    print(\"NEPHROTIC SYNDROME GENETICS DATABASE - CORRECTED VERSION\")\n",
    "    print(\"=\"*70)\n",
    "    print(\"\\n📏 Using Significance Criteria:\")\n",
    "    print(f\"  • Chi-square: p ≤ {CHISQ_P_THRESHOLD} AND AF ≥ {CHISQ_AF_THRESHOLD} (in EITHER population)\")\n",
    "    print(f\"  • Bonferroni: p ≤ {BONFERRONI_P_THRESHOLD}\")\n",
    "    print(f\"  • FST: FST ≥ {FST_THRESHOLD}\")\n",
    "    \n",
    "    # Initialize database\n",
    "    db = NephroticSyndromeDB('nephrotic_syndrome.db')\n",
    "    db.connect()\n",
    "    \n",
    "    # Create schema\n",
    "    db.create_schema()\n",
    "    \n",
    "    # ==================== LOAD DATA ====================\n",
    "    # Update these paths to match your file locations\n",
    "    \n",
    "    # 1. Load gene information from Excel\n",
    "    db.load_gene_data('Final_gene_list.xlsx')\n",
    "    \n",
    "    # 2. Load variant data from main TSV (with dbSNP links)\n",
    "    db.load_variant_data('Final_with_dbsnp.tsv')\n",
    "    \n",
    "    # 3. Mark LD-pruned variants from After_LD folder\n",
    "    db.load_ld_pruned_variants('After_LD/')\n",
    "    \n",
    "    # 4. Update gene positions from variant data (only if missing)\n",
    "    db.update_gene_positions()\n",
    "    \n",
    "    # 5. Show statistics\n",
    "    db.show_statistics()\n",
    "    \n",
    "    # Close connection\n",
    "    db.close()\n",
    "    \n",
    "    print(\"\\n\" + \"=\"*70)\n",
    "    print(\"✅ DATABASE SETUP COMPLETE!\")\n",
    "    print(\"=\"*70)\n",
    "    print(f\"\\nDatabase file: nephrotic_syndrome.db\")\n",
    "    print(\"\\nNext steps:\")\n",
    "    print(\"1. Start Flask API: python app.py\")\n",
    "    print(\"2. Open dashboard in browser\")\n",
    "    print(\"3. Test queries and explore your data\")\n",
    "\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "131e11d3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬\n",
      "MULTI-POPULATION SIGNIFICANT VARIANTS ANALYSIS\n",
      "🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬🧬\n",
      "\n",
      "====================================================================================================\n",
      "VARIANTS SIGNIFICANT IN MULTIPLE POPULATIONS - SUMMARY\n",
      "====================================================================================================\n",
      "Criteria: p ≤ 0.01 AND AF ≥ 0.05 (either pop) AND FST ≥ 0.15\n",
      "====================================================================================================\n",
      "\n",
      "+-------------+-----------+----------------+------------------+-------------------------------------------+-------------+---------+\n",
      "|    rs_id    | gene_name |    position    | population_count |                populations                | avg_p_value | avg_fst |\n",
      "+-------------+-----------+----------------+------------------+-------------------------------------------+-------------+---------+\n",
      "|  rs4709451  |    LPA    | chr6:160633972 |        10        | AMR,AMR1,AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU | 0.00018405  |  0.318  |\n",
      "|  rs4646440  |  CYP3A4   | chr7:99763247  |        9         |   AMR,AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU    |  2.78e-06   | 0.4332  |\n",
      "| rs138749777 |   BRAF    | chr7:140728535 |        9         |   AMR,AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU    |  0.000311   | 0.3407  |\n",
      "|  rs873966   |   SYNPO   | chr5:150613013 |        9         |   AFR,AMR,AMR1,AMR2,BEB,GIH,ITU,PJL,STU   | 0.00289032  |  0.314  |\n",
      "| rs79898317  | ARHGAP24  | chr4:85877037  |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00080038  |  0.372  |\n",
      "| rs34875318  |   RUNX3   | chr1:24907890  |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      |     0.0     | 0.3559  |\n",
      "|  rs9561177  |   GPC5    | chr13:92846123 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3382  |\n",
      "| rs141394110 |   GPC5    | chr13:92845865 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3366  |\n",
      "|  rs9561176  |   GPC5    | chr13:92846103 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3366  |\n",
      "|  rs9556212  |   GPC5    | chr13:92781168 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3365  |\n",
      "|  rs7984465  |   GPC5    | chr13:92838340 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3359  |\n",
      "|  rs8000315  |   GPC5    | chr13:92838587 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3359  |\n",
      "| rs12427983  |   GPC5    | chr13:92795245 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  |  0.335  |\n",
      "|  rs7333121  |   GPC5    | chr13:92787074 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3349  |\n",
      "| rs79651017  |   GPC5    | chr13:92789175 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3349  |\n",
      "|  rs9556213  |   GPC5    | chr13:92788224 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3349  |\n",
      "|  rs9561153  |   GPC5    | chr13:92788922 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3349  |\n",
      "|  rs9556227  |   GPC5    | chr13:92844871 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3347  |\n",
      "|  rs9561151  |   GPC5    | chr13:92784453 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3347  |\n",
      "|  rs9561152  |   GPC5    | chr13:92785460 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3347  |\n",
      "|  rs9561174  |   GPC5    | chr13:92836114 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00015333  | 0.3345  |\n",
      "|  rs2065584  |   GPC5    | chr13:92792192 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00015333  | 0.3344  |\n",
      "|  rs9301834  |   GPC5    | chr13:92797304 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00015333  | 0.3344  |\n",
      "|  rs9561156  |   GPC5    | chr13:92793102 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00015333  | 0.3344  |\n",
      "|  rs9561160  |   GPC5    | chr13:92796387 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00015333  | 0.3344  |\n",
      "|  rs9561175  |   GPC5    | chr13:92842897 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00015333  | 0.3344  |\n",
      "| rs35400222  |   GPC5    | chr13:92830416 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3342  |\n",
      "|  rs9561179  |   GPC5    | chr13:92848760 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00015333  | 0.3342  |\n",
      "|  rs9556215  |   GPC5    | chr13:92799198 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3341  |\n",
      "|  rs9561150  |   GPC5    | chr13:92782701 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00015333  |  0.334  |\n",
      "|  rs9561161  |   GPC5    | chr13:92798338 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00015333  | 0.3337  |\n",
      "|  rs9561163  |   GPC5    | chr13:92800304 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00015333  | 0.3337  |\n",
      "|  rs9561173  |   GPC5    | chr13:92834763 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3337  |\n",
      "| rs12429947  |   GPC5    | chr13:92792429 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00015333  | 0.3335  |\n",
      "|  rs9561155  |   GPC5    | chr13:92789753 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00015333  | 0.3335  |\n",
      "|  rs9561180  |   GPC5    | chr13:92849818 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3334  |\n",
      "| rs137945550 |   GPC5    | chr13:92819274 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3333  |\n",
      "|  rs1890198  |   GPC5    | chr13:92786030 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00015333  | 0.3333  |\n",
      "|  rs9556219  |   GPC5    | chr13:92817373 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3333  |\n",
      "|  rs9561165  |   GPC5    | chr13:92817426 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3333  |\n",
      "|  rs9561166  |   GPC5    | chr13:92821755 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3333  |\n",
      "|  rs9556224  |   GPC5    | chr13:92828239 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00015333  | 0.3331  |\n",
      "|  rs9561182  |   GPC5    | chr13:92850796 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3331  |\n",
      "|  rs9556222  |   GPC5    | chr13:92824750 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00015334  | 0.3329  |\n",
      "|  rs9556223  |   GPC5    | chr13:92824768 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00015334  | 0.3329  |\n",
      "|  rs9556216  |   GPC5    | chr13:92804757 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00015333  | 0.3327  |\n",
      "|  rs9561164  |   GPC5    | chr13:92806686 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3325  |\n",
      "|  rs9561172  |   GPC5    | chr13:92833568 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00015334  | 0.3325  |\n",
      "| rs12428589  |   GPC5    | chr13:92827411 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3324  |\n",
      "|  rs5805764  |   GPC5    | chr13:92807142 |        8         |     AMR2,EAS,SAS,BEB,GIH,ITU,PJL,STU      | 0.00011506  | 0.3324  |\n",
      "+-------------+-----------+----------------+------------------+-------------------------------------------+-------------+---------+\n",
      "\n",
      "✓ Found 50 variants significant in 2+ populations\n",
      "✓ Top variant: rs4709451 in 10 populations\n",
      "\n",
      "====================================================================================================\n",
      "DISTRIBUTION: HOW MANY POPULATIONS ARE VARIANTS SIGNIFICANT IN?\n",
      "====================================================================================================\n",
      "\n",
      "+-----------------------+---------------+------------+\n",
      "| Number_of_Populations | Variant_Count | Percentage |\n",
      "+-----------------------+---------------+------------+\n",
      "|         10.0          |      1.0      |    0.01    |\n",
      "|          9.0          |      3.0      |    0.02    |\n",
      "|          8.0          |     102.0     |    0.66    |\n",
      "|          7.0          |     61.0      |    0.39    |\n",
      "|          6.0          |     124.0     |    0.8     |\n",
      "|          5.0          |     244.0     |    1.58    |\n",
      "|          4.0          |     406.0     |    2.63    |\n",
      "|          3.0          |     652.0     |    4.22    |\n",
      "|          2.0          |    1451.0     |    9.39    |\n",
      "|          1.0          |    12411.0    |    80.3    |\n",
      "+-----------------------+---------------+------------+\n",
      "\n",
      "📊 Summary:\n",
      "  • Total significant variants: 15,455\n",
      "  • Multi-population (2+): 3,044 (19.7%)\n",
      "  • Single population only: 12,411 (80.3%)\n",
      "\n",
      "====================================================================================================\n",
      "GENES WITH MOST MULTI-POPULATION SIGNIFICANT VARIANTS\n",
      "====================================================================================================\n",
      "\n",
      "+-----------+--------------------+----------------------+-----------------+----------------+\n",
      "|   Gene    | Multi_Pop_Variants | Avg_Pops_Per_Variant | Max_Populations | Total_Pop_Hits |\n",
      "+-----------+--------------------+----------------------+-----------------+----------------+\n",
      "|   PTPRD   |        290         |         2.5          |        8        |      724       |\n",
      "|   GPC5    |        189         |         6.23         |        8        |      1177      |\n",
      "|   DLC1    |        145         |         2.8          |        6        |      406       |\n",
      "|   KANK1   |        139         |         2.73         |        6        |      379       |\n",
      "|   MAGI2   |        114         |         2.94         |        6        |      335       |\n",
      "| HLA-DQB1  |        111         |         4.45         |        7        |      494       |\n",
      "|   TRPC6   |        111         |         3.48         |        6        |      386       |\n",
      "|   FBN1    |        107         |         2.0          |        2        |      214       |\n",
      "|  MORF4L1  |         77         |         2.97         |        4        |      229       |\n",
      "|  GAPVD1   |         63         |         2.03         |        3        |      128       |\n",
      "|   IFIH1   |         58         |         2.98         |        5        |      173       |\n",
      "|   CLVS1   |         51         |         4.57         |        8        |      233       |\n",
      "| LINC02531 |         51         |         2.16         |        5        |      110       |\n",
      "|  NUP188   |         48         |         2.0          |        2        |       96       |\n",
      "|    TTR    |         43         |         4.95         |        8        |      213       |\n",
      "|  CREBBP   |         41         |         4.07         |        7        |      167       |\n",
      "|    GCA    |         40         |         2.8          |        5        |      112       |\n",
      "|   KALRN   |         40         |         2.17         |        4        |       87       |\n",
      "|  THSD7A   |         40         |         2.17         |        4        |       87       |\n",
      "|  CCDC66   |         34         |         3.03         |        4        |      103       |\n",
      "+-----------+--------------------+----------------------+-----------------+----------------+\n",
      "\n",
      "✓ 20 genes have variants significant in 2+ populations\n",
      "\n",
      "====================================================================================================\n",
      "DETAILED ANALYSIS FOR rs4709451 - ALL POPULATIONS\n",
      "====================================================================================================\n",
      "\n",
      "+------------+-------------+--------+--------+--------+---------+----------+-----------+---------+\n",
      "| population |   p_value   |  fst   | af_eur | af_pop | p_check | af_check | fst_check | result  |\n",
      "+------------+-------------+--------+--------+--------+---------+----------+-----------+---------+\n",
      "|    GIH     |   1.2e-09   | 0.5167 | 0.0016 | 0.2427 |    ✓    |    ✓     |     ✓     | ✅ PASS |\n",
      "|    STU     |  7.75e-07   | 0.4277 | 0.0016 | 0.1974 |    ✓    |    ✓     |     ✓     | ✅ PASS |\n",
      "|    ITU     |  1.52e-05   | 0.3985 | 0.0016 | 0.1729 |    ✓    |    ✓     |     ✓     | ✅ PASS |\n",
      "|    AMR2    |    1e-10    | 0.394  | 0.0016 | 0.258  |    ✓    |    ✓     |     ✓     | ✅ PASS |\n",
      "|    BEB     |  4.53e-06   | 0.3783 | 0.0016 | 0.1832 |    ✓    |    ✓     |     ✓     | ✅ PASS |\n",
      "|    PJL     | 0.001549282 | 0.2608 | 0.0016 | 0.1267 |    ✓    |    ✓     |     ✓     | ✅ PASS |\n",
      "|    AMR     |  9.33e-07   | 0.2127 | 0.0016 | 0.1959 |    ✓    |    ✓     |     ✓     | ✅ PASS |\n",
      "|    AMR1    | 0.00026834  | 0.2118 | 0.0016 | 0.1458 |    ✓    |    ✓     |     ✓     | ✅ PASS |\n",
      "|    EAS     |  6.87e-07   | 0.1998 | 0.0016 | 0.1983 |    ✓    |    ✓     |     ✓     | ✅ PASS |\n",
      "|    SAS     |  7.75e-07   | 0.1795 | 0.0016 | 0.1974 |    ✓    |    ✓     |     ✓     | ✅ PASS |\n",
      "|    AFR     | 0.678686019 | 0.0109 | 0.0016 | 0.0174 |    ✗    |    ✗     |     ✗     | ❌ FAIL |\n",
      "+------------+-------------+--------+--------+--------+---------+----------+-----------+---------+\n",
      "\n",
      "📊 Summary for rs4709451:\n",
      "  • Populations tested: 11\n",
      "  • Populations passing ALL THREE criteria: 10\n",
      "  • Passing populations: GIH, STU, ITU, AMR2, BEB, PJL, AMR, AMR1, EAS, SAS\n",
      "\n",
      "====================================================================================================\n",
      "VARIANTS SIGNIFICANT IN MULTIPLE SOUTH ASIAN POPULATIONS\n",
      "====================================================================================================\n",
      "Looking at: BEB, GIH, ITU, PJL, STU\n",
      "====================================================================================================\n",
      "\n",
      "+--------------+--------------+---------------+---------------------+---------+\n",
      "|    rs_id     |  gene_name   | sas_pop_count |   sas_populations   | avg_fst |\n",
      "+--------------+--------------+---------------+---------------------+---------+\n",
      "|  rs72258732  |    CYP3A4    |       5       | BEB,GIH,ITU,PJL,STU | 0.5298  |\n",
      "|  rs4646440   |    CYP3A4    |       5       | BEB,GIH,ITU,PJL,STU | 0.5233  |\n",
      "|  rs3743930   |     MEFV     |       5       | BEB,GIH,ITU,PJL,STU | 0.5204  |\n",
      "|  rs79898317  |   ARHGAP24   |       5       | BEB,GIH,ITU,PJL,STU | 0.4698  |\n",
      "|  rs9374602   |     DSE      |       5       | BEB,GIH,ITU,PJL,STU | 0.4665  |\n",
      "| rs372009731  |    RUNX3     |       5       | BEB,GIH,ITU,PJL,STU | 0.4566  |\n",
      "|   rs13612    |     MCAM     |       5       | BEB,GIH,ITU,PJL,STU | 0.4564  |\n",
      "|  rs35564277  |    CYP3A4    |       5       | BEB,GIH,ITU,PJL,STU | 0.4476  |\n",
      "|  rs10485908  |    MAGI2     |       5       | BEB,GIH,ITU,PJL,STU | 0.4472  |\n",
      "| rs138749777  |     BRAF     |       5       | BEB,GIH,ITU,PJL,STU | 0.4315  |\n",
      "|  rs9273522   |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU | 0.4298  |\n",
      "| rs796913782  |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU | 0.4298  |\n",
      "| rs535728917  |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU | 0.4298  |\n",
      "| rs117016875  |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU | 0.4298  |\n",
      "|  rs78621519  |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU | 0.4295  |\n",
      "|  rs7800480   |    MAGI2     |       5       | BEB,GIH,ITU,PJL,STU | 0.4287  |\n",
      "|  rs73378222  |    MAGI2     |       5       | BEB,GIH,ITU,PJL,STU | 0.4286  |\n",
      "|  rs9273741   |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU |  0.428  |\n",
      "| rs1311807036 |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU |  0.428  |\n",
      "|  rs73378221  |    MAGI2     |       5       | BEB,GIH,ITU,PJL,STU | 0.4261  |\n",
      "|  rs35409710  |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU | 0.4258  |\n",
      "|  rs7812172   |    MAGI2     |       5       | BEB,GIH,ITU,PJL,STU | 0.4249  |\n",
      "|  rs35702610  |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU | 0.4245  |\n",
      "|  rs9273721   |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU | 0.4235  |\n",
      "| rs117509699  |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU | 0.4235  |\n",
      "| rs578067733  |     KLK3     |       5       | BEB,GIH,ITU,PJL,STU | 0.4231  |\n",
      "| rs560492067  |     KLK3     |       5       | BEB,GIH,ITU,PJL,STU | 0.4231  |\n",
      "| rs117783863  |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU | 0.4203  |\n",
      "|  rs1130429   |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU | 0.4168  |\n",
      "|  rs34875318  |    RUNX3     |       5       | BEB,GIH,ITU,PJL,STU | 0.4162  |\n",
      "| rs1554163802 |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU |  0.415  |\n",
      "| rs200955452  |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU | 0.4134  |\n",
      "| rs200177238  |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU | 0.4134  |\n",
      "|  rs3204379   |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU | 0.4132  |\n",
      "|  rs3735060   |     EGFR     |       5       | BEB,GIH,ITU,PJL,STU | 0.4102  |\n",
      "|  rs17172443  |     EGFR     |       5       | BEB,GIH,ITU,PJL,STU | 0.4102  |\n",
      "| rs118117192  |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU |  0.41   |\n",
      "| rs281864242  |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU | 0.4069  |\n",
      "|  rs9561177   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4068  |\n",
      "| rs281864239  |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU | 0.4055  |\n",
      "|  rs9556212   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4049  |\n",
      "|  rs9561176   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4046  |\n",
      "| rs141394110  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4046  |\n",
      "| rs117488551  |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU | 0.4037  |\n",
      "|  rs8000315   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4034  |\n",
      "|  rs7984465   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4034  |\n",
      "| rs139676734  |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU | 0.4024  |\n",
      "|  rs3735061   |     EGFR     |       5       | BEB,GIH,ITU,PJL,STU | 0.4023  |\n",
      "|  rs9561179   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4022  |\n",
      "|  rs9561175   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4022  |\n",
      "|  rs9561174   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4022  |\n",
      "|  rs9556227   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4022  |\n",
      "|  rs9561160   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4021  |\n",
      "|  rs9561156   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4021  |\n",
      "|  rs9561153   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4021  |\n",
      "|  rs9561152   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4021  |\n",
      "|  rs9561151   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4021  |\n",
      "|  rs9556213   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4021  |\n",
      "|  rs9301834   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4021  |\n",
      "|  rs79651017  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4021  |\n",
      "|  rs7333121   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4021  |\n",
      "|  rs2065584   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4021  |\n",
      "|  rs12427983  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4021  |\n",
      "|  rs9561150   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4019  |\n",
      "|  rs35400222  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4011  |\n",
      "|  rs9561173   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4009  |\n",
      "|  rs9556224   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4009  |\n",
      "|  rs9561163   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4007  |\n",
      "|  rs9561161   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4007  |\n",
      "|  rs9561155   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4007  |\n",
      "|  rs9556215   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4007  |\n",
      "| rs281864454  | HLA-DQB1-AS1 |       5       | BEB,GIH,ITU,PJL,STU | 0.4007  |\n",
      "|  rs1890198   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4007  |\n",
      "|  rs12429947  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4007  |\n",
      "|  rs9561180   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.4001  |\n",
      "|  rs9561172   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3999  |\n",
      "|  rs9561166   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3999  |\n",
      "|  rs9561165   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3999  |\n",
      "|  rs9556223   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3999  |\n",
      "|  rs9556222   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3999  |\n",
      "|  rs9556219   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3999  |\n",
      "|  rs74909357  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3999  |\n",
      "| rs137945550  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3999  |\n",
      "|  rs3204373   |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU | 0.3998  |\n",
      "|  rs9561182   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3997  |\n",
      "|  rs9556216   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3995  |\n",
      "| rs200160235  | HLA-DQB1-AS1 |       5       | BEB,GIH,ITU,PJL,STU | 0.3992  |\n",
      "|  rs9561164   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3986  |\n",
      "|  rs5805764   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3986  |\n",
      "|  rs9561181   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3985  |\n",
      "|  rs9561169   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3985  |\n",
      "|  rs9561168   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3985  |\n",
      "|  rs77143890  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3985  |\n",
      "|  rs67558042  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3985  |\n",
      "|  rs35226443  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3985  |\n",
      "| rs141535075  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3985  |\n",
      "|  rs12722107  |   HLA-DQB1   |       5       | BEB,GIH,ITU,PJL,STU | 0.3985  |\n",
      "|  rs12428589  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3985  |\n",
      "|  rs9556217   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3972  |\n",
      "|  rs7792998   |    MAGI2     |       5       | BEB,GIH,ITU,PJL,STU |  0.397  |\n",
      "|  rs4709451   |     LPA      |       5       | BEB,GIH,ITU,PJL,STU | 0.3964  |\n",
      "|  rs60326929  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3931  |\n",
      "|  rs34750967  | HLA-DQB1-AS1 |       5       | BEB,GIH,ITU,PJL,STU | 0.3923  |\n",
      "| rs372610941  |  LINC02182   |       5       | BEB,GIH,ITU,PJL,STU |  0.392  |\n",
      "| rs149892231  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3886  |\n",
      "| rs879711264  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3856  |\n",
      "|  rs58675009  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU |  0.385  |\n",
      "|  rs4465482   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU |  0.385  |\n",
      "| rs190044054  |    DAAM2     |       5       | BEB,GIH,ITU,PJL,STU | 0.3842  |\n",
      "| rs369906286  |  LINC02182   |       5       | BEB,GIH,ITU,PJL,STU | 0.3838  |\n",
      "|  rs12431303  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3838  |\n",
      "|  rs2182367   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3836  |\n",
      "| rs118162574  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3816  |\n",
      "|  rs1791186   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3776  |\n",
      "|  rs9561170   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU |  0.377  |\n",
      "|  rs73312162  |    SULF2     |       5       | BEB,GIH,ITU,PJL,STU | 0.3769  |\n",
      "|   rs130002   |    CREBBP    |       5       | BEB,GIH,ITU,PJL,STU | 0.3767  |\n",
      "|  rs1667252   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3754  |\n",
      "|  rs78144647  |    PTPRD     |       5       | BEB,GIH,ITU,PJL,STU | 0.3753  |\n",
      "|  rs1667253   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3749  |\n",
      "|  rs36228615  |   SERPINE1   |       5       | BEB,GIH,ITU,PJL,STU | 0.3745  |\n",
      "| rs140904684  |    PSMD14    |       5       | BEB,GIH,ITU,PJL,STU | 0.3729  |\n",
      "| rs181427081  |    PRKCD     |       5       | BEB,GIH,ITU,PJL,STU | 0.3727  |\n",
      "|  rs9304103   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3716  |\n",
      "|   rs948526   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3706  |\n",
      "| rs146105005  |    PTPRD     |       5       | BEB,GIH,ITU,PJL,STU | 0.3704  |\n",
      "|  rs75886888  |    TRPC6     |       5       | BEB,GIH,ITU,PJL,STU | 0.3694  |\n",
      "|  rs35197841  |     TTR      |       5       | BEB,GIH,ITU,PJL,STU |  0.369  |\n",
      "|  rs1667245   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU |  0.369  |\n",
      "|  rs10538221  |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3687  |\n",
      "|  rs9561099   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3684  |\n",
      "|  rs2704061   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3683  |\n",
      "|  rs16918872  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU | 0.3681  |\n",
      "|  rs1667243   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3675  |\n",
      "| rs142786948  |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3673  |\n",
      "| rs377729424  |    PRKCD     |       5       | BEB,GIH,ITU,PJL,STU | 0.3669  |\n",
      "|  rs7321917   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3663  |\n",
      "|  rs1667251   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3661  |\n",
      "|  rs1667242   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3659  |\n",
      "|  rs9561116   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3651  |\n",
      "|  rs9561115   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3651  |\n",
      "|  rs7491566   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3651  |\n",
      "|  rs7334046   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3651  |\n",
      "|  rs7318597   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3651  |\n",
      "|  rs1999857   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3651  |\n",
      "|  rs9561136   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3645  |\n",
      "|  rs9561131   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3645  |\n",
      "|  rs9561130   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3645  |\n",
      "|  rs9301832   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3645  |\n",
      "|  rs1667246   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3645  |\n",
      "|  rs12429124  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3645  |\n",
      "| rs150381668  |    CREBBP    |       5       | BEB,GIH,ITU,PJL,STU | 0.3639  |\n",
      "|  rs68118265  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3632  |\n",
      "|  rs1791188   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3631  |\n",
      "|  rs1791187   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3631  |\n",
      "|  rs1631020   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3631  |\n",
      "|  rs9561128   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3627  |\n",
      "|  rs9556204   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3627  |\n",
      "|  rs7992127   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3627  |\n",
      "|  rs7987140   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3627  |\n",
      "|  rs78956231  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3627  |\n",
      "| rs149564265  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3625  |\n",
      "|  rs34682191  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3619  |\n",
      "| rs201592336  |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3616  |\n",
      "|  rs9561110   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3611  |\n",
      "|  rs1449083   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3599  |\n",
      "|  rs1791227   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3592  |\n",
      "|  rs1667250   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3592  |\n",
      "|  rs2704059   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3556  |\n",
      "|  rs1791191   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3556  |\n",
      "|  rs1667241   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3556  |\n",
      "|  rs1667238   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3556  |\n",
      "|  rs17766555  |    IL1R1     |       5       | BEB,GIH,ITU,PJL,STU | 0.3509  |\n",
      "|  rs75131341  |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3508  |\n",
      "|  rs73418151  |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3508  |\n",
      "|  rs57017056  |    CREBBP    |       5       | BEB,GIH,ITU,PJL,STU | 0.3495  |\n",
      "| rs188755786  |    PRKCD     |       5       | BEB,GIH,ITU,PJL,STU | 0.3493  |\n",
      "| rs138669340  |    PRKCD     |       5       | BEB,GIH,ITU,PJL,STU | 0.3488  |\n",
      "|  rs7338542   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3486  |\n",
      "|  rs9556206   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3485  |\n",
      "|  rs12430678  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3481  |\n",
      "| rs188517097  |    PRKCD     |       5       | BEB,GIH,ITU,PJL,STU | 0.3473  |\n",
      "| rs118177693  |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3472  |\n",
      "|  rs9561142   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3468  |\n",
      "|  rs71746500  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3468  |\n",
      "|  rs3735062   |     EGFR     |       5       | BEB,GIH,ITU,PJL,STU | 0.3467  |\n",
      "|  rs9561127   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3462  |\n",
      "|  rs1933194   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3462  |\n",
      "| rs577260058  |    PRKCD     |       5       | BEB,GIH,ITU,PJL,STU | 0.3454  |\n",
      "| rs1398581272 |    PRKCD     |       5       | BEB,GIH,ITU,PJL,STU | 0.3447  |\n",
      "|  rs7338330   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3439  |\n",
      "| rs568550170  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3433  |\n",
      "| rs201277328  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3433  |\n",
      "| rs146996211  |    PRKCD     |       5       | BEB,GIH,ITU,PJL,STU | 0.3427  |\n",
      "| rs114562702  |     CPB2     |       5       | BEB,GIH,ITU,PJL,STU | 0.3424  |\n",
      "| rs374327270  |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU |  0.342  |\n",
      "| rs189058853  |   SMARCAL1   |       5       | BEB,GIH,ITU,PJL,STU | 0.3418  |\n",
      "|  rs9561124   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3412  |\n",
      "|  rs9561147   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3408  |\n",
      "|  rs7327371   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3408  |\n",
      "| rs111639717  |    CREBBP    |       5       | BEB,GIH,ITU,PJL,STU | 0.3399  |\n",
      "|  rs12429761  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3398  |\n",
      "|  rs12429760  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3398  |\n",
      "| rs570499702  |    PRKCD     |       5       | BEB,GIH,ITU,PJL,STU | 0.3397  |\n",
      "|  rs9561143   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3395  |\n",
      "|  rs9561141   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3395  |\n",
      "|  rs9561139   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3395  |\n",
      "|  rs9556211   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3395  |\n",
      "|  rs9556205   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3395  |\n",
      "| rs201457074  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3395  |\n",
      "| rs150501879  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3395  |\n",
      "|  rs60451764  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3388  |\n",
      "| rs148625478  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3381  |\n",
      "|  rs1620287   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU |  0.338  |\n",
      "| rs184186499  |   SMARCAL1   |       5       | BEB,GIH,ITU,PJL,STU | 0.3378  |\n",
      "| rs112425593  |    PRKCD     |       5       | BEB,GIH,ITU,PJL,STU | 0.3373  |\n",
      "| rs142234364  |    IL2RA     |       5       | BEB,GIH,ITU,PJL,STU | 0.3356  |\n",
      "| rs149072572  |    PRKCD     |       5       | BEB,GIH,ITU,PJL,STU | 0.3353  |\n",
      "|  rs17335912  |     EGFR     |       5       | BEB,GIH,ITU,PJL,STU | 0.3349  |\n",
      "|  rs12538116  |     EGFR     |       5       | BEB,GIH,ITU,PJL,STU | 0.3349  |\n",
      "| rs374818109  |    RUNX3     |       5       | BEB,GIH,ITU,PJL,STU | 0.3342  |\n",
      "| rs142028154  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3342  |\n",
      "|  rs9561126   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU |  0.334  |\n",
      "| rs373871206  |    IFIH1     |       5       | BEB,GIH,ITU,PJL,STU | 0.3336  |\n",
      "|  rs7800086   |     EGFR     |       5       | BEB,GIH,ITU,PJL,STU |  0.333  |\n",
      "|  rs17335905  |     EGFR     |       5       | BEB,GIH,ITU,PJL,STU |  0.333  |\n",
      "| rs149118532  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3326  |\n",
      "|  rs12428255  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.3326  |\n",
      "|  rs76539632  |     EGFR     |       5       | BEB,GIH,ITU,PJL,STU |  0.332  |\n",
      "|  rs12538513  |     EGFR     |       5       | BEB,GIH,ITU,PJL,STU | 0.3315  |\n",
      "|  rs79093999  |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU | 0.3301  |\n",
      "|  rs4947966   |     EGFR     |       5       | BEB,GIH,ITU,PJL,STU | 0.3301  |\n",
      "| rs189849462  |    PTPRD     |       5       | BEB,GIH,ITU,PJL,STU | 0.3301  |\n",
      "| rs116645015  |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU | 0.3301  |\n",
      "| rs115395612  |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU | 0.3301  |\n",
      "|   rs873966   |    SYNPO     |       5       | BEB,GIH,ITU,PJL,STU | 0.3273  |\n",
      "| rs111610803  |    PRKCD     |       5       | BEB,GIH,ITU,PJL,STU | 0.3264  |\n",
      "|  rs78721874  |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU | 0.3253  |\n",
      "| rs140018962  |    PTPRD     |       5       | BEB,GIH,ITU,PJL,STU | 0.3232  |\n",
      "|  rs73682257  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU | 0.3205  |\n",
      "|  rs3792163   |    PLA2R1    |       5       | BEB,GIH,ITU,PJL,STU | 0.3201  |\n",
      "|   rs508958   |     DLC1     |       5       | BEB,GIH,ITU,PJL,STU | 0.3181  |\n",
      "|  rs75893245  |   RAPGEF2    |       5       | BEB,GIH,ITU,PJL,STU |  0.317  |\n",
      "|  rs10092878  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU | 0.3154  |\n",
      "|  rs76961617  |   RAPGEF2    |       5       | BEB,GIH,ITU,PJL,STU | 0.3141  |\n",
      "|  rs79575131  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU |  0.314  |\n",
      "|  rs79535962  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU |  0.314  |\n",
      "|  rs78799427  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU |  0.314  |\n",
      "|  rs77036827  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU |  0.314  |\n",
      "|  rs76009155  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU |  0.314  |\n",
      "|  rs75848424  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU |  0.314  |\n",
      "|  rs73682221  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU |  0.314  |\n",
      "|  rs57724912  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU |  0.314  |\n",
      "|  rs57290263  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU |  0.314  |\n",
      "|  rs16927189  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU |  0.314  |\n",
      "|  rs79648847  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU | 0.3123  |\n",
      "|  rs9947284   |     TTR      |       5       | BEB,GIH,ITU,PJL,STU | 0.3115  |\n",
      "| rs571913884  |     GCA      |       5       | BEB,GIH,ITU,PJL,STU | 0.3108  |\n",
      "|  rs77179901  |    PLA2R1    |       5       | BEB,GIH,ITU,PJL,STU | 0.3105  |\n",
      "|  rs78338826  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU | 0.3103  |\n",
      "|  rs75661549  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU | 0.3103  |\n",
      "|  rs16927172  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU | 0.3103  |\n",
      "| rs149650003  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU | 0.3088  |\n",
      "|  rs73967964  |    PLA2R1    |       5       | BEB,GIH,ITU,PJL,STU | 0.3079  |\n",
      "|  rs3828322   |    PLA2R1    |       5       | BEB,GIH,ITU,PJL,STU | 0.3079  |\n",
      "|  rs34505557  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU | 0.3074  |\n",
      "| rs187945460  |    PRKCD     |       5       | BEB,GIH,ITU,PJL,STU | 0.3048  |\n",
      "| rs186686581  |    PRKCD     |       5       | BEB,GIH,ITU,PJL,STU | 0.3048  |\n",
      "|  rs80015738  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU | 0.3034  |\n",
      "|  rs73682260  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU | 0.3034  |\n",
      "|  rs77456134  |    PLA2R1    |       5       | BEB,GIH,ITU,PJL,STU | 0.3033  |\n",
      "|  rs75633297  |    PLA2R1    |       5       | BEB,GIH,ITU,PJL,STU | 0.3033  |\n",
      "|  rs75168090  |    PLA2R1    |       5       | BEB,GIH,ITU,PJL,STU | 0.3033  |\n",
      "|  rs75160143  |    PLA2R1    |       5       | BEB,GIH,ITU,PJL,STU | 0.3033  |\n",
      "| rs149861457  |    PLA2R1    |       5       | BEB,GIH,ITU,PJL,STU | 0.3033  |\n",
      "| rs111633585  |    PLA2R1    |       5       | BEB,GIH,ITU,PJL,STU | 0.3033  |\n",
      "| rs111630731  |    PLA2R1    |       5       | BEB,GIH,ITU,PJL,STU | 0.3033  |\n",
      "| rs193019090  |    PRKCD     |       5       | BEB,GIH,ITU,PJL,STU | 0.3031  |\n",
      "|  rs72632990  |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.3028  |\n",
      "|  rs67771121  |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.3028  |\n",
      "|  rs11024190  |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.3028  |\n",
      "|  rs59250592  |     TJP1     |       5       | BEB,GIH,ITU,PJL,STU | 0.3021  |\n",
      "| rs143850044  |   RAPGEF2    |       5       | BEB,GIH,ITU,PJL,STU | 0.3017  |\n",
      "|  rs73493507  |    CREBBP    |       5       | BEB,GIH,ITU,PJL,STU | 0.3007  |\n",
      "| rs151141305  |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.3001  |\n",
      "| rs139353436  |  LINC02182   |       5       | BEB,GIH,ITU,PJL,STU | 0.2985  |\n",
      "| rs114965558  |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU |  0.298  |\n",
      "|  rs76755468  |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU | 0.2958  |\n",
      "|  rs74450190  |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU | 0.2958  |\n",
      "| rs114636226  |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU | 0.2958  |\n",
      "| rs114025159  |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU | 0.2958  |\n",
      "|  rs80255205  |    TRPC6     |       5       | BEB,GIH,ITU,PJL,STU | 0.2956  |\n",
      "|  rs77587713  |    TRPC6     |       5       | BEB,GIH,ITU,PJL,STU | 0.2956  |\n",
      "|  rs74890898  |    TRPC6     |       5       | BEB,GIH,ITU,PJL,STU | 0.2956  |\n",
      "|  rs74836390  |    TRPC6     |       5       | BEB,GIH,ITU,PJL,STU | 0.2956  |\n",
      "| rs150743817  |    TRPC6     |       5       | BEB,GIH,ITU,PJL,STU | 0.2956  |\n",
      "| rs113938755  |    TRPC6     |       5       | BEB,GIH,ITU,PJL,STU | 0.2956  |\n",
      "| rs113827498  |    TRPC6     |       5       | BEB,GIH,ITU,PJL,STU | 0.2956  |\n",
      "| rs113103264  |    TRPC6     |       5       | BEB,GIH,ITU,PJL,STU | 0.2956  |\n",
      "| rs113073524  |    TRPC6     |       5       | BEB,GIH,ITU,PJL,STU | 0.2956  |\n",
      "| rs112511997  |    TRPC6     |       5       | BEB,GIH,ITU,PJL,STU | 0.2956  |\n",
      "|  rs76589774  |    TRPC6     |       5       | BEB,GIH,ITU,PJL,STU | 0.2944  |\n",
      "| rs138234844  |    TRPC6     |       5       | BEB,GIH,ITU,PJL,STU | 0.2944  |\n",
      "| rs112943669  |    TRPC6     |       5       | BEB,GIH,ITU,PJL,STU | 0.2944  |\n",
      "|  rs72632989  |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2934  |\n",
      "|  rs72632988  |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2934  |\n",
      "|  rs7122571   |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2934  |\n",
      "|  rs7113011   |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2934  |\n",
      "|  rs4757499   |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2934  |\n",
      "|  rs12225195  |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2934  |\n",
      "|  rs11024186  |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2934  |\n",
      "| rs372970860  |     GAD1     |       5       | BEB,GIH,ITU,PJL,STU | 0.2933  |\n",
      "| rs377374573  |    PTPRD     |       5       | BEB,GIH,ITU,PJL,STU | 0.2932  |\n",
      "| rs376485313  |     CMIP     |       5       | BEB,GIH,ITU,PJL,STU | 0.2931  |\n",
      "| rs375442681  |    MAGI2     |       5       | BEB,GIH,ITU,PJL,STU | 0.2922  |\n",
      "|  rs7945756   |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2915  |\n",
      "|  rs7934596   |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2915  |\n",
      "| rs372989421  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2913  |\n",
      "| rs1368691747 |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU | 0.2912  |\n",
      "| rs145891570  |  LINC02182   |       5       | BEB,GIH,ITU,PJL,STU | 0.2883  |\n",
      "|  rs77162704  |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU | 0.2868  |\n",
      "| rs370228955  |    PTPRD     |       5       | BEB,GIH,ITU,PJL,STU | 0.2868  |\n",
      "|  rs12664261  |    RUNX2     |       5       | BEB,GIH,ITU,PJL,STU | 0.2866  |\n",
      "| rs200290437  |    TRPC6     |       5       | BEB,GIH,ITU,PJL,STU | 0.2864  |\n",
      "|  rs10924994  |    NLRP3     |       5       | BEB,GIH,ITU,PJL,STU | 0.2864  |\n",
      "| rs1554721428 |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU | 0.2855  |\n",
      "|  rs77891883  |  LINC02740   |       5       | BEB,GIH,ITU,PJL,STU |  0.285  |\n",
      "| rs376063641  |   KIRREL1    |       5       | BEB,GIH,ITU,PJL,STU |  0.285  |\n",
      "|  rs78008826  |  TSBP1-AS1   |       5       | BEB,GIH,ITU,PJL,STU | 0.2845  |\n",
      "|  rs7744293   |  TSBP1-AS1   |       5       | BEB,GIH,ITU,PJL,STU | 0.2845  |\n",
      "|  rs76746547  |    BTNL2     |       5       | BEB,GIH,ITU,PJL,STU | 0.2845  |\n",
      "|  rs76318143  |  TSBP1-AS1   |       5       | BEB,GIH,ITU,PJL,STU | 0.2845  |\n",
      "|  rs74833452  |    PSMD14    |       5       | BEB,GIH,ITU,PJL,STU | 0.2845  |\n",
      "| rs149697550  |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU | 0.2842  |\n",
      "| rs143766226  |    TRPC6     |       5       | BEB,GIH,ITU,PJL,STU | 0.2837  |\n",
      "|  rs10977225  |    PTPRD     |       5       | BEB,GIH,ITU,PJL,STU | 0.2836  |\n",
      "|  rs59152435  |    TRPC6     |       5       | BEB,GIH,ITU,PJL,STU | 0.2831  |\n",
      "|  rs12289931  |    TRPC6     |       5       | BEB,GIH,ITU,PJL,STU | 0.2831  |\n",
      "|  rs77176359  |    PSMD14    |       5       | BEB,GIH,ITU,PJL,STU | 0.2827  |\n",
      "| rs375663789  |   SMARCAL1   |       5       | BEB,GIH,ITU,PJL,STU | 0.2817  |\n",
      "|  rs79032534  |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU | 0.2811  |\n",
      "|  rs76954985  |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU | 0.2811  |\n",
      "|  rs7047249   |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU | 0.2811  |\n",
      "| rs368704043  |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU | 0.2811  |\n",
      "| rs151202180  |  LINC02182   |       5       | BEB,GIH,ITU,PJL,STU | 0.2811  |\n",
      "| rs140797650  |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU | 0.2811  |\n",
      "|  rs78346230  |    PRKCD     |       5       | BEB,GIH,ITU,PJL,STU | 0.2803  |\n",
      "| rs563561786  |     LPA      |       5       | BEB,GIH,ITU,PJL,STU | 0.2798  |\n",
      "| rs377417508  |  LINC02182   |       5       | BEB,GIH,ITU,PJL,STU | 0.2797  |\n",
      "|  rs78101631  |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU | 0.2796  |\n",
      "| rs143998330  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2789  |\n",
      "| rs150446000  |    TRPC6     |       5       | BEB,GIH,ITU,PJL,STU | 0.2787  |\n",
      "|  rs35052563  |    IGFBP3    |       5       | BEB,GIH,ITU,PJL,STU | 0.2785  |\n",
      "|  rs34508316  |    IGFBP3    |       5       | BEB,GIH,ITU,PJL,STU | 0.2785  |\n",
      "| rs375748003  |  LINC02182   |       5       | BEB,GIH,ITU,PJL,STU | 0.2779  |\n",
      "|  rs75351282  |    MAGI2     |       5       | BEB,GIH,ITU,PJL,STU | 0.2773  |\n",
      "|  rs7759853   |    RUNX2     |       5       | BEB,GIH,ITU,PJL,STU |  0.277  |\n",
      "| rs372521158  |     HRG      |       5       | BEB,GIH,ITU,PJL,STU | 0.2753  |\n",
      "|  rs79431890  |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU | 0.2722  |\n",
      "| rs375638877  |    PTPRD     |       5       | BEB,GIH,ITU,PJL,STU | 0.2719  |\n",
      "| rs149137540  |    RUNX2     |       5       | BEB,GIH,ITU,PJL,STU | 0.2719  |\n",
      "| rs143735685  |    IL2RA     |       5       | BEB,GIH,ITU,PJL,STU | 0.2712  |\n",
      "|  rs77434329  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU | 0.2711  |\n",
      "|  rs79226477  |   ARHGAP24   |       5       | BEB,GIH,ITU,PJL,STU | 0.2708  |\n",
      "| rs142426101  |     TNXB     |       5       | BEB,GIH,ITU,PJL,STU | 0.2702  |\n",
      "| rs189484236  |    IL2RA     |       5       | BEB,GIH,ITU,PJL,STU | 0.2698  |\n",
      "|  rs12664292  |    RUNX2     |       5       | BEB,GIH,ITU,PJL,STU | 0.2694  |\n",
      "| rs117875982  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2684  |\n",
      "| rs184921197  |    CLVS1     |       5       | BEB,GIH,ITU,PJL,STU | 0.2682  |\n",
      "| rs370688654  |   KIRREL3    |       5       | BEB,GIH,ITU,PJL,STU | 0.2679  |\n",
      "| rs556046080  |    PLCG2     |       5       | BEB,GIH,ITU,PJL,STU | 0.2675  |\n",
      "|  rs9357487   |    RUNX2     |       5       | BEB,GIH,ITU,PJL,STU | 0.2636  |\n",
      "| rs376361533  |    KIF21A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2626  |\n",
      "| rs542387097  |     LEPR     |       5       | BEB,GIH,ITU,PJL,STU | 0.2617  |\n",
      "|  rs73394296  |    PTPRD     |       5       | BEB,GIH,ITU,PJL,STU | 0.2613  |\n",
      "| rs191292048  |  LINC02531   |       5       | BEB,GIH,ITU,PJL,STU | 0.2613  |\n",
      "| rs190768396  |    MAGI2     |       5       | BEB,GIH,ITU,PJL,STU | 0.2598  |\n",
      "| rs144658674  |    SYNPO     |       5       | BEB,GIH,ITU,PJL,STU | 0.2593  |\n",
      "| rs144595812  |    SYNPO     |       5       | BEB,GIH,ITU,PJL,STU | 0.2573  |\n",
      "|  rs78287258  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2569  |\n",
      "| rs142445767  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2569  |\n",
      "| rs552968234  |  LINC02182   |       5       | BEB,GIH,ITU,PJL,STU | 0.2568  |\n",
      "| rs373951192  |    PTPRD     |       5       | BEB,GIH,ITU,PJL,STU | 0.2563  |\n",
      "| rs369924239  |     LIPC     |       5       | BEB,GIH,ITU,PJL,STU | 0.2542  |\n",
      "|  rs56890485  |  LINC02182   |       5       | BEB,GIH,ITU,PJL,STU | 0.2533  |\n",
      "|  rs77930411  |    IQGAP1    |       5       | BEB,GIH,ITU,PJL,STU | 0.2527  |\n",
      "| rs142117244  |    SYNPO     |       5       | BEB,GIH,ITU,PJL,STU | 0.2526  |\n",
      "|  rs75146996  |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2518  |\n",
      "| rs145666620  |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2518  |\n",
      "| rs138114382  |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2518  |\n",
      "| rs114052084  |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2518  |\n",
      "|  rs74302394  |    MAGI2     |       5       | BEB,GIH,ITU,PJL,STU | 0.2514  |\n",
      "| rs199500769  |    MAGI2     |       5       | BEB,GIH,ITU,PJL,STU | 0.2514  |\n",
      "|  rs16947878  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2506  |\n",
      "| rs148614433  |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2504  |\n",
      "| rs368765304  |    PTPRD     |       5       | BEB,GIH,ITU,PJL,STU | 0.2501  |\n",
      "|  rs6462762   |    ELMO1     |       5       | BEB,GIH,ITU,PJL,STU | 0.2499  |\n",
      "| rs1224810930 |  LINC02188   |       5       | BEB,GIH,ITU,PJL,STU | 0.2493  |\n",
      "|  rs74818649  |    SYNPO     |       5       | BEB,GIH,ITU,PJL,STU | 0.2492  |\n",
      "| rs146833543  |    SYNPO     |       5       | BEB,GIH,ITU,PJL,STU | 0.2492  |\n",
      "| rs368300562  |     LEPR     |       5       | BEB,GIH,ITU,PJL,STU | 0.2484  |\n",
      "| rs141941403  |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2482  |\n",
      "|  rs73693564  |    ELMO1     |       5       | BEB,GIH,ITU,PJL,STU | 0.2476  |\n",
      "|  rs61763083  |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2463  |\n",
      "|  rs17150575  |    MAGI2     |       5       | BEB,GIH,ITU,PJL,STU |  0.246  |\n",
      "|  rs78644741  |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2442  |\n",
      "| rs117479943  |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2439  |\n",
      "| rs368483685  |    PTPRD     |       5       | BEB,GIH,ITU,PJL,STU | 0.2424  |\n",
      "|  rs17847725  |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2422  |\n",
      "|   rs955221   |    DOCK8     |       5       | BEB,GIH,ITU,PJL,STU | 0.2411  |\n",
      "|   rs755026   |    SYNPO     |       5       | BEB,GIH,ITU,PJL,STU | 0.2404  |\n",
      "|   rs755025   |    SYNPO     |       5       | BEB,GIH,ITU,PJL,STU | 0.2404  |\n",
      "| rs183505079  |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2401  |\n",
      "| rs139816927  |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2401  |\n",
      "|  rs6486351   |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2399  |\n",
      "| rs375894329  |    PTPRD     |       5       | BEB,GIH,ITU,PJL,STU | 0.2387  |\n",
      "| rs146003828  |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU |  0.238  |\n",
      "|  rs75879108  |     EGFR     |       5       | BEB,GIH,ITU,PJL,STU | 0.2374  |\n",
      "| rs150656339  |    SYNPO     |       5       | BEB,GIH,ITU,PJL,STU | 0.2364  |\n",
      "|  rs7933926   |   PIK3C2A    |       5       | BEB,GIH,ITU,PJL,STU | 0.2318  |\n",
      "| rs868104828  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2316  |\n",
      "|  rs75417090  |    SYNPO     |       5       | BEB,GIH,ITU,PJL,STU | 0.2316  |\n",
      "| rs373884400  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2316  |\n",
      "| rs973025594  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "|  rs9589226   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "|  rs9589223   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "|  rs9583931   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "|  rs9583930   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "|  rs9583929   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "|  rs9583926   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "|  rs9583925   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "| rs954409947  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "|  rs77255798  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "|  rs7319643   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "| rs576454877  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "| rs570533339  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "| rs548609973  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "|  rs28583407  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "|  rs28540075  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "| rs192705480  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "| rs191451134  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "| rs188124977  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "| rs182909056  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "| rs149229247  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "| rs149211127  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "| rs147718511  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "| rs141913669  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "|  rs13378856  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "|  rs13378154  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "| rs115072843  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "| rs114375303  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "| rs1037166992 |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2299  |\n",
      "|  rs9589236   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2281  |\n",
      "|  rs9589235   |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2281  |\n",
      "|  rs78496244  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2281  |\n",
      "|  rs76934952  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2281  |\n",
      "| rs369580789  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2281  |\n",
      "| rs181626263  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2272  |\n",
      "| rs548262831  |     CPB2     |       5       | BEB,GIH,ITU,PJL,STU | 0.2262  |\n",
      "|  rs76510113  |    PODXL     |       5       | BEB,GIH,ITU,PJL,STU | 0.2258  |\n",
      "| rs369533315  |    IL2RA     |       5       | BEB,GIH,ITU,PJL,STU | 0.2219  |\n",
      "| rs563520699  |     WT1      |       5       | BEB,GIH,ITU,PJL,STU | 0.2218  |\n",
      "| rs373280044  |     GPC5     |       5       | BEB,GIH,ITU,PJL,STU | 0.2202  |\n",
      "| rs368298783  |   SERPINE1   |       5       | BEB,GIH,ITU,PJL,STU | 0.2185  |\n",
      "| rs375746392  |    NUP160    |       5       | BEB,GIH,ITU,PJL,STU |  0.213  |\n",
      "| rs572608010  |  LINC02531   |       5       | BEB,GIH,ITU,PJL,STU | 0.2127  |\n",
      "| rs147815625  |    IGFBP2    |       5       | BEB,GIH,ITU,PJL,STU | 0.2125  |\n",
      "| rs146880775  |    IGFBP2    |       5       | BEB,GIH,ITU,PJL,STU | 0.2125  |\n",
      "| rs550248971  |    KANK4     |       5       | BEB,GIH,ITU,PJL,STU |  0.212  |\n",
      "| rs150849396  |     FAT1     |       5       | BEB,GIH,ITU,PJL,STU | 0.2114  |\n",
      "| rs372807672  |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU | 0.2104  |\n",
      "| rs566271675  |    KANK4     |       5       | BEB,GIH,ITU,PJL,STU | 0.2089  |\n",
      "| rs145587520  |    KANK4     |       5       | BEB,GIH,ITU,PJL,STU | 0.2047  |\n",
      "| rs371651956  |     DLC1     |       5       | BEB,GIH,ITU,PJL,STU |  0.204  |\n",
      "| rs374985830  |     ANLN     |       5       | BEB,GIH,ITU,PJL,STU | 0.2002  |\n",
      "| rs529271854  |    PTPRD     |       5       | BEB,GIH,ITU,PJL,STU | 0.1979  |\n",
      "| rs532675710  |    MUC16     |       5       | BEB,GIH,ITU,PJL,STU | 0.1974  |\n",
      "| rs114192862  |    KANK1     |       5       | BEB,GIH,ITU,PJL,STU | 0.1887  |\n",
      "| rs550890118  |    PTPRD     |       5       | BEB,GIH,ITU,PJL,STU | 0.1779  |\n",
      "| rs201685439  |   SNORD3A    |       4       |   BEB,GIH,ITU,STU   | 0.4848  |\n",
      "|  rs16887217  |     STAR     |       4       |   BEB,GIH,ITU,STU   |  0.468  |\n",
      "| rs141507870  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.4363  |\n",
      "|  rs9274350   |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.4353  |\n",
      "| rs150457662  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.4346  |\n",
      "|  rs11246324  |    CD151     |       4       |   BEB,ITU,PJL,STU   | 0.4324  |\n",
      "| rs755717699  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.4179  |\n",
      "| rs281861730  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.4163  |\n",
      "| rs281861725  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.4157  |\n",
      "| rs138584745  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.412  |\n",
      "| rs151109211  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.4108  |\n",
      "| rs117476746  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.4102  |\n",
      "|  rs34692792  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.4095  |\n",
      "|  rs35908913  | HLA-DQB1-AS1 |       4       |   BEB,GIH,ITU,PJL   | 0.4094  |\n",
      "| rs1397849352 | HLA-DQB1-AS1 |       4       |   BEB,GIH,ITU,PJL   | 0.4094  |\n",
      "| rs1321318532 | HLA-DQB1-AS1 |       4       |   BEB,GIH,ITU,PJL   | 0.4094  |\n",
      "| rs281874782  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.4052  |\n",
      "| rs879705868  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.4048  |\n",
      "| rs116903963  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.4048  |\n",
      "|  rs9273499   | HLA-DQB1-AS1 |       4       |   BEB,GIH,ITU,PJL   | 0.4037  |\n",
      "| rs182275627  | HLA-DQB1-AS1 |       4       |   BEB,GIH,ITU,PJL   | 0.4037  |\n",
      "| rs199987488  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.403  |\n",
      "| rs116984222  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.403  |\n",
      "| rs281864446  | HLA-DQB1-AS1 |       4       |   BEB,GIH,ITU,PJL   | 0.4021  |\n",
      "| rs201814598  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.4018  |\n",
      "| rs117578558  | HLA-DQB1-AS1 |       4       |   BEB,GIH,ITU,PJL   | 0.4012  |\n",
      "| rs117337127  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.4007  |\n",
      "| rs281864445  | HLA-DQB1-AS1 |       4       |   BEB,GIH,ITU,PJL   | 0.4004  |\n",
      "|  rs9274203   |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.4003  |\n",
      "| rs117390145  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3974  |\n",
      "|  rs34983499  | HLA-DQB1-AS1 |       4       |   BEB,GIH,ITU,PJL   | 0.3967  |\n",
      "|  rs9273580   |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3962  |\n",
      "| rs117899140  | HLA-DQB1-AS1 |       4       |   BEB,GIH,ITU,PJL   |  0.396  |\n",
      "| rs117706971  | HLA-DQB1-AS1 |       4       |   BEB,GIH,ITU,PJL   |  0.396  |\n",
      "| rs117531835  | HLA-DQB1-AS1 |       4       |   BEB,GIH,ITU,PJL   | 0.3951  |\n",
      "| rs116937403  | HLA-DQB1-AS1 |       4       |   BEB,GIH,ITU,PJL   | 0.3951  |\n",
      "|  rs9273588   |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3941  |\n",
      "| rs117069212  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3939  |\n",
      "| rs281861320  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3928  |\n",
      "| rs187940467  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3928  |\n",
      "|   rs129982   |    CREBBP    |       4       |   BEB,GIH,ITU,STU   | 0.3926  |\n",
      "|  rs9273609   |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3923  |\n",
      "| rs281864289  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3923  |\n",
      "| rs281864287  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3923  |\n",
      "| rs1237867150 |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3917  |\n",
      "| rs117626117  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3912  |\n",
      "|  rs34029741  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3907  |\n",
      "| rs281861181  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3907  |\n",
      "| rs200544780  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3907  |\n",
      "| rs117677058  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3907  |\n",
      "| rs117672109  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3907  |\n",
      "| rs116969302  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3907  |\n",
      "| rs281860801  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3903  |\n",
      "| rs144101223  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3903  |\n",
      "|  rs9282081   |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "|  rs9274418   |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "|  rs9273861   |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs548946394  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs541430730  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "|  rs36222416  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "|  rs34236112  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "|  rs3210146   |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs281860823  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs281860822  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs281860814  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs202088642  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs201478149  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs200098349  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs183607390  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs146003607  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs143370221  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs138033214  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "|  rs12722106  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs118132027  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs118120568  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs117992092  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs117988605  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs117912793  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs117867884  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs117829012  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs117789582  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs117780937  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs117734503  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs117621674  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs117570697  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs117407437  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs117403290  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs117384891  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs117360622  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs117233242  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs117067420  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs117000147  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs116850050  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs1019543462 |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.389  |\n",
      "| rs117425276  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3887  |\n",
      "| rs281861148  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3885  |\n",
      "| rs116882546  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3877  |\n",
      "| rs117898330  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3871  |\n",
      "| rs370470191  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   |  0.387  |\n",
      "| rs142491022  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3865  |\n",
      "| rs201930518  |   HLA-DQB1   |       4       |   BEB,GIH,ITU,PJL   | 0.3842  |\n",
      "|  rs2287290   |    IFIH1     |       4       |   BEB,GIH,ITU,STU   | 0.3814  |\n",
      "|  rs79497955  |    CREBBP    |       4       |   BEB,GIH,ITU,STU   | 0.3723  |\n",
      "|  rs78676523  |   HLA-DQA1   |       4       |   BEB,GIH,ITU,PJL   | 0.3635  |\n",
      "|  rs3025694   |    CREBBP    |       4       |   BEB,GIH,ITU,STU   |  0.363  |\n",
      "|  rs28653457  |    CREBBP    |       4       |   BEB,GIH,ITU,STU   |  0.361  |\n",
      "|  rs78487339  |    CREBBP    |       4       |   BEB,GIH,ITU,STU   | 0.3588  |\n",
      "|   rs508097   |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.3569  |\n",
      "|   rs642691   |     DLC1     |       4       |   BEB,GIH,ITU,STU   |  0.356  |\n",
      "| rs201813623  |     DLC1     |       4       |   BEB,GIH,ITU,STU   |  0.356  |\n",
      "| rs143071348  |    IL1R1     |       4       |   BEB,GIH,PJL,STU   | 0.3282  |\n",
      "|  rs77039968  |   RAPGEF2    |       4       |   BEB,GIH,ITU,STU   | 0.3129  |\n",
      "| rs137882436  |   RAPGEF2    |       4       |   BEB,GIH,ITU,STU   | 0.3129  |\n",
      "| rs113345623  |   RAPGEF2    |       4       |   BEB,GIH,ITU,STU   | 0.3129  |\n",
      "| rs147293380  |    DICER1    |       4       |   BEB,ITU,PJL,STU   | 0.3116  |\n",
      "| rs140838258  |    IGFBP2    |       4       |   BEB,GIH,ITU,STU   | 0.3108  |\n",
      "|  rs78210562  |   RAPGEF2    |       4       |   BEB,GIH,ITU,STU   | 0.3105  |\n",
      "|  rs80142809  |    CREBBP    |       4       |   BEB,GIH,ITU,STU   | 0.3093  |\n",
      "| rs117682612  |   RAPGEF2    |       4       |   BEB,GIH,ITU,STU   |  0.299  |\n",
      "|  rs7526937   |     SELE     |       4       |   BEB,GIH,ITU,STU   | 0.2986  |\n",
      "|  rs75009340  |     SELE     |       4       |   BEB,GIH,ITU,STU   | 0.2986  |\n",
      "|  rs12023533  |     SELE     |       4       |   BEB,GIH,ITU,STU   | 0.2986  |\n",
      "|  rs79302224  |    ELMO1     |       4       |   BEB,GIH,ITU,STU   | 0.2896  |\n",
      "| rs182646184  |     GPC5     |       4       |   BEB,GIH,ITU,STU   | 0.2882  |\n",
      "| rs144409895  |    ALOX5     |       4       |   BEB,GIH,ITU,STU   | 0.2833  |\n",
      "|  rs80324514  |  LINC02740   |       4       |   BEB,ITU,PJL,STU   | 0.2809  |\n",
      "|  rs78697484  |  LINC02740   |       4       |   BEB,ITU,PJL,STU   | 0.2809  |\n",
      "|  rs78150728  |  LINC02740   |       4       |   BEB,ITU,PJL,STU   | 0.2809  |\n",
      "|  rs57862527  |  LINC02740   |       4       |   BEB,ITU,PJL,STU   | 0.2809  |\n",
      "|  rs16936409  |  LINC02740   |       4       |   BEB,ITU,PJL,STU   | 0.2809  |\n",
      "|  rs16936406  |  LINC02740   |       4       |   BEB,ITU,PJL,STU   | 0.2809  |\n",
      "|  rs16936404  |  LINC02740   |       4       |   BEB,ITU,PJL,STU   | 0.2809  |\n",
      "| rs114347768  |  LINC02740   |       4       |   BEB,ITU,PJL,STU   | 0.2809  |\n",
      "|  rs10977214  |    PTPRD     |       4       |   BEB,GIH,ITU,STU   | 0.2793  |\n",
      "|  rs77339496  |    PTPRD     |       4       |   BEB,ITU,PJL,STU   | 0.2787  |\n",
      "| rs374737917  |     SELE     |       4       |   BEB,GIH,ITU,STU   | 0.2781  |\n",
      "|  rs4742544   |    PTPRD     |       4       |   BEB,ITU,PJL,STU   | 0.2773  |\n",
      "| rs373266474  |     AGT      |       4       |   BEB,GIH,ITU,PJL   | 0.2761  |\n",
      "| rs375823270  |     LRP2     |       4       |   BEB,GIH,ITU,STU   | 0.2722  |\n",
      "| rs374891141  |     LRP2     |       4       |   BEB,GIH,ITU,STU   | 0.2722  |\n",
      "| rs370718773  |     LRP2     |       4       |   BEB,GIH,ITU,STU   | 0.2722  |\n",
      "|  rs76910536  |  LINC02182   |       4       |   BEB,GIH,ITU,STU   | 0.2715  |\n",
      "| rs563212156  |     CUBN     |       4       |   BEB,GIH,ITU,STU   | 0.2715  |\n",
      "| rs141156598  |   KIRREL3    |       4       |   GIH,ITU,PJL,STU   | 0.2709  |\n",
      "| rs373435375  |    TNFSF4    |       4       |   BEB,GIH,ITU,STU   | 0.2692  |\n",
      "|  rs59379824  |  LINC02182   |       4       |   BEB,GIH,ITU,STU   | 0.2688  |\n",
      "|  rs55845661  |  LINC02182   |       4       |   BEB,GIH,ITU,STU   | 0.2676  |\n",
      "|  rs78396511  |  LINC02182   |       4       |   BEB,GIH,ITU,STU   | 0.2675  |\n",
      "| rs374271891  |     LRP2     |       4       |   BEB,GIH,ITU,STU   | 0.2674  |\n",
      "| rs375893791  |    IL1R1     |       4       |   BEB,GIH,PJL,STU   | 0.2652  |\n",
      "|  rs75840024  |  LINC02182   |       4       |   BEB,GIH,ITU,STU   | 0.2631  |\n",
      "| rs376955159  |     GPC5     |       4       |   GIH,ITU,PJL,STU   | 0.2626  |\n",
      "| rs548449531  |     A2M      |       4       |   BEB,GIH,ITU,STU   | 0.2617  |\n",
      "| rs866039547  |    KALRN     |       4       |   BEB,GIH,ITU,STU   | 0.2608  |\n",
      "| rs376155209  |  LINC02182   |       4       |   BEB,GIH,ITU,STU   | 0.2604  |\n",
      "| rs375724699  |    KALRN     |       4       |   BEB,GIH,ITU,STU   | 0.2604  |\n",
      "| rs376371786  |      GC      |       4       |   BEB,GIH,ITU,STU   | 0.2599  |\n",
      "|  rs11985350  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.2597  |\n",
      "|  rs57501450  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.2596  |\n",
      "|  rs9637534   |     TFRC     |       4       |   BEB,GIH,ITU,STU   | 0.2586  |\n",
      "| rs369862903  |     GPC5     |       4       |   GIH,ITU,PJL,STU   | 0.2578  |\n",
      "| rs540220202  |    SLC9C2    |       4       |   BEB,GIH,ITU,STU   | 0.2537  |\n",
      "| rs879318071  |   ANKRD36    |       4       |   BEB,GIH,ITU,STU   | 0.2499  |\n",
      "| rs554190749  |     IL7      |       4       |   BEB,ITU,PJL,STU   | 0.2494  |\n",
      "| rs147206408  |   ARHGAP24   |       4       |   GIH,ITU,PJL,STU   |  0.249  |\n",
      "| rs373111827  |    SFMBT2    |       4       |   BEB,GIH,ITU,STU   | 0.2481  |\n",
      "| rs141529952  |    SYNPO     |       4       |   BEB,GIH,PJL,STU   | 0.2481  |\n",
      "| rs370184472  |    THSD7A    |       4       |   BEB,GIH,ITU,STU   |  0.247  |\n",
      "| rs148536312  |    RUNX2     |       4       |   BEB,GIH,ITU,STU   |  0.247  |\n",
      "|  rs17170857  |    ELMO1     |       4       |   BEB,GIH,ITU,STU   | 0.2468  |\n",
      "| rs369017753  |    CD79A     |       4       |   BEB,GIH,ITU,STU   | 0.2465  |\n",
      "| rs188038872  |   UNC93B5    |       4       |   BEB,GIH,ITU,STU   | 0.2463  |\n",
      "| rs369215180  |    PTPRO     |       4       |   BEB,GIH,ITU,STU   | 0.2452  |\n",
      "| rs192642877  |    SYNPO     |       4       |   BEB,GIH,PJL,STU   | 0.2442  |\n",
      "| rs370928127  |    IL2RA     |       4       |   GIH,ITU,PJL,STU   | 0.2438  |\n",
      "| rs367786935  |   UNC93B5    |       4       |   BEB,GIH,ITU,STU   | 0.2437  |\n",
      "| rs567160822  |     CMIP     |       4       |   BEB,GIH,ITU,STU   | 0.2433  |\n",
      "| rs530529968  |    TNFSF4    |       4       |   BEB,GIH,ITU,STU   | 0.2414  |\n",
      "| rs374139325  |     LPA      |       4       |   BEB,GIH,ITU,STU   | 0.2399  |\n",
      "|  rs74985287  |    MAGI2     |       4       |   BEB,GIH,ITU,STU   | 0.2397  |\n",
      "| rs371654344  |    SFMBT2    |       4       |   BEB,GIH,ITU,STU   | 0.2394  |\n",
      "| rs550202962  |    SLC9C2    |       4       |   BEB,GIH,ITU,STU   | 0.2388  |\n",
      "| rs568645141  |    NR3C2     |       4       |   BEB,ITU,PJL,STU   | 0.2384  |\n",
      "| rs533928911  |     A2M      |       4       |   BEB,GIH,ITU,STU   | 0.2376  |\n",
      "| rs375632917  |     A2M      |       4       |   BEB,GIH,ITU,STU   | 0.2376  |\n",
      "|  rs7728814   |   SLC6A18    |       4       |   BEB,GIH,ITU,STU   | 0.2374  |\n",
      "|  rs72491338  |   SLC6A18    |       4       |   BEB,GIH,ITU,STU   | 0.2374  |\n",
      "|  rs33945178  |   SLC6A18    |       4       |   BEB,GIH,ITU,STU   | 0.2374  |\n",
      "| rs374572100  |     LPA      |       4       |   BEB,GIH,ITU,STU   | 0.2354  |\n",
      "|  rs59710276  |    ANXA5     |       4       |   BEB,GIH,ITU,STU   | 0.2349  |\n",
      "| rs114137012  |    ANXA5     |       4       |   BEB,GIH,ITU,STU   | 0.2349  |\n",
      "| rs111567789  |    ANXA5     |       4       |   BEB,GIH,ITU,STU   | 0.2349  |\n",
      "| rs541494466  |    HMG20A    |       4       |   BEB,GIH,ITU,STU   | 0.2346  |\n",
      "| rs1369874225 |   SLC6A18    |       4       |   BEB,GIH,ITU,STU   | 0.2346  |\n",
      "| rs375217845  |    LTA4H     |       4       |   BEB,ITU,PJL,STU   | 0.2341  |\n",
      "| rs373236965  |   SLC6A18    |       4       |   BEB,GIH,ITU,STU   | 0.2334  |\n",
      "| rs116536415  |     GPC5     |       4       |   BEB,GIH,ITU,STU   | 0.2328  |\n",
      "|  rs75179490  |    MAGI2     |       4       |   BEB,ITU,PJL,STU   | 0.2314  |\n",
      "| rs369521653  |    PTPRO     |       4       |   BEB,GIH,ITU,STU   | 0.2307  |\n",
      "|  rs6579796   |    SYNPO     |       4       |   BEB,ITU,PJL,STU   | 0.2301  |\n",
      "| rs147984220  |    SYNPO     |       4       |   BEB,ITU,PJL,STU   | 0.2301  |\n",
      "| rs374067734  |     TFRC     |       4       |   BEB,GIH,ITU,STU   | 0.2299  |\n",
      "| rs374757622  |     GPC5     |       4       |   GIH,ITU,PJL,STU   | 0.2294  |\n",
      "| rs570445244  |    KALRN     |       4       |   BEB,GIH,ITU,STU   | 0.2286  |\n",
      "| rs551126527  |    MAGI2     |       4       |   BEB,GIH,ITU,STU   |  0.228  |\n",
      "| rs368884651  |     STAR     |       4       |   BEB,GIH,ITU,STU   | 0.2279  |\n",
      "| rs531760550  |    PTPRD     |       4       |   BEB,GIH,ITU,STU   | 0.2275  |\n",
      "| rs147359445  |    TRPC6     |       4       |   BEB,ITU,PJL,STU   | 0.2271  |\n",
      "| rs143112356  |    TRPC6     |       4       |   BEB,ITU,PJL,STU   | 0.2271  |\n",
      "| rs373649208  |    NUP93     |       4       |   BEB,GIH,ITU,STU   | 0.2266  |\n",
      "|  rs59634181  |   SLC6A18    |       4       |   BEB,GIH,ITU,STU   | 0.2265  |\n",
      "|  rs58136877  |   SLC6A18    |       4       |   BEB,GIH,ITU,STU   | 0.2265  |\n",
      "|  rs80078576  |    TRPC6     |       4       |   BEB,ITU,PJL,STU   | 0.2255  |\n",
      "|  rs7935364   |    TRPC6     |       4       |   BEB,ITU,PJL,STU   | 0.2255  |\n",
      "|  rs78775411  |    TRPC6     |       4       |   BEB,ITU,PJL,STU   | 0.2255  |\n",
      "|  rs78624923  |    TRPC6     |       4       |   BEB,ITU,PJL,STU   | 0.2255  |\n",
      "|  rs77467446  |    TRPC6     |       4       |   BEB,ITU,PJL,STU   | 0.2255  |\n",
      "|  rs61700184  |    TRPC6     |       4       |   BEB,ITU,PJL,STU   | 0.2255  |\n",
      "|  rs58720814  |    TRPC6     |       4       |   BEB,ITU,PJL,STU   | 0.2255  |\n",
      "|  rs58546739  |    TRPC6     |       4       |   BEB,ITU,PJL,STU   | 0.2255  |\n",
      "|  rs57222897  |    TRPC6     |       4       |   BEB,ITU,PJL,STU   | 0.2255  |\n",
      "| rs371140465  |    TRPC6     |       4       |   BEB,ITU,PJL,STU   | 0.2255  |\n",
      "| rs200562572  |    TRPC6     |       4       |   BEB,ITU,PJL,STU   | 0.2255  |\n",
      "| rs148583192  |    TRPC6     |       4       |   BEB,ITU,PJL,STU   | 0.2255  |\n",
      "| rs145836025  |    TRPC6     |       4       |   BEB,ITU,PJL,STU   | 0.2255  |\n",
      "| rs144529493  |    TRPC6     |       4       |   BEB,ITU,PJL,STU   | 0.2255  |\n",
      "|  rs7718167   |   SLC6A18    |       4       |   BEB,GIH,ITU,STU   | 0.2243  |\n",
      "| rs372454657  |   ADAMTS13   |       4       |   BEB,GIH,ITU,STU   | 0.2243  |\n",
      "| rs140237602  |     GPC5     |       4       |   BEB,GIH,ITU,STU   | 0.2242  |\n",
      "| rs548138267  |  LINC02531   |       4       |   BEB,GIH,ITU,STU   | 0.2234  |\n",
      "| rs370970803  |     MCAM     |       4       |   BEB,GIH,ITU,STU   | 0.2232  |\n",
      "| rs376552553  |     ZHX2     |       4       |   BEB,GIH,ITU,STU   | 0.2222  |\n",
      "|  rs7938237   |    TRPC6     |       4       |   BEB,ITU,PJL,STU   | 0.2221  |\n",
      "| rs1328767928 |     GPC5     |       4       |   BEB,GIH,ITU,STU   | 0.2221  |\n",
      "| rs114744588  |    TRPC6     |       4       |   BEB,ITU,PJL,STU   | 0.2221  |\n",
      "| rs114140359  |    TRPC6     |       4       |   BEB,ITU,PJL,STU   | 0.2221  |\n",
      "|  rs79628155  |    TRPC6     |       4       |   BEB,ITU,PJL,STU   | 0.2217  |\n",
      "| rs371937088  |    PODXL     |       4       |   BEB,GIH,ITU,STU   |  0.221  |\n",
      "| rs113899944  |    PTPRD     |       4       |   GIH,ITU,PJL,STU   | 0.2205  |\n",
      "| rs104895140  |     MEFV     |       4       |   BEB,ITU,PJL,STU   | 0.2188  |\n",
      "| rs373510303  |    PTPRD     |       4       |   BEB,GIH,ITU,STU   | 0.2182  |\n",
      "| rs373150972  |    PTPRD     |       4       |   BEB,GIH,ITU,STU   | 0.2182  |\n",
      "| rs574413252  |     CUBN     |       4       |   BEB,GIH,ITU,STU   | 0.2176  |\n",
      "|   rs352006   |     PF4      |       4       |   BEB,GIH,ITU,STU   | 0.2156  |\n",
      "|   rs352005   |     PF4      |       4       |   BEB,GIH,ITU,STU   | 0.2156  |\n",
      "|   rs352004   |     PF4      |       4       |   BEB,GIH,ITU,STU   | 0.2156  |\n",
      "|  rs75738259  |     TFRC     |       4       |   BEB,GIH,ITU,STU   | 0.2149  |\n",
      "| rs201188114  |    IL2RA     |       4       |   BEB,GIH,ITU,STU   | 0.2141  |\n",
      "|  rs77192918  |    PTPRD     |       4       |   BEB,GIH,ITU,STU   | 0.2124  |\n",
      "| rs367996244  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.2123  |\n",
      "|  rs79433112  |     TFRC     |       4       |   BEB,GIH,ITU,STU   | 0.2117  |\n",
      "| rs147096572  |     TFRC     |       4       |   BEB,GIH,ITU,STU   | 0.2117  |\n",
      "| rs143927887  |  LINC02664   |       4       |   BEB,GIH,ITU,PJL   | 0.2101  |\n",
      "|  rs77284806  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.2097  |\n",
      "| rs376766693  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.2097  |\n",
      "| rs376608109  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.2097  |\n",
      "| rs374884445  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.2097  |\n",
      "| rs374132698  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.2097  |\n",
      "| rs373627706  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.2097  |\n",
      "| rs372647700  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.2097  |\n",
      "| rs372196440  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.2097  |\n",
      "| rs370117393  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.2097  |\n",
      "| rs370116576  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.2097  |\n",
      "| rs369252050  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.2097  |\n",
      "| rs368665338  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.2097  |\n",
      "| rs368587977  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.2097  |\n",
      "|  rs34591797  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.2097  |\n",
      "| rs151255328  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.2097  |\n",
      "| rs113009638  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.2097  |\n",
      "| rs190234931  |     CUBN     |       4       |   BEB,GIH,ITU,STU   | 0.2096  |\n",
      "| rs143669336  |    TGFB1     |       4       |   BEB,GIH,ITU,STU   | 0.2087  |\n",
      "| rs373183078  |   TMEM131L   |       4       |   BEB,GIH,PJL,STU   | 0.2084  |\n",
      "| rs372584314  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.2077  |\n",
      "| rs541971231  |    KITLG     |       4       |   BEB,GIH,ITU,STU   | 0.2066  |\n",
      "|  rs11039413  |    NUP160    |       4       |   BEB,GIH,ITU,STU   | 0.2064  |\n",
      "|  rs74895246  |    IL2RA     |       4       |   BEB,GIH,ITU,STU   | 0.2056  |\n",
      "| rs542002526  |    HDAC2     |       4       |   BEB,ITU,PJL,STU   | 0.2056  |\n",
      "|  rs34164172  |    IL2RA     |       4       |   BEB,GIH,ITU,STU   | 0.2056  |\n",
      "|  rs79007970  |    MAGI2     |       4       |   BEB,GIH,ITU,STU   | 0.2051  |\n",
      "|  rs78236545  |    ELMO1     |       4       |   BEB,GIH,PJL,STU   |  0.205  |\n",
      "| rs186941477  |     VWF      |       4       |   BEB,GIH,ITU,STU   | 0.2047  |\n",
      "| rs149074070  |    KANK1     |       4       |   BEB,GIH,ITU,PJL   |  0.204  |\n",
      "| rs116977214  |     CUBN     |       4       |   BEB,GIH,PJL,STU   | 0.2039  |\n",
      "| rs372397897  |    KITLG     |       4       |   BEB,GIH,ITU,STU   | 0.2033  |\n",
      "|  rs17150146  |    MAGI2     |       4       |   BEB,GIH,ITU,STU   | 0.2016  |\n",
      "| rs558773759  |     ZHX2     |       4       |   GIH,ITU,PJL,STU   | 0.2011  |\n",
      "| rs374122505  |    PTPRD     |       4       |   BEB,GIH,ITU,STU   | 0.2007  |\n",
      "| rs374553604  |    CCDC66    |       4       |   BEB,GIH,ITU,STU   | 0.1999  |\n",
      "| rs372108513  |     MCAM     |       4       |   BEB,GIH,PJL,STU   |  0.199  |\n",
      "| rs570686873  |    AXDND1    |       4       |   BEB,GIH,ITU,STU   | 0.1981  |\n",
      "| rs375208451  |     PMM2     |       4       |   GIH,ITU,PJL,STU   | 0.1981  |\n",
      "| rs375553717  |    CD247     |       4       |   BEB,GIH,ITU,STU   | 0.1979  |\n",
      "| rs138384292  |    KANK1     |       4       |   BEB,GIH,ITU,PJL   | 0.1979  |\n",
      "|  rs9282640   |     CD80     |       4       |   BEB,GIH,ITU,STU   | 0.1978  |\n",
      "|  rs74592427  |     CD80     |       4       |   BEB,GIH,ITU,STU   | 0.1978  |\n",
      "|  rs74468571  |     CD80     |       4       |   BEB,GIH,ITU,STU   | 0.1978  |\n",
      "|   rs13532    |     CD80     |       4       |   BEB,GIH,ITU,STU   | 0.1978  |\n",
      "| rs113991263  |     CD80     |       4       |   BEB,GIH,ITU,STU   | 0.1978  |\n",
      "| rs111359458  |     CD80     |       4       |   BEB,GIH,ITU,STU   | 0.1978  |\n",
      "| rs543173662  |     LEPR     |       4       |   BEB,GIH,ITU,PJL   | 0.1977  |\n",
      "| rs373787313  |    PODXL     |       4       |   BEB,GIH,ITU,STU   | 0.1959  |\n",
      "|  rs8111982   |     LDLR     |       4       |   GIH,ITU,PJL,STU   | 0.1951  |\n",
      "|  rs74938451  |     LDLR     |       4       |   GIH,ITU,PJL,STU   | 0.1951  |\n",
      "|  rs74516607  |     LDLR     |       4       |   GIH,ITU,PJL,STU   | 0.1951  |\n",
      "| rs376440702  |     IL4R     |       4       |   GIH,ITU,PJL,STU   |  0.193  |\n",
      "|  rs75686372  |     BMP6     |       4       |   BEB,GIH,PJL,STU   | 0.1927  |\n",
      "|  rs74507307  |    MAGI2     |       4       |   BEB,GIH,ITU,STU   | 0.1922  |\n",
      "| rs368845261  |    DOCK8     |       4       |   BEB,GIH,ITU,STU   | 0.1916  |\n",
      "|  rs79538864  |    MAGI2     |       4       |   BEB,GIH,ITU,STU   | 0.1912  |\n",
      "|  rs79292329  |    MAGI2     |       4       |   BEB,GIH,ITU,STU   | 0.1912  |\n",
      "| rs371814972  |    PLA2R1    |       4       |   BEB,GIH,ITU,STU   | 0.1905  |\n",
      "| rs112400586  |    KANK1     |       4       |   BEB,GIH,ITU,PJL   | 0.1901  |\n",
      "| rs572598267  |    MFF-DT    |       4       |   BEB,GIH,ITU,STU   | 0.1897  |\n",
      "| rs534924657  |     LDLR     |       4       |   GIH,ITU,PJL,STU   | 0.1891  |\n",
      "| rs368980367  |     LDLR     |       4       |   GIH,ITU,PJL,STU   | 0.1891  |\n",
      "|  rs17248833  |     LDLR     |       4       |   GIH,ITU,PJL,STU   | 0.1891  |\n",
      "|  rs75645908  |     DLC1     |       4       |   BEB,GIH,ITU,STU   |  0.189  |\n",
      "|  rs60902636  |     DLC1     |       4       |   BEB,GIH,ITU,STU   |  0.189  |\n",
      "|  rs59354870  |     DLC1     |       4       |   BEB,GIH,ITU,STU   |  0.189  |\n",
      "| rs145355059  |     DLC1     |       4       |   BEB,GIH,ITU,STU   |  0.189  |\n",
      "| rs143994132  |     DLC1     |       4       |   BEB,GIH,ITU,STU   |  0.189  |\n",
      "| rs138799331  |     DLC1     |       4       |   BEB,GIH,ITU,STU   |  0.189  |\n",
      "| rs138138219  |     DLC1     |       4       |   BEB,GIH,ITU,STU   |  0.189  |\n",
      "| rs117839457  |     DLC1     |       4       |   BEB,GIH,ITU,STU   |  0.189  |\n",
      "| rs117056195  |     DLC1     |       4       |   BEB,GIH,ITU,STU   |  0.189  |\n",
      "| rs376953887  |    NR3C2     |       4       |   BEB,GIH,ITU,STU   | 0.1887  |\n",
      "|  rs73430547  |    DAAM2     |       4       |   BEB,GIH,ITU,STU   |  0.188  |\n",
      "|  rs57607132  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.1869  |\n",
      "| rs571873539  |    NR3C2     |       4       |   BEB,GIH,ITU,STU   | 0.1866  |\n",
      "| rs377284624  |   PIK3C2A    |       4       |   BEB,GIH,ITU,STU   | 0.1856  |\n",
      "|  rs75083675  |    SULF2     |       4       |   BEB,GIH,ITU,STU   | 0.1842  |\n",
      "| rs377236377  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.1826  |\n",
      "| rs370921205  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.1826  |\n",
      "| rs151117354  |     DLC1     |       4       |   BEB,GIH,ITU,STU   | 0.1826  |\n",
      "|  rs76295247  |    MAGI2     |       4       |   BEB,ITU,PJL,STU   | 0.1812  |\n",
      "| rs200484943  |    MAGI2     |       4       |   BEB,ITU,PJL,STU   | 0.1812  |\n",
      "|  rs11086221  |    SULF2     |       4       |   BEB,GIH,ITU,STU   | 0.1812  |\n",
      "| rs561425388  |    CLIC4     |       4       |   BEB,GIH,ITU,STU   | 0.1798  |\n",
      "| rs540802499  |    CLIC4     |       4       |   BEB,GIH,ITU,STU   | 0.1798  |\n",
      "| rs562308102  |     EXT1     |       4       |   BEB,ITU,PJL,STU   | 0.1784  |\n",
      "|  rs73430540  |    DAAM2     |       4       |   BEB,GIH,ITU,STU   | 0.1781  |\n",
      "| rs559058978  |  LINC02605   |       4       |   BEB,ITU,PJL,STU   | 0.1757  |\n",
      "| rs540935606  |  LINC02605   |       4       |   BEB,ITU,PJL,STU   | 0.1757  |\n",
      "| rs201412253  |     IL7      |       4       |   BEB,ITU,PJL,STU   | 0.1757  |\n",
      "| rs200030259  |     IL7      |       4       |   BEB,ITU,PJL,STU   | 0.1757  |\n",
      "| rs140461559  |    MAGI2     |       4       |   BEB,GIH,ITU,STU   | 0.1757  |\n",
      "| rs373443263  |    ABCA1     |       4       |   BEB,GIH,ITU,STU   | 0.1756  |\n",
      "| rs145023077  |    HLA-G     |       4       |   BEB,GIH,ITU,STU   | 0.1753  |\n",
      "| rs369156680  |     GPC5     |       4       |   BEB,GIH,ITU,STU   | 0.1751  |\n",
      "| rs199611102  |     DSE      |       4       |   BEB,GIH,ITU,PJL   | 0.1741  |\n",
      "|  rs11086222  |    SULF2     |       4       |   BEB,GIH,ITU,STU   |  0.173  |\n",
      "| rs553468484  |    SEMA3E    |       4       |   BEB,GIH,ITU,STU   | 0.1722  |\n",
      "| rs373995422  |    MAGI2     |       4       |   BEB,GIH,ITU,STU   | 0.1719  |\n",
      "| rs376451326  |     IL4R     |       4       |   BEB,GIH,ITU,PJL   | 0.1715  |\n",
      "| rs570690323  |   SLC6A18    |       4       |   BEB,GIH,PJL,STU   | 0.1699  |\n",
      "|  rs7951175   |    TRPC6     |       4       |   BEB,ITU,PJL,STU   |  0.169  |\n",
      "|  rs7949151   |    TRPC6     |       4       |   BEB,ITU,PJL,STU   |  0.169  |\n",
      "| rs201577601  |    TRPC6     |       4       |   BEB,ITU,PJL,STU   |  0.169  |\n",
      "| rs200133377  |    TRPC6     |       4       |   BEB,ITU,PJL,STU   |  0.169  |\n",
      "| rs200028321  |    TRPC6     |       4       |   BEB,ITU,PJL,STU   |  0.169  |\n",
      "| rs575430486  |    ABCB1     |       4       |   BEB,GIH,ITU,STU   | 0.1683  |\n",
      "| rs377498385  |     BRAF     |       4       |   BEB,GIH,ITU,STU   | 0.1651  |\n",
      "| rs372736519  |     CMIP     |       4       |   BEB,GIH,ITU,STU   |  0.161  |\n",
      "|  rs11170342  |    KRT18     |       3       |     BEB,ITU,STU     | 0.4947  |\n",
      "| rs1346725310 |   HLA-DQB1   |       3       |     GIH,ITU,PJL     | 0.4572  |\n",
      "|  rs9274093   |   HLA-DQB1   |       3       |     GIH,ITU,PJL     | 0.4428  |\n",
      "|  rs75823142  |    CREBBP    |       3       |     GIH,ITU,STU     | 0.4418  |\n",
      "|  rs9635541   |    CREBBP    |       3       |     GIH,ITU,STU     | 0.4109  |\n",
      "|  rs75520688  |    CREBBP    |       3       |     BEB,GIH,ITU     | 0.4071  |\n",
      "| rs1446957270 |   HLA-DQB1   |       3       |     GIH,ITU,PJL     | 0.3977  |\n",
      "| rs867811256  |     GCA      |       3       |     GIH,ITU,STU     | 0.3951  |\n",
      "|  rs76530479  |    CREBBP    |       3       |     GIH,ITU,STU     | 0.3847  |\n",
      "|   rs130051   |    CREBBP    |       3       |     GIH,ITU,STU     | 0.3847  |\n",
      "|  rs9931922   |    CREBBP    |       3       |     GIH,ITU,STU     | 0.3812  |\n",
      "|  rs76447783  |    CREBBP    |       3       |     GIH,ITU,STU     | 0.3812  |\n",
      "| rs149484322  |    CREBBP    |       3       |     GIH,ITU,STU     | 0.3789  |\n",
      "| rs138275226  |    DOCK8     |       3       |     BEB,GIH,ITU     | 0.3786  |\n",
      "|  rs17151016  |    MAGI2     |       3       |     BEB,GIH,PJL     | 0.3765  |\n",
      "| rs117966543  |   HLA-DQA1   |       3       |     GIH,ITU,PJL     | 0.3755  |\n",
      "|  rs79182296  |    CREBBP    |       3       |     GIH,ITU,STU     | 0.3753  |\n",
      "| rs201661149  |    MAGI2     |       3       |     BEB,GIH,PJL     | 0.3728  |\n",
      "|   rs129974   |    CREBBP    |       3       |     GIH,ITU,STU     | 0.3725  |\n",
      "|  rs80202046  |    CREBBP    |       3       |     GIH,ITU,STU     | 0.3723  |\n",
      "| rs147724013  |  LINC02182   |       3       |     BEB,GIH,ITU     | 0.3715  |\n",
      "| rs145898228  |  LINC02182   |       3       |     BEB,GIH,ITU     | 0.3689  |\n",
      "| rs138380147  |  LINC02182   |       3       |     BEB,GIH,ITU     | 0.3672  |\n",
      "|  rs78949699  |    CREBBP    |       3       |     GIH,ITU,STU     | 0.3671  |\n",
      "|  rs7845684   |    CLVS1     |       3       |     BEB,GIH,ITU     | 0.3572  |\n",
      "|  rs7832496   |    CLVS1     |       3       |     BEB,GIH,ITU     | 0.3572  |\n",
      "|  rs6991850   |    CLVS1     |       3       |     BEB,GIH,ITU     | 0.3572  |\n",
      "|  rs57878945  |    CLVS1     |       3       |     BEB,GIH,ITU     | 0.3572  |\n",
      "|  rs55974482  |    CLVS1     |       3       |     BEB,GIH,ITU     | 0.3572  |\n",
      "|  rs16927178  |    CLVS1     |       3       |     BEB,GIH,ITU     | 0.3572  |\n",
      "|  rs10429447  |    CLVS1     |       3       |     BEB,GIH,ITU     | 0.3572  |\n",
      "|  rs76768539  |    MAGI2     |       3       |     BEB,PJL,STU     | 0.3505  |\n",
      "| rs1554559744 |    CLVS1     |       3       |     BEB,GIH,ITU     | 0.3393  |\n",
      "|  rs10104760  |    CLVS1     |       3       |     BEB,GIH,ITU     | 0.3393  |\n",
      "|  rs10089827  |    CLVS1     |       3       |     BEB,GIH,ITU     | 0.3393  |\n",
      "|  rs10958971  |    PTPRD     |       3       |     BEB,ITU,STU     |  0.304  |\n",
      "|  rs2925998   |     CMIP     |       3       |     BEB,ITU,STU     |  0.299  |\n",
      "|  rs77072651  |     WNK4     |       3       |     BEB,GIH,STU     | 0.2986  |\n",
      "| rs113105976  |     WNK4     |       3       |     BEB,GIH,STU     | 0.2986  |\n",
      "|  rs79877866  |    PTPRD     |       3       |     BEB,ITU,STU     | 0.2941  |\n",
      "|  rs78513083  |    PTPRD     |       3       |     BEB,ITU,STU     | 0.2941  |\n",
      "|  rs12686059  |    PTPRD     |       3       |     BEB,ITU,STU     | 0.2941  |\n",
      "|  rs12685598  |    PTPRD     |       3       |     BEB,ITU,STU     | 0.2941  |\n",
      "|  rs12683930  |    PTPRD     |       3       |     BEB,ITU,STU     | 0.2941  |\n",
      "|  rs12683206  |    PTPRD     |       3       |     BEB,ITU,STU     | 0.2941  |\n",
      "|  rs4142336   |    ELMO1     |       3       |     BEB,ITU,STU     |  0.294  |\n",
      "| rs147525031  |  LINC02740   |       3       |     BEB,ITU,PJL     | 0.2925  |\n",
      "| rs138220451  |  LINC02740   |       3       |     BEB,ITU,PJL     | 0.2925  |\n",
      "|  rs74366457  |    PTPRD     |       3       |     BEB,ITU,STU     | 0.2917  |\n",
      "|  rs1036276   |    PTPRD     |       3       |     BEB,ITU,STU     | 0.2917  |\n",
      "|  rs76175827  |    ELMO1     |       3       |     BEB,ITU,STU     | 0.2915  |\n",
      "|  rs59672764  |    ELMO1     |       3       |     BEB,ITU,STU     | 0.2915  |\n",
      "| rs116761890  |    ELMO1     |       3       |     BEB,ITU,STU     | 0.2915  |\n",
      "| rs367677005  |  LINC02740   |       3       |     BEB,ITU,PJL     | 0.2903  |\n",
      "|  rs78777018  |    PTPRD     |       3       |     BEB,ITU,STU     | 0.2892  |\n",
      "| rs146460879  |  LINC02740   |       3       |     BEB,ITU,PJL     | 0.2892  |\n",
      "| rs142972372  |  LINC02740   |       3       |     BEB,ITU,PJL     | 0.2892  |\n",
      "| rs114153666  |  LINC02740   |       3       |     BEB,ITU,PJL     | 0.2892  |\n",
      "|  rs7105287   |    TRPC6     |       3       |     BEB,ITU,PJL     | 0.2864  |\n",
      "|  rs76613339  |    TRPC6     |       3       |     BEB,ITU,PJL     | 0.2843  |\n",
      "|   rs476553   |     DLC1     |       3       |     BEB,ITU,STU     | 0.2843  |\n",
      "|  rs12275495  |    TRPC6     |       3       |     BEB,ITU,PJL     | 0.2843  |\n",
      "|  rs1662078   |     DLC1     |       3       |     BEB,ITU,STU     | 0.2813  |\n",
      "| rs113578431  |    TRPC6     |       3       |     ITU,PJL,STU     | 0.2794  |\n",
      "| rs571662262  |     VWF      |       3       |     BEB,GIH,STU     | 0.2756  |\n",
      "|  rs75459238  |  LINC02182   |       3       |     GIH,ITU,STU     | 0.2739  |\n",
      "|  rs77656105  |    NUP85     |       3       |     BEB,PJL,STU     | 0.2723  |\n",
      "| rs146201831  |    NUP85     |       3       |     BEB,PJL,STU     | 0.2723  |\n",
      "|  rs75668023  |  LINC02182   |       3       |     GIH,ITU,STU     | 0.2707  |\n",
      "| rs141180606  |     ACE      |       3       |     BEB,GIH,ITU     | 0.2705  |\n",
      "|  rs10400967  |  LINC02182   |       3       |     GIH,ITU,STU     | 0.2705  |\n",
      "|  rs78600753  |  LINC02182   |       3       |     GIH,ITU,STU     | 0.2675  |\n",
      "| rs1568000166 |  LINC02182   |       3       |     GIH,ITU,STU     | 0.2675  |\n",
      "|  rs10400968  |  LINC02182   |       3       |     GIH,ITU,STU     | 0.2675  |\n",
      "|  rs7039402   |    DOCK8     |       3       |     GIH,ITU,STU     | 0.2648  |\n",
      "|  rs60587915  |    DOCK8     |       3       |     GIH,ITU,STU     | 0.2648  |\n",
      "|  rs75953759  |  LINC02182   |       3       |     GIH,ITU,STU     | 0.2638  |\n",
      "| rs376234050  |  LINC02182   |       3       |     GIH,ITU,STU     | 0.2561  |\n",
      "| rs373271645  |   KIRREL3    |       3       |     BEB,ITU,STU     | 0.2542  |\n",
      "|  rs6676704   |    NOS1AP    |       3       |     BEB,ITU,STU     | 0.2515  |\n",
      "| rs568413368  |    CREBBP    |       3       |     BEB,GIH,ITU     | 0.2502  |\n",
      "| rs201953731  |    WDR73     |       3       |     GIH,ITU,STU     | 0.2501  |\n",
      "|  rs75323021  |      TF      |       3       |     ITU,PJL,STU     | 0.2495  |\n",
      "|   rs527601   |     DLC1     |       3       |     BEB,ITU,STU     | 0.2491  |\n",
      "| rs552487824  |   SLC6A18    |       3       |     BEB,ITU,STU     | 0.2487  |\n",
      "| rs183186926  |     DSE      |       3       |     GIH,PJL,STU     | 0.2481  |\n",
      "| rs377566318  |     DCN      |       3       |     GIH,ITU,STU     | 0.2478  |\n",
      "| rs562787663  |    PTPRD     |       3       |     BEB,ITU,STU     | 0.2468  |\n",
      "| rs568510038  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2462  |\n",
      "|   rs882284   |    NOS1AP    |       3       |     BEB,ITU,STU     | 0.2431  |\n",
      "|  rs79076222  |    NOS1AP    |       3       |     BEB,ITU,STU     | 0.2431  |\n",
      "|  rs7555808   |    NOS1AP    |       3       |     BEB,ITU,STU     | 0.2431  |\n",
      "|  rs74933662  |    NOS1AP    |       3       |     BEB,ITU,STU     | 0.2431  |\n",
      "| rs138578097  |    NOS1AP    |       3       |     BEB,ITU,STU     | 0.2431  |\n",
      "| rs115292973  |    NOS1AP    |       3       |     BEB,ITU,STU     | 0.2431  |\n",
      "|  rs78716699  |    NOS1AP    |       3       |     BEB,ITU,STU     | 0.2392  |\n",
      "|  rs76362438  |    NOS1AP    |       3       |     BEB,ITU,STU     | 0.2392  |\n",
      "|  rs74803915  |    NOS1AP    |       3       |     BEB,ITU,STU     | 0.2392  |\n",
      "|  rs6427669   |    NOS1AP    |       3       |     BEB,ITU,STU     | 0.2392  |\n",
      "| rs116185888  |    NOS1AP    |       3       |     BEB,ITU,STU     | 0.2392  |\n",
      "| rs114828566  |    NOS1AP    |       3       |     BEB,ITU,STU     | 0.2388  |\n",
      "| rs116709803  |   HSD11B1    |       3       |     BEB,GIH,ITU     | 0.2371  |\n",
      "| rs1221552085 |    PLA2R1    |       3       |     BEB,GIH,STU     | 0.2357  |\n",
      "| rs368221767  |    SEMA3E    |       3       |     GIH,PJL,STU     | 0.2349  |\n",
      "| rs367689139  |    SGPL1     |       3       |     BEB,GIH,STU     | 0.2344  |\n",
      "| rs147807356  |  LINC02188   |       3       |     ITU,PJL,STU     |  0.232  |\n",
      "| rs371366595  |  LINC03056   |       3       |     BEB,ITU,STU     | 0.2297  |\n",
      "| rs144379626  |    DELEC1    |       3       |     BEB,ITU,STU     | 0.2295  |\n",
      "|  rs7261563   |    SULF2     |       3       |     GIH,ITU,STU     | 0.2294  |\n",
      "| rs370324232  |    SGPL1     |       3       |     BEB,GIH,STU     | 0.2285  |\n",
      "| rs567898887  |   KIRREL3    |       3       |     BEB,ITU,STU     | 0.2274  |\n",
      "| rs188103793  |     JEB7     |       3       |     GIH,ITU,STU     | 0.2274  |\n",
      "| rs577694601  |     EGFR     |       3       |     GIH,ITU,STU     | 0.2269  |\n",
      "| rs369742244  |    GSDMB     |       3       |     GIH,ITU,STU     | 0.2253  |\n",
      "| rs546172207  |     CUBN     |       3       |     GIH,ITU,STU     |  0.225  |\n",
      "| rs370446490  |    ELMO1     |       3       |     GIH,ITU,STU     | 0.2244  |\n",
      "| rs572992755  |  TRIM31-AS1  |       3       |     ITU,PJL,STU     | 0.2228  |\n",
      "| rs139893876  |     ATF1     |       3       |     GIH,ITU,STU     | 0.2222  |\n",
      "|  rs78738080  |    TRPC6     |       3       |     BEB,PJL,STU     | 0.2216  |\n",
      "|  rs7530938   |    FCGR2A    |       3       |     GIH,ITU,PJL     | 0.2213  |\n",
      "| rs375853017  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2211  |\n",
      "| rs370224805  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2211  |\n",
      "| rs368681742  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2211  |\n",
      "| rs367837710  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2211  |\n",
      "| rs376447550  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2208  |\n",
      "| rs573997147  |     LEP      |       3       |     BEB,GIH,ITU     | 0.2201  |\n",
      "| rs572146155  |     LEP      |       3       |     BEB,GIH,ITU     | 0.2201  |\n",
      "| rs571251915  |     LEP      |       3       |     BEB,GIH,ITU     | 0.2201  |\n",
      "| rs562645794  |     LEP      |       3       |     BEB,GIH,ITU     | 0.2201  |\n",
      "| rs554048528  |     LEP      |       3       |     BEB,GIH,ITU     | 0.2201  |\n",
      "| rs549755515  |     LEP      |       3       |     BEB,GIH,ITU     | 0.2201  |\n",
      "| rs549666685  |     LEP      |       3       |     BEB,GIH,ITU     | 0.2201  |\n",
      "| rs545511953  |     LEP      |       3       |     BEB,GIH,ITU     | 0.2201  |\n",
      "| rs544172584  |     LEP      |       3       |     BEB,GIH,ITU     | 0.2201  |\n",
      "| rs538843747  |     LEP      |       3       |     BEB,GIH,ITU     | 0.2201  |\n",
      "| rs533213771  |     LEP      |       3       |     BEB,GIH,ITU     | 0.2201  |\n",
      "| rs200893300  |     LEP      |       3       |     BEB,GIH,ITU     | 0.2201  |\n",
      "| rs369737779  |     CMIP     |       3       |     GIH,ITU,STU     | 0.2197  |\n",
      "| rs376238443  |    FRG1JP    |       3       |     BEB,ITU,STU     | 0.2192  |\n",
      "| rs543813876  |   CCDC162P   |       3       |     GIH,ITU,STU     |  0.219  |\n",
      "| rs200158301  |    PCSK9     |       3       |     BEB,ITU,STU     | 0.2187  |\n",
      "|  rs12022044  |    CD247     |       3       |     BEB,ITU,STU     | 0.2185  |\n",
      "| rs531749798  |    PLCG2     |       3       |     GIH,ITU,STU     | 0.2179  |\n",
      "| rs368276401  |   TRAPPC3L   |       3       |     GIH,ITU,STU     | 0.2178  |\n",
      "| rs376362092  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2177  |\n",
      "| rs375758995  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2177  |\n",
      "| rs375086594  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2177  |\n",
      "| rs373703177  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2177  |\n",
      "| rs372695352  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2177  |\n",
      "| rs372664692  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2177  |\n",
      "| rs372196480  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2177  |\n",
      "| rs369687484  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2177  |\n",
      "| rs368719849  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2177  |\n",
      "| rs182119604  |     DSE      |       3       |     GIH,PJL,STU     | 0.2177  |\n",
      "| rs114004357  |   HNRNPKP3   |       3       |     BEB,GIH,STU     | 0.2172  |\n",
      "|  rs74328720  |    SULF2     |       3       |     GIH,ITU,STU     | 0.2169  |\n",
      "|  rs6122605   |    SULF2     |       3       |     GIH,ITU,STU     | 0.2169  |\n",
      "| rs372952994  |    PTPRD     |       3       |     BEB,GIH,ITU     | 0.2166  |\n",
      "| rs370755691  |    FRG1JP    |       3       |     BEB,ITU,STU     | 0.2156  |\n",
      "| rs185510462  |    IGFBP2    |       3       |     BEB,PJL,STU     | 0.2155  |\n",
      "| rs573756843  |    LIMA1     |       3       |     BEB,GIH,ITU     | 0.2145  |\n",
      "|  rs34927999  |   ANGPTL3    |       3       |     ITU,PJL,STU     | 0.2145  |\n",
      "|    rs266     |     LPL      |       3       |     BEB,GIH,ITU     | 0.2136  |\n",
      "|  rs4134948   |     E2F3     |       3       |     BEB,ITU,STU     | 0.2135  |\n",
      "| rs1566625573 |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2135  |\n",
      "| rs1314239742 |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2135  |\n",
      "| rs569910813  |    ARID3B    |       3       |     GIH,ITU,STU     | 0.2133  |\n",
      "| rs368421382  |    DOCK8     |       3       |     GIH,ITU,STU     | 0.2133  |\n",
      "| rs115588976  |     NEU1     |       3       |     GIH,ITU,PJL     | 0.2121  |\n",
      "| rs186132504  |    NLRP3     |       3       |     BEB,ITU,STU     |  0.212  |\n",
      "| rs373553459  |     KC6      |       3       |     ITU,PJL,STU     | 0.2115  |\n",
      "| rs370308910  |   ANKRD36    |       3       |     BEB,GIH,STU     | 0.2112  |\n",
      "| rs573133118  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2103  |\n",
      "| rs530964991  |     PAX2     |       3       |     GIH,ITU,STU     | 0.2103  |\n",
      "| rs138545464  |    UCHL1     |       3       |     GIH,ITU,STU     | 0.2103  |\n",
      "| rs370526413  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2102  |\n",
      "| rs374639540  |      GC      |       3       |     BEB,ITU,STU     | 0.2095  |\n",
      "| rs1471792074 |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2095  |\n",
      "| rs1210865577 |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2095  |\n",
      "| rs562914347  |     GPC5     |       3       |     ITU,PJL,STU     | 0.2089  |\n",
      "| rs562301550  |     DGKE     |       3       |     BEB,GIH,ITU     | 0.2084  |\n",
      "| rs528568279  |     PDCN     |       3       |     BEB,ITU,STU     | 0.2079  |\n",
      "| rs376505382  |   TMEM131L   |       3       |     GIH,ITU,STU     | 0.2073  |\n",
      "|  rs8104576   |     LDLR     |       3       |     GIH,ITU,STU     | 0.2064  |\n",
      "|  rs8104549   |     LDLR     |       3       |     GIH,ITU,STU     | 0.2064  |\n",
      "|  rs17242766  |     LDLR     |       3       |     GIH,ITU,STU     | 0.2064  |\n",
      "| rs560132284  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2056  |\n",
      "| rs192615582  |   TMEM131L   |       3       |     GIH,ITU,STU     | 0.2045  |\n",
      "|  rs66894954  |     CD4      |       3       |     BEB,ITU,STU     | 0.2041  |\n",
      "| rs557576978  |     MC1R     |       3       |     GIH,ITU,STU     | 0.2024  |\n",
      "| rs376303978  |  LINC02182   |       3       |     GIH,ITU,STU     | 0.2024  |\n",
      "| rs558727932  |     DSE      |       3       |     BEB,GIH,ITU     | 0.2021  |\n",
      "| rs377403246  |    DOCK8     |       3       |     GIH,ITU,STU     |  0.202  |\n",
      "| rs377102697  |     CETP     |       3       |     GIH,ITU,STU     |  0.202  |\n",
      "| rs372850206  |    DOCK8     |       3       |     GIH,ITU,STU     |  0.202  |\n",
      "|    rs251     |     LPL      |       3       |     BEB,GIH,ITU     |  0.202  |\n",
      "| rs879699766  |     LDLR     |       3       |     GIH,ITU,STU     | 0.2017  |\n",
      "| rs543947886  |     LDLR     |       3       |     GIH,ITU,STU     | 0.2017  |\n",
      "| rs200312084  |     LDLR     |       3       |     GIH,ITU,STU     | 0.2017  |\n",
      "| rs192678460  |     LDLR     |       3       |     GIH,ITU,STU     | 0.2017  |\n",
      "| rs188618890  |    NPHP4     |       3       |     GIH,ITU,STU     | 0.2017  |\n",
      "|  rs17249330  |     LDLR     |       3       |     GIH,ITU,STU     | 0.2017  |\n",
      "| rs114085323  |     LDLR     |       3       |     GIH,ITU,STU     | 0.2017  |\n",
      "|  rs1557546   |     DSE      |       3       |     GIH,PJL,STU     | 0.2016  |\n",
      "| rs374612708  |   TMEM131L   |       3       |     GIH,ITU,STU     | 0.2009  |\n",
      "| rs375451552  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2005  |\n",
      "| rs374017038  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2005  |\n",
      "| rs372929829  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2005  |\n",
      "| rs368918950  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.2005  |\n",
      "|  rs77155065  |    SULF2     |       3       |     GIH,ITU,STU     |   0.2   |\n",
      "| rs540781595  |   KIRREL3    |       3       |     GIH,ITU,PJL     | 0.1998  |\n",
      "| rs369635755  |    ANXA5     |       3       |     GIH,ITU,STU     | 0.1995  |\n",
      "| rs372697006  |     FAT1     |       3       |     ITU,PJL,STU     | 0.1987  |\n",
      "| rs145184030  |   KIRREL3    |       3       |     BEB,ITU,STU     | 0.1982  |\n",
      "| rs375135782  |     LDLR     |       3       |     GIH,ITU,STU     | 0.1981  |\n",
      "|  rs17242962  |     LDLR     |       3       |     GIH,ITU,STU     | 0.1981  |\n",
      "| rs150781410  |     LDLR     |       3       |     GIH,ITU,STU     | 0.1981  |\n",
      "| rs145104510  |     LDLR     |       3       |     GIH,ITU,STU     | 0.1981  |\n",
      "| rs140478111  |     LDLR     |       3       |     GIH,ITU,STU     | 0.1981  |\n",
      "| rs373072671  |     EGFR     |       3       |     GIH,ITU,STU     | 0.1979  |\n",
      "| rs148377741  |   TMEM131L   |       3       |     GIH,ITU,STU     | 0.1975  |\n",
      "| rs370508020  |     LIPC     |       3       |     BEB,ITU,STU     | 0.1964  |\n",
      "| rs377383645  |    ELMO1     |       3       |     BEB,ITU,STU     | 0.1961  |\n",
      "| rs372770208  |    ELMO1     |       3       |     BEB,ITU,STU     | 0.1961  |\n",
      "| rs372591832  |    MAGI2     |       3       |     BEB,ITU,STU     | 0.1961  |\n",
      "| rs563999282  |    LTA4H     |       3       |     GIH,ITU,STU     | 0.1957  |\n",
      "| rs551773925  |    LTA4H     |       3       |     GIH,ITU,STU     | 0.1957  |\n",
      "| rs532280496  |    LTA4H     |       3       |     GIH,ITU,STU     | 0.1957  |\n",
      "| rs150378663  |    MAGI2     |       3       |     GIH,ITU,STU     | 0.1949  |\n",
      "| rs139039475  |    MAGI2     |       3       |     GIH,ITU,STU     | 0.1949  |\n",
      "| rs539867326  |  LINC02182   |       3       |     GIH,ITU,STU     | 0.1946  |\n",
      "| rs368234869  |     GPC5     |       3       |     GIH,ITU,STU     | 0.1942  |\n",
      "|  rs8182748   |    SULF2     |       3       |     GIH,ITU,STU     | 0.1941  |\n",
      "|  rs6125080   |    SULF2     |       3       |     GIH,ITU,STU     | 0.1941  |\n",
      "| rs570121787  |     DLC1     |       3       |     BEB,PJL,STU     | 0.1933  |\n",
      "| rs376984675  |     AQP2     |       3       |     GIH,ITU,STU     |  0.193  |\n",
      "| rs377141403  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.1929  |\n",
      "| rs374728233  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.1929  |\n",
      "| rs369652814  |    KALRN     |       3       |     BEB,ITU,STU     | 0.1929  |\n",
      "| rs1262926711 |     LIPC     |       3       |     BEB,ITU,STU     | 0.1928  |\n",
      "| rs560129547  |    NR3C2     |       3       |     BEB,ITU,STU     | 0.1924  |\n",
      "| rs550247422  |    CCDC66    |       3       |     BEB,GIH,PJL     |  0.192  |\n",
      "|  rs13432480  |     NPPC     |       3       |     GIH,ITU,STU     | 0.1909  |\n",
      "|  rs75210409  |     DLC1     |       3       |     BEB,ITU,STU     | 0.1897  |\n",
      "|  rs58463115  |     DLC1     |       3       |     BEB,ITU,STU     | 0.1897  |\n",
      "| rs200539632  |     DLC1     |       3       |     BEB,ITU,STU     | 0.1897  |\n",
      "| rs185448330  |     DLC1     |       3       |     BEB,ITU,STU     | 0.1897  |\n",
      "| rs182655435  |     DLC1     |       3       |     BEB,ITU,STU     | 0.1897  |\n",
      "| rs182299274  |     DLC1     |       3       |     BEB,ITU,STU     | 0.1897  |\n",
      "| rs180708245  |     DLC1     |       3       |     BEB,ITU,STU     | 0.1897  |\n",
      "| rs139831921  |     DLC1     |       3       |     BEB,ITU,STU     | 0.1897  |\n",
      "|  rs11996047  |     DLC1     |       3       |     BEB,ITU,STU     | 0.1897  |\n",
      "| rs118114297  |     DLC1     |       3       |     BEB,ITU,STU     | 0.1897  |\n",
      "| rs118042050  |     DLC1     |       3       |     BEB,ITU,STU     | 0.1897  |\n",
      "| rs117582544  |     DLC1     |       3       |     BEB,ITU,STU     | 0.1897  |\n",
      "| rs374784833  |    PTPRO     |       3       |     BEB,ITU,STU     | 0.1896  |\n",
      "| rs572186824  |     KRT8     |       3       |     BEB,ITU,STU     | 0.1895  |\n",
      "| rs373331730  |    CD247     |       3       |     GIH,ITU,STU     |  0.189  |\n",
      "| rs370778480  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.1887  |\n",
      "| rs370420088  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.1887  |\n",
      "|  rs75668813  |  TSBP1-AS1   |       3       |     BEB,PJL,STU     | 0.1877  |\n",
      "|  rs74464187  |  TSBP1-AS1   |       3       |     BEB,PJL,STU     | 0.1877  |\n",
      "| rs146011548  |     RBP4     |       3       |     BEB,PJL,STU     | 0.1873  |\n",
      "| rs527372078  |    NR3C2     |       3       |     GIH,ITU,STU     |  0.187  |\n",
      "| rs377752759  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.1868  |\n",
      "|  rs6807408   |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "|  rs6768855   |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs575379372  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs574783352  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs574529253  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs572187034  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs570557391  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs570381866  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs570319103  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs568477744  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs567145483  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs566191677  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs564660313  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs563304976  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs563159726  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs562853369  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs560820090  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs558425203  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs556903159  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs555045710  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs550588674  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs545984034  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs537896390  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs534836207  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs534538641  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs533469104  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs532301897  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs528684191  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs202139142  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs200227044  |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "| rs1482592820 |    CCDC66    |       3       |     BEB,GIH,PJL     | 0.1867  |\n",
      "|  rs77032411  |     DLC1     |       3       |     BEB,ITU,STU     | 0.1862  |\n",
      "|  rs76591447  |     DLC1     |       3       |     BEB,ITU,STU     | 0.1862  |\n",
      "|  rs60839480  |     DLC1     |       3       |     BEB,ITU,STU     | 0.1862  |\n",
      "| rs563822584  |    PLCG2     |       3       |     GIH,ITU,STU     | 0.1861  |\n",
      "| rs368977447  |   ADAMTS13   |       3       |     BEB,ITU,STU     | 0.1856  |\n",
      "|  rs78834018  |  TSBP1-AS1   |       3       |     BEB,PJL,STU     | 0.1852  |\n",
      "|  rs74714696  |  TSBP1-AS1   |       3       |     BEB,PJL,STU     | 0.1852  |\n",
      "| rs375972297  |  TSBP1-AS1   |       3       |     BEB,PJL,STU     | 0.1852  |\n",
      "| rs145968068  |  TSBP1-AS1   |       3       |     BEB,PJL,STU     | 0.1852  |\n",
      "| rs574680317  |    PTPRD     |       3       |     GIH,PJL,STU     | 0.1851  |\n",
      "| rs373872331  |    MAGI2     |       3       |     BEB,ITU,STU     | 0.1848  |\n",
      "|  rs74463718  |    PTPRD     |       3       |     BEB,GIH,STU     |  0.184  |\n",
      "| rs373581252  |     TJP1     |       3       |     GIH,ITU,STU     | 0.1826  |\n",
      "| rs372878780  |    NR3C2     |       3       |     GIH,ITU,STU     | 0.1826  |\n",
      "| rs374488702  |    MAGI2     |       3       |     BEB,GIH,STU     | 0.1823  |\n",
      "| rs533233868  |     EXT1     |       3       |     BEB,ITU,STU     | 0.1818  |\n",
      "| rs369378776  |     GPC5     |       3       |     GIH,ITU,STU     | 0.1817  |\n",
      "|  rs76945503  |    SULF2     |       3       |     GIH,ITU,STU     | 0.1815  |\n",
      "|  rs76653185  |    SULF2     |       3       |     GIH,ITU,STU     | 0.1812  |\n",
      "| rs554978716  |    PTPRO     |       3       |     GIH,ITU,STU     | 0.1811  |\n",
      "|  rs76697536  |    MAGI2     |       3       |     BEB,ITU,STU     | 0.1809  |\n",
      "|  rs76094414  |    PTPRD     |       3       |     BEB,GIH,STU     | 0.1804  |\n",
      "| rs563560543  |    RRP12     |       3       |     BEB,ITU,STU     | 0.1804  |\n",
      "| rs201414377  |    PTPRD     |       3       |     BEB,GIH,STU     | 0.1804  |\n",
      "| rs148183971  |    PTPRD     |       3       |     BEB,GIH,STU     | 0.1804  |\n",
      "| rs147753642  |    PTPRD     |       3       |     BEB,GIH,STU     | 0.1804  |\n",
      "| rs1434377732 |     RAC1     |       3       |     ITU,PJL,STU     | 0.1802  |\n",
      "| rs374534163  |     PLG      |       3       |     BEB,ITU,PJL     | 0.1793  |\n",
      "| rs371303737  |    NR3C2     |       3       |     BEB,GIH,STU     | 0.1787  |\n",
      "| rs374525376  |     PAX2     |       3       |     BEB,GIH,ITU     |  0.178  |\n",
      "|  rs80113980  |    MAGI2     |       3       |     BEB,ITU,STU     | 0.1774  |\n",
      "|  rs78883177  |    MAGI2     |       3       |     BEB,ITU,STU     | 0.1774  |\n",
      "|  rs76093776  |    MAGI2     |       3       |     BEB,ITU,STU     | 0.1774  |\n",
      "| rs150297684  |    MAGI2     |       3       |     BEB,ITU,STU     | 0.1774  |\n",
      "| rs538602645  |     LPA      |       3       |     BEB,ITU,PJL     | 0.1772  |\n",
      "|  rs73644466  |    PTPRD     |       3       |     BEB,GIH,STU     | 0.1768  |\n",
      "| rs541636970  |    ELMO1     |       3       |     BEB,ITU,STU     | 0.1761  |\n",
      "|  rs7951375   |    TRPC6     |       3       |     ITU,PJL,STU     | 0.1758  |\n",
      "| rs373652743  |    CCDC66    |       3       |     BEB,GIH,STU     | 0.1757  |\n",
      "| rs372407587  |     KDR      |       3       |     BEB,PJL,STU     |  0.175  |\n",
      "| rs375165166  |    ELMO1     |       3       |     BEB,ITU,STU     | 0.1749  |\n",
      "| rs146036418  |     DLC1     |       3       |     BEB,ITU,STU     | 0.1749  |\n",
      "| rs374932322  |    GAPVD1    |       3       |     GIH,ITU,STU     | 0.1742  |\n",
      "| rs367637143  |    GAPVD1    |       3       |     GIH,ITU,STU     | 0.1742  |\n",
      "|  rs77820233  |    MAGI2     |       3       |     BEB,GIH,STU     | 0.1741  |\n",
      "| rs150407759  |    CLIC4     |       3       |     BEB,ITU,PJL     | 0.1732  |\n",
      "|  rs12036492  |    CLIC4     |       3       |     BEB,ITU,PJL     | 0.1732  |\n",
      "| rs562945491  |   MICA-AS1   |       3       |     BEB,ITU,STU     | 0.1731  |\n",
      "| rs371551614  |    MAPK14    |       3       |     BEB,ITU,STU     | 0.1731  |\n",
      "| rs367685759  |     TNXB     |       3       |     GIH,ITU,PJL     | 0.1731  |\n",
      "| rs367917073  |    MAPK14    |       3       |     BEB,ITU,STU     | 0.1729  |\n",
      "|  rs12567076  |     AGT      |       3       |     BEB,GIH,STU     | 0.1707  |\n",
      "| rs371400031  |     LPA      |       3       |     BEB,PJL,STU     | 0.1704  |\n",
      "| rs568841423  |     CMIP     |       3       |     GIH,PJL,STU     | 0.1703  |\n",
      "|  rs77478433  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.1697  |\n",
      "| rs576160450  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.1697  |\n",
      "| rs571708212  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.1697  |\n",
      "| rs557071701  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.1697  |\n",
      "| rs532719110  |    PTPRD     |       3       |     GIH,ITU,STU     | 0.1697  |\n",
      "| rs528845040  |    PSMD14    |       3       |     GIH,ITU,STU     | 0.1697  |\n",
      "| rs540736678  |    MAPK14    |       3       |     BEB,ITU,STU     | 0.1691  |\n",
      "| rs312262682  |     AGT      |       3       |     BEB,ITU,STU     |  0.169  |\n",
      "| rs148373970  |    TRPC6     |       3       |     ITU,PJL,STU     | 0.1687  |\n",
      "| rs141339822  |    TRPC6     |       3       |     ITU,PJL,STU     | 0.1687  |\n",
      "| rs116038364  |    TRPC6     |       3       |     ITU,PJL,STU     | 0.1687  |\n",
      "|  rs73430510  |    DAAM2     |       3       |     BEB,GIH,STU     | 0.1686  |\n",
      "| rs561187919  |    PTPRD     |       3       |     BEB,GIH,STU     |  0.167  |\n",
      "|  rs2159367   |     VWF      |       3       |     BEB,ITU,STU     | 0.1666  |\n",
      "|  rs77042093  |     DLC1     |       3       |     BEB,ITU,STU     | 0.1662  |\n",
      "| rs557374550  |    PTPRO     |       3       |     BEB,GIH,STU     | 0.1654  |\n",
      "| rs368882915  |    MAGI2     |       3       |     BEB,GIH,ITU     | 0.1635  |\n",
      "| rs867481901  |    CREBBP    |       2       |       GIH,ITU       | 0.4533  |\n",
      "|  rs2380332   |    CREBBP    |       2       |       GIH,ITU       | 0.4419  |\n",
      "|   rs129993   |    CREBBP    |       2       |       GIH,ITU       | 0.4419  |\n",
      "|  rs12051375  |    CREBBP    |       2       |       GIH,ITU       | 0.4136  |\n",
      "|  rs58001875  |    CLVS1     |       2       |       BEB,STU       | 0.4119  |\n",
      "|  rs74091262  |     IGHE     |       2       |       BEB,PJL       | 0.4105  |\n",
      "|  rs61737471  |     IGHE     |       2       |       BEB,PJL       | 0.4105  |\n",
      "|   rs130032   |    CREBBP    |       2       |       GIH,ITU       | 0.4057  |\n",
      "|   rs129964   |    CREBBP    |       2       |       GIH,ITU       | 0.4055  |\n",
      "|  rs12051217  |    CREBBP    |       2       |       GIH,ITU       | 0.4053  |\n",
      "|  rs12682406  |    CLVS1     |       2       |       BEB,STU       | 0.4023  |\n",
      "|  rs79115753  |     EGFR     |       2       |       GIH,STU       | 0.4019  |\n",
      "|  rs7792218   |     EGFR     |       2       |       GIH,STU       | 0.4019  |\n",
      "|   rs759165   |     EGFR     |       2       |       GIH,STU       | 0.4019  |\n",
      "|  rs73420783  |     EGFR     |       2       |       GIH,STU       | 0.4019  |\n",
      "|  rs73420781  |     EGFR     |       2       |       GIH,STU       | 0.4019  |\n",
      "|  rs60351580  |     EGFR     |       2       |       GIH,STU       | 0.4019  |\n",
      "|  rs2110285   |     EGFR     |       2       |       GIH,STU       | 0.4019  |\n",
      "|  rs17172429  |     EGFR     |       2       |       GIH,STU       | 0.4019  |\n",
      "|  rs17172427  |     EGFR     |       2       |       GIH,STU       | 0.4019  |\n",
      "|  rs12533495  |     EGFR     |       2       |       GIH,STU       | 0.4019  |\n",
      "|  rs11975526  |     EGFR     |       2       |       GIH,STU       | 0.4019  |\n",
      "|  rs11974707  |     EGFR     |       2       |       GIH,STU       | 0.4019  |\n",
      "| rs113558289  |     EGFR     |       2       |       GIH,STU       | 0.4019  |\n",
      "| rs111511218  |     EGFR     |       2       |       GIH,STU       | 0.4019  |\n",
      "|  rs79247998  |    CREBBP    |       2       |       GIH,ITU       | 0.3973  |\n",
      "| rs199861716  |   SNORD3A    |       2       |       ITU,STU       | 0.3925  |\n",
      "|  rs75063259  |    THSD7A    |       2       |       ITU,STU       | 0.3924  |\n",
      "|  rs79663342  |     CRB2     |       2       |       BEB,ITU       | 0.3916  |\n",
      "|  rs75350650  |    CLVS1     |       2       |       BEB,STU       | 0.3902  |\n",
      "|   rs884320   |     CRB2     |       2       |       BEB,ITU       | 0.3885  |\n",
      "|  rs3751845   |    CREBBP    |       2       |       GIH,ITU       |  0.38   |\n",
      "| rs768967412  |    PTPRD     |       2       |       ITU,STU       | 0.3685  |\n",
      "|  rs78698578  |    CREBBP    |       2       |       GIH,ITU       | 0.3668  |\n",
      "|  rs3789027   |    CREBBP    |       2       |       GIH,ITU       | 0.3487  |\n",
      "| rs553496511  |     MC1R     |       2       |       GIH,STU       | 0.3377  |\n",
      "| rs796603331  |    BTNL2     |       2       |       BEB,ITU       | 0.3364  |\n",
      "| rs375097616  |    BTNL2     |       2       |       BEB,ITU       | 0.3364  |\n",
      "|  rs61764777  |    NR3C2     |       2       |       BEB,ITU       | 0.3356  |\n",
      "| rs144459979  |     MC1R     |       2       |       GIH,STU       | 0.3286  |\n",
      "|  rs77002389  |  LINC02740   |       2       |       BEB,ITU       | 0.3236  |\n",
      "|  rs6590914   |    TRPC6     |       2       |       ITU,PJL       | 0.3111  |\n",
      "|  rs11446413  |    TRPC6     |       2       |       ITU,PJL       | 0.3111  |\n",
      "|  rs7950896   |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs7943262   |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs76298608  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs73590250  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs73590249  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs73590245  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs73588290  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs7125654   |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs7122060   |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs7121103   |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs7118524   |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs7107036   |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs7106033   |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs7103795   |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs6590915   |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs6590913   |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs6590912   |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs60185950  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs57091433  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "| rs199649728  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs17097034  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs12292681  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs12289778  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs12288874  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs12288516  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs12288139  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs12285105  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs12283513  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs12282424  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs12279049  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs12278947  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs12274640  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs12273006  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs12271790  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs12270263  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "| rs115369221  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs11500941  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "| rs113980190  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "| rs113506896  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "| rs113377159  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs11224981  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs11224980  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs11224971  |    TRPC6     |       2       |       ITU,PJL       | 0.3079  |\n",
      "|  rs78172251  |    TRPC6     |       2       |       ITU,STU       | 0.3041  |\n",
      "| rs143671629  |    PTPRO     |       2       |       BEB,STU       | 0.3023  |\n",
      "| rs140426805  |    PTPRO     |       2       |       BEB,STU       | 0.3023  |\n",
      "| rs117393207  |    PTPRO     |       2       |       BEB,STU       | 0.3023  |\n",
      "| rs115010966  |    PTPRO     |       2       |       BEB,STU       | 0.3023  |\n",
      "| rs375911294  |  TRIM31-AS1  |       2       |       BEB,PJL       | 0.3021  |\n",
      "|  rs79675667  |    PTPRD     |       2       |       BEB,GIH       | 0.3012  |\n",
      "|  rs76083241  |  LINC02740   |       2       |       BEB,ITU       | 0.3011  |\n",
      "|  rs73433136  |    PTPRD     |       2       |       BEB,STU       | 0.2982  |\n",
      "|  rs11024185  |   PIK3C2A    |       2       |       BEB,PJL       | 0.2981  |\n",
      "|  rs78870387  |  LINC02740   |       2       |       BEB,ITU       |  0.296  |\n",
      "|  rs78366383  |  LINC02740   |       2       |       BEB,ITU       |  0.296  |\n",
      "|  rs76158430  |  LINC02740   |       2       |       BEB,ITU       |  0.296  |\n",
      "|  rs74695740  |  LINC02740   |       2       |       BEB,ITU       |  0.296  |\n",
      "| rs150161936  |  LINC02740   |       2       |       BEB,ITU       |  0.296  |\n",
      "| rs138838623  |  LINC02740   |       2       |       BEB,ITU       |  0.296  |\n",
      "| rs114807246  |  LINC02740   |       2       |       BEB,ITU       |  0.296  |\n",
      "| rs115365585  |    PTPRD     |       2       |       BEB,GIH       | 0.2957  |\n",
      "| rs112507468  |    PTPRD     |       2       |       BEB,STU       | 0.2937  |\n",
      "| rs375779050  |  TSBP1-AS1   |       2       |       BEB,ITU       | 0.2929  |\n",
      "| rs1192126857 |    BTNL2     |       2       |       BEB,ITU       | 0.2914  |\n",
      "|  rs77533798  |    MAGI2     |       2       |       PJL,STU       |  0.29   |\n",
      "|  rs76531962  |    MAGI2     |       2       |       PJL,STU       |  0.29   |\n",
      "| rs147897017  |     CD28     |       2       |       ITU,STU       | 0.2869  |\n",
      "| rs146086865  |    PTPRD     |       2       |       ITU,STU       | 0.2863  |\n",
      "|  rs2285314   |    DELEC1    |       2       |       BEB,ITU       | 0.2857  |\n",
      "|  rs12089371  |    LAMB3     |       2       |       ITU,STU       | 0.2851  |\n",
      "|  rs17164940  |    THSD7A    |       2       |       BEB,ITU       | 0.2829  |\n",
      "| rs374229093  |     ZHX2     |       2       |       GIH,STU       | 0.2828  |\n",
      "|  rs76940994  |    PTPRD     |       2       |       BEB,STU       | 0.2804  |\n",
      "| rs193135952  |    PTPRC     |       2       |       GIH,ITU       | 0.2796  |\n",
      "| rs557629200  |    LAMA5     |       2       |       BEB,GIH       | 0.2779  |\n",
      "|  rs73676041  |    THSD7A    |       2       |       BEB,ITU       | 0.2778  |\n",
      "|   rs769240   |    CCNA2     |       2       |       BEB,STU       | 0.2754  |\n",
      "|  rs1507987   |    CCNA2     |       2       |       BEB,STU       | 0.2754  |\n",
      "| rs376231296  |     CD28     |       2       |       ITU,STU       | 0.2747  |\n",
      "|  rs2217656   |    TGFB1     |       2       |       ITU,STU       | 0.2744  |\n",
      "| rs192936892  |     GPC5     |       2       |       GIH,ITU       | 0.2744  |\n",
      "|  rs16975164  |    TGFB1     |       2       |       ITU,STU       | 0.2744  |\n",
      "| rs116523265  |    PTPRD     |       2       |       BEB,STU       |  0.272  |\n",
      "|  rs11536810  |    DELEC1    |       2       |       BEB,ITU       | 0.2693  |\n",
      "|  rs10982646  |    DELEC1    |       2       |       BEB,ITU       | 0.2693  |\n",
      "|  rs12526388  |    RUNX2     |       2       |       ITU,STU       | 0.2689  |\n",
      "|  rs79110762  |  LINC02182   |       2       |       BEB,ITU       | 0.2673  |\n",
      "|  rs74386324  |  LINC02182   |       2       |       BEB,ITU       | 0.2673  |\n",
      "|  rs76553830  |  LINC02182   |       2       |       GIH,STU       |  0.267  |\n",
      "|  rs9555454   |   TNFSF13B   |       2       |       BEB,STU       | 0.2614  |\n",
      "|  rs74486673  |    PTPRD     |       2       |       BEB,STU       | 0.2589  |\n",
      "| rs538379016  |    LAMA5     |       2       |       ITU,STU       | 0.2585  |\n",
      "| rs553248564  |    NUP160    |       2       |       ITU,STU       | 0.2584  |\n",
      "| rs557192953  |    FCGR2A    |       2       |       GIH,ITU       | 0.2574  |\n",
      "|  rs77482535  |    RUNX2     |       2       |       ITU,STU       | 0.2533  |\n",
      "| rs150214123  |    KALRN     |       2       |       ITU,STU       | 0.2515  |\n",
      "| rs117209838  |    PTPRD     |       2       |       BEB,STU       | 0.2498  |\n",
      "| rs535349719  |  LINC00494   |       2       |       GIH,ITU       | 0.2493  |\n",
      "| rs112966471  |     DLC1     |       2       |       GIH,STU       | 0.2451  |\n",
      "| rs370776378  |   CCDC162P   |       2       |       GIH,ITU       | 0.2443  |\n",
      "| rs370036376  |   CCDC162P   |       2       |       GIH,ITU       | 0.2443  |\n",
      "| rs373439195  |    CD247     |       2       |       GIH,ITU       | 0.2442  |\n",
      "|  rs74977089  |   SERPINA1   |       2       |       ITU,STU       | 0.2438  |\n",
      "|  rs7145047   |   SERPINA1   |       2       |       ITU,STU       | 0.2438  |\n",
      "|  rs55695365  |   SERPINA1   |       2       |       ITU,STU       | 0.2438  |\n",
      "|  rs28444556  |   SERPINA1   |       2       |       ITU,STU       | 0.2438  |\n",
      "|  rs1049800   |   SERPINA1   |       2       |       ITU,STU       | 0.2438  |\n",
      "|  rs8178762   |     PLAT     |       2       |       ITU,STU       | 0.2436  |\n",
      "| rs546195502  |    TNFSF4    |       2       |       ITU,STU       | 0.2427  |\n",
      "| rs139327106  |     JEB7     |       2       |       BEB,GIH       | 0.2406  |\n",
      "| rs552115269  |   TRAPPC3L   |       2       |       BEB,GIH       | 0.2405  |\n",
      "| rs377571783  |   TRAPPC3L   |       2       |       BEB,GIH       | 0.2405  |\n",
      "|  rs17078021  |   TRAPPC3L   |       2       |       BEB,GIH       | 0.2405  |\n",
      "|  rs11969533  |   TRAPPC3L   |       2       |       BEB,GIH       | 0.2405  |\n",
      "| rs367795335  |   CCDC162P   |       2       |       GIH,ITU       | 0.2389  |\n",
      "|  rs77282042  |    DELEC1    |       2       |       ITU,STU       | 0.2383  |\n",
      "| rs542938394  |     LEP      |       2       |       GIH,ITU       | 0.2374  |\n",
      "| rs117176622  |     DLC1     |       2       |       GIH,STU       | 0.2368  |\n",
      "|  rs78587639  |    CD247     |       2       |       ITU,STU       |  0.236  |\n",
      "| rs112675346  |    CD247     |       2       |       ITU,STU       |  0.236  |\n",
      "| rs373924263  |     TFRC     |       2       |       GIH,STU       | 0.2359  |\n",
      "| rs181525628  |    CLVS1     |       2       |       BEB,GIH       | 0.2351  |\n",
      "| rs560214464  |    NPHP1     |       2       |       GIH,ITU       | 0.2345  |\n",
      "| rs115301603  |   TRAPPC3L   |       2       |       BEB,GIH       | 0.2344  |\n",
      "| rs879896694  |     DLC1     |       2       |       GIH,STU       | 0.2309  |\n",
      "| rs879743565  |     DLC1     |       2       |       GIH,STU       | 0.2309  |\n",
      "| rs879281291  |     DLC1     |       2       |       GIH,STU       | 0.2309  |\n",
      "|  rs77789564  |     DLC1     |       2       |       GIH,STU       | 0.2309  |\n",
      "|  rs60234062  |     DLC1     |       2       |       GIH,STU       | 0.2309  |\n",
      "|  rs57616143  |     DLC1     |       2       |       GIH,STU       | 0.2309  |\n",
      "|  rs56728281  |     DLC1     |       2       |       GIH,STU       | 0.2309  |\n",
      "|  rs56718000  |     DLC1     |       2       |       GIH,STU       | 0.2309  |\n",
      "|  rs2289555   |     DLC1     |       2       |       GIH,STU       | 0.2309  |\n",
      "| rs200876805  |     DLC1     |       2       |       GIH,STU       | 0.2309  |\n",
      "| rs146882977  |     DLC1     |       2       |       GIH,STU       | 0.2309  |\n",
      "|  rs11997163  |     DLC1     |       2       |       GIH,STU       | 0.2309  |\n",
      "|  rs77152739  |    CLVS1     |       2       |       BEB,GIH       | 0.2296  |\n",
      "|  rs75228565  |    CLVS1     |       2       |       BEB,GIH       | 0.2296  |\n",
      "| rs140985572  |    SFMBT2    |       2       |       GIH,ITU       | 0.2285  |\n",
      "| rs140361251  |    STAT6     |       2       |       ITU,STU       | 0.2281  |\n",
      "| rs116947293  |     EZR      |       2       |       BEB,ITU       | 0.2281  |\n",
      "| rs528768802  |    NR3C2     |       2       |       GIH,ITU       | 0.2278  |\n",
      "| rs370201246  |     DSE      |       2       |       GIH,STU       | 0.2246  |\n",
      "| rs201666212  |     LEP      |       2       |       GIH,ITU       | 0.2227  |\n",
      "| rs367773435  |   KIRREL3    |       2       |       GIH,ITU       | 0.2226  |\n",
      "| rs370452567  |  TTC21B-AS1  |       2       |       GIH,ITU       | 0.2224  |\n",
      "| rs367883410  |     FAN1     |       2       |       ITU,STU       | 0.2223  |\n",
      "| rs374367179  |    MAGI2     |       2       |       GIH,STU       | 0.2221  |\n",
      "| rs372345008  |   TMEM131L   |       2       |       BEB,GIH       | 0.2209  |\n",
      "| rs143070390  |   HLA-DQA1   |       2       |       GIH,ITU       | 0.2208  |\n",
      "| rs549651967  |     LRP2     |       2       |       ITU,STU       | 0.2207  |\n",
      "|  rs79598606  |     DLC1     |       2       |       GIH,STU       |  0.218  |\n",
      "|  rs77119421  |     DLC1     |       2       |       GIH,STU       |  0.218  |\n",
      "|  rs75504964  |     DLC1     |       2       |       GIH,STU       |  0.218  |\n",
      "|  rs74430243  |     DLC1     |       2       |       GIH,STU       |  0.218  |\n",
      "|  rs61653688  |     DLC1     |       2       |       GIH,STU       |  0.218  |\n",
      "|  rs59645086  |     DLC1     |       2       |       GIH,STU       |  0.218  |\n",
      "|  rs3217507   |     DLC1     |       2       |       GIH,STU       |  0.218  |\n",
      "| rs200502981  |     DLC1     |       2       |       GIH,STU       |  0.218  |\n",
      "| rs147872198  |     DLC1     |       2       |       GIH,STU       |  0.218  |\n",
      "| rs117910789  |     DLC1     |       2       |       GIH,STU       |  0.218  |\n",
      "| rs137872346  |    TNFSF4    |       2       |       ITU,STU       | 0.2177  |\n",
      "| rs143140947  |    NLRP3     |       2       |       ITU,STU       | 0.2172  |\n",
      "| rs552745564  |    DELEC1    |       2       |       ITU,STU       | 0.2169  |\n",
      "| rs201666040  |     LEP      |       2       |       GIH,ITU       | 0.2169  |\n",
      "| rs201335762  |     LEP      |       2       |       GIH,ITU       | 0.2169  |\n",
      "| rs181974050  |    PTPRD     |       2       |       GIH,ITU       | 0.2145  |\n",
      "| rs368180885  |    COQ8B     |       2       |       GIH,STU       | 0.2137  |\n",
      "| rs199940927  |   KIRREL3    |       2       |       BEB,ITU       | 0.2135  |\n",
      "|  rs57097928  |     DLC1     |       2       |       BEB,STU       |  0.212  |\n",
      "| rs372732831  |    MAGI2     |       2       |       ITU,STU       | 0.2109  |\n",
      "| rs367605627  |    LAMA5     |       2       |       GIH,ITU       | 0.2108  |\n",
      "| rs551367542  |    NOS1AP    |       2       |       GIH,ITU       | 0.2101  |\n",
      "| rs375433865  |   CLEC16A    |       2       |       BEB,ITU       |  0.21   |\n",
      "| rs376662771  |  TSBP1-AS1   |       2       |       ITU,STU       | 0.2099  |\n",
      "|  rs78693717  |    TRPC6     |       2       |       ITU,STU       | 0.2091  |\n",
      "| rs150739465  |    SULF2     |       2       |       ITU,STU       | 0.2091  |\n",
      "| rs367569113  |    PDSS2     |       2       |       ITU,STU       |  0.209  |\n",
      "| rs372170571  |     TFRC     |       2       |       GIH,STU       | 0.2081  |\n",
      "| rs576749080  |   TMEM131L   |       2       |       ITU,STU       | 0.2068  |\n",
      "| rs375249566  |   TMEM131L   |       2       |       ITU,STU       | 0.2068  |\n",
      "| rs368349899  |   TMEM131L   |       2       |       ITU,STU       | 0.2068  |\n",
      "| rs531585961  |    TNFSF4    |       2       |       GIH,STU       | 0.2066  |\n",
      "| rs369099047  |    SYNPO     |       2       |       ITU,STU       | 0.2062  |\n",
      "| rs184505188  |    ACTN4     |       2       |       BEB,STU       | 0.2062  |\n",
      "| rs568139391  |   RAPGEF2    |       2       |       ITU,STU       | 0.2051  |\n",
      "| rs561509497  |   KIRREL3    |       2       |       GIH,ITU       | 0.2048  |\n",
      "|  rs77597874  |    PTPRD     |       2       |       ITU,STU       | 0.2047  |\n",
      "| rs549007538  |    NR3C2     |       2       |       GIH,ITU       | 0.2047  |\n",
      "| rs530491963  |  LINC02531   |       2       |       ITU,STU       | 0.2047  |\n",
      "| rs763034646  |   HLA-DRB1   |       2       |       BEB,ITU       | 0.2045  |\n",
      "|  rs59406271  |   HLA-DRB9   |       2       |       BEB,ITU       | 0.2045  |\n",
      "| rs373939101  |   HLA-DRB9   |       2       |       BEB,ITU       | 0.2045  |\n",
      "| rs182341441  |   HLA-DRB9   |       2       |       BEB,ITU       | 0.2045  |\n",
      "| rs1390197854 |   HLA-DRB1   |       2       |       BEB,ITU       | 0.2045  |\n",
      "| rs1267506090 |   HLA-DRB1   |       2       |       BEB,ITU       | 0.2045  |\n",
      "|  rs80352123  |     KC6      |       2       |       PJL,STU       | 0.2043  |\n",
      "| rs370530508  |    PLA2G7    |       2       |       BEB,ITU       | 0.2043  |\n",
      "| rs537120662  |    PLCG2     |       2       |       ITU,PJL       | 0.2042  |\n",
      "| rs544523859  |    CLVS1     |       2       |       ITU,STU       | 0.2038  |\n",
      "|  rs74782900  |    PTPRD     |       2       |       ITU,STU       | 0.2035  |\n",
      "|  rs28631003  |   TMEM131L   |       2       |       ITU,STU       | 0.2035  |\n",
      "| rs533361036  |    SULF2     |       2       |       ITU,PJL       | 0.2024  |\n",
      "| rs114098053  |    ADGRE1    |       2       |       ITU,PJL       | 0.2024  |\n",
      "| rs536499832  |    NOS1AP    |       2       |       GIH,STU       |  0.202  |\n",
      "|  rs75030401  |     FBN1     |       2       |       BEB,STU       | 0.2014  |\n",
      "| rs545079627  |    STAT6     |       2       |       ITU,STU       |  0.201  |\n",
      "| rs193078009  |    KANK1     |       2       |       ITU,STU       | 0.2008  |\n",
      "| rs181018565  |    KANK1     |       2       |       ITU,STU       | 0.2008  |\n",
      "| rs139404960  |    KANK1     |       2       |       ITU,STU       | 0.2008  |\n",
      "| rs138926059  |    KANK1     |       2       |       ITU,STU       | 0.2008  |\n",
      "|  rs75359142  |     FBN1     |       2       |       ITU,STU       | 0.2001  |\n",
      "| rs183550827  |     GPC5     |       2       |       ITU,STU       | 0.1999  |\n",
      "| rs373350560  |  LINC02531   |       2       |       ITU,STU       | 0.1996  |\n",
      "| rs145743886  |  LINC02531   |       2       |       ITU,STU       | 0.1996  |\n",
      "| rs142957088  |  LINC02531   |       2       |       ITU,STU       | 0.1996  |\n",
      "| rs140462197  |  LINC02531   |       2       |       ITU,STU       | 0.1996  |\n",
      "| rs137870219  |  LINC02531   |       2       |       ITU,STU       | 0.1996  |\n",
      "| rs557115522  |    CLVS1     |       2       |       GIH,ITU       | 0.1987  |\n",
      "| rs550774665  |    CLVS1     |       2       |       GIH,ITU       | 0.1987  |\n",
      "| rs149343151  |    CTLA4     |       2       |       GIH,ITU       | 0.1987  |\n",
      "| rs550075142  |     DLC1     |       2       |       BEB,ITU       | 0.1986  |\n",
      "| rs368563270  |    SEMA3E    |       2       |       PJL,STU       | 0.1983  |\n",
      "| rs542465461  |     NWD2     |       2       |       BEB,GIH       | 0.1982  |\n",
      "| rs373953208  |    PTPRD     |       2       |       GIH,ITU       |  0.198  |\n",
      "| rs370396422  |     E2F3     |       2       |       BEB,STU       | 0.1979  |\n",
      "| rs376213225  |    RUNX3     |       2       |       BEB,STU       | 0.1969  |\n",
      "| rs148788741  |    KITLG     |       2       |       ITU,STU       | 0.1965  |\n",
      "| rs139899069  |    KITLG     |       2       |       ITU,STU       | 0.1965  |\n",
      "|  rs75899147  |    MAGI2     |       2       |       GIH,STU       | 0.1963  |\n",
      "| rs370733312  |    HMG20A    |       2       |       GIH,STU       |  0.196  |\n",
      "| rs367806805  |     KRT8     |       2       |       BEB,ITU       | 0.1957  |\n",
      "| rs189892797  |   ZMPSTE24   |       2       |       BEB,GIH       | 0.1956  |\n",
      "| rs375065959  |     BRAF     |       2       |       GIH,STU       | 0.1955  |\n",
      "|  rs75820040  |     FBN1     |       2       |       BEB,STU       | 0.1954  |\n",
      "| rs535337237  |    ACTN4     |       2       |       BEB,STU       | 0.1953  |\n",
      "| rs183262099  |     CETP     |       2       |       ITU,STU       | 0.1952  |\n",
      "| rs375065545  |    PTPRD     |       2       |       ITU,STU       | 0.1951  |\n",
      "| rs374538528  |    ELMO1     |       2       |       ITU,STU       | 0.1951  |\n",
      "| rs374886427  |     SELP     |       2       |       ITU,PJL       |  0.195  |\n",
      "|  rs78114450  |     FBN1     |       2       |       ITU,STU       | 0.1947  |\n",
      "|  rs77732998  |     FBN1     |       2       |       ITU,STU       | 0.1947  |\n",
      "|  rs77581118  |     FBN1     |       2       |       ITU,STU       | 0.1947  |\n",
      "|  rs76951679  |     FBN1     |       2       |       ITU,STU       | 0.1947  |\n",
      "| rs375705879  |    PTPRD     |       2       |       GIH,STU       | 0.1947  |\n",
      "| rs533765322  |    NPHP1     |       2       |       ITU,STU       | 0.1943  |\n",
      "| rs573482042  |     BMP6     |       2       |       BEB,GIH       |  0.194  |\n",
      "| rs544940657  |     BMP6     |       2       |       BEB,GIH       |  0.194  |\n",
      "| rs370553670  |    LIMA1     |       2       |       ITU,STU       | 0.1939  |\n",
      "| rs201490178  |    PLCG2     |       2       |       GIH,ITU       | 0.1926  |\n",
      "| rs375817894  |    PTPRD     |       2       |       GIH,ITU       | 0.1925  |\n",
      "| rs560511374  |     NOS3     |       2       |       GIH,ITU       | 0.1924  |\n",
      "| rs536532435  |     DCN      |       2       |       GIH,ITU       | 0.1922  |\n",
      "| rs556857719  |     ZHX2     |       2       |       GIH,PJL       | 0.1919  |\n",
      "| rs147783220  |    ACTN4     |       2       |       BEB,STU       | 0.1915  |\n",
      "| rs141692021  |    ACTN4     |       2       |       BEB,STU       | 0.1915  |\n",
      "| rs376445738  |   CLEC16A    |       2       |       GIH,STU       | 0.1914  |\n",
      "| rs141329338  |     LCP1     |       2       |       BEB,ITU       | 0.1913  |\n",
      "| rs189950845  |    MYO1E     |       2       |       GIH,ITU       |  0.191  |\n",
      "| rs200259478  |    MAPK14    |       2       |       BEB,ITU       | 0.1908  |\n",
      "| rs569027758  |    PTPRD     |       2       |       GIH,STU       | 0.1906  |\n",
      "| rs534472762  |     ALG1     |       2       |       GIH,STU       | 0.1902  |\n",
      "|  rs77030234  |     TFRC     |       2       |       BEB,STU       |  0.19   |\n",
      "|  rs41303539  |     TFRC     |       2       |       BEB,STU       |  0.19   |\n",
      "|  rs41303521  |     TFRC     |       2       |       BEB,STU       |  0.19   |\n",
      "|  rs41301369  |     TFRC     |       2       |       BEB,STU       |  0.19   |\n",
      "|  rs41297481  |     TFRC     |       2       |       BEB,STU       |  0.19   |\n",
      "|  rs41297473  |     TFRC     |       2       |       BEB,STU       |  0.19   |\n",
      "|  rs3804142   |     TFRC     |       2       |       BEB,STU       |  0.19   |\n",
      "| rs367634655  |    MAGI2     |       2       |       BEB,STU       |  0.19   |\n",
      "|  rs2300776   |     TFRC     |       2       |       BEB,STU       |  0.19   |\n",
      "|  rs17091585  |     TFRC     |       2       |       BEB,STU       |  0.19   |\n",
      "|  rs17091582  |     TFRC     |       2       |       BEB,STU       |  0.19   |\n",
      "| rs148436820  |     TFRC     |       2       |       BEB,STU       |  0.19   |\n",
      "| rs118084882  |     TFRC     |       2       |       BEB,STU       |  0.19   |\n",
      "| rs577761420  |    SGPL1     |       2       |       BEB,GIH       | 0.1898  |\n",
      "| rs559466614  |     LCP1     |       2       |       BEB,ITU       | 0.1898  |\n",
      "| rs577877399  |     BRAF     |       2       |       ITU,STU       | 0.1893  |\n",
      "| rs537444126  |    SYNPO     |       2       |       ITU,STU       | 0.1892  |\n",
      "| rs369343719  |     GPC5     |       2       |       ITU,STU       |  0.189  |\n",
      "| rs376496994  |     EXT1     |       2       |       BEB,ITU       | 0.1887  |\n",
      "| rs377548740  |     ZHX2     |       2       |       GIH,STU       | 0.1885  |\n",
      "| rs572073879  |     GPC5     |       2       |       GIH,STU       | 0.1883  |\n",
      "| rs544492214  |     GPC5     |       2       |       GIH,STU       | 0.1883  |\n",
      "| rs542777130  |     GPC5     |       2       |       GIH,STU       | 0.1883  |\n",
      "|  rs77373715  |    MAGI2     |       2       |       ITU,STU       | 0.1875  |\n",
      "| rs545323408  |    LTA4H     |       2       |       ITU,STU       | 0.1875  |\n",
      "| rs569901029  |  TSBP1-AS1   |       2       |       GIH,PJL       | 0.1869  |\n",
      "| rs183484070  |  TSBP1-AS1   |       2       |       GIH,PJL       | 0.1869  |\n",
      "| rs186163911  |    ZNF469    |       2       |       BEB,STU       | 0.1866  |\n",
      "| rs578030585  |    DELEC1    |       2       |       PJL,STU       | 0.1865  |\n",
      "| rs546767880  |    DELEC1    |       2       |       PJL,STU       | 0.1865  |\n",
      "| rs370019300  |    RUNX3     |       2       |       ITU,STU       | 0.1864  |\n",
      "| rs561526243  |   PIK3C2A    |       2       |       GIH,ITU       | 0.1863  |\n",
      "| rs370214142  |    PTPRD     |       2       |       GIH,ITU       | 0.1862  |\n",
      "| rs566869021  |     BMP6     |       2       |       GIH,STU       | 0.1861  |\n",
      "| rs372811698  |     GPC5     |       2       |       BEB,STU       | 0.1855  |\n",
      "| rs149365718  |    PSMD14    |       2       |       BEB,ITU       | 0.1854  |\n",
      "|  rs12471658  |    PSMD14    |       2       |       BEB,ITU       | 0.1854  |\n",
      "| rs147390865  |    ALOX5     |       2       |       BEB,GIH       | 0.1853  |\n",
      "| rs367661953  |     EGFR     |       2       |       GIH,ITU       | 0.1851  |\n",
      "| rs527513653  |    ELMO1     |       2       |       GIH,STU       | 0.1841  |\n",
      "| rs375813168  |     LIPC     |       2       |       BEB,ITU       | 0.1841  |\n",
      "| rs576767813  |    MAPK14    |       2       |       BEB,ITU       |  0.184  |\n",
      "| rs539682445  |    MAPK14    |       2       |       BEB,ITU       |  0.184  |\n",
      "| rs377618354  |    MAPK14    |       2       |       BEB,ITU       |  0.184  |\n",
      "| rs376482229  |    MAPK14    |       2       |       BEB,ITU       |  0.184  |\n",
      "| rs375742009  |    MAPK14    |       2       |       BEB,ITU       |  0.184  |\n",
      "| rs374563248  |    MAPK14    |       2       |       BEB,ITU       |  0.184  |\n",
      "| rs374121687  |    MAPK14    |       2       |       BEB,ITU       |  0.184  |\n",
      "| rs373790831  |    MAPK14    |       2       |       BEB,ITU       |  0.184  |\n",
      "| rs373379803  |    MAPK14    |       2       |       BEB,ITU       |  0.184  |\n",
      "| rs373335380  |    MAPK14    |       2       |       BEB,ITU       |  0.184  |\n",
      "| rs373328059  |    MAPK14    |       2       |       BEB,ITU       |  0.184  |\n",
      "| rs372677054  |    MAPK14    |       2       |       BEB,ITU       |  0.184  |\n",
      "| rs372374942  |    MAPK14    |       2       |       BEB,ITU       |  0.184  |\n",
      "| rs372079027  |    MAPK14    |       2       |       BEB,ITU       |  0.184  |\n",
      "| rs371834848  |    MAPK14    |       2       |       BEB,ITU       |  0.184  |\n",
      "| rs370401484  |    MAPK14    |       2       |       BEB,ITU       |  0.184  |\n",
      "| rs370386133  |    MAPK14    |       2       |       BEB,ITU       |  0.184  |\n",
      "| rs369835216  |    MAPK14    |       2       |       BEB,ITU       |  0.184  |\n",
      "| rs369058889  |    MAPK14    |       2       |       BEB,ITU       |  0.184  |\n",
      "| rs368250176  |    MAPK14    |       2       |       BEB,ITU       |  0.184  |\n",
      "| rs368181243  |    MAPK14    |       2       |       BEB,ITU       |  0.184  |\n",
      "| rs147744046  |    MAPK14    |       2       |       BEB,ITU       |  0.184  |\n",
      "|  rs78930696  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs78734256  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs78650653  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs77850716  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs76551117  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs76378657  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs74588528  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs550600914  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs529414627  |    PTPRD     |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs375286177  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs3750324   |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs374725639  |    PTPRD     |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs2302814   |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs201150944  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs200054067  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs17481372  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs17481358  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs17481310  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs17481303  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs17481282  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs17481268  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs17481247  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs17481205  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs17481198  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs17452541  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs17452534  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs17452520  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs17452463  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs17452372  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "|  rs17452365  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs150590461  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs149934477  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs149647402  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs149460910  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs147643853  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs145830282  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs143855056  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs143124086  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs141740293  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs139843056  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs117977801  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs117930501  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs117732155  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs117657952  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs117456940  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs117364376  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs117182970  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs117066852  |    NUP188    |       2       |       ITU,STU       | 0.1837  |\n",
      "| rs571953875  |    TRPC6     |       2       |       ITU,STU       | 0.1835  |\n",
      "| rs542381370  |    SULF2     |       2       |       ITU,PJL       | 0.1835  |\n",
      "| rs374918067  |    CCNA2     |       2       |       BEB,GIH       | 0.1835  |\n",
      "| rs372026286  |     GGT1     |       2       |       ITU,STU       | 0.1835  |\n",
      "| rs373643679  |    ELMO1     |       2       |       BEB,STU       | 0.1832  |\n",
      "| rs144452245  |    PTPRD     |       2       |       GIH,STU       | 0.1832  |\n",
      "| rs377712311  |    ITSN2     |       2       |       GIH,STU       |  0.183  |\n",
      "| rs377119152  |    KALRN     |       2       |       ITU,STU       | 0.1828  |\n",
      "| rs578095054  |   MICA-AS1   |       2       |       BEB,STU       | 0.1822  |\n",
      "| rs565815653  |   MICA-AS1   |       2       |       BEB,STU       | 0.1822  |\n",
      "| rs373681960  |   MICA-AS1   |       2       |       BEB,STU       | 0.1822  |\n",
      "| rs371964376  |    ALOX5     |       2       |       BEB,GIH       | 0.1808  |\n",
      "| rs375098627  |    MAGI2     |       2       |       GIH,PJL       | 0.1806  |\n",
      "| rs190405095  |     ZHX2     |       2       |       BEB,STU       | 0.1804  |\n",
      "|  rs75936328  |    RUNX2     |       2       |       BEB,STU       | 0.1803  |\n",
      "| rs368905447  |     EXT1     |       2       |       BEB,ITU       | 0.1803  |\n",
      "| rs117152785  |    RUNX2     |       2       |       BEB,STU       | 0.1803  |\n",
      "| rs528702447  |     SELP     |       2       |       GIH,ITU       | 0.1801  |\n",
      "| rs367654878  |    COL4A3    |       2       |       BEB,ITU       | 0.1801  |\n",
      "| rs373435183  |    PTPRD     |       2       |       GIH,ITU       | 0.1799  |\n",
      "| rs370066835  |    PTPRD     |       2       |       GIH,ITU       | 0.1799  |\n",
      "| rs368990659  |    PTPRD     |       2       |       GIH,ITU       | 0.1799  |\n",
      "| rs536024346  |     FAT1     |       2       |       GIH,ITU       | 0.1798  |\n",
      "| rs564465206  |    PTPRD     |       2       |       GIH,ITU       | 0.1797  |\n",
      "|  rs74435318  |    CLVS1     |       2       |       BEB,GIH       |  0.179  |\n",
      "| rs557625258  |    ZAP70     |       2       |       BEB,GIH       |  0.179  |\n",
      "| rs569534937  |    PTPRD     |       2       |       GIH,ITU       | 0.1788  |\n",
      "| rs546309954  |     DSE      |       2       |       GIH,STU       | 0.1782  |\n",
      "| rs377055323  |   HLA-DRB1   |       2       |       BEB,ITU       | 0.1782  |\n",
      "| rs375420496  |     DSE      |       2       |       GIH,STU       | 0.1782  |\n",
      "| rs200203992  |     FGA      |       2       |       ITU,STU       | 0.1781  |\n",
      "| rs186222179  |    NUP188    |       2       |       ITU,STU       |  0.178  |\n",
      "|  rs17452456  |    NUP188    |       2       |       ITU,STU       |  0.178  |\n",
      "| rs560643931  |    MAPK14    |       2       |       BEB,ITU       | 0.1779  |\n",
      "|  rs77198894  |     FBN1     |       2       |       ITU,STU       | 0.1777  |\n",
      "| rs567885422  |    COL4A4    |       2       |       GIH,STU       | 0.1776  |\n",
      "| rs542286828  |    SYNPO     |       2       |       ITU,STU       | 0.1776  |\n",
      "| rs543349141  |    PTPRD     |       2       |       ITU,PJL       | 0.1773  |\n",
      "| rs374859262  |    PTPRD     |       2       |       BEB,GIH       |  0.177  |\n",
      "| rs531934946  |   KIRREL3    |       2       |       BEB,GIH       | 0.1769  |\n",
      "| rs571722911  |     LIPC     |       2       |       ITU,STU       | 0.1767  |\n",
      "| rs537256261  |     LIPC     |       2       |       ITU,STU       | 0.1767  |\n",
      "|  rs11467138  |    SULF2     |       2       |       ITU,STU       | 0.1767  |\n",
      "| rs571380066  |     LCP1     |       2       |       ITU,STU       | 0.1766  |\n",
      "| rs373161856  |     DSE      |       2       |       GIH,STU       | 0.1763  |\n",
      "| rs367836933  |     DSE      |       2       |       GIH,STU       | 0.1763  |\n",
      "| rs370344585  |   KIRREL3    |       2       |       ITU,STU       |  0.176  |\n",
      "| rs566599159  |    SFMBT2    |       2       |       ITU,STU       | 0.1758  |\n",
      "| rs555277816  |    SFMBT2    |       2       |       ITU,STU       | 0.1758  |\n",
      "|  rs34141664  |   HLA-DRB1   |       2       |       ITU,STU       |  0.175  |\n",
      "| rs552019470  |     EGFR     |       2       |       BEB,ITU       | 0.1749  |\n",
      "| rs574391014  |     DSE      |       2       |       BEB,GIH       | 0.1747  |\n",
      "| rs1373787805 |     ZHX2     |       2       |       BEB,STU       | 0.1746  |\n",
      "| rs1310118202 |     ZHX2     |       2       |       BEB,STU       | 0.1746  |\n",
      "| rs577681403  |    MAGI2     |       2       |       GIH,ITU       | 0.1736  |\n",
      "| rs537831625  |   ZMPSTE24   |       2       |       GIH,ITU       | 0.1734  |\n",
      "|  rs2297403   |    ABCA1     |       2       |       BEB,ITU       | 0.1734  |\n",
      "| rs560670201  |    ELMO1     |       2       |       BEB,STU       | 0.1733  |\n",
      "| rs118155348  |    COQ8A     |       2       |       BEB,ITU       | 0.1733  |\n",
      "| rs533294775  |   MORF4L1    |       2       |       GIH,ITU       | 0.1731  |\n",
      "| rs533225501  |     DLC1     |       2       |       GIH,ITU       | 0.1731  |\n",
      "| rs370764654  |    NUP133    |       2       |       GIH,ITU       | 0.1731  |\n",
      "| rs113676351  |     DLC1     |       2       |       GIH,ITU       | 0.1731  |\n",
      "| rs567969753  |    PTPRD     |       2       |       GIH,ITU       | 0.1727  |\n",
      "| rs558400226  |    PTPRD     |       2       |       GIH,ITU       | 0.1727  |\n",
      "| rs551863397  |    PTPRD     |       2       |       GIH,ITU       | 0.1727  |\n",
      "| rs544653404  |    PTPRD     |       2       |       GIH,ITU       | 0.1727  |\n",
      "| rs544079874  |    PTPRD     |       2       |       GIH,ITU       | 0.1727  |\n",
      "| rs540746318  |    PTPRD     |       2       |       GIH,ITU       | 0.1727  |\n",
      "| rs539737889  |    PTPRD     |       2       |       GIH,ITU       | 0.1727  |\n",
      "| rs534987989  |    PTPRD     |       2       |       GIH,ITU       | 0.1727  |\n",
      "| rs530240258  |    PTPRD     |       2       |       GIH,ITU       | 0.1727  |\n",
      "| rs527257019  |    PTPRD     |       2       |       GIH,ITU       | 0.1727  |\n",
      "| rs143109660  |    PTPRD     |       2       |       GIH,ITU       | 0.1727  |\n",
      "| rs138137718  |    PTPRD     |       2       |       GIH,ITU       | 0.1727  |\n",
      "| rs116725391  |    PTPRD     |       2       |       GIH,ITU       | 0.1727  |\n",
      "| rs371548920  |     GPC5     |       2       |       ITU,STU       | 0.1722  |\n",
      "| rs578174336  |     LIPC     |       2       |       ITU,STU       |  0.172  |\n",
      "| rs375687638  |    MAGI2     |       2       |       ITU,STU       |  0.172  |\n",
      "| rs139218071  |     DLC1     |       2       |       BEB,STU       |  0.172  |\n",
      "| rs573308344  |     WT1      |       2       |       ITU,STU       | 0.1718  |\n",
      "| rs375661609  |     NWD2     |       2       |       ITU,STU       | 0.1716  |\n",
      "| rs373491112  |     NWD2     |       2       |       ITU,STU       | 0.1716  |\n",
      "| rs371470653  |     NWD2     |       2       |       ITU,STU       | 0.1716  |\n",
      "| rs376876145  |    ZNF469    |       2       |       BEB,STU       | 0.1714  |\n",
      "| rs368284669  |   SLC22A12   |       2       |       GIH,STU       | 0.1711  |\n",
      "| rs552334082  |     TUG1     |       2       |       ITU,STU       | 0.1704  |\n",
      "|  rs4927709   |     TFRC     |       2       |       BEB,STU       | 0.1704  |\n",
      "| rs531131652  |     GPC5     |       2       |       PJL,STU       | 0.1701  |\n",
      "| rs529018197  |   KIRREL3    |       2       |       BEB,ITU       | 0.1697  |\n",
      "| rs565643095  |     DLC1     |       2       |       BEB,ITU       | 0.1691  |\n",
      "| rs377734148  |     BRAF     |       2       |       BEB,ITU       | 0.1691  |\n",
      "| rs181895258  |     DLC1     |       2       |       BEB,ITU       | 0.1691  |\n",
      "| rs116725349  |     DLC1     |       2       |       BEB,ITU       | 0.1691  |\n",
      "| rs370099504  |    ANXA5     |       2       |       BEB,GIH       | 0.1681  |\n",
      "| rs534615581  |    PSMD14    |       2       |       BEB,STU       | 0.1679  |\n",
      "| rs112419938  |  LINC02182   |       2       |       GIH,PJL       | 0.1673  |\n",
      "| rs549329118  |    MAGI2     |       2       |       GIH,ITU       |  0.167  |\n",
      "| rs536276024  |     XPO5     |       2       |       BEB,PJL       |  0.167  |\n",
      "| rs370332329  |    NR3C2     |       2       |       GIH,PJL       | 0.1667  |\n",
      "| rs560043988  |    ACTR2     |       2       |       ITU,STU       | 0.1663  |\n",
      "| rs541583844  |      C3      |       2       |       ITU,STU       | 0.1663  |\n",
      "| rs537365949  |   SCGB1A1    |       2       |       ITU,STU       | 0.1663  |\n",
      "| rs374626523  |    KANK1     |       2       |       ITU,STU       | 0.1663  |\n",
      "| rs372083034  |    KALRN     |       2       |       ITU,STU       | 0.1663  |\n",
      "| rs573582615  |     CUBN     |       2       |       ITU,STU       | 0.1658  |\n",
      "|  rs74469947  |      GC      |       2       |       ITU,STU       | 0.1656  |\n",
      "| rs527476820  |     GPC5     |       2       |       ITU,STU       | 0.1656  |\n",
      "| rs375443504  |    MAGI2     |       2       |       ITU,STU       | 0.1656  |\n",
      "|  rs16847024  |      GC      |       2       |       ITU,STU       | 0.1656  |\n",
      "| rs115862008  |      GC      |       2       |       ITU,STU       | 0.1656  |\n",
      "| rs112043289  |     EXT1     |       2       |       ITU,STU       | 0.1656  |\n",
      "|  rs10014541  |      GC      |       2       |       ITU,STU       | 0.1656  |\n",
      "|  rs10008328  |      GC      |       2       |       ITU,STU       | 0.1656  |\n",
      "|  rs76124556  |    DAAM2     |       2       |       GIH,STU       | 0.1654  |\n",
      "|  rs76090263  |    DAAM2     |       2       |       GIH,STU       | 0.1654  |\n",
      "|  rs73414819  |    DAAM2     |       2       |       GIH,STU       | 0.1654  |\n",
      "|  rs3736829   |    DAAM2     |       2       |       GIH,STU       | 0.1654  |\n",
      "|  rs3736827   |    DAAM2     |       2       |       GIH,STU       | 0.1654  |\n",
      "|  rs2295884   |    DAAM2     |       2       |       GIH,STU       | 0.1654  |\n",
      "| rs144102312  |    DAAM2     |       2       |       GIH,STU       | 0.1654  |\n",
      "| rs537933858  |    PTPRD     |       2       |       GIH,STU       | 0.1652  |\n",
      "| rs376241377  |    NOS1AP    |       2       |       GIH,STU       |  0.165  |\n",
      "| rs541999307  |     LIPC     |       2       |       ITU,STU       | 0.1649  |\n",
      "| rs376484243  |     FBN1     |       2       |       ITU,STU       | 0.1647  |\n",
      "| rs374805364  |     FBN1     |       2       |       ITU,STU       | 0.1647  |\n",
      "| rs570234842  |    THSD7A    |       2       |       GIH,STU       | 0.1644  |\n",
      "| rs542334688  |    PTPRD     |       2       |       GIH,STU       | 0.1644  |\n",
      "| rs373536942  |     EGFR     |       2       |       GIH,STU       | 0.1644  |\n",
      "| rs371317372  |    TGFB1     |       2       |       GIH,STU       | 0.1644  |\n",
      "| rs546734818  |    PTPRD     |       2       |       PJL,STU       | 0.1643  |\n",
      "| rs374499460  |    NUP160    |       2       |       GIH,STU       | 0.1642  |\n",
      "| rs376723078  |  LINC02182   |       2       |       GIH,PJL       | 0.1636  |\n",
      "| rs574744747  |     IL4R     |       2       |       BEB,STU       | 0.1634  |\n",
      "| rs139187947  |     GPC5     |       2       |       PJL,STU       | 0.1628  |\n",
      "|  rs79719567  |     PAX2     |       2       |       BEB,ITU       | 0.1627  |\n",
      "| rs371756999  |     FAT1     |       2       |       PJL,STU       | 0.1625  |\n",
      "| rs547327167  |     DLC1     |       2       |       BEB,ITU       | 0.1608  |\n",
      "| rs545342718  |     DLC1     |       2       |       BEB,ITU       | 0.1608  |\n",
      "| rs530495349  |     DLC1     |       2       |       BEB,ITU       | 0.1608  |\n",
      "| rs537596921  |    CREBBP    |       2       |       BEB,ITU       | 0.1607  |\n",
      "| rs552547254  |     ATIC     |       2       |       GIH,ITU       | 0.1606  |\n",
      "| rs547852835  |    KITLG     |       2       |       GIH,ITU       | 0.1606  |\n",
      "| rs533141997  |     DLC1     |       2       |       GIH,ITU       | 0.1606  |\n",
      "| rs373227293  |   KIRREL3    |       2       |       GIH,ITU       | 0.1604  |\n",
      "| rs183847042  |    SCARB2    |       2       |       GIH,PJL       | 0.1602  |\n",
      "| rs562434261  |   RAPGEF2    |       2       |       ITU,STU       | 0.1601  |\n",
      "| rs368364935  |    PLA2R1    |       2       |       BEB,STU       |  0.16   |\n",
      "| rs182614691  |     EXT1     |       2       |       ITU,STU       | 0.1599  |\n",
      "| rs139395604  |     GSN      |       2       |       ITU,STU       | 0.1599  |\n",
      "| rs549609189  |     BRAF     |       2       |       ITU,STU       | 0.1597  |\n",
      "| rs371613143  |     GGT1     |       2       |       ITU,STU       | 0.1595  |\n",
      "| rs373705423  |     EXT1     |       2       |       ITU,STU       | 0.1593  |\n",
      "| rs574789556  |     CUBN     |       2       |       ITU,STU       | 0.1592  |\n",
      "| rs573387074  |     LRP2     |       2       |       ITU,STU       | 0.1592  |\n",
      "| rs377580327  |    PTPRD     |       2       |       ITU,STU       | 0.1592  |\n",
      "| rs377519981  |    PTPRD     |       2       |       ITU,STU       | 0.1592  |\n",
      "| rs374467301  |    PTPRD     |       2       |       ITU,STU       | 0.1592  |\n",
      "|  rs76660617  |    MUC16     |       2       |       BEB,STU       | 0.1591  |\n",
      "| rs569342466  |     PMM2     |       2       |       ITU,STU       | 0.1591  |\n",
      "| rs562835399  |     DLC1     |       2       |       ITU,PJL       | 0.1588  |\n",
      "| rs533478137  |    HDAC2     |       2       |       PJL,STU       | 0.1587  |\n",
      "| rs559471410  |     DLC1     |       2       |       BEB,ITU       | 0.1564  |\n",
      "| rs371902390  |    AXDND1    |       2       |       ITU,PJL       | 0.1552  |\n",
      "| rs368920031  |   PIK3C2A    |       2       |       ITU,PJL       |  0.155  |\n",
      "| rs551222239  |     NEU1     |       2       |       BEB,STU       | 0.1547  |\n",
      "| rs561451939  |     XPO5     |       2       |       ITU,STU       | 0.1538  |\n",
      "| rs528543711  |     LRP2     |       2       |       ITU,STU       | 0.1536  |\n",
      "| rs191390165  |     EXT1     |       2       |       ITU,STU       | 0.1536  |\n",
      "+--------------+--------------+---------------+---------------------+---------+\n",
      "\n",
      "✓ 1794 variants significant in 2+ South Asian populations\n",
      "\n",
      "====================================================================================================\n",
      "EXPORT OPTIONS\n",
      "====================================================================================================\n",
      "\n",
      "====================================================================================================\n",
      "✅ ANALYSIS COMPLETE!\n",
      "====================================================================================================\n",
      "\n",
      "📊 Key Findings:\n",
      "  • Variants in 2+ populations: 50\n",
      "  • Top variant: rs4709451 (LPA) in 10 populations\n"
     ]
    }
   ],
   "source": [
    "\"\"\"\n",
    "Multi-Population Significant Variants Analysis\n",
    "Find variants that are significant in 2 or more populations\n",
    "\"\"\"\n",
    "\n",
    "import sqlite3\n",
    "import pandas as pd\n",
    "from tabulate import tabulate\n",
    "\n",
    "DATABASE = 'nephrotic_syndrome.db'\n",
    "\n",
    "def get_connection():\n",
    "    return sqlite3.connect(DATABASE)\n",
    "\n",
    "def query1_multi_pop_summary():\n",
    "    \"\"\"Variants significant in multiple populations - Summary\"\"\"\n",
    "    print(\"\\n\" + \"=\"*100)\n",
    "    print(\"VARIANTS SIGNIFICANT IN MULTIPLE POPULATIONS - SUMMARY\")\n",
    "    print(\"=\"*100)\n",
    "    print(\"Criteria: p ≤ 0.01 AND AF ≥ 0.05 (either pop) AND FST ≥ 0.15\")\n",
    "    print(\"=\"*100 + \"\\n\")\n",
    "    \n",
    "    query = \"\"\"\n",
    "    SELECT \n",
    "        v.rs_id,\n",
    "        v.gene_name,\n",
    "        v.chrom || ':' || v.position AS position,\n",
    "        COUNT(DISTINCT st.population) AS population_count,\n",
    "        GROUP_CONCAT(DISTINCT st.population) AS populations,\n",
    "        ROUND(AVG(st.chi_square_p_value), 8) AS avg_p_value,\n",
    "        ROUND(AVG(st.fst_value), 4) AS avg_fst\n",
    "    FROM statistical_tests st\n",
    "    INNER JOIN variants v ON st.variant_id = v.variant_id\n",
    "    INNER JOIN allele_frequencies af_eur \n",
    "        ON st.variant_id = af_eur.variant_id \n",
    "        AND af_eur.population = 'EUR'\n",
    "    INNER JOIN allele_frequencies af_pop \n",
    "        ON st.variant_id = af_pop.variant_id \n",
    "        AND af_pop.population = st.population\n",
    "    WHERE \n",
    "        st.chi_square_p_value <= 0.01\n",
    "        AND (af_eur.allele_frequency >= 0.05 OR af_pop.allele_frequency >= 0.05)\n",
    "        AND st.fst_value >= 0.15\n",
    "    GROUP BY v.rs_id, v.gene_name, v.chrom, v.position\n",
    "    HAVING COUNT(DISTINCT st.population) >= 2\n",
    "    ORDER BY COUNT(DISTINCT st.population) DESC, avg_fst DESC\n",
    "    LIMIT 50\n",
    "    \"\"\"\n",
    "    \n",
    "    conn = get_connection()\n",
    "    df = pd.read_sql_query(query, conn)\n",
    "    conn.close()\n",
    "    \n",
    "    if len(df) == 0:\n",
    "        print(\"⚠️  No variants found that are significant in 2+ populations!\")\n",
    "        print(\"This could mean:\")\n",
    "        print(\"  • The criteria are very strict (which is good for quality)\")\n",
    "        print(\"  • Most variants show population-specific effects\")\n",
    "        return None\n",
    "    \n",
    "    # Truncate long population lists\n",
    "    df['populations'] = df['populations'].apply(\n",
    "        lambda x: x if len(str(x)) < 60 else str(x)[:57] + '...'\n",
    "    )\n",
    "    \n",
    "    print(tabulate(df, headers='keys', tablefmt='pretty', showindex=False))\n",
    "    print(f\"\\n✓ Found {len(df)} variants significant in 2+ populations\")\n",
    "    print(f\"✓ Top variant: {df.iloc[0]['rs_id']} in {df.iloc[0]['population_count']} populations\")\n",
    "    \n",
    "    return df\n",
    "\n",
    "def query2_distribution():\n",
    "    \"\"\"Distribution of variants by population count\"\"\"\n",
    "    print(\"\\n\" + \"=\"*100)\n",
    "    print(\"DISTRIBUTION: HOW MANY POPULATIONS ARE VARIANTS SIGNIFICANT IN?\")\n",
    "    print(\"=\"*100 + \"\\n\")\n",
    "    \n",
    "    query = \"\"\"\n",
    "    WITH variant_pop_counts AS (\n",
    "        SELECT \n",
    "            v.rs_id,\n",
    "            COUNT(DISTINCT st.population) AS pop_count\n",
    "        FROM statistical_tests st\n",
    "        INNER JOIN variants v ON st.variant_id = v.variant_id\n",
    "        INNER JOIN allele_frequencies af_eur \n",
    "            ON st.variant_id = af_eur.variant_id \n",
    "            AND af_eur.population = 'EUR'\n",
    "        INNER JOIN allele_frequencies af_pop \n",
    "            ON st.variant_id = af_pop.variant_id \n",
    "            AND af_pop.population = st.population\n",
    "        WHERE \n",
    "            st.chi_square_p_value <= 0.01\n",
    "            AND (af_eur.allele_frequency >= 0.05 OR af_pop.allele_frequency >= 0.05)\n",
    "            AND st.fst_value >= 0.15\n",
    "        GROUP BY v.rs_id\n",
    "    )\n",
    "    SELECT \n",
    "        pop_count AS 'Number_of_Populations',\n",
    "        COUNT(*) AS 'Variant_Count',\n",
    "        ROUND(COUNT(*) * 100.0 / (SELECT COUNT(*) FROM variant_pop_counts), 2) AS 'Percentage'\n",
    "    FROM variant_pop_counts\n",
    "    GROUP BY pop_count\n",
    "    ORDER BY pop_count DESC\n",
    "    \"\"\"\n",
    "    \n",
    "    conn = get_connection()\n",
    "    df = pd.read_sql_query(query, conn)\n",
    "    conn.close()\n",
    "    \n",
    "    print(tabulate(df, headers='keys', tablefmt='pretty', showindex=False))\n",
    "    \n",
    "    total = df['Variant_Count'].sum()\n",
    "    multi_pop = df[df['Number_of_Populations'] >= 2]['Variant_Count'].sum()\n",
    "    \n",
    "    print(f\"\\n📊 Summary:\")\n",
    "    print(f\"  • Total significant variants: {total:,}\")\n",
    "    print(f\"  • Multi-population (2+): {multi_pop:,} ({multi_pop/total*100:.1f}%)\")\n",
    "    print(f\"  • Single population only: {total-multi_pop:,} ({(total-multi_pop)/total*100:.1f}%)\")\n",
    "    \n",
    "    return df\n",
    "\n",
    "def query3_genes_with_multi_pop_variants():\n",
    "    \"\"\"Genes with most multi-population significant variants\"\"\"\n",
    "    print(\"\\n\" + \"=\"*100)\n",
    "    print(\"GENES WITH MOST MULTI-POPULATION SIGNIFICANT VARIANTS\")\n",
    "    print(\"=\"*100 + \"\\n\")\n",
    "    \n",
    "    query = \"\"\"\n",
    "    WITH multi_pop_variants AS (\n",
    "        SELECT \n",
    "            v.gene_name,\n",
    "            v.rs_id,\n",
    "            COUNT(DISTINCT st.population) AS pop_count\n",
    "        FROM statistical_tests st\n",
    "        INNER JOIN variants v ON st.variant_id = v.variant_id\n",
    "        INNER JOIN allele_frequencies af_eur \n",
    "            ON st.variant_id = af_eur.variant_id \n",
    "            AND af_eur.population = 'EUR'\n",
    "        INNER JOIN allele_frequencies af_pop \n",
    "            ON st.variant_id = af_pop.variant_id \n",
    "            AND af_pop.population = st.population\n",
    "        WHERE \n",
    "            st.chi_square_p_value <= 0.01\n",
    "            AND (af_eur.allele_frequency >= 0.05 OR af_pop.allele_frequency >= 0.05)\n",
    "            AND st.fst_value >= 0.15\n",
    "        GROUP BY v.gene_name, v.rs_id\n",
    "        HAVING COUNT(DISTINCT st.population) >= 2\n",
    "    )\n",
    "    SELECT \n",
    "        gene_name AS 'Gene',\n",
    "        COUNT(*) AS 'Multi_Pop_Variants',\n",
    "        ROUND(AVG(pop_count), 2) AS 'Avg_Pops_Per_Variant',\n",
    "        MAX(pop_count) AS 'Max_Populations',\n",
    "        SUM(pop_count) AS 'Total_Pop_Hits'\n",
    "    FROM multi_pop_variants\n",
    "    GROUP BY gene_name\n",
    "    ORDER BY COUNT(*) DESC, AVG(pop_count) DESC\n",
    "    LIMIT 20\n",
    "    \"\"\"\n",
    "    \n",
    "    conn = get_connection()\n",
    "    df = pd.read_sql_query(query, conn)\n",
    "    conn.close()\n",
    "    \n",
    "    if len(df) == 0:\n",
    "        print(\"⚠️  No genes found with multi-population variants\")\n",
    "        return None\n",
    "    \n",
    "    print(tabulate(df, headers='keys', tablefmt='pretty', showindex=False))\n",
    "    print(f\"\\n✓ {len(df)} genes have variants significant in 2+ populations\")\n",
    "    \n",
    "    return df\n",
    "\n",
    "def query4_specific_variant(rs_id):\n",
    "    \"\"\"Detailed analysis for a specific variant across all populations\"\"\"\n",
    "    print(\"\\n\" + \"=\"*100)\n",
    "    print(f\"DETAILED ANALYSIS FOR {rs_id} - ALL POPULATIONS\")\n",
    "    print(\"=\"*100 + \"\\n\")\n",
    "    \n",
    "    query = \"\"\"\n",
    "    SELECT \n",
    "        st.population,\n",
    "        ROUND(st.chi_square_p_value, 10) AS p_value,\n",
    "        ROUND(st.fst_value, 4) AS fst,\n",
    "        ROUND(af_eur.allele_frequency, 4) AS af_eur,\n",
    "        ROUND(af_pop.allele_frequency, 4) AS af_pop,\n",
    "        CASE WHEN st.chi_square_p_value <= 0.01 THEN '✓' ELSE '✗' END AS p_check,\n",
    "        CASE WHEN (af_eur.allele_frequency >= 0.05 OR af_pop.allele_frequency >= 0.05) \n",
    "             THEN '✓' ELSE '✗' END AS af_check,\n",
    "        CASE WHEN st.fst_value >= 0.15 THEN '✓' ELSE '✗' END AS fst_check,\n",
    "        CASE \n",
    "            WHEN st.chi_square_p_value <= 0.01 \n",
    "             AND (af_eur.allele_frequency >= 0.05 OR af_pop.allele_frequency >= 0.05)\n",
    "             AND st.fst_value >= 0.15 \n",
    "            THEN '✅ PASS'\n",
    "            ELSE '❌ FAIL'\n",
    "        END AS result\n",
    "    FROM statistical_tests st\n",
    "    INNER JOIN variants v ON st.variant_id = v.variant_id\n",
    "    INNER JOIN allele_frequencies af_eur \n",
    "        ON st.variant_id = af_eur.variant_id \n",
    "        AND af_eur.population = 'EUR'\n",
    "    INNER JOIN allele_frequencies af_pop \n",
    "        ON st.variant_id = af_pop.variant_id \n",
    "        AND af_pop.population = st.population\n",
    "    WHERE v.rs_id = ?\n",
    "    ORDER BY \n",
    "        CASE \n",
    "            WHEN st.chi_square_p_value <= 0.01 \n",
    "             AND (af_eur.allele_frequency >= 0.05 OR af_pop.allele_frequency >= 0.05)\n",
    "             AND st.fst_value >= 0.15 \n",
    "            THEN 1 ELSE 2 \n",
    "        END,\n",
    "        st.fst_value DESC\n",
    "    \"\"\"\n",
    "    \n",
    "    conn = get_connection()\n",
    "    df = pd.read_sql_query(query, conn, params=(rs_id,))\n",
    "    conn.close()\n",
    "    \n",
    "    if len(df) == 0:\n",
    "        print(f\"⚠️  Variant {rs_id} not found in database\")\n",
    "        return None\n",
    "    \n",
    "    print(tabulate(df, headers='keys', tablefmt='pretty', showindex=False))\n",
    "    \n",
    "    passed = df[df['result'] == '✅ PASS']\n",
    "    print(f\"\\n📊 Summary for {rs_id}:\")\n",
    "    print(f\"  • Populations tested: {len(df)}\")\n",
    "    print(f\"  • Populations passing ALL THREE criteria: {len(passed)}\")\n",
    "    \n",
    "    if len(passed) > 0:\n",
    "        print(f\"  • Passing populations: {', '.join(passed['population'].tolist())}\")\n",
    "    \n",
    "    return df\n",
    "\n",
    "def query5_sas_analysis():\n",
    "    \"\"\"Variants significant across South Asian populations\"\"\"\n",
    "    print(\"\\n\" + \"=\"*100)\n",
    "    print(\"VARIANTS SIGNIFICANT IN MULTIPLE SOUTH ASIAN POPULATIONS\")\n",
    "    print(\"=\"*100)\n",
    "    print(\"Looking at: BEB, GIH, ITU, PJL, STU\")\n",
    "    print(\"=\"*100 + \"\\n\")\n",
    "    \n",
    "    query = \"\"\"\n",
    "    WITH sas_significant AS (\n",
    "        SELECT \n",
    "            v.rs_id,\n",
    "            v.gene_name,\n",
    "            st.population,\n",
    "            st.fst_value\n",
    "        FROM statistical_tests st\n",
    "        INNER JOIN variants v ON st.variant_id = v.variant_id\n",
    "        INNER JOIN allele_frequencies af_eur \n",
    "            ON st.variant_id = af_eur.variant_id \n",
    "            AND af_eur.population = 'EUR'\n",
    "        INNER JOIN allele_frequencies af_pop \n",
    "            ON st.variant_id = af_pop.variant_id \n",
    "            AND af_pop.population = st.population\n",
    "        WHERE \n",
    "            st.population IN ('BEB', 'GIH', 'ITU', 'PJL', 'STU')\n",
    "            AND st.chi_square_p_value <= 0.01\n",
    "            AND (af_eur.allele_frequency >= 0.05 OR af_pop.allele_frequency >= 0.05)\n",
    "            AND st.fst_value >= 0.15\n",
    "    )\n",
    "    SELECT \n",
    "        rs_id,\n",
    "        gene_name,\n",
    "        COUNT(DISTINCT population) AS sas_pop_count,\n",
    "        GROUP_CONCAT(population) AS sas_populations,\n",
    "        ROUND(AVG(fst_value), 4) AS avg_fst\n",
    "    FROM sas_significant\n",
    "    GROUP BY rs_id, gene_name\n",
    "    HAVING COUNT(DISTINCT population) >= 2\n",
    "    ORDER BY COUNT(DISTINCT population) DESC, avg_fst DESC\n",
    "    \"\"\"\n",
    "    \n",
    "    conn = get_connection()\n",
    "    df = pd.read_sql_query(query, conn)\n",
    "    conn.close()\n",
    "    \n",
    "    if len(df) == 0:\n",
    "        print(\"⚠️  No variants found significant in 2+ South Asian populations\")\n",
    "        return None\n",
    "    \n",
    "    print(tabulate(df, headers='keys', tablefmt='pretty', showindex=False))\n",
    "    print(f\"\\n✓ {len(df)} variants significant in 2+ South Asian populations\")\n",
    "    \n",
    "    return df\n",
    "\n",
    "def export_multi_pop_variants(min_populations=2, filename=None):\n",
    "    \"\"\"Export multi-population variants to CSV\"\"\"\n",
    "    print(\"\\n\" + \"=\"*100)\n",
    "    print(f\"EXPORTING VARIANTS SIGNIFICANT IN {min_populations}+ POPULATIONS\")\n",
    "    print(\"=\"*100 + \"\\n\")\n",
    "    \n",
    "    query = \"\"\"\n",
    "    WITH multi_pop_counts AS (\n",
    "        SELECT \n",
    "            v.variant_id,\n",
    "            v.rs_id,\n",
    "            COUNT(DISTINCT st.population) AS pop_count\n",
    "        FROM statistical_tests st\n",
    "        INNER JOIN variants v ON st.variant_id = v.variant_id\n",
    "        INNER JOIN allele_frequencies af_eur \n",
    "            ON st.variant_id = af_eur.variant_id \n",
    "            AND af_eur.population = 'EUR'\n",
    "        INNER JOIN allele_frequencies af_pop \n",
    "            ON st.variant_id = af_pop.variant_id \n",
    "            AND af_pop.population = st.population\n",
    "        WHERE \n",
    "            st.chi_square_p_value <= 0.01\n",
    "            AND (af_eur.allele_frequency >= 0.05 OR af_pop.allele_frequency >= 0.05)\n",
    "            AND st.fst_value >= 0.15\n",
    "        GROUP BY v.variant_id, v.rs_id\n",
    "        HAVING COUNT(DISTINCT st.population) >= ?\n",
    "    )\n",
    "    SELECT \n",
    "        v.rs_id,\n",
    "        v.gene_name,\n",
    "        v.chrom,\n",
    "        v.position,\n",
    "        v.ref_allele,\n",
    "        v.alt_allele,\n",
    "        mpc.pop_count AS total_significant_populations,\n",
    "        st.population,\n",
    "        ROUND(st.chi_square_p_value, 10) AS p_value,\n",
    "        ROUND(st.fst_value, 6) AS fst,\n",
    "        ROUND(af_eur.allele_frequency, 6) AS af_eur,\n",
    "        ROUND(af_pop.allele_frequency, 6) AS af_population,\n",
    "        CASE WHEN ld.ld_pruned = 1 THEN 'Yes' ELSE 'No' END AS passed_ld_pruning,\n",
    "        v.dbsnp_link\n",
    "    FROM multi_pop_counts mpc\n",
    "    INNER JOIN variants v ON mpc.variant_id = v.variant_id\n",
    "    INNER JOIN statistical_tests st ON v.variant_id = st.variant_id\n",
    "    INNER JOIN allele_frequencies af_eur \n",
    "        ON st.variant_id = af_eur.variant_id \n",
    "        AND af_eur.population = 'EUR'\n",
    "    INNER JOIN allele_frequencies af_pop \n",
    "        ON st.variant_id = af_pop.variant_id \n",
    "        AND af_pop.population = st.population\n",
    "    LEFT JOIN ld_status ld ON v.variant_id = ld.variant_id\n",
    "    WHERE \n",
    "        st.chi_square_p_value <= 0.01\n",
    "        AND (af_eur.allele_frequency >= 0.05 OR af_pop.allele_frequency >= 0.05)\n",
    "        AND st.fst_value >= 0.15\n",
    "    ORDER BY mpc.pop_count DESC, v.rs_id, st.fst_value DESC\n",
    "    \"\"\"\n",
    "    \n",
    "    conn = get_connection()\n",
    "    df = pd.read_sql_query(query, conn, params=(min_populations,))\n",
    "    conn.close()\n",
    "    \n",
    "    if len(df) == 0:\n",
    "        print(f\"⚠️  No variants found significant in {min_populations}+ populations\")\n",
    "        return None\n",
    "    \n",
    "    if filename is None:\n",
    "        filename = f'multi_population_variants_{min_populations}plus.csv'\n",
    "    \n",
    "    df.to_csv(filename, index=False)\n",
    "    \n",
    "    unique_variants = df['rs_id'].nunique()\n",
    "    print(f\"✓ Exported {len(df)} rows ({unique_variants} unique variants)\")\n",
    "    print(f\"✓ Saved to: {filename}\")\n",
    "    \n",
    "    return df\n",
    "\n",
    "def main():\n",
    "    \"\"\"Run all analyses\"\"\"\n",
    "    print(\"\\n\" + \"🧬\"*50)\n",
    "    print(\"MULTI-POPULATION SIGNIFICANT VARIANTS ANALYSIS\")\n",
    "    print(\"🧬\"*50)\n",
    "    \n",
    "    # Query 1: Summary\n",
    "    df1 = query1_multi_pop_summary()\n",
    "    \n",
    "    # Query 2: Distribution\n",
    "    df2 = query2_distribution()\n",
    "    \n",
    "    # Query 3: Genes\n",
    "    df3 = query3_genes_with_multi_pop_variants()\n",
    "    \n",
    "    # Query 4: Specific variant example (if exists)\n",
    "    if df1 is not None and len(df1) > 0:\n",
    "        example_variant = df1.iloc[0]['rs_id']\n",
    "        df4 = query4_specific_variant(example_variant)\n",
    "    \n",
    "    # Query 5: South Asian analysis\n",
    "    df5 = query5_sas_analysis()\n",
    "    \n",
    "    # Export option\n",
    "    print(\"\\n\" + \"=\"*100)\n",
    "    print(\"EXPORT OPTIONS\")\n",
    "    print(\"=\"*100)\n",
    "    \n",
    "    export_choice = input(\"\\nExport multi-population variants to CSV? (yes/no): \").lower().strip()\n",
    "    \n",
    "    if export_choice == 'yes':\n",
    "        min_pops = input(\"Minimum number of populations (default=2): \").strip()\n",
    "        min_pops = int(min_pops) if min_pops.isdigit() else 2\n",
    "        export_multi_pop_variants(min_populations=min_pops)\n",
    "    \n",
    "    print(\"\\n\" + \"=\"*100)\n",
    "    print(\"✅ ANALYSIS COMPLETE!\")\n",
    "    print(\"=\"*100)\n",
    "    \n",
    "    # Summary\n",
    "    if df1 is not None:\n",
    "        print(f\"\\n📊 Key Findings:\")\n",
    "        print(f\"  • Variants in 2+ populations: {len(df1['rs_id'].unique())}\")\n",
    "        if len(df1) > 0:\n",
    "            print(f\"  • Top variant: {df1.iloc[0]['rs_id']} ({df1.iloc[0]['gene_name']}) in {df1.iloc[0]['population_count']} populations\")\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    try:\n",
    "        main()\n",
    "    except Exception as e:\n",
    "        print(f\"\\n❌ Error: {e}\")\n",
    "        print(\"\\nMake sure:\")\n",
    "        print(\"1. Database file 'nephrotic_syndrome.db' exists\")\n",
    "        print(\"2. Install required packages: pip install pandas tabulate\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c4872e06",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
