{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "f3de1a83",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Merged file created successfully!\n",
      "Total rows: 1060588\n",
      "\n",
      "First few rows:\n",
      "   CHROM      POS            ID REF ALT        AF  GENE    AF_AFR    AF_AMR  \\\n",
      "0  chr19  1248349   rs112363642   G   A       NaN   NaN  0.019037  0.217347   \n",
      "1  chr19  1248372   rs145305690   C   T  0.000468  MIDN  0.000000  0.000000   \n",
      "2  chr19  1248399  rs1404444736   G   C  0.000312  MIDN  0.000000  0.000000   \n",
      "3  chr19  1248405   rs535078794   C   T  0.000468  MIDN  0.000000  0.000000   \n",
      "4  chr19  1248419   rs371906837   C   A  0.003748  MIDN  0.012878  0.001020   \n",
      "\n",
      "   AF_AMR.1  ...  P_EUR_AMR  P_EUR_AMR1  P_EUR_AMR2  P_EUR_BEB  P_EUR_EAS  \\\n",
      "0  0.241697  ...   0.973014    0.969612    0.897267    0.99913   0.603798   \n",
      "1  0.000000  ...   1.000000    1.000000    1.000000    1.00000   0.000000   \n",
      "2  0.000000  ...   0.000000    0.000000    0.000000    0.00000   0.000000   \n",
      "3  0.000000  ...   1.000000    1.000000    1.000000    1.00000   0.000000   \n",
      "4  0.000000  ...   0.000000    1.000000    0.000000    1.00000   1.000000   \n",
      "\n",
      "   P_EUR_GIH  P_EUR_ITU  P_EUR_PJL  P_EUR_SAS  P_EUR_STU  \n",
      "0   0.987329     0.8941   0.926662   0.967103   0.967103  \n",
      "1   1.000000     1.0000   1.000000   1.000000   1.000000  \n",
      "2   0.000000     0.0000   0.000000   0.000000   0.000000  \n",
      "3   1.000000     1.0000   1.000000   1.000000   1.000000  \n",
      "4   1.000000     1.0000   1.000000   1.000000   1.000000  \n",
      "\n",
      "[5 rows x 41 columns]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Read the two TSV files\n",
    "final_chi = pd.read_csv('Final_chi_sq_fst_result.tsv', sep='\\t')\n",
    "pop_af = pd.read_csv('pop_AF_with_gene.raw.tsv', sep=' ', comment='#')\n",
    "\n",
    "# The pop_af file has a comment line, so we need to set proper column names\n",
    "# Based on the comment line in the file\n",
    "pop_af.columns = ['CHROM', 'POS', 'ID', 'REF', 'ALT', 'AF', 'AF_EUR', 'AF_AFR', \n",
    "                  'AF_EAS', 'AF_AMR', 'AF_AMR1', 'AF_AMR2', 'AF_SAS', 'AF_GIH', \n",
    "                  'AF_ITU', 'AF_BEB', 'AF_PJL', 'AF_STU', 'GENE']\n",
    "\n",
    "# Extract only AF and GENE columns along with merge keys\n",
    "pop_af_subset = pop_af[['CHROM', 'POS', 'ID', 'AF', 'GENE']]\n",
    "\n",
    "# Merge the dataframes on CHROM, POS, and ID\n",
    "merged = pd.merge(final_chi, pop_af_subset, on=['CHROM', 'POS', 'ID'], how='left')\n",
    "\n",
    "# Reorder columns to put AF and GENE after the basic variant info (CHROM, POS, ID, REF, ALT)\n",
    "cols = merged.columns.tolist()\n",
    "# Move AF and GENE to position 5 (right after ALT)\n",
    "basic_cols = ['CHROM', 'POS', 'ID', 'REF', 'ALT', 'AF', 'GENE']\n",
    "other_cols = [col for col in cols if col not in basic_cols]\n",
    "final_cols = basic_cols + other_cols\n",
    "\n",
    "merged = merged[final_cols]\n",
    "\n",
    "# Save the result\n",
    "merged.to_csv('Final_chi_sq_fst_result_with_AF_GENE.tsv', sep='\\t', index=False)\n",
    "\n",
    "print(f\"Merged file created successfully!\")\n",
    "print(f\"Total rows: {len(merged)}\")\n",
    "print(f\"\\nFirst few rows:\")\n",
    "print(merged.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "ee5d55e0",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"Final_chi_sq_fst_result_with_AF_GENE.tsv\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "590fbf94",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "CHROM",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "POS",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "ID",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "REF",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "ALT",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "AF",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "GENE",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "AF_AFR",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "AF_AMR",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "AF_AMR.1",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "AF_AMR.2",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "AF_EAS",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "AF_EUR",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "AF_SAS",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "AF_BEB",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "AF_GIH",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "AF_ITU",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "AF_PJL",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "AF_STU",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "FST_EUR_VS_AFR",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "FST_EUR_VS_AMR",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "FST_EUR_VS_AMR1",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "FST_EUR_VS_AMR2",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "FST_EUR_VS_BEB",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "FST_EUR_VS_EAS",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "FST_EUR_VS_GIH",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "FST_EUR_VS_ITU",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "FST_EUR_VS_PJL",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "FST_EUR_VS_SAS",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "FST_EUR_VS_STU",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "P_EUR_AFR",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "P_EUR_AMR",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "P_EUR_AMR1",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "P_EUR_AMR2",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "P_EUR_BEB",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "P_EUR_EAS",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "P_EUR_GIH",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "P_EUR_ITU",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "P_EUR_PJL",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "P_EUR_SAS",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "P_EUR_STU",
         "rawType": "float64",
         "type": "float"
        }
       ],
       "ref": "75fe48e8-a618-4b6e-980a-cbde824bcff8",
       "rows": [
        [
         "0",
         "chr19",
         "1248349",
         "rs112363642",
         "G",
         "A",
         null,
         null,
         "0.019037",
         "0.217347",
         "0.241697",
         "0.187215",
         "0.42906",
         "0.228639",
         "0.214912",
         "0.229008",
         "0.223301",
         "0.275701",
         "0.19863",
         "0.214912",
         "0.206717",
         "0.0",
         "0.0",
         "0.00348294",
         "0.0",
         "0.0867858",
         "0.0",
         "0.00336793",
         "0.000420678",
         "0.0",
         "0.0",
         "0.113852149",
         "0.97301411",
         "0.969611797",
         "0.897266583",
         "0.999129575",
         "0.603798001",
         "0.987328726",
         "0.894099823",
         "0.926661539",
         "0.967102804",
         "0.967102804"
        ],
        [
         "1",
         "chr19",
         "1248372",
         "rs145305690",
         "C",
         "T",
         "0.000468457",
         "MIDN",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0025641",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.00184687",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0",
         "0.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0"
        ],
        [
         "2",
         "chr19",
         "1248399",
         "rs1404444736",
         "G",
         "C",
         "0.000312305",
         "MIDN",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.00158228",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.00123435",
         "0.000500099",
         "0.0",
         "0.0",
         "0.0",
         "0.00070182",
         "0.0",
         "0.0",
         "0.0",
         "0.000733212",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0"
        ],
        [
         "3",
         "chr19",
         "1248405",
         "rs535078794",
         "C",
         "T",
         "0.000468457",
         "MIDN",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0025641",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.00184687",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0",
         "0.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0"
        ],
        [
         "4",
         "chr19",
         "1248419",
         "rs371906837",
         "C",
         "A",
         "0.00374766",
         "MIDN",
         "0.0128779",
         "0.00102041",
         "0.0",
         "0.0022831",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0102978",
         "0.000262689",
         "0.0",
         "0.002897",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "1.0",
         "0.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0",
         "1.0"
        ]
       ],
       "shape": {
        "columns": 41,
        "rows": 5
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AF</th>\n",
       "      <th>GENE</th>\n",
       "      <th>AF_AFR</th>\n",
       "      <th>AF_AMR</th>\n",
       "      <th>AF_AMR.1</th>\n",
       "      <th>...</th>\n",
       "      <th>P_EUR_AMR</th>\n",
       "      <th>P_EUR_AMR1</th>\n",
       "      <th>P_EUR_AMR2</th>\n",
       "      <th>P_EUR_BEB</th>\n",
       "      <th>P_EUR_EAS</th>\n",
       "      <th>P_EUR_GIH</th>\n",
       "      <th>P_EUR_ITU</th>\n",
       "      <th>P_EUR_PJL</th>\n",
       "      <th>P_EUR_SAS</th>\n",
       "      <th>P_EUR_STU</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248349</td>\n",
       "      <td>rs112363642</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.019037</td>\n",
       "      <td>0.217347</td>\n",
       "      <td>0.241697</td>\n",
       "      <td>...</td>\n",
       "      <td>0.973014</td>\n",
       "      <td>0.969612</td>\n",
       "      <td>0.897267</td>\n",
       "      <td>0.99913</td>\n",
       "      <td>0.603798</td>\n",
       "      <td>0.987329</td>\n",
       "      <td>0.8941</td>\n",
       "      <td>0.926662</td>\n",
       "      <td>0.967103</td>\n",
       "      <td>0.967103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248372</td>\n",
       "      <td>rs145305690</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000468</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248399</td>\n",
       "      <td>rs1404444736</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000312</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248405</td>\n",
       "      <td>rs535078794</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000468</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248419</td>\n",
       "      <td>rs371906837</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.003748</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.012878</td>\n",
       "      <td>0.001020</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM      POS            ID REF ALT        AF  GENE    AF_AFR    AF_AMR  \\\n",
       "0  chr19  1248349   rs112363642   G   A       NaN   NaN  0.019037  0.217347   \n",
       "1  chr19  1248372   rs145305690   C   T  0.000468  MIDN  0.000000  0.000000   \n",
       "2  chr19  1248399  rs1404444736   G   C  0.000312  MIDN  0.000000  0.000000   \n",
       "3  chr19  1248405   rs535078794   C   T  0.000468  MIDN  0.000000  0.000000   \n",
       "4  chr19  1248419   rs371906837   C   A  0.003748  MIDN  0.012878  0.001020   \n",
       "\n",
       "   AF_AMR.1  ...  P_EUR_AMR  P_EUR_AMR1  P_EUR_AMR2  P_EUR_BEB  P_EUR_EAS  \\\n",
       "0  0.241697  ...   0.973014    0.969612    0.897267    0.99913   0.603798   \n",
       "1  0.000000  ...   1.000000    1.000000    1.000000    1.00000   0.000000   \n",
       "2  0.000000  ...   0.000000    0.000000    0.000000    0.00000   0.000000   \n",
       "3  0.000000  ...   1.000000    1.000000    1.000000    1.00000   0.000000   \n",
       "4  0.000000  ...   0.000000    1.000000    0.000000    1.00000   1.000000   \n",
       "\n",
       "   P_EUR_GIH  P_EUR_ITU  P_EUR_PJL  P_EUR_SAS  P_EUR_STU  \n",
       "0   0.987329     0.8941   0.926662   0.967103   0.967103  \n",
       "1   1.000000     1.0000   1.000000   1.000000   1.000000  \n",
       "2   0.000000     0.0000   0.000000   0.000000   0.000000  \n",
       "3   1.000000     1.0000   1.000000   1.000000   1.000000  \n",
       "4   1.000000     1.0000   1.000000   1.000000   1.000000  \n",
       "\n",
       "[5 rows x 41 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6fa694dd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "0",
         "rawType": "int64",
         "type": "integer"
        }
       ],
       "ref": "c3a1d824-ee8f-4933-a232-5fbcdd25a8b7",
       "rows": [
        [
         "CHROM",
         "0"
        ],
        [
         "POS",
         "0"
        ],
        [
         "ID",
         "0"
        ],
        [
         "REF",
         "0"
        ],
        [
         "ALT",
         "0"
        ],
        [
         "AF",
         "1"
        ],
        [
         "GENE",
         "1"
        ],
        [
         "AF_AFR",
         "0"
        ],
        [
         "AF_AMR",
         "0"
        ],
        [
         "AF_AMR.1",
         "0"
        ],
        [
         "AF_AMR.2",
         "0"
        ],
        [
         "AF_EAS",
         "0"
        ],
        [
         "AF_EUR",
         "0"
        ],
        [
         "AF_SAS",
         "0"
        ],
        [
         "AF_BEB",
         "0"
        ],
        [
         "AF_GIH",
         "0"
        ],
        [
         "AF_ITU",
         "0"
        ],
        [
         "AF_PJL",
         "0"
        ],
        [
         "AF_STU",
         "0"
        ],
        [
         "FST_EUR_VS_AFR",
         "0"
        ],
        [
         "FST_EUR_VS_AMR",
         "0"
        ],
        [
         "FST_EUR_VS_AMR1",
         "0"
        ],
        [
         "FST_EUR_VS_AMR2",
         "0"
        ],
        [
         "FST_EUR_VS_BEB",
         "0"
        ],
        [
         "FST_EUR_VS_EAS",
         "0"
        ],
        [
         "FST_EUR_VS_GIH",
         "0"
        ],
        [
         "FST_EUR_VS_ITU",
         "0"
        ],
        [
         "FST_EUR_VS_PJL",
         "0"
        ],
        [
         "FST_EUR_VS_SAS",
         "0"
        ],
        [
         "FST_EUR_VS_STU",
         "0"
        ],
        [
         "P_EUR_AFR",
         "0"
        ],
        [
         "P_EUR_AMR",
         "0"
        ],
        [
         "P_EUR_AMR1",
         "0"
        ],
        [
         "P_EUR_AMR2",
         "0"
        ],
        [
         "P_EUR_BEB",
         "0"
        ],
        [
         "P_EUR_EAS",
         "0"
        ],
        [
         "P_EUR_GIH",
         "0"
        ],
        [
         "P_EUR_ITU",
         "0"
        ],
        [
         "P_EUR_PJL",
         "0"
        ],
        [
         "P_EUR_SAS",
         "0"
        ],
        [
         "P_EUR_STU",
         "0"
        ]
       ],
       "shape": {
        "columns": 1,
        "rows": 41
       }
      },
      "text/plain": [
       "CHROM              0\n",
       "POS                0\n",
       "ID                 0\n",
       "REF                0\n",
       "ALT                0\n",
       "AF                 1\n",
       "GENE               1\n",
       "AF_AFR             0\n",
       "AF_AMR             0\n",
       "AF_AMR.1           0\n",
       "AF_AMR.2           0\n",
       "AF_EAS             0\n",
       "AF_EUR             0\n",
       "AF_SAS             0\n",
       "AF_BEB             0\n",
       "AF_GIH             0\n",
       "AF_ITU             0\n",
       "AF_PJL             0\n",
       "AF_STU             0\n",
       "FST_EUR_VS_AFR     0\n",
       "FST_EUR_VS_AMR     0\n",
       "FST_EUR_VS_AMR1    0\n",
       "FST_EUR_VS_AMR2    0\n",
       "FST_EUR_VS_BEB     0\n",
       "FST_EUR_VS_EAS     0\n",
       "FST_EUR_VS_GIH     0\n",
       "FST_EUR_VS_ITU     0\n",
       "FST_EUR_VS_PJL     0\n",
       "FST_EUR_VS_SAS     0\n",
       "FST_EUR_VS_STU     0\n",
       "P_EUR_AFR          0\n",
       "P_EUR_AMR          0\n",
       "P_EUR_AMR1         0\n",
       "P_EUR_AMR2         0\n",
       "P_EUR_BEB          0\n",
       "P_EUR_EAS          0\n",
       "P_EUR_GIH          0\n",
       "P_EUR_ITU          0\n",
       "P_EUR_PJL          0\n",
       "P_EUR_SAS          0\n",
       "P_EUR_STU          0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.isnull().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "1ca30276",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "2c3271b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_excel(\"Final_gene_list.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "c8d3c855",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Protein</th>\n",
       "      <th>Population</th>\n",
       "      <th>Gene Function</th>\n",
       "      <th>Primary Pathway</th>\n",
       "      <th>Cellular Location</th>\n",
       "      <th>Interacting Genes</th>\n",
       "      <th>SRNS/SSNS</th>\n",
       "      <th>Phenotypes</th>\n",
       "      <th>Inheritance</th>\n",
       "      <th>Pediatric (Y/N)</th>\n",
       "      <th>Adult (Y/N)</th>\n",
       "      <th>Both (Y/N)</th>\n",
       "      <th>Database</th>\n",
       "      <th>Database Link</th>\n",
       "      <th>Source Paper 1</th>\n",
       "      <th>Source Paper 2</th>\n",
       "      <th>Source Paper 3</th>\n",
       "      <th>Category</th>\n",
       "      <th>Variation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NPHS1</td>\n",
       "      <td>nephrin</td>\n",
       "      <td>Finnish type , highest in Finland but occurs i...</td>\n",
       "      <td>mutations in the gene encoSeems to play a role...</td>\n",
       "      <td>Cell-Cell communication , Nephrin family inter...</td>\n",
       "      <td>The gene is primarily expressed in renal tissu...</td>\n",
       "      <td>IQGAP1, NCK1, ARRB2 , CANX , FYN</td>\n",
       "      <td>SRNS</td>\n",
       "      <td>CNS, SRNS (NPHS1)</td>\n",
       "      <td>AR</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>Gwas</td>\n",
       "      <td>https://www.ebi.ac.uk/gwas/variants/rs2285450</td>\n",
       "      <td>https://pmc.ncbi.nlm.nih.gov/articles/PMC53836...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NPHS2</td>\n",
       "      <td>podocin</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Plays a role in the regulation of glomerular p...</td>\n",
       "      <td>Cell-Cell communication , Nephrin family inter...</td>\n",
       "      <td>Almost exclusively expressed in the podocytes ...</td>\n",
       "      <td>bcs1l , yme1l1 ,sgp7, afg3l2</td>\n",
       "      <td>SRNS</td>\n",
       "      <td>CNS, SRNS (NPHS2) &amp; Adult both</td>\n",
       "      <td>AR</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>Omim</td>\n",
       "      <td>https://www.omim.org/entry/604766?search=%22ne...</td>\n",
       "      <td>https://pmc.ncbi.nlm.nih.gov/articles/PMC53836...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PLCE1</td>\n",
       "      <td>phospholipase C, ε1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The production of the second messenger molecul...</td>\n",
       "      <td>GPR40 pathway, Nephrotic syndrome, Phosphoinos...</td>\n",
       "      <td>Expressed in podocytes</td>\n",
       "      <td>RHOA, HRAS, RAP1A, RAP2A, RAP2B, RRAS, AVIL , ...</td>\n",
       "      <td>SRNS</td>\n",
       "      <td>DMS, SRNS (NPHS3)</td>\n",
       "      <td>AR</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>Omim</td>\n",
       "      <td>https://www.omim.org/entry/610725?search=%22ne...</td>\n",
       "      <td>https://pmc.ncbi.nlm.nih.gov/articles/PMC53836...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CD2AP</td>\n",
       "      <td>CD2-associated protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seems to act as an adapter protein between mem...</td>\n",
       "      <td>GDNF signaling,Genes controlling nephrogenesis...</td>\n",
       "      <td>Widely expressed in fetal and adult tissues.</td>\n",
       "      <td>PKD2, NPHS1, NPHS2, WTIP, DDN, CBL BCAR1/p130...</td>\n",
       "      <td>SRNS</td>\n",
       "      <td>SRNS (FSGS3)</td>\n",
       "      <td>AD/AR</td>\n",
       "      <td>yes(recessive mutation)</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Omim</td>\n",
       "      <td>https://www.omim.org/entry/607832?search=%22ne...</td>\n",
       "      <td>https://pmc.ncbi.nlm.nih.gov/articles/PMC53836...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FAT1</td>\n",
       "      <td>FAT1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NS, Ciliopathy</td>\n",
       "      <td>AR</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>Omim</td>\n",
       "      <td>https://www.omim.org/entry/600976?search=%22ne...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene                 Protein  \\\n",
       "0  NPHS1                 nephrin   \n",
       "1  NPHS2                 podocin   \n",
       "2  PLCE1     phospholipase C, ε1   \n",
       "3  CD2AP  CD2-associated protein   \n",
       "4   FAT1                    FAT1   \n",
       "\n",
       "                                          Population  \\\n",
       "0  Finnish type , highest in Finland but occurs i...   \n",
       "1                                                NaN   \n",
       "2                                                NaN   \n",
       "3                                                NaN   \n",
       "4                                                NaN   \n",
       "\n",
       "                                       Gene Function  \\\n",
       "0  mutations in the gene encoSeems to play a role...   \n",
       "1  Plays a role in the regulation of glomerular p...   \n",
       "2  The production of the second messenger molecul...   \n",
       "3  Seems to act as an adapter protein between mem...   \n",
       "4                                                NaN   \n",
       "\n",
       "                                     Primary Pathway  \\\n",
       "0  Cell-Cell communication , Nephrin family inter...   \n",
       "1  Cell-Cell communication , Nephrin family inter...   \n",
       "2  GPR40 pathway, Nephrotic syndrome, Phosphoinos...   \n",
       "3  GDNF signaling,Genes controlling nephrogenesis...   \n",
       "4                                                NaN   \n",
       "\n",
       "                                   Cellular Location  \\\n",
       "0  The gene is primarily expressed in renal tissu...   \n",
       "1  Almost exclusively expressed in the podocytes ...   \n",
       "2                             Expressed in podocytes   \n",
       "3       Widely expressed in fetal and adult tissues.   \n",
       "4                                                NaN   \n",
       "\n",
       "                                   Interacting Genes SRNS/SSNS  \\\n",
       "0                   IQGAP1, NCK1, ARRB2 , CANX , FYN      SRNS   \n",
       "1                       bcs1l , yme1l1 ,sgp7, afg3l2      SRNS   \n",
       "2  RHOA, HRAS, RAP1A, RAP2A, RAP2B, RRAS, AVIL , ...      SRNS   \n",
       "3   PKD2, NPHS1, NPHS2, WTIP, DDN, CBL BCAR1/p130...      SRNS   \n",
       "4                                                NaN       NaN   \n",
       "\n",
       "                       Phenotypes Inheritance          Pediatric (Y/N)  \\\n",
       "0               CNS, SRNS (NPHS1)          AR                      yes   \n",
       "1  CNS, SRNS (NPHS2) & Adult both          AR                      yes   \n",
       "2               DMS, SRNS (NPHS3)          AR                      yes   \n",
       "3                    SRNS (FSGS3)       AD/AR  yes(recessive mutation)   \n",
       "4                  NS, Ciliopathy          AR                     yes    \n",
       "\n",
       "  Adult (Y/N) Both (Y/N) Database  \\\n",
       "0          no         no     Gwas   \n",
       "1          no        yes     Omim   \n",
       "2          no         no     Omim   \n",
       "3          no        NaN     Omim   \n",
       "4          no         no     Omim   \n",
       "\n",
       "                                       Database Link  \\\n",
       "0      https://www.ebi.ac.uk/gwas/variants/rs2285450   \n",
       "1  https://www.omim.org/entry/604766?search=%22ne...   \n",
       "2  https://www.omim.org/entry/610725?search=%22ne...   \n",
       "3  https://www.omim.org/entry/607832?search=%22ne...   \n",
       "4  https://www.omim.org/entry/600976?search=%22ne...   \n",
       "\n",
       "                                      Source Paper 1 Source Paper 2  \\\n",
       "0  https://pmc.ncbi.nlm.nih.gov/articles/PMC53836...            NaN   \n",
       "1  https://pmc.ncbi.nlm.nih.gov/articles/PMC53836...            NaN   \n",
       "2  https://pmc.ncbi.nlm.nih.gov/articles/PMC53836...            NaN   \n",
       "3  https://pmc.ncbi.nlm.nih.gov/articles/PMC53836...            NaN   \n",
       "4                                                NaN            NaN   \n",
       "\n",
       "   Source Paper 3  Category  Variation  \n",
       "0             NaN       NaN        NaN  \n",
       "1             NaN       NaN        NaN  \n",
       "2             NaN       NaN        NaN  \n",
       "3             NaN       NaN        NaN  \n",
       "4             NaN       NaN        NaN  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "4c0c9aed",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Gene', 'Protein', 'Population', 'Gene Function', 'Primary Pathway',\n",
       "       'Cellular Location', 'Interacting Genes', 'SRNS/SSNS', 'Phenotypes',\n",
       "       'Inheritance', ' Pediatric (Y/N)', 'Adult (Y/N)', 'Both (Y/N)',\n",
       "       'Database', 'Database Link', 'Source Paper 1', 'Source Paper 2',\n",
       "       'Source Paper 3', 'Category', 'Variation'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "b627a507",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 493 entries, 0 to 492\n",
      "Data columns (total 20 columns):\n",
      " #   Column             Non-Null Count  Dtype  \n",
      "---  ------             --------------  -----  \n",
      " 0   Gene               493 non-null    object \n",
      " 1   Protein            54 non-null     object \n",
      " 2   Population         6 non-null      object \n",
      " 3   Gene Function      24 non-null     object \n",
      " 4   Primary Pathway    18 non-null     object \n",
      " 5   Cellular Location  15 non-null     object \n",
      " 6   Interacting Genes  18 non-null     object \n",
      " 7   SRNS/SSNS          89 non-null     object \n",
      " 8   Phenotypes         166 non-null    object \n",
      " 9   Inheritance        82 non-null     object \n",
      " 10   Pediatric (Y/N)   152 non-null    object \n",
      " 11  Adult (Y/N)        154 non-null    object \n",
      " 12  Both (Y/N)         137 non-null    object \n",
      " 13  Database           468 non-null    object \n",
      " 14  Database Link      183 non-null    object \n",
      " 15  Source Paper 1     34 non-null     object \n",
      " 16  Source Paper 2     15 non-null     object \n",
      " 17  Source Paper 3     0 non-null      float64\n",
      " 18  Category           0 non-null      float64\n",
      " 19  Variation          0 non-null      float64\n",
      "dtypes: float64(3), object(17)\n",
      "memory usage: 77.2+ KB\n"
     ]
    }
   ],
   "source": [
    "df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "e14e9be5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(493, 20)"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "08bffa74",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"Final_chi_sq_fst_result_with_AF_GENE.tsv\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "37b6bb4b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AF</th>\n",
       "      <th>GENE</th>\n",
       "      <th>AF_AFR</th>\n",
       "      <th>AF_AMR</th>\n",
       "      <th>AF_AMR1</th>\n",
       "      <th>...</th>\n",
       "      <th>P_EUR_AMR</th>\n",
       "      <th>P_EUR_AMR1</th>\n",
       "      <th>P_EUR_AMR2</th>\n",
       "      <th>P_EUR_BEB</th>\n",
       "      <th>P_EUR_EAS</th>\n",
       "      <th>P_EUR_GIH</th>\n",
       "      <th>P_EUR_ITU</th>\n",
       "      <th>P_EUR_PJL</th>\n",
       "      <th>P_EUR_SAS</th>\n",
       "      <th>P_EUR_STU</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248349</td>\n",
       "      <td>rs112363642</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.204560</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.019037</td>\n",
       "      <td>0.217347</td>\n",
       "      <td>0.241697</td>\n",
       "      <td>...</td>\n",
       "      <td>0.973014</td>\n",
       "      <td>0.969612</td>\n",
       "      <td>0.897267</td>\n",
       "      <td>0.99913</td>\n",
       "      <td>0.603798</td>\n",
       "      <td>0.987329</td>\n",
       "      <td>0.8941</td>\n",
       "      <td>0.926662</td>\n",
       "      <td>0.967103</td>\n",
       "      <td>0.967103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248372</td>\n",
       "      <td>rs145305690</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000468</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248399</td>\n",
       "      <td>rs1404444736</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000312</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248405</td>\n",
       "      <td>rs535078794</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000468</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248419</td>\n",
       "      <td>rs371906837</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.003748</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.012878</td>\n",
       "      <td>0.001020</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM      POS            ID REF ALT        AF  GENE    AF_AFR    AF_AMR  \\\n",
       "0  chr19  1248349   rs112363642   G   A  0.204560  MIDN  0.019037  0.217347   \n",
       "1  chr19  1248372   rs145305690   C   T  0.000468  MIDN  0.000000  0.000000   \n",
       "2  chr19  1248399  rs1404444736   G   C  0.000312  MIDN  0.000000  0.000000   \n",
       "3  chr19  1248405   rs535078794   C   T  0.000468  MIDN  0.000000  0.000000   \n",
       "4  chr19  1248419   rs371906837   C   A  0.003748  MIDN  0.012878  0.001020   \n",
       "\n",
       "    AF_AMR1  ...  P_EUR_AMR  P_EUR_AMR1  P_EUR_AMR2  P_EUR_BEB  P_EUR_EAS  \\\n",
       "0  0.241697  ...   0.973014    0.969612    0.897267    0.99913   0.603798   \n",
       "1  0.000000  ...   1.000000    1.000000    1.000000    1.00000   0.000000   \n",
       "2  0.000000  ...   0.000000    0.000000    0.000000    0.00000   0.000000   \n",
       "3  0.000000  ...   1.000000    1.000000    1.000000    1.00000   0.000000   \n",
       "4  0.000000  ...   0.000000    1.000000    0.000000    1.00000   1.000000   \n",
       "\n",
       "   P_EUR_GIH  P_EUR_ITU  P_EUR_PJL  P_EUR_SAS  P_EUR_STU  \n",
       "0   0.987329     0.8941   0.926662   0.967103   0.967103  \n",
       "1   1.000000     1.0000   1.000000   1.000000   1.000000  \n",
       "2   0.000000     0.0000   0.000000   0.000000   0.000000  \n",
       "3   1.000000     1.0000   1.000000   1.000000   1.000000  \n",
       "4   1.000000     1.0000   1.000000   1.000000   1.000000  \n",
       "\n",
       "[5 rows x 41 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "55d0b061",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POS</th>\n",
       "      <th>AF</th>\n",
       "      <th>AF_AFR</th>\n",
       "      <th>AF_AMR</th>\n",
       "      <th>AF_AMR1</th>\n",
       "      <th>AF_AMR2</th>\n",
       "      <th>AF_EAS</th>\n",
       "      <th>AF_EUR</th>\n",
       "      <th>AF_SAS</th>\n",
       "      <th>AF_BEB</th>\n",
       "      <th>...</th>\n",
       "      <th>P_EUR_AMR</th>\n",
       "      <th>P_EUR_AMR1</th>\n",
       "      <th>P_EUR_AMR2</th>\n",
       "      <th>P_EUR_BEB</th>\n",
       "      <th>P_EUR_EAS</th>\n",
       "      <th>P_EUR_GIH</th>\n",
       "      <th>P_EUR_ITU</th>\n",
       "      <th>P_EUR_PJL</th>\n",
       "      <th>P_EUR_SAS</th>\n",
       "      <th>P_EUR_STU</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>...</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "      <td>1.048575e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>7.477768e+07</td>\n",
       "      <td>5.146699e-02</td>\n",
       "      <td>5.649458e-02</td>\n",
       "      <td>4.984452e-02</td>\n",
       "      <td>4.969536e-02</td>\n",
       "      <td>5.002973e-02</td>\n",
       "      <td>5.040034e-02</td>\n",
       "      <td>4.831791e-02</td>\n",
       "      <td>4.991778e-02</td>\n",
       "      <td>4.983308e-02</td>\n",
       "      <td>...</td>\n",
       "      <td>6.961169e-01</td>\n",
       "      <td>7.321370e-01</td>\n",
       "      <td>7.289671e-01</td>\n",
       "      <td>7.194738e-01</td>\n",
       "      <td>6.229560e-01</td>\n",
       "      <td>7.395275e-01</td>\n",
       "      <td>7.295889e-01</td>\n",
       "      <td>7.366329e-01</td>\n",
       "      <td>7.245362e-01</td>\n",
       "      <td>7.245362e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>5.793876e+07</td>\n",
       "      <td>1.451645e-01</td>\n",
       "      <td>1.494555e-01</td>\n",
       "      <td>1.500193e-01</td>\n",
       "      <td>1.477009e-01</td>\n",
       "      <td>1.556383e-01</td>\n",
       "      <td>1.604505e-01</td>\n",
       "      <td>1.497605e-01</td>\n",
       "      <td>1.525278e-01</td>\n",
       "      <td>1.521766e-01</td>\n",
       "      <td>...</td>\n",
       "      <td>4.256450e-01</td>\n",
       "      <td>4.114855e-01</td>\n",
       "      <td>4.053948e-01</td>\n",
       "      <td>4.148443e-01</td>\n",
       "      <td>4.523035e-01</td>\n",
       "      <td>4.054333e-01</td>\n",
       "      <td>4.098429e-01</td>\n",
       "      <td>4.074041e-01</td>\n",
       "      <td>4.126985e-01</td>\n",
       "      <td>4.126985e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>1.160080e+05</td>\n",
       "      <td>2.080730e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>1.993037e+07</td>\n",
       "      <td>4.161460e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6.600250e-01</td>\n",
       "      <td>5.785506e-01</td>\n",
       "      <td>4.894919e-01</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6.324544e-01</td>\n",
       "      <td>5.688892e-01</td>\n",
       "      <td>6.369542e-01</td>\n",
       "      <td>5.226347e-01</td>\n",
       "      <td>5.226347e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>7.362441e+07</td>\n",
       "      <td>1.249220e-03</td>\n",
       "      <td>1.679730e-03</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>9.634765e-01</td>\n",
       "      <td>9.870043e-01</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>9.646361e-01</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>1.129737e+08</td>\n",
       "      <td>1.389760e-02</td>\n",
       "      <td>2.407610e-02</td>\n",
       "      <td>6.613760e-03</td>\n",
       "      <td>9.225090e-03</td>\n",
       "      <td>4.566210e-03</td>\n",
       "      <td>2.564100e-03</td>\n",
       "      <td>3.164560e-03</td>\n",
       "      <td>4.385960e-03</td>\n",
       "      <td>7.633590e-03</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>2.474497e+08</td>\n",
       "      <td>9.996880e-01</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows × 36 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                POS            AF        AF_AFR        AF_AMR       AF_AMR1  \\\n",
       "count  1.048575e+06  1.048575e+06  1.048575e+06  1.048575e+06  1.048575e+06   \n",
       "mean   7.477768e+07  5.146699e-02  5.649458e-02  4.984452e-02  4.969536e-02   \n",
       "std    5.793876e+07  1.451645e-01  1.494555e-01  1.500193e-01  1.477009e-01   \n",
       "min    1.160080e+05  2.080730e-04  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "25%    1.993037e+07  4.161460e-04  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "50%    7.362441e+07  1.249220e-03  1.679730e-03  0.000000e+00  0.000000e+00   \n",
       "75%    1.129737e+08  1.389760e-02  2.407610e-02  6.613760e-03  9.225090e-03   \n",
       "max    2.474497e+08  9.996880e-01  1.000000e+00  1.000000e+00  1.000000e+00   \n",
       "\n",
       "            AF_AMR2        AF_EAS        AF_EUR        AF_SAS        AF_BEB  \\\n",
       "count  1.048575e+06  1.048575e+06  1.048575e+06  1.048575e+06  1.048575e+06   \n",
       "mean   5.002973e-02  5.040034e-02  4.831791e-02  4.991778e-02  4.983308e-02   \n",
       "std    1.556383e-01  1.604505e-01  1.497605e-01  1.525278e-01  1.521766e-01   \n",
       "min    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "25%    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "50%    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "75%    4.566210e-03  2.564100e-03  3.164560e-03  4.385960e-03  7.633590e-03   \n",
       "max    1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00   \n",
       "\n",
       "       ...     P_EUR_AMR    P_EUR_AMR1    P_EUR_AMR2     P_EUR_BEB  \\\n",
       "count  ...  1.048575e+06  1.048575e+06  1.048575e+06  1.048575e+06   \n",
       "mean   ...  6.961169e-01  7.321370e-01  7.289671e-01  7.194738e-01   \n",
       "std    ...  4.256450e-01  4.114855e-01  4.053948e-01  4.148443e-01   \n",
       "min    ...  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "25%    ...  0.000000e+00  6.600250e-01  5.785506e-01  4.894919e-01   \n",
       "50%    ...  9.634765e-01  9.870043e-01  1.000000e+00  1.000000e+00   \n",
       "75%    ...  1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00   \n",
       "max    ...  1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00   \n",
       "\n",
       "          P_EUR_EAS     P_EUR_GIH     P_EUR_ITU     P_EUR_PJL     P_EUR_SAS  \\\n",
       "count  1.048575e+06  1.048575e+06  1.048575e+06  1.048575e+06  1.048575e+06   \n",
       "mean   6.229560e-01  7.395275e-01  7.295889e-01  7.366329e-01  7.245362e-01   \n",
       "std    4.523035e-01  4.054333e-01  4.098429e-01  4.074041e-01  4.126985e-01   \n",
       "min    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "25%    0.000000e+00  6.324544e-01  5.688892e-01  6.369542e-01  5.226347e-01   \n",
       "50%    9.646361e-01  1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00   \n",
       "75%    1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00   \n",
       "max    1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00   \n",
       "\n",
       "          P_EUR_STU  \n",
       "count  1.048575e+06  \n",
       "mean   7.245362e-01  \n",
       "std    4.126985e-01  \n",
       "min    0.000000e+00  \n",
       "25%    5.226347e-01  \n",
       "50%    1.000000e+00  \n",
       "75%    1.000000e+00  \n",
       "max    1.000000e+00  \n",
       "\n",
       "[8 rows x 36 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "fb439149",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 1048575 entries, 0 to 1048574\n",
      "Data columns (total 41 columns):\n",
      " #   Column           Non-Null Count    Dtype  \n",
      "---  ------           --------------    -----  \n",
      " 0   CHROM            1048575 non-null  object \n",
      " 1   POS              1048575 non-null  int64  \n",
      " 2   ID               1048575 non-null  object \n",
      " 3   REF              1048575 non-null  object \n",
      " 4   ALT              1048575 non-null  object \n",
      " 5   AF               1048575 non-null  float64\n",
      " 6   GENE             1048575 non-null  object \n",
      " 7   AF_AFR           1048575 non-null  float64\n",
      " 8   AF_AMR           1048575 non-null  float64\n",
      " 9   AF_AMR1          1048575 non-null  float64\n",
      " 10  AF_AMR2          1048575 non-null  float64\n",
      " 11  AF_EAS           1048575 non-null  float64\n",
      " 12  AF_EUR           1048575 non-null  float64\n",
      " 13  AF_SAS           1048575 non-null  float64\n",
      " 14  AF_BEB           1048575 non-null  float64\n",
      " 15  AF_GIH           1048575 non-null  float64\n",
      " 16  AF_ITU           1048575 non-null  float64\n",
      " 17  AF_PJL           1048575 non-null  float64\n",
      " 18  AF_STU           1048575 non-null  float64\n",
      " 19  FST_EUR_VS_AFR   1048575 non-null  float64\n",
      " 20  FST_EUR_VS_AMR   1048575 non-null  float64\n",
      " 21  FST_EUR_VS_AMR1  1048575 non-null  float64\n",
      " 22  FST_EUR_VS_AMR2  1048575 non-null  float64\n",
      " 23  FST_EUR_VS_BEB   1048575 non-null  float64\n",
      " 24  FST_EUR_VS_EAS   1048575 non-null  float64\n",
      " 25  FST_EUR_VS_GIH   1048575 non-null  float64\n",
      " 26  FST_EUR_VS_ITU   1048575 non-null  float64\n",
      " 27  FST_EUR_VS_PJL   1048575 non-null  float64\n",
      " 28  FST_EUR_VS_SAS   1048575 non-null  float64\n",
      " 29  FST_EUR_VS_STU   1048575 non-null  float64\n",
      " 30  P_EUR_AFR        1048575 non-null  float64\n",
      " 31  P_EUR_AMR        1048575 non-null  float64\n",
      " 32  P_EUR_AMR1       1048575 non-null  float64\n",
      " 33  P_EUR_AMR2       1048575 non-null  float64\n",
      " 34  P_EUR_BEB        1048575 non-null  float64\n",
      " 35  P_EUR_EAS        1048575 non-null  float64\n",
      " 36  P_EUR_GIH        1048575 non-null  float64\n",
      " 37  P_EUR_ITU        1048575 non-null  float64\n",
      " 38  P_EUR_PJL        1048575 non-null  float64\n",
      " 39  P_EUR_SAS        1048575 non-null  float64\n",
      " 40  P_EUR_STU        1048575 non-null  float64\n",
      "dtypes: float64(35), int64(1), object(5)\n",
      "memory usage: 328.0+ MB\n"
     ]
    }
   ],
   "source": [
    "df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "c73e4748",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1048575, 41)"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "f7fd1c0c",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"After_LD/Significant_Chisq_result/AFR_significant_variants.tsv\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "a105eed1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AF_EUR</th>\n",
       "      <th>AF_AFR</th>\n",
       "      <th>P_EUR_AFR</th>\n",
       "      <th>COUNT</th>\n",
       "      <th>PASS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1248572</td>\n",
       "      <td>rs10425167</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001582</td>\n",
       "      <td>0.092945</td>\n",
       "      <td>0.020542</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1250398</td>\n",
       "      <td>rs3746107</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.013449</td>\n",
       "      <td>0.533595</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1251398</td>\n",
       "      <td>rs60724761</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.010285</td>\n",
       "      <td>0.210526</td>\n",
       "      <td>0.043161</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1251409</td>\n",
       "      <td>rs58898131</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.013449</td>\n",
       "      <td>0.530235</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1253350</td>\n",
       "      <td>rs113173533</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.155095</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM      POS           ID REF ALT    AF_EUR    AF_AFR  P_EUR_AFR  COUNT  \\\n",
       "0  chr19  1248572   rs10425167   G   C  0.001582  0.092945   0.020542      1   \n",
       "1  chr19  1250398    rs3746107   C   T  0.013449  0.533595   0.000006      1   \n",
       "2  chr19  1251398   rs60724761   C   T  0.010285  0.210526   0.043161      1   \n",
       "3  chr19  1251409   rs58898131   G   T  0.013449  0.530235   0.000006      1   \n",
       "4  chr19  1253350  rs113173533   C   T  0.000000  0.155095   0.000000      1   \n",
       "\n",
       "   PASS  \n",
       "0     1  \n",
       "1     1  \n",
       "2     1  \n",
       "3     1  \n",
       "4     1  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "a6f8019f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 40098 entries, 0 to 40097\n",
      "Data columns (total 10 columns):\n",
      " #   Column     Non-Null Count  Dtype  \n",
      "---  ------     --------------  -----  \n",
      " 0   CHROM      40098 non-null  object \n",
      " 1   POS        40098 non-null  int64  \n",
      " 2   ID         40098 non-null  object \n",
      " 3   REF        40098 non-null  object \n",
      " 4   ALT        40098 non-null  object \n",
      " 5   AF_EUR     40098 non-null  float64\n",
      " 6   AF_AFR     40098 non-null  float64\n",
      " 7   P_EUR_AFR  40098 non-null  float64\n",
      " 8   COUNT      40098 non-null  int64  \n",
      " 9   PASS       40098 non-null  int64  \n",
      "dtypes: float64(3), int64(3), object(4)\n",
      "memory usage: 3.1+ MB\n"
     ]
    }
   ],
   "source": [
    "df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "406fcbcd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(40098, 10)"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ef27fdd9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "============================================================\n",
      "Nephrotic Syndrome Genetics Database Setup\n",
      "============================================================\n",
      "✓ Connected to database: nephrotic_syndrome.db\n",
      "✓ Database schema created successfully\n",
      "\n",
      "============================================================\n",
      "LOADING DATA\n",
      "============================================================\n",
      "\n",
      "📊 Loading gene data...\n",
      "Error inserting gene NPHS1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NPHS2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PLCE1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CD2AP: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene FAT1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ACTN4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene INF2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MYH9: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MYO1E: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ARHGDIA: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ARHGAP24: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ANLN: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LAMB2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ITGB4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CD151: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene EXT1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene COL4A3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene COL4A4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene COL4A5: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TRPC6: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene EMP2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene WT1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LMX1B: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SMARCAL1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PAX2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MAFB: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LMNA: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NXF5: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GATA3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NUP93: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NUP107: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene COQ2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene COQ6: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PDSS2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MT-TL1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ADCK4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SCARB2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NEU1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene APOL1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PTPRO: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CRB2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene DGKE: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ZMPSTE24: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PMM2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ALG1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CUBN: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TTC21B: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene WDR73: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene E2F3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SYNPO: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GPC5: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PODXL: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CTLA4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TNFSF15: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LINC03079: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LINC00494: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene HLA-DQB1-AS1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene BTC: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene DSE: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CBX3P9: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene HLA-DRB1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TSBP1-AS1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene BTNL2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MTCO3P1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PARM1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PTPRC: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene KALRN: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LINC01991: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene BCL6-AS1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NWD2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LINC02531: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SNORD3A: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene AHI1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ELMO1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PTPRD: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LINC02664: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ZNF345: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene HLA-DQA1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene HLA-DQB1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TNXB: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TPT1P13: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MIR3672: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ALPG: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SLC9C2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MUC16: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene APOL4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TNXA: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene FRG1JP: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene HLA-DRB5: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene HLA-DRB9: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PHC1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MORF4L1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TRIM31-AS1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene RNF39: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MICA-AS1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene DELEC1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CD28: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene HCG15: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene OR2AD1P: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CLEC16A: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TRAPPC3L: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CALHM6: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NFKBIL1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene RUNX3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CLIC4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene OPN5: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene RPL27AP7: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene KC6: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LINC02181: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LINC02188: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene RPL12P5: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene COCH: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene RRP12: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CCDC66: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CCDC162P: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MICA: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GSDMB: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TNFSF4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene RNF10P1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene UBTFL8: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SCYGR5: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SLC19A3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SLC6A18: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SEMA3E: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GHET1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PDIA4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene DOCK8: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SFMBT2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LINC02740: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene HNRNPKP3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene RPL37P2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene UNC93B5: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene KCNA5: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LINC02443: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LIMA1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene HNRNPA1P69: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LINC03056: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ARID3B: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene HMG20A: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LINC02182: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ZNF469: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MIDN: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene JEB7: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SGPL1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MAGI2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene KANK2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NUP85: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NUP133: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NUP160: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene COQ8B: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NUP205: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TBC1D8B: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene DKC1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene AVIL: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NOS1AP: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene KIRREL1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene DAAM2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LAMA5: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NOP10: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PDCN: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ITGA3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene FGA: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene XPO5: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene OSGEP: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TP53RK: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TPRKB: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LAGE3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GAPVD1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GSN: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene FN1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SLC17A5: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ZAP70: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GFND1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ANKFY1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PRKCD: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ABCB1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CYP3A5: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CYP3A4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NR3C1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene EDNRA: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ANKRD36: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GAD1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TMEM131L: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CFH: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene DLC1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ITSN1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ITSN2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene KANK1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene KANK4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LCAT: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MMACHC: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NPHP4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene OCRL: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TNS2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CLCN5: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene KIRREL2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MFF-DT: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TTC21B-AS1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene AXDND1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene WNK4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ARHGEF17: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IFIH1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ADGRE1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene FAN1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IL36G: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NUMBL: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene H19: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GLA: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CTNS: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene AGXT: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PLCE1-AS1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LOC112997540: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene RPL36A-HNRNPH2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ATIC: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ANGPTL3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MPV17: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ALB: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene RUNX2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LOC112272600: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ACE: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SERPINC1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CMIP: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene APOB: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene B2M: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NPPA: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene APOA1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene APOE: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CD79A: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene AMBP: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LPA: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IL13: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TF: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene F9: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene HMGCR: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene HPSE: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene RBP4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SERPINF2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene A2M: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene AGT: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IL2RA: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LPL: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GUCA2B: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NAGLU: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GC: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SERPINA1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CETP: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene REN: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TNF: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IL17A: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene AQP3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene VTN: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TGFB1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CLU: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene EZR: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SMPDL3B: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene C3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IL4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LRP2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IL6: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene F2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PPBP: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PON1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CRP: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TFRC: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ANGPTL4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene F3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IL1B: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IL2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IL4R: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PLG: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CD80: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene AQP2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IGHE: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LCN2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene COQ8A: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NFKBIA: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CD4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene EPO: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IL18: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PCSK9: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TTR: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CCL2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MPO: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene INS: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene EDN1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IFNG: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene AGTR1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IGFBP2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GPT: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene COQ7: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LDLR: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene STOM: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IL10: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CXCL8: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SERPINE1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CDC42: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene APOC3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LEPR: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TJP1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MIR193A: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene VWF: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SERPINA7: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GON7: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene FOXP3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IFNA2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene THBD: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ACE2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LEP: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MEFV: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IFNA1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TNFSF13B: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MIF: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PLAT: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NLRP3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SAA4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IL1A: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TLR4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene FGF23: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LIPC: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NR3C2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene APOA2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CD40LG: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ADAMTS13: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ADIPOQ: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NOS3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SELP: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ABCA1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GUSB: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene AQP1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene KIRREL3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IGF1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PLA2G7: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CP: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MTHFR: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GLCCI1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene APOH: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NUP37: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene HSD11B2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PLAUR: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene KDR: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MIR30A: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NPHP1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GPX3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene HLA-A: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CCL5: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene RELA: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CD44: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene STAT3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IQGAP1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MAF: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IL1RN: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TLR3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SELE: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MC1R: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene STAT6: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene YRDC: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ZHX3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene EGFR: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PIK3C2A: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PF4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ANXA5: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CLVS1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene A1BG: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PLCG2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ADA: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IGFBP3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ZHX2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ZHX1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NUP188: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CXCL16: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SMAD4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene VIM: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MIR21: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene UCHL1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene WDR4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PCNA: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ITGB3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CDK20: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GSTM1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene BRAF: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene FCER1G: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LAMB3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LGALS3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IL7: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene RAC1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene DCN: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NEIL1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CYP11B2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NPPC: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene FCGR2A: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IL1R1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SLC22A12: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CYP11A1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MIR30C1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PDPN: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene KIF21A: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene F5: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene KLK3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SCGB1A1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LGALS1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene HLA-G: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MAPK14: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CDKN1C: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NOX1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MIR17: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene FLNA: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CCL4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TSFM: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CPB2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene DICER1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene RETN: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ACTR2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ALOX5: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GSTP1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IPO7: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GSR: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene RAPGEF2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene HSD11B1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene DROSHA: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GP1BA: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MLXIPL: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ADM: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene KRT8: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MIR34C: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LTA: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GGT1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CDKN1A: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IGFBP1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TUG1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene HDAC2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CCNA2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CREBBP: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MAL: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CYBA: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene HRG: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MCAM: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GH1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TGM1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene FBN1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene HAVCR2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LCP1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NOS2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene G6PD: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene KRT18: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene HLA-S: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene STAR: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IL1RL1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SULF2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CENPI: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CLDN2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene KITLG: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SH3BP2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GCA: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IL1RAP: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IL13RA2: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MIR16-1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene FBP1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene HADHB: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PPIB: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IL23A: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LTA4H: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PLA2R1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MIR1915: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MIR192: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene THSD7A: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene HP: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IGHE: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene NOC3L: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ENTPD5: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SOD1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene GSTT1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene BMP6: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CAT: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CERNA3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LSL: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene TRA-TGC7-1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CBS: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ACTR3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene IL12B: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LINC02605: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene OSGEP: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene PSMD14: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene RAB11B: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene STS: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene VEGFA: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene SNORD15A: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LCN1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene LOC126862264: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene MAGI2-AS3: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ACAT1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene AQP4: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene ATF1: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "Error inserting gene CD247: Incorrect number of bindings supplied. The current statement uses 13, and there are 14 supplied.\n",
      "✓ Loaded 0 genes\n",
      "\n",
      "📊 Loading variant data (this may take a few minutes)...\n",
      "  Processed chunk 1 (10000 variants)\n",
      "  Processed chunk 2 (10000 variants)\n",
      "  Processed chunk 3 (10000 variants)\n",
      "  Processed chunk 4 (10000 variants)\n",
      "  Processed chunk 5 (10000 variants)\n",
      "  Processed chunk 6 (10000 variants)\n",
      "  Processed chunk 7 (10000 variants)\n",
      "  Processed chunk 8 (10000 variants)\n",
      "  Processed chunk 9 (10000 variants)\n",
      "  Processed chunk 10 (10000 variants)\n",
      "  Processed chunk 11 (10000 variants)\n",
      "  Processed chunk 12 (10000 variants)\n",
      "  Processed chunk 13 (10000 variants)\n",
      "  Processed chunk 14 (10000 variants)\n",
      "  Processed chunk 15 (10000 variants)\n",
      "  Processed chunk 16 (10000 variants)\n",
      "  Processed chunk 17 (10000 variants)\n",
      "  Processed chunk 18 (10000 variants)\n",
      "  Processed chunk 19 (10000 variants)\n",
      "  Processed chunk 20 (10000 variants)\n",
      "  Processed chunk 21 (10000 variants)\n",
      "  Processed chunk 22 (10000 variants)\n",
      "  Processed chunk 23 (10000 variants)\n",
      "  Processed chunk 24 (10000 variants)\n",
      "  Processed chunk 25 (10000 variants)\n",
      "  Processed chunk 26 (10000 variants)\n",
      "  Processed chunk 27 (10000 variants)\n",
      "  Processed chunk 28 (10000 variants)\n",
      "  Processed chunk 29 (10000 variants)\n",
      "  Processed chunk 30 (10000 variants)\n",
      "  Processed chunk 31 (10000 variants)\n",
      "  Processed chunk 32 (10000 variants)\n",
      "  Processed chunk 33 (10000 variants)\n",
      "  Processed chunk 34 (10000 variants)\n",
      "  Processed chunk 35 (10000 variants)\n",
      "  Processed chunk 36 (10000 variants)\n",
      "  Processed chunk 37 (10000 variants)\n",
      "  Processed chunk 38 (10000 variants)\n",
      "  Processed chunk 39 (10000 variants)\n",
      "  Processed chunk 40 (10000 variants)\n",
      "  Processed chunk 41 (10000 variants)\n",
      "  Processed chunk 42 (10000 variants)\n",
      "  Processed chunk 43 (10000 variants)\n",
      "  Processed chunk 44 (10000 variants)\n",
      "  Processed chunk 45 (10000 variants)\n",
      "  Processed chunk 46 (10000 variants)\n",
      "  Processed chunk 47 (10000 variants)\n",
      "  Processed chunk 48 (10000 variants)\n",
      "  Processed chunk 49 (10000 variants)\n",
      "  Processed chunk 50 (10000 variants)\n",
      "  Processed chunk 51 (10000 variants)\n",
      "  Processed chunk 52 (10000 variants)\n",
      "  Processed chunk 53 (10000 variants)\n",
      "  Processed chunk 54 (10000 variants)\n",
      "  Processed chunk 55 (10000 variants)\n",
      "  Processed chunk 56 (10000 variants)\n",
      "  Processed chunk 57 (10000 variants)\n",
      "  Processed chunk 58 (10000 variants)\n",
      "  Processed chunk 59 (10000 variants)\n",
      "  Processed chunk 60 (10000 variants)\n"
     ]
    }
   ],
   "source": [
    "\"\"\"\n",
    "Nephrotic Syndrome Genetic Diversity Database Setup\n",
    "This script creates and populates a SQLite database for the NS genetics dashboard\n",
    "\"\"\"\n",
    "\n",
    "import sqlite3\n",
    "import pandas as pd\n",
    "import os\n",
    "from pathlib import Path\n",
    "\n",
    "# Database Schema Design\n",
    "DATABASE_SCHEMA = \"\"\"\n",
    "-- Table 1: Genes - Core gene information\n",
    "CREATE TABLE IF NOT EXISTS genes (\n",
    "    gene_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    gene_name TEXT UNIQUE NOT NULL,\n",
    "    ensembl_id TEXT,\n",
    "    protein_name TEXT,\n",
    "    chromosome TEXT,\n",
    "    start_position INTEGER,\n",
    "    end_position INTEGER,\n",
    "    strand TEXT,\n",
    "    gene_function TEXT,\n",
    "    primary_pathway TEXT,\n",
    "    cellular_location TEXT,\n",
    "    interacting_genes TEXT,\n",
    "    srns_ssns TEXT,\n",
    "    phenotypes TEXT,\n",
    "    inheritance TEXT,\n",
    "    pediatric TEXT,\n",
    "    adult TEXT,\n",
    "    both TEXT,\n",
    "    population_specific TEXT,\n",
    "    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP\n",
    ");\n",
    "\n",
    "-- Table 2: Variants - All variant information\n",
    "CREATE TABLE IF NOT EXISTS variants (\n",
    "    variant_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    rs_id TEXT UNIQUE NOT NULL,\n",
    "    chrom TEXT NOT NULL,\n",
    "    position INTEGER NOT NULL,\n",
    "    ref_allele TEXT NOT NULL,\n",
    "    alt_allele TEXT NOT NULL,\n",
    "    gene_name TEXT NOT NULL,\n",
    "    global_af REAL,\n",
    "    FOREIGN KEY (gene_name) REFERENCES genes(gene_name),\n",
    "    UNIQUE(chrom, position, ref_allele, alt_allele)\n",
    ");\n",
    "\n",
    "-- Table 3: Allele Frequencies - Population-specific frequencies\n",
    "CREATE TABLE IF NOT EXISTS allele_frequencies (\n",
    "    af_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    variant_id INTEGER NOT NULL,\n",
    "    population TEXT NOT NULL,\n",
    "    allele_frequency REAL NOT NULL,\n",
    "    FOREIGN KEY (variant_id) REFERENCES variants(variant_id),\n",
    "    UNIQUE(variant_id, population)\n",
    ");\n",
    "\n",
    "-- Table 4: Statistical Tests - FST and Chi-square results\n",
    "CREATE TABLE IF NOT EXISTS statistical_tests (\n",
    "    test_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    variant_id INTEGER NOT NULL,\n",
    "    population TEXT NOT NULL,\n",
    "    fst_value REAL,\n",
    "    chi_square_p_value REAL,\n",
    "    is_significant_chisq BOOLEAN DEFAULT 0,\n",
    "    is_significant_bonferroni BOOLEAN DEFAULT 0,\n",
    "    is_significant_fst BOOLEAN DEFAULT 0,\n",
    "    FOREIGN KEY (variant_id) REFERENCES variants(variant_id),\n",
    "    UNIQUE(variant_id, population)\n",
    ");\n",
    "\n",
    "-- Table 5: LD Status - Linkage Disequilibrium information\n",
    "CREATE TABLE IF NOT EXISTS ld_status (\n",
    "    ld_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    variant_id INTEGER NOT NULL,\n",
    "    population TEXT NOT NULL,\n",
    "    in_ld BOOLEAN DEFAULT 0,\n",
    "    ld_pruned BOOLEAN DEFAULT 0,\n",
    "    FOREIGN KEY (variant_id) REFERENCES variants(variant_id),\n",
    "    UNIQUE(variant_id, population)\n",
    ");\n",
    "\n",
    "-- Table 6: Gene Sources - Database and paper references\n",
    "CREATE TABLE IF NOT EXISTS gene_sources (\n",
    "    source_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    gene_name TEXT NOT NULL,\n",
    "    database_name TEXT,\n",
    "    database_link TEXT,\n",
    "    paper_1_link TEXT,\n",
    "    paper_2_link TEXT,\n",
    "    paper_3_link TEXT,\n",
    "    category TEXT,\n",
    "    FOREIGN KEY (gene_name) REFERENCES genes(gene_name)\n",
    ");\n",
    "\n",
    "-- Indexes for faster queries\n",
    "CREATE INDEX IF NOT EXISTS idx_variants_gene ON variants(gene_name);\n",
    "CREATE INDEX IF NOT EXISTS idx_variants_rs ON variants(rs_id);\n",
    "CREATE INDEX IF NOT EXISTS idx_variants_chrom_pos ON variants(chrom, position);\n",
    "CREATE INDEX IF NOT EXISTS idx_af_variant ON allele_frequencies(variant_id);\n",
    "CREATE INDEX IF NOT EXISTS idx_af_population ON allele_frequencies(population);\n",
    "CREATE INDEX IF NOT EXISTS idx_stats_variant ON statistical_tests(variant_id);\n",
    "CREATE INDEX IF NOT EXISTS idx_ld_variant ON ld_status(variant_id);\n",
    "\"\"\"\n",
    "\n",
    "class NephroticSyndromeDB:\n",
    "    def __init__(self, db_path='nephrotic_syndrome.db'):\n",
    "        \"\"\"Initialize database connection\"\"\"\n",
    "        self.db_path = db_path\n",
    "        self.conn = None\n",
    "        self.cursor = None\n",
    "        \n",
    "    def connect(self):\n",
    "        \"\"\"Connect to SQLite database\"\"\"\n",
    "        self.conn = sqlite3.connect(self.db_path)\n",
    "        self.cursor = self.conn.cursor()\n",
    "        print(f\"✓ Connected to database: {self.db_path}\")\n",
    "        \n",
    "    def create_schema(self):\n",
    "        \"\"\"Create database schema\"\"\"\n",
    "        self.cursor.executescript(DATABASE_SCHEMA)\n",
    "        self.conn.commit()\n",
    "        print(\"✓ Database schema created successfully\")\n",
    "        \n",
    "    def load_gene_data(self, excel_path):\n",
    "        \"\"\"Load gene information from Excel file\"\"\"\n",
    "        print(\"\\n📊 Loading gene data...\")\n",
    "        df = pd.read_excel(excel_path)\n",
    "        \n",
    "        # Clean column names\n",
    "        df.columns = df.columns.str.strip()\n",
    "        \n",
    "        gene_count = 0\n",
    "        for idx, row in df.iterrows():\n",
    "            try:\n",
    "                # Insert gene information\n",
    "                self.cursor.execute(\"\"\"\n",
    "                    INSERT OR IGNORE INTO genes (\n",
    "                        gene_name, protein_name, gene_function, primary_pathway,\n",
    "                        cellular_location, interacting_genes, srns_ssns, phenotypes,\n",
    "                        inheritance, pediatric, adult, both, population_specific\n",
    "                    ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)\n",
    "                \"\"\", (\n",
    "                    row.get('Gene'),\n",
    "                    row.get('Ensembl_ID'),\n",
    "                    row.get('Protein'),\n",
    "                    row.get('Gene Function'),\n",
    "                    row.get('Primary Pathway'),\n",
    "                    row.get('Cellular Location'),\n",
    "                    row.get('Interacting Genes'),\n",
    "                    row.get('SRNS/SSNS'),\n",
    "                    row.get('Phenotypes'),\n",
    "                    row.get('Inheritance'),\n",
    "                    row.get('Pediatric (Y/N)'),\n",
    "                    row.get('Adult (Y/N)'),\n",
    "                    row.get('Both (Y/N)'),\n",
    "                    row.get('Population')\n",
    "                ))\n",
    "                \n",
    "                # Insert source information\n",
    "                self.cursor.execute(\"\"\"\n",
    "                    INSERT INTO gene_sources (\n",
    "                        gene_name, database_name, database_link,\n",
    "                        paper_1_link, paper_2_link, paper_3_link, category\n",
    "                    ) VALUES (?, ?, ?, ?, ?, ?, ?)\n",
    "                \"\"\", (\n",
    "                    row.get('Gene'),\n",
    "                    row.get('Database'),\n",
    "                    row.get('Database Link'),\n",
    "                    row.get('Source Paper 1'),\n",
    "                    row.get('Source Paper 2'),\n",
    "                    row.get('Source Paper 3'),\n",
    "                    row.get('Category')\n",
    "                ))\n",
    "                \n",
    "                gene_count += 1\n",
    "                \n",
    "            except Exception as e:\n",
    "                print(f\"Error inserting gene {row.get('Gene')}: {e}\")\n",
    "                \n",
    "        self.conn.commit()\n",
    "        print(f\"✓ Loaded {gene_count} genes\")\n",
    "        \n",
    "    def load_variant_data(self, tsv_path):\n",
    "        \"\"\"Load variant and statistical test data from main TSV file\"\"\"\n",
    "        print(\"\\n📊 Loading variant data (this may take a few minutes)...\")\n",
    "        \n",
    "        # Read in chunks to handle large file\n",
    "        chunk_size = 10000\n",
    "        chunks_processed = 0\n",
    "        \n",
    "        for chunk in pd.read_csv(tsv_path, sep='\\t', chunksize=chunk_size):\n",
    "            variant_count = 0\n",
    "            \n",
    "            for idx, row in chunk.iterrows():\n",
    "                try:\n",
    "                    # Insert variant\n",
    "                    self.cursor.execute(\"\"\"\n",
    "                        INSERT OR IGNORE INTO variants (\n",
    "                            rs_id, chrom, position, ref_allele, alt_allele, gene_name, global_af\n",
    "                        ) VALUES (?, ?, ?, ?, ?, ?, ?)\n",
    "                    \"\"\", (\n",
    "                        row['ID'],\n",
    "                        row['CHROM'],\n",
    "                        row['POS'],\n",
    "                        row['REF'],\n",
    "                        row['ALT'],\n",
    "                        row['GENE'],\n",
    "                        row['AF']\n",
    "                    ))\n",
    "                    \n",
    "                    # Get variant_id\n",
    "                    variant_id = self.cursor.execute(\n",
    "                        \"SELECT variant_id FROM variants WHERE rs_id = ?\",\n",
    "                        (row['ID'],)\n",
    "                    ).fetchone()[0]\n",
    "                    \n",
    "                    # Insert allele frequencies for all populations\n",
    "                    populations = {\n",
    "                        'AFR': row['AF_AFR'], 'AMR': row['AF_AMR'], 'AMR1': row['AF_AMR1'],\n",
    "                        'AMR2': row['AF_AMR2'], 'EAS': row['AF_EAS'], 'EUR': row['AF_EUR'],\n",
    "                        'SAS': row['AF_SAS'], 'BEB': row['AF_BEB'], 'GIH': row['AF_GIH'],\n",
    "                        'ITU': row['AF_ITU'], 'PJL': row['AF_PJL'], 'STU': row['AF_STU']\n",
    "                    }\n",
    "                    \n",
    "                    for pop, freq in populations.items():\n",
    "                        self.cursor.execute(\"\"\"\n",
    "                            INSERT OR IGNORE INTO allele_frequencies (variant_id, population, allele_frequency)\n",
    "                            VALUES (?, ?, ?)\n",
    "                        \"\"\", (variant_id, pop, freq))\n",
    "                    \n",
    "                    # Insert statistical test results\n",
    "                    stat_populations = ['AFR', 'AMR', 'AMR1', 'AMR2', 'EAS', 'SAS', 'BEB', 'GIH', 'ITU', 'PJL', 'STU']\n",
    "                    \n",
    "                    for pop in stat_populations:\n",
    "                        fst_col = f'FST_EUR_VS_{pop}'\n",
    "                        p_col = f'P_EUR_{pop}'\n",
    "                        \n",
    "                        self.cursor.execute(\"\"\"\n",
    "                            INSERT OR IGNORE INTO statistical_tests (\n",
    "                                variant_id, population, fst_value, chi_square_p_value\n",
    "                            ) VALUES (?, ?, ?, ?)\n",
    "                        \"\"\", (variant_id, pop, row[fst_col], row[p_col]))\n",
    "                    \n",
    "                    variant_count += 1\n",
    "                    \n",
    "                except Exception as e:\n",
    "                    print(f\"Error processing variant {row.get('ID')}: {e}\")\n",
    "                    continue\n",
    "            \n",
    "            chunks_processed += 1\n",
    "            self.conn.commit()\n",
    "            print(f\"  Processed chunk {chunks_processed} ({variant_count} variants)\")\n",
    "            \n",
    "        print(f\"✓ Loaded all variants\")\n",
    "        \n",
    "    def load_significant_variants(self, folder_path, ld_status='before'):\n",
    "        \"\"\"Load significant variants and mark them in the database\"\"\"\n",
    "        print(f\"\\n📊 Loading significant variants ({ld_status} LD)...\")\n",
    "        \n",
    "        test_types = {\n",
    "            'Significant Chisq result': 'chisq',\n",
    "            'Significant_Chisq_result': 'chisq',\n",
    "            'Significant_Chisq_result_Bonferroni': 'bonferroni',\n",
    "            'Significant_Bonferroni_with_FST': 'bonferroni_fst',\n",
    "            'Significant_Fst': 'fst'\n",
    "        }\n",
    "        \n",
    "        populations = ['AFR', 'AMR', 'AMR1', 'AMR2', 'EAS', 'SAS', 'BEB', 'GIH', 'ITU', 'PJL', 'STU']\n",
    "        \n",
    "        for test_folder, test_type in test_types.items():\n",
    "            test_path = os.path.join(folder_path, test_folder)\n",
    "            if not os.path.exists(test_path):\n",
    "                continue\n",
    "                \n",
    "            for pop in populations:\n",
    "                # Find matching file\n",
    "                files = list(Path(test_path).glob(f\"{pop}_*.tsv\"))\n",
    "                if not files:\n",
    "                    continue\n",
    "                    \n",
    "                file_path = files[0]\n",
    "                \n",
    "                try:\n",
    "                    df = pd.read_csv(file_path, sep='\\t')\n",
    "                    \n",
    "                    for _, row in df.iterrows():\n",
    "                        # Update significance flags\n",
    "                        update_query = f\"\"\"\n",
    "                            UPDATE statistical_tests\n",
    "                            SET is_significant_{test_type.split('_')[0]} = 1\n",
    "                            WHERE variant_id = (SELECT variant_id FROM variants WHERE rs_id = ?)\n",
    "                            AND population = ?\n",
    "                        \"\"\"\n",
    "                        self.cursor.execute(update_query, (row['ID'], pop))\n",
    "                        \n",
    "                        # Update LD status\n",
    "                        self.cursor.execute(\"\"\"\n",
    "                            INSERT OR REPLACE INTO ld_status (variant_id, population, in_ld, ld_pruned)\n",
    "                            VALUES (\n",
    "                                (SELECT variant_id FROM variants WHERE rs_id = ?),\n",
    "                                ?, ?, ?\n",
    "                            )\n",
    "                        \"\"\", (row['ID'], pop, 1 if ld_status == 'after' else 0, ld_status == 'after'))\n",
    "                        \n",
    "                except Exception as e:\n",
    "                    print(f\"Error processing {file_path}: {e}\")\n",
    "                    \n",
    "        self.conn.commit()\n",
    "        print(f\"✓ Loaded significant variants ({ld_status} LD)\")\n",
    "        \n",
    "    def close(self):\n",
    "        \"\"\"Close database connection\"\"\"\n",
    "        if self.conn:\n",
    "            self.conn.close()\n",
    "            print(\"\\n✓ Database connection closed\")\n",
    "\n",
    "\n",
    "# Main execution function\n",
    "def main():\n",
    "    print(\"=\" * 60)\n",
    "    print(\"Nephrotic Syndrome Genetics Database Setup\")\n",
    "    print(\"=\" * 60)\n",
    "    \n",
    "    # Initialize database\n",
    "    db = NephroticSyndromeDB('nephrotic_syndrome.db')\n",
    "    db.connect()\n",
    "    \n",
    "    # Create schema\n",
    "    db.create_schema()\n",
    "    \n",
    "    # Load data (Update these paths to your actual file locations)\n",
    "    print(\"\\n\" + \"=\" * 60)\n",
    "    print(\"LOADING DATA\")\n",
    "    print(\"=\" * 60)\n",
    "    \n",
    "    # 1. Load gene information\n",
    "    db.load_gene_data('Final_gene_list.xlsx')\n",
    "    \n",
    "    # 2. Load variant data\n",
    "    db.load_variant_data('Final_chi_sq_fst_result_with_AF_GENE.tsv')\n",
    "    \n",
    "    # 3. Load significant variants (Before LD)\n",
    "    db.load_significant_variants('Before_LD', ld_status='before')\n",
    "    \n",
    "    # 4. Load significant variants (After LD)\n",
    "    db.load_significant_variants('After_LD', ld_status='after')\n",
    "    \n",
    "    # Close connection\n",
    "    db.close()\n",
    "    \n",
    "    print(\"\\n\" + \"=\" * 60)\n",
    "    print(\"DATABASE SETUP COMPLETE!\")\n",
    "    print(\"=\" * 60)\n",
    "    print(f\"\\nYour database 'nephrotic_syndrome.db' is ready to use!\")\n",
    "    print(\"\\nNext steps:\")\n",
    "    print(\"1. Test queries with the query examples script\")\n",
    "    print(\"2. Set up your web dashboard using Flask/Django\")\n",
    "    print(\"3. Deploy your application\")\n",
    "\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "7b34daae",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "============================================================\n",
      "Nephrotic Syndrome Genetics Database Setup\n",
      "============================================================\n",
      "✓ Connected to database: nephrotic_syndrome.db\n",
      "✓ Database schema created successfully\n",
      "\n",
      "============================================================\n",
      "LOADING DATA\n",
      "============================================================\n",
      "\n",
      "📊 Loading gene data...\n",
      "✓ Loaded 493 genes\n",
      "\n",
      "📊 Loading variant data (this may take a few minutes)...\n",
      "  Processed chunk 1 (10000 variants)\n",
      "  Processed chunk 2 (10000 variants)\n",
      "  Processed chunk 3 (10000 variants)\n",
      "  Processed chunk 4 (10000 variants)\n",
      "  Processed chunk 5 (10000 variants)\n",
      "  Processed chunk 6 (10000 variants)\n",
      "  Processed chunk 7 (10000 variants)\n",
      "  Processed chunk 8 (10000 variants)\n",
      "  Processed chunk 9 (10000 variants)\n",
      "  Processed chunk 10 (10000 variants)\n",
      "  Processed chunk 11 (10000 variants)\n",
      "  Processed chunk 12 (10000 variants)\n",
      "  Processed chunk 13 (10000 variants)\n",
      "  Processed chunk 14 (10000 variants)\n",
      "  Processed chunk 15 (10000 variants)\n",
      "  Processed chunk 16 (10000 variants)\n",
      "  Processed chunk 17 (10000 variants)\n",
      "  Processed chunk 18 (10000 variants)\n",
      "  Processed chunk 19 (10000 variants)\n",
      "  Processed chunk 20 (10000 variants)\n",
      "  Processed chunk 21 (10000 variants)\n",
      "  Processed chunk 22 (10000 variants)\n",
      "  Processed chunk 23 (10000 variants)\n",
      "  Processed chunk 24 (10000 variants)\n",
      "  Processed chunk 25 (10000 variants)\n",
      "  Processed chunk 26 (10000 variants)\n",
      "  Processed chunk 27 (10000 variants)\n",
      "  Processed chunk 28 (10000 variants)\n",
      "  Processed chunk 29 (10000 variants)\n",
      "  Processed chunk 30 (10000 variants)\n",
      "  Processed chunk 31 (10000 variants)\n",
      "  Processed chunk 32 (10000 variants)\n",
      "  Processed chunk 33 (10000 variants)\n",
      "  Processed chunk 34 (10000 variants)\n",
      "  Processed chunk 35 (10000 variants)\n",
      "  Processed chunk 36 (10000 variants)\n",
      "  Processed chunk 37 (10000 variants)\n",
      "  Processed chunk 38 (10000 variants)\n",
      "  Processed chunk 39 (10000 variants)\n",
      "  Processed chunk 40 (10000 variants)\n",
      "  Processed chunk 41 (10000 variants)\n",
      "  Processed chunk 42 (10000 variants)\n",
      "  Processed chunk 43 (10000 variants)\n",
      "  Processed chunk 44 (10000 variants)\n",
      "  Processed chunk 45 (10000 variants)\n",
      "  Processed chunk 46 (10000 variants)\n",
      "  Processed chunk 47 (10000 variants)\n",
      "  Processed chunk 48 (10000 variants)\n",
      "  Processed chunk 49 (10000 variants)\n",
      "  Processed chunk 50 (10000 variants)\n",
      "  Processed chunk 51 (10000 variants)\n",
      "  Processed chunk 52 (10000 variants)\n",
      "  Processed chunk 53 (10000 variants)\n",
      "  Processed chunk 54 (10000 variants)\n",
      "  Processed chunk 55 (10000 variants)\n",
      "  Processed chunk 56 (10000 variants)\n",
      "  Processed chunk 57 (10000 variants)\n",
      "  Processed chunk 58 (10000 variants)\n",
      "  Processed chunk 59 (10000 variants)\n",
      "  Processed chunk 60 (10000 variants)\n",
      "  Processed chunk 61 (10000 variants)\n",
      "  Processed chunk 62 (10000 variants)\n",
      "  Processed chunk 63 (10000 variants)\n",
      "  Processed chunk 64 (10000 variants)\n",
      "  Processed chunk 65 (10000 variants)\n",
      "  Processed chunk 66 (10000 variants)\n",
      "  Processed chunk 67 (10000 variants)\n",
      "  Processed chunk 68 (10000 variants)\n",
      "  Processed chunk 69 (10000 variants)\n",
      "  Processed chunk 70 (10000 variants)\n",
      "  Processed chunk 71 (10000 variants)\n",
      "  Processed chunk 72 (10000 variants)\n",
      "  Processed chunk 73 (10000 variants)\n",
      "  Processed chunk 74 (10000 variants)\n",
      "  Processed chunk 75 (10000 variants)\n",
      "  Processed chunk 76 (10000 variants)\n",
      "  Processed chunk 77 (10000 variants)\n",
      "  Processed chunk 78 (10000 variants)\n",
      "  Processed chunk 79 (10000 variants)\n",
      "  Processed chunk 80 (10000 variants)\n",
      "  Processed chunk 81 (10000 variants)\n",
      "  Processed chunk 82 (10000 variants)\n",
      "  Processed chunk 83 (10000 variants)\n",
      "  Processed chunk 84 (10000 variants)\n",
      "  Processed chunk 85 (10000 variants)\n",
      "  Processed chunk 86 (10000 variants)\n",
      "  Processed chunk 87 (10000 variants)\n",
      "  Processed chunk 88 (10000 variants)\n",
      "  Processed chunk 89 (10000 variants)\n",
      "  Processed chunk 90 (10000 variants)\n",
      "  Processed chunk 91 (10000 variants)\n",
      "  Processed chunk 92 (10000 variants)\n",
      "  Processed chunk 93 (10000 variants)\n",
      "  Processed chunk 94 (10000 variants)\n",
      "  Processed chunk 95 (10000 variants)\n",
      "  Processed chunk 96 (10000 variants)\n",
      "  Processed chunk 97 (10000 variants)\n",
      "  Processed chunk 98 (10000 variants)\n",
      "  Processed chunk 99 (10000 variants)\n",
      "  Processed chunk 100 (10000 variants)\n",
      "  Processed chunk 101 (10000 variants)\n",
      "  Processed chunk 102 (10000 variants)\n",
      "  Processed chunk 103 (10000 variants)\n",
      "  Processed chunk 104 (10000 variants)\n",
      "  Processed chunk 105 (8575 variants)\n",
      "✓ Loaded all variants\n",
      "\n",
      "📊 Loading significant variants (before LD)...\n",
      "✓ Loaded significant variants (before LD)\n",
      "  Updated: 214167 variant-population combinations\n",
      "  Skipped: 1322 variants not found in main dataset\n",
      "\n",
      "📊 Loading significant variants (after LD)...\n",
      "✓ Loaded significant variants (after LD)\n",
      "  Updated: 169000 variant-population combinations\n",
      "  Skipped: 1329 variants not found in main dataset\n",
      "\n",
      "✓ Database connection closed\n",
      "\n",
      "============================================================\n",
      "DATABASE SETUP COMPLETE!\n",
      "============================================================\n",
      "\n",
      "Your database 'nephrotic_syndrome.db' is ready to use!\n",
      "\n",
      "Next steps:\n",
      "1. Test queries with the query examples script\n",
      "2. Set up your web dashboard using Flask/Django\n",
      "3. Deploy your application\n"
     ]
    }
   ],
   "source": [
    "\"\"\"\n",
    "Nephrotic Syndrome Genetic Diversity Database Setup\n",
    "This script creates and populates a SQLite database for the NS genetics dashboard\n",
    "\"\"\"\n",
    "\n",
    "import sqlite3\n",
    "import pandas as pd\n",
    "import os\n",
    "from pathlib import Path\n",
    "\n",
    "# Database Schema Design\n",
    "DATABASE_SCHEMA = \"\"\"\n",
    "-- Table 1: Genes - Core gene information\n",
    "CREATE TABLE IF NOT EXISTS genes (\n",
    "    gene_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    gene_name TEXT UNIQUE NOT NULL,\n",
    "    ensembl_id TEXT,\n",
    "    protein_name TEXT,\n",
    "    chromosome TEXT,\n",
    "    start_position INTEGER,\n",
    "    end_position INTEGER,\n",
    "    strand TEXT,\n",
    "    gene_function TEXT,\n",
    "    primary_pathway TEXT,\n",
    "    cellular_location TEXT,\n",
    "    interacting_genes TEXT,\n",
    "    srns_ssns TEXT,\n",
    "    phenotypes TEXT,\n",
    "    inheritance TEXT,\n",
    "    pediatric TEXT,\n",
    "    adult TEXT,\n",
    "    both TEXT,\n",
    "    population_specific TEXT,\n",
    "    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP\n",
    ");\n",
    "\n",
    "-- Table 2: Variants - All variant information\n",
    "CREATE TABLE IF NOT EXISTS variants (\n",
    "    variant_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    rs_id TEXT UNIQUE NOT NULL,\n",
    "    chrom TEXT NOT NULL,\n",
    "    position INTEGER NOT NULL,\n",
    "    ref_allele TEXT NOT NULL,\n",
    "    alt_allele TEXT NOT NULL,\n",
    "    gene_name TEXT NOT NULL,\n",
    "    global_af REAL,\n",
    "    FOREIGN KEY (gene_name) REFERENCES genes(gene_name),\n",
    "    UNIQUE(chrom, position, ref_allele, alt_allele)\n",
    ");\n",
    "\n",
    "-- Table 3: Allele Frequencies - Population-specific frequencies\n",
    "CREATE TABLE IF NOT EXISTS allele_frequencies (\n",
    "    af_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    variant_id INTEGER NOT NULL,\n",
    "    population TEXT NOT NULL,\n",
    "    allele_frequency REAL NOT NULL,\n",
    "    FOREIGN KEY (variant_id) REFERENCES variants(variant_id),\n",
    "    UNIQUE(variant_id, population)\n",
    ");\n",
    "\n",
    "-- Table 4: Statistical Tests - FST and Chi-square results\n",
    "CREATE TABLE IF NOT EXISTS statistical_tests (\n",
    "    test_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    variant_id INTEGER NOT NULL,\n",
    "    population TEXT NOT NULL,\n",
    "    fst_value REAL,\n",
    "    chi_square_p_value REAL,\n",
    "    is_significant_chisq BOOLEAN DEFAULT 0,\n",
    "    is_significant_bonferroni BOOLEAN DEFAULT 0,\n",
    "    is_significant_fst BOOLEAN DEFAULT 0,\n",
    "    FOREIGN KEY (variant_id) REFERENCES variants(variant_id),\n",
    "    UNIQUE(variant_id, population)\n",
    ");\n",
    "\n",
    "-- Table 5: LD Status - Linkage Disequilibrium information\n",
    "CREATE TABLE IF NOT EXISTS ld_status (\n",
    "    ld_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    variant_id INTEGER NOT NULL,\n",
    "    population TEXT NOT NULL,\n",
    "    in_ld BOOLEAN DEFAULT 0,\n",
    "    ld_pruned BOOLEAN DEFAULT 0,\n",
    "    FOREIGN KEY (variant_id) REFERENCES variants(variant_id),\n",
    "    UNIQUE(variant_id, population)\n",
    ");\n",
    "\n",
    "-- Table 6: Gene Sources - Database and paper references\n",
    "CREATE TABLE IF NOT EXISTS gene_sources (\n",
    "    source_id INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    gene_name TEXT NOT NULL,\n",
    "    database_name TEXT,\n",
    "    database_link TEXT,\n",
    "    paper_1_link TEXT,\n",
    "    paper_2_link TEXT,\n",
    "    paper_3_link TEXT,\n",
    "    category TEXT,\n",
    "    FOREIGN KEY (gene_name) REFERENCES genes(gene_name)\n",
    ");\n",
    "\n",
    "-- Indexes for faster queries\n",
    "CREATE INDEX IF NOT EXISTS idx_variants_gene ON variants(gene_name);\n",
    "CREATE INDEX IF NOT EXISTS idx_variants_rs ON variants(rs_id);\n",
    "CREATE INDEX IF NOT EXISTS idx_variants_chrom_pos ON variants(chrom, position);\n",
    "CREATE INDEX IF NOT EXISTS idx_af_variant ON allele_frequencies(variant_id);\n",
    "CREATE INDEX IF NOT EXISTS idx_af_population ON allele_frequencies(population);\n",
    "CREATE INDEX IF NOT EXISTS idx_stats_variant ON statistical_tests(variant_id);\n",
    "CREATE INDEX IF NOT EXISTS idx_ld_variant ON ld_status(variant_id);\n",
    "\"\"\"\n",
    "\n",
    "class NephroticSyndromeDB:\n",
    "    def __init__(self, db_path='nephrotic_syndrome.db'):\n",
    "        \"\"\"Initialize database connection\"\"\"\n",
    "        self.db_path = db_path\n",
    "        self.conn = None\n",
    "        self.cursor = None\n",
    "        \n",
    "    def connect(self):\n",
    "        \"\"\"Connect to SQLite database\"\"\"\n",
    "        self.conn = sqlite3.connect(self.db_path)\n",
    "        self.cursor = self.conn.cursor()\n",
    "        print(f\"✓ Connected to database: {self.db_path}\")\n",
    "        \n",
    "    def create_schema(self):\n",
    "        \"\"\"Create database schema\"\"\"\n",
    "        self.cursor.executescript(DATABASE_SCHEMA)\n",
    "        self.conn.commit()\n",
    "        print(\"✓ Database schema created successfully\")\n",
    "        \n",
    "    def load_gene_data(self, excel_path):\n",
    "        \"\"\"Load gene information from Excel file\"\"\"\n",
    "        print(\"\\n📊 Loading gene data...\")\n",
    "        df = pd.read_excel(excel_path)\n",
    "        \n",
    "        # Clean column names\n",
    "        df.columns = df.columns.str.strip()\n",
    "        \n",
    "        gene_count = 0\n",
    "        for idx, row in df.iterrows():\n",
    "            try:\n",
    "                # Insert gene information - FIXED: Added ensembl_id to column list\n",
    "                self.cursor.execute(\"\"\"\n",
    "                    INSERT OR IGNORE INTO genes (\n",
    "                        gene_name, ensembl_id, protein_name, gene_function, primary_pathway,\n",
    "                        cellular_location, interacting_genes, srns_ssns, phenotypes,\n",
    "                        inheritance, pediatric, adult, both, population_specific\n",
    "                    ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)\n",
    "                \"\"\", (\n",
    "                    row.get('Gene'),\n",
    "                    row.get('Ensembl_ID'),\n",
    "                    row.get('Protein'),\n",
    "                    row.get('Gene Function'),\n",
    "                    row.get('Primary Pathway'),\n",
    "                    row.get('Cellular Location'),\n",
    "                    row.get('Interacting Genes'),\n",
    "                    row.get('SRNS/SSNS'),\n",
    "                    row.get('Phenotypes'),\n",
    "                    row.get('Inheritance'),\n",
    "                    row.get('Pediatric (Y/N)'),\n",
    "                    row.get('Adult (Y/N)'),\n",
    "                    row.get('Both (Y/N)'),\n",
    "                    row.get('Population')\n",
    "                ))\n",
    "                \n",
    "                # Insert source information\n",
    "                self.cursor.execute(\"\"\"\n",
    "                    INSERT INTO gene_sources (\n",
    "                        gene_name, database_name, database_link,\n",
    "                        paper_1_link, paper_2_link, paper_3_link, category\n",
    "                    ) VALUES (?, ?, ?, ?, ?, ?, ?)\n",
    "                \"\"\", (\n",
    "                    row.get('Gene'),\n",
    "                    row.get('Database'),\n",
    "                    row.get('Database Link'),\n",
    "                    row.get('Source Paper 1'),\n",
    "                    row.get('Source Paper 2'),\n",
    "                    row.get('Source Paper 3'),\n",
    "                    row.get('Category')\n",
    "                ))\n",
    "                \n",
    "                gene_count += 1\n",
    "                \n",
    "            except Exception as e:\n",
    "                print(f\"Error inserting gene {row.get('Gene')}: {e}\")\n",
    "                \n",
    "        self.conn.commit()\n",
    "        print(f\"✓ Loaded {gene_count} genes\")\n",
    "        \n",
    "    def load_variant_data(self, tsv_path):\n",
    "        \"\"\"Load variant and statistical test data from main TSV file\"\"\"\n",
    "        print(\"\\n📊 Loading variant data (this may take a few minutes)...\")\n",
    "        \n",
    "        # Read in chunks to handle large file\n",
    "        chunk_size = 10000\n",
    "        chunks_processed = 0\n",
    "        \n",
    "        for chunk in pd.read_csv(tsv_path, sep='\\t', chunksize=chunk_size):\n",
    "            variant_count = 0\n",
    "            \n",
    "            for idx, row in chunk.iterrows():\n",
    "                try:\n",
    "                    # Insert variant\n",
    "                    self.cursor.execute(\"\"\"\n",
    "                        INSERT OR IGNORE INTO variants (\n",
    "                            rs_id, chrom, position, ref_allele, alt_allele, gene_name, global_af\n",
    "                        ) VALUES (?, ?, ?, ?, ?, ?, ?)\n",
    "                    \"\"\", (\n",
    "                        row['ID'],\n",
    "                        row['CHROM'],\n",
    "                        row['POS'],\n",
    "                        row['REF'],\n",
    "                        row['ALT'],\n",
    "                        row['GENE'],\n",
    "                        row['AF']\n",
    "                    ))\n",
    "                    \n",
    "                    # Get variant_id\n",
    "                    variant_id = self.cursor.execute(\n",
    "                        \"SELECT variant_id FROM variants WHERE rs_id = ?\",\n",
    "                        (row['ID'],)\n",
    "                    ).fetchone()[0]\n",
    "                    \n",
    "                    # Insert allele frequencies for all populations\n",
    "                    populations = {\n",
    "                        'AFR': row['AF_AFR'], 'AMR': row['AF_AMR'], 'AMR1': row['AF_AMR1'],\n",
    "                        'AMR2': row['AF_AMR2'], 'EAS': row['AF_EAS'], 'EUR': row['AF_EUR'],\n",
    "                        'SAS': row['AF_SAS'], 'BEB': row['AF_BEB'], 'GIH': row['AF_GIH'],\n",
    "                        'ITU': row['AF_ITU'], 'PJL': row['AF_PJL'], 'STU': row['AF_STU']\n",
    "                    }\n",
    "                    \n",
    "                    for pop, freq in populations.items():\n",
    "                        self.cursor.execute(\"\"\"\n",
    "                            INSERT OR IGNORE INTO allele_frequencies (variant_id, population, allele_frequency)\n",
    "                            VALUES (?, ?, ?)\n",
    "                        \"\"\", (variant_id, pop, freq))\n",
    "                    \n",
    "                    # Insert statistical test results\n",
    "                    stat_populations = ['AFR', 'AMR', 'AMR1', 'AMR2', 'EAS', 'SAS', 'BEB', 'GIH', 'ITU', 'PJL', 'STU']\n",
    "                    \n",
    "                    for pop in stat_populations:\n",
    "                        fst_col = f'FST_EUR_VS_{pop}'\n",
    "                        p_col = f'P_EUR_{pop}'\n",
    "                        \n",
    "                        self.cursor.execute(\"\"\"\n",
    "                            INSERT OR IGNORE INTO statistical_tests (\n",
    "                                variant_id, population, fst_value, chi_square_p_value\n",
    "                            ) VALUES (?, ?, ?, ?)\n",
    "                        \"\"\", (variant_id, pop, row[fst_col], row[p_col]))\n",
    "                    \n",
    "                    variant_count += 1\n",
    "                    \n",
    "                except Exception as e:\n",
    "                    print(f\"Error processing variant {row.get('ID')}: {e}\")\n",
    "                    continue\n",
    "            \n",
    "            chunks_processed += 1\n",
    "            self.conn.commit()\n",
    "            print(f\"  Processed chunk {chunks_processed} ({variant_count} variants)\")\n",
    "            \n",
    "        print(f\"✓ Loaded all variants\")\n",
    "        \n",
    "    def load_significant_variants(self, folder_path, ld_status='before'):\n",
    "        \"\"\"Load significant variants and mark them in the database\"\"\"\n",
    "        print(f\"\\n📊 Loading significant variants ({ld_status} LD)...\")\n",
    "        \n",
    "        test_types = {\n",
    "            'Significant Chisq result': 'chisq',\n",
    "            'Significant_Chisq_result': 'chisq',\n",
    "            'Significant_Chisq_result_Bonferroni': 'bonferroni',\n",
    "            'Significant_Bonferroni_with_FST': 'bonferroni',\n",
    "            'Significant_Fst': 'fst'\n",
    "        }\n",
    "        \n",
    "        populations = ['AFR', 'AMR', 'AMR1', 'AMR2', 'EAS', 'SAS', 'BEB', 'GIH', 'ITU', 'PJL', 'STU']\n",
    "        \n",
    "        total_updated = 0\n",
    "        total_skipped = 0\n",
    "        \n",
    "        for test_folder, test_type in test_types.items():\n",
    "            test_path = os.path.join(folder_path, test_folder)\n",
    "            if not os.path.exists(test_path):\n",
    "                continue\n",
    "                \n",
    "            for pop in populations:\n",
    "                # Find matching file\n",
    "                files = list(Path(test_path).glob(f\"{pop}_*.tsv\"))\n",
    "                if not files:\n",
    "                    continue\n",
    "                    \n",
    "                file_path = files[0]\n",
    "                \n",
    "                try:\n",
    "                    df = pd.read_csv(file_path, sep='\\t')\n",
    "                    \n",
    "                    for _, row in df.iterrows():\n",
    "                        rs_id = row['ID']\n",
    "                        \n",
    "                        # Check if variant exists in database\n",
    "                        result = self.cursor.execute(\n",
    "                            \"SELECT variant_id FROM variants WHERE rs_id = ?\",\n",
    "                            (rs_id,)\n",
    "                        ).fetchone()\n",
    "                        \n",
    "                        if result is None:\n",
    "                            total_skipped += 1\n",
    "                            continue\n",
    "                        \n",
    "                        variant_id = result[0]\n",
    "                        \n",
    "                        # Update significance flags\n",
    "                        update_query = f\"\"\"\n",
    "                            UPDATE statistical_tests\n",
    "                            SET is_significant_{test_type} = 1\n",
    "                            WHERE variant_id = ? AND population = ?\n",
    "                        \"\"\"\n",
    "                        self.cursor.execute(update_query, (variant_id, pop))\n",
    "                        \n",
    "                        # Update LD status\n",
    "                        self.cursor.execute(\"\"\"\n",
    "                            INSERT OR REPLACE INTO ld_status (variant_id, population, in_ld, ld_pruned)\n",
    "                            VALUES (?, ?, ?, ?)\n",
    "                        \"\"\", (variant_id, pop, 1 if ld_status == 'after' else 0, ld_status == 'after'))\n",
    "                        \n",
    "                        total_updated += 1\n",
    "                        \n",
    "                except Exception as e:\n",
    "                    print(f\"  Warning: Error processing {file_path.name}: {e}\")\n",
    "                    \n",
    "        self.conn.commit()\n",
    "        print(f\"✓ Loaded significant variants ({ld_status} LD)\")\n",
    "        print(f\"  Updated: {total_updated} variant-population combinations\")\n",
    "        if total_skipped > 0:\n",
    "            print(f\"  Skipped: {total_skipped} variants not found in main dataset\")\n",
    "        \n",
    "    def close(self):\n",
    "        \"\"\"Close database connection\"\"\"\n",
    "        if self.conn:\n",
    "            self.conn.close()\n",
    "            print(\"\\n✓ Database connection closed\")\n",
    "\n",
    "\n",
    "# Main execution function\n",
    "def main():\n",
    "    print(\"=\" * 60)\n",
    "    print(\"Nephrotic Syndrome Genetics Database Setup\")\n",
    "    print(\"=\" * 60)\n",
    "    \n",
    "    # Initialize database\n",
    "    db = NephroticSyndromeDB('nephrotic_syndrome.db')\n",
    "    db.connect()\n",
    "    \n",
    "    # Create schema\n",
    "    db.create_schema()\n",
    "    \n",
    "    # Load data (Update these paths to your actual file locations)\n",
    "    print(\"\\n\" + \"=\" * 60)\n",
    "    print(\"LOADING DATA\")\n",
    "    print(\"=\" * 60)\n",
    "    \n",
    "    # 1. Load gene information\n",
    "    db.load_gene_data('Final_gene_list.xlsx')\n",
    "    \n",
    "    # 2. Load variant data\n",
    "    db.load_variant_data('Final_chi_sq_fst_result_with_AF_GENE.tsv')\n",
    "    \n",
    "    # 3. Load significant variants (Before LD)\n",
    "    db.load_significant_variants('Before_LD', ld_status='before')\n",
    "    \n",
    "    # 4. Load significant variants (After LD)\n",
    "    db.load_significant_variants('After_LD', ld_status='after')\n",
    "    \n",
    "    # Close connection\n",
    "    db.close()\n",
    "    \n",
    "    print(\"\\n\" + \"=\" * 60)\n",
    "    print(\"DATABASE SETUP COMPLETE!\")\n",
    "    print(\"=\" * 60)\n",
    "    print(f\"\\nYour database 'nephrotic_syndrome.db' is ready to use!\")\n",
    "    print(\"\\nNext steps:\")\n",
    "    print(\"1. Test queries with the query examples script\")\n",
    "    print(\"2. Set up your web dashboard using Flask/Django\")\n",
    "    print(\"3. Deploy your application\")\n",
    "\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b742b01d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "======================================================================\n",
      "Nephrotic Syndrome Genetics Database - Query Examples\n",
      "======================================================================\n",
      "\n",
      "1. Getting all genes...\n",
      "   Found 491 genes\n",
      "   Example: A1BG - 201 variants\n",
      "\n",
      "2. Getting details for gene: A1BG\n",
      "   Ensembl ID: ENSG00000121410\n",
      "   SRNS/SSNS: None\n",
      "   Database: malacards\n",
      "\n",
      "3. Getting variants for A1BG...\n",
      "   Found 201 variants\n",
      "   Example: rs2051907652 at chrchr19:58345239\n",
      "   Significant: Chisq=0, Bonferroni=0, FST=0\n",
      "   In LD: No\n",
      "\n",
      "4. Getting complete info for rs2051907652...\n",
      "   Allele frequencies in 12 populations\n",
      "   Statistical tests for 11 populations\n",
      "\n",
      "   Sample frequencies:\n",
      "      AFR: 0.0000\n",
      "      AMR: 0.0000\n",
      "      AMR1: 0.0000\n",
      "\n",
      "5. Getting significant variants (Bonferroni, after LD)...\n",
      "   Found 54694 significant variants\n",
      "\n",
      "6. Database Statistics:\n",
      "   total_genes: 491\n",
      "   total_variants: 876075\n",
      "   significant_variants: 42463\n",
      "   avg_variants_per_gene: 1844.3684210526317\n",
      "   most_significant_population: AFR\n",
      "   most_significant_count: 33573\n",
      "\n",
      "7. Testing search (searching for 'NPHS')...\n",
      "   Found 100 results\n",
      "   1. rs74509387 - NPHS1 (chrchr19:35825395)\n",
      "   2. rs554324381 - NPHS1 (chrchr19:35825421)\n",
      "   3. rs181405524 - NPHS1 (chrchr19:35825474)\n",
      "\n",
      "======================================================================\n",
      "All queries executed successfully!\n",
      "======================================================================\n"
     ]
    }
   ],
   "source": [
    "\"\"\"\n",
    "Query Examples for Nephrotic Syndrome Genetics Database\n",
    "This script demonstrates all the queries you'll need for your dashboard\n",
    "\"\"\"\n",
    "\n",
    "import sqlite3\n",
    "import pandas as pd\n",
    "from typing import Dict, List, Tuple\n",
    "\n",
    "\n",
    "class NSGeneticsQueries:\n",
    "    def __init__(self, db_path='nephrotic_syndrome.db'):\n",
    "        self.conn = sqlite3.connect(db_path)\n",
    "        self.conn.row_factory = sqlite3.Row  # Return rows as dictionaries\n",
    "        \n",
    "    def get_all_genes(self) -> List[Dict]:\n",
    "        \"\"\"\n",
    "        Query 1: Get list of all genes for homepage\n",
    "        Returns: List of gene names with basic info\n",
    "        \"\"\"\n",
    "        query = \"\"\"\n",
    "            SELECT \n",
    "                g.gene_name,\n",
    "                g.ensembl_id,\n",
    "                g.srns_ssns,\n",
    "                g.chromosome,\n",
    "                g.protein_name,\n",
    "                COUNT(DISTINCT v.variant_id) as variant_count\n",
    "            FROM genes g\n",
    "            LEFT JOIN variants v ON g.gene_name = v.gene_name\n",
    "            GROUP BY g.gene_name\n",
    "            ORDER BY g.gene_name\n",
    "        \"\"\"\n",
    "        cursor = self.conn.execute(query)\n",
    "        return [dict(row) for row in cursor.fetchall()]\n",
    "    \n",
    "    def get_gene_details(self, gene_name: str) -> Dict:\n",
    "        \"\"\"\n",
    "        Query 2: Get complete gene information when user clicks a gene\n",
    "        \"\"\"\n",
    "        query = \"\"\"\n",
    "            SELECT \n",
    "                g.*,\n",
    "                gs.database_name,\n",
    "                gs.database_link,\n",
    "                gs.paper_1_link,\n",
    "                gs.paper_2_link,\n",
    "                gs.paper_3_link\n",
    "            FROM genes g\n",
    "            LEFT JOIN gene_sources gs ON g.gene_name = gs.gene_name\n",
    "            WHERE g.gene_name = ?\n",
    "        \"\"\"\n",
    "        cursor = self.conn.execute(query, (gene_name,))\n",
    "        result = cursor.fetchone()\n",
    "        return dict(result) if result else None\n",
    "    \n",
    "    def get_gene_variants(self, gene_name: str) -> List[Dict]:\n",
    "        \"\"\"\n",
    "        Query 3: Get all variants for a specific gene with all population frequencies\n",
    "        This is the main query for displaying variant information\n",
    "        \"\"\"\n",
    "        query = \"\"\"\n",
    "            SELECT \n",
    "                v.variant_id,\n",
    "                v.rs_id,\n",
    "                v.chrom,\n",
    "                v.position,\n",
    "                v.ref_allele,\n",
    "                v.alt_allele,\n",
    "                v.global_af,\n",
    "                -- Allele frequencies (will be pivoted)\n",
    "                GROUP_CONCAT(DISTINCT af.population || ':' || af.allele_frequency) as population_frequencies,\n",
    "                -- Significance markers\n",
    "                MAX(CASE WHEN st.is_significant_chisq = 1 THEN 1 ELSE 0 END) as is_sig_chisq,\n",
    "                MAX(CASE WHEN st.is_significant_bonferroni = 1 THEN 1 ELSE 0 END) as is_sig_bonferroni,\n",
    "                MAX(CASE WHEN st.is_significant_fst = 1 THEN 1 ELSE 0 END) as is_sig_fst,\n",
    "                -- LD status\n",
    "                MAX(CASE WHEN ld.ld_pruned = 1 THEN 'Yes' ELSE 'No' END) as in_ld\n",
    "            FROM variants v\n",
    "            LEFT JOIN allele_frequencies af ON v.variant_id = af.variant_id\n",
    "            LEFT JOIN statistical_tests st ON v.variant_id = st.variant_id\n",
    "            LEFT JOIN ld_status ld ON v.variant_id = ld.variant_id\n",
    "            WHERE v.gene_name = ?\n",
    "            GROUP BY v.variant_id\n",
    "            ORDER BY v.chrom, v.position\n",
    "        \"\"\"\n",
    "        cursor = self.conn.execute(query, (gene_name,))\n",
    "        return [dict(row) for row in cursor.fetchall()]\n",
    "    \n",
    "    def get_variant_complete_info(self, rs_id: str) -> Dict:\n",
    "        \"\"\"\n",
    "        Query 4: Get complete information for a specific variant\n",
    "        Including all populations, statistics, and significance\n",
    "        \"\"\"\n",
    "        variant_info = {}\n",
    "        \n",
    "        # Basic variant info\n",
    "        basic_query = \"\"\"\n",
    "            SELECT v.*, g.gene_name, g.ensembl_id, g.srns_ssns\n",
    "            FROM variants v\n",
    "            JOIN genes g ON v.gene_name = g.gene_name\n",
    "            WHERE v.rs_id = ?\n",
    "        \"\"\"\n",
    "        cursor = self.conn.execute(basic_query, (rs_id,))\n",
    "        variant_info['basic'] = dict(cursor.fetchone())\n",
    "        \n",
    "        # Allele frequencies\n",
    "        af_query = \"\"\"\n",
    "            SELECT population, allele_frequency\n",
    "            FROM allele_frequencies\n",
    "            WHERE variant_id = (SELECT variant_id FROM variants WHERE rs_id = ?)\n",
    "            ORDER BY population\n",
    "        \"\"\"\n",
    "        cursor = self.conn.execute(af_query, (rs_id,))\n",
    "        variant_info['allele_frequencies'] = {row['population']: row['allele_frequency'] \n",
    "                                              for row in cursor.fetchall()}\n",
    "        \n",
    "        # Statistical tests\n",
    "        stats_query = \"\"\"\n",
    "            SELECT \n",
    "                population,\n",
    "                fst_value,\n",
    "                chi_square_p_value,\n",
    "                is_significant_chisq,\n",
    "                is_significant_bonferroni,\n",
    "                is_significant_fst\n",
    "            FROM statistical_tests\n",
    "            WHERE variant_id = (SELECT variant_id FROM variants WHERE rs_id = ?)\n",
    "            ORDER BY population\n",
    "        \"\"\"\n",
    "        cursor = self.conn.execute(stats_query, (rs_id,))\n",
    "        variant_info['statistics'] = [dict(row) for row in cursor.fetchall()]\n",
    "        \n",
    "        # LD status\n",
    "        ld_query = \"\"\"\n",
    "            SELECT population, in_ld, ld_pruned\n",
    "            FROM ld_status\n",
    "            WHERE variant_id = (SELECT variant_id FROM variants WHERE rs_id = ?)\n",
    "        \"\"\"\n",
    "        cursor = self.conn.execute(ld_query, (rs_id,))\n",
    "        variant_info['ld_status'] = [dict(row) for row in cursor.fetchall()]\n",
    "        \n",
    "        return variant_info\n",
    "    \n",
    "    def get_population_comparison(self, gene_name: str, populations: List[str]) -> pd.DataFrame:\n",
    "        \"\"\"\n",
    "        Query 5: Compare allele frequencies across specified populations for a gene\n",
    "        \"\"\"\n",
    "        placeholders = ','.join(['?' for _ in populations])\n",
    "        query = f\"\"\"\n",
    "            SELECT \n",
    "                v.rs_id,\n",
    "                v.chrom,\n",
    "                v.position,\n",
    "                af.population,\n",
    "                af.allele_frequency,\n",
    "                st.fst_value,\n",
    "                st.chi_square_p_value\n",
    "            FROM variants v\n",
    "            JOIN allele_frequencies af ON v.variant_id = af.variant_id\n",
    "            LEFT JOIN statistical_tests st ON v.variant_id = st.variant_id \n",
    "                AND af.population = st.population\n",
    "            WHERE v.gene_name = ?\n",
    "            AND af.population IN ({placeholders})\n",
    "            ORDER BY v.position, af.population\n",
    "        \"\"\"\n",
    "        return pd.read_sql_query(query, self.conn, params=[gene_name] + populations)\n",
    "    \n",
    "    def get_significant_variants(self, test_type='bonferroni', population='all', \n",
    "                                 after_ld=True) -> List[Dict]:\n",
    "        \"\"\"\n",
    "        Query 6: Get all significant variants with filters\n",
    "        test_type: 'chisq', 'bonferroni', or 'fst'\n",
    "        population: specific population or 'all'\n",
    "        after_ld: True for after LD pruning, False for before\n",
    "        \"\"\"\n",
    "        ld_condition = \"ld.ld_pruned = 1\" if after_ld else \"ld.ld_pruned = 0\"\n",
    "        pop_condition = f\"AND st.population = '{population}'\" if population != 'all' else \"\"\n",
    "        \n",
    "        significance_column = f\"is_significant_{test_type}\"\n",
    "        \n",
    "        query = f\"\"\"\n",
    "            SELECT \n",
    "                v.rs_id,\n",
    "                v.chrom,\n",
    "                v.position,\n",
    "                v.gene_name,\n",
    "                st.population,\n",
    "                st.fst_value,\n",
    "                st.chi_square_p_value,\n",
    "                af.allele_frequency\n",
    "            FROM variants v\n",
    "            JOIN statistical_tests st ON v.variant_id = st.variant_id\n",
    "            JOIN allele_frequencies af ON v.variant_id = af.variant_id \n",
    "                AND st.population = af.population\n",
    "            LEFT JOIN ld_status ld ON v.variant_id = ld.variant_id \n",
    "                AND st.population = ld.population\n",
    "            WHERE st.{significance_column} = 1\n",
    "            {pop_condition}\n",
    "            AND ({ld_condition} OR ld.ld_pruned IS NULL)\n",
    "            ORDER BY st.chi_square_p_value\n",
    "        \"\"\"\n",
    "        cursor = self.conn.execute(query)\n",
    "        return [dict(row) for row in cursor.fetchall()]\n",
    "    \n",
    "    def get_database_statistics(self) -> Dict:\n",
    "        \"\"\"\n",
    "        Query 7: Get overall database statistics\n",
    "        \"\"\"\n",
    "        stats = {}\n",
    "        \n",
    "        # Total genes\n",
    "        stats['total_genes'] = self.conn.execute(\n",
    "            \"SELECT COUNT(DISTINCT gene_name) FROM genes\"\n",
    "        ).fetchone()[0]\n",
    "        \n",
    "        # Total variants\n",
    "        stats['total_variants'] = self.conn.execute(\n",
    "            \"SELECT COUNT(*) FROM variants\"\n",
    "        ).fetchone()[0]\n",
    "        \n",
    "        # Significant variants (bonferroni, after LD)\n",
    "        stats['significant_variants'] = self.conn.execute(\"\"\"\n",
    "            SELECT COUNT(DISTINCT v.variant_id)\n",
    "            FROM variants v\n",
    "            JOIN statistical_tests st ON v.variant_id = st.variant_id\n",
    "            JOIN ld_status ld ON v.variant_id = ld.variant_id\n",
    "            WHERE st.is_significant_bonferroni = 1 AND ld.ld_pruned = 1\n",
    "        \"\"\").fetchone()[0]\n",
    "        \n",
    "        # Variants per gene\n",
    "        cursor = self.conn.execute(\"\"\"\n",
    "            SELECT AVG(variant_count) as avg_variants\n",
    "            FROM (\n",
    "                SELECT gene_name, COUNT(*) as variant_count\n",
    "                FROM variants\n",
    "                GROUP BY gene_name\n",
    "            )\n",
    "        \"\"\")\n",
    "        stats['avg_variants_per_gene'] = cursor.fetchone()[0]\n",
    "        \n",
    "        # Population with most significant variants\n",
    "        cursor = self.conn.execute(\"\"\"\n",
    "            SELECT population, COUNT(*) as sig_count\n",
    "            FROM statistical_tests\n",
    "            WHERE is_significant_bonferroni = 1\n",
    "            GROUP BY population\n",
    "            ORDER BY sig_count DESC\n",
    "            LIMIT 1\n",
    "        \"\"\")\n",
    "        result = cursor.fetchone()\n",
    "        stats['most_significant_population'] = result[0] if result else None\n",
    "        stats['most_significant_count'] = result[1] if result else 0\n",
    "        \n",
    "        return stats\n",
    "    \n",
    "    def search_variants(self, search_term: str) -> List[Dict]:\n",
    "        \"\"\"\n",
    "        Query 8: Search for variants by rs_id, gene name, or chromosome position\n",
    "        \"\"\"\n",
    "        query = \"\"\"\n",
    "            SELECT \n",
    "                v.rs_id,\n",
    "                v.chrom,\n",
    "                v.position,\n",
    "                v.gene_name,\n",
    "                g.ensembl_id,\n",
    "                g.srns_ssns,\n",
    "                COUNT(DISTINCT st.population) as populations_tested\n",
    "            FROM variants v\n",
    "            JOIN genes g ON v.gene_name = g.gene_name\n",
    "            LEFT JOIN statistical_tests st ON v.variant_id = st.variant_id\n",
    "            WHERE v.rs_id LIKE ?\n",
    "            OR v.gene_name LIKE ?\n",
    "            OR v.chrom LIKE ?\n",
    "            GROUP BY v.variant_id\n",
    "            LIMIT 100\n",
    "        \"\"\"\n",
    "        search_pattern = f\"%{search_term}%\"\n",
    "        cursor = self.conn.execute(query, (search_pattern, search_pattern, search_pattern))\n",
    "        return [dict(row) for row in cursor.fetchall()]\n",
    "    \n",
    "    def close(self):\n",
    "        \"\"\"Close database connection\"\"\"\n",
    "        self.conn.close()\n",
    "\n",
    "\n",
    "# Example usage and testing\n",
    "def main():\n",
    "    print(\"=\" * 70)\n",
    "    print(\"Nephrotic Syndrome Genetics Database - Query Examples\")\n",
    "    print(\"=\" * 70)\n",
    "    \n",
    "    db = NSGeneticsQueries('nephrotic_syndrome.db')\n",
    "    \n",
    "    # Test 1: Get all genes\n",
    "    print(\"\\n1. Getting all genes...\")\n",
    "    genes = db.get_all_genes()\n",
    "    print(f\"   Found {len(genes)} genes\")\n",
    "    if genes:\n",
    "        print(f\"   Example: {genes[0]['gene_name']} - {genes[0]['variant_count']} variants\")\n",
    "    \n",
    "    # Test 2: Get gene details\n",
    "    if genes:\n",
    "        gene_name = genes[0]['gene_name']\n",
    "        print(f\"\\n2. Getting details for gene: {gene_name}\")\n",
    "        gene_info = db.get_gene_details(gene_name)\n",
    "        if gene_info:\n",
    "            print(f\"   Ensembl ID: {gene_info.get('ensembl_id', 'N/A')}\")\n",
    "            print(f\"   SRNS/SSNS: {gene_info.get('srns_ssns', 'N/A')}\")\n",
    "            print(f\"   Database: {gene_info.get('database_name', 'N/A')}\")\n",
    "    \n",
    "    # Test 3: Get variants for a gene\n",
    "        print(f\"\\n3. Getting variants for {gene_name}...\")\n",
    "        variants = db.get_gene_variants(gene_name)\n",
    "        print(f\"   Found {len(variants)} variants\")\n",
    "        if variants:\n",
    "            v = variants[0]\n",
    "            print(f\"   Example: {v['rs_id']} at chr{v['chrom']}:{v['position']}\")\n",
    "            print(f\"   Significant: Chisq={v['is_sig_chisq']}, \"\n",
    "                  f\"Bonferroni={v['is_sig_bonferroni']}, FST={v['is_sig_fst']}\")\n",
    "            print(f\"   In LD: {v['in_ld']}\")\n",
    "    \n",
    "    # Test 4: Get complete variant info\n",
    "            rs_id = v['rs_id']\n",
    "            print(f\"\\n4. Getting complete info for {rs_id}...\")\n",
    "            variant_info = db.get_variant_complete_info(rs_id)\n",
    "            print(f\"   Allele frequencies in {len(variant_info['allele_frequencies'])} populations\")\n",
    "            print(f\"   Statistical tests for {len(variant_info['statistics'])} populations\")\n",
    "            \n",
    "            # Show example frequencies\n",
    "            if variant_info['allele_frequencies']:\n",
    "                print(\"\\n   Sample frequencies:\")\n",
    "                for pop, freq in list(variant_info['allele_frequencies'].items())[:3]:\n",
    "                    print(f\"      {pop}: {freq:.4f}\")\n",
    "    \n",
    "    # Test 5: Get significant variants\n",
    "    print(\"\\n5. Getting significant variants (Bonferroni, after LD)...\")\n",
    "    sig_variants = db.get_significant_variants('bonferroni', 'all', True)\n",
    "    print(f\"   Found {len(sig_variants)} significant variants\")\n",
    "    \n",
    "    # Test 6: Database statistics\n",
    "    print(\"\\n6. Database Statistics:\")\n",
    "    stats = db.get_database_statistics()\n",
    "    for key, value in stats.items():\n",
    "        print(f\"   {key}: {value}\")\n",
    "    \n",
    "    # Test 7: Search functionality\n",
    "    print(\"\\n7. Testing search (searching for 'NPHS')...\")\n",
    "    results = db.search_variants('NPHS')\n",
    "    print(f\"   Found {len(results)} results\")\n",
    "    if results:\n",
    "        for i, r in enumerate(results[:3], 1):\n",
    "            print(f\"   {i}. {r['rs_id']} - {r['gene_name']} (chr{r['chrom']}:{r['position']})\")\n",
    "    \n",
    "    db.close()\n",
    "    print(\"\\n\" + \"=\" * 70)\n",
    "    print(\"All queries executed successfully!\")\n",
    "    print(\"=\" * 70)\n",
    "\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e59cd3d9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "e7c2645f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "89d7f1c6",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"Final_with_dbsnp.tsv\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d2f3a835",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CHROM', 'POS', 'ID', 'REF', 'ALT', 'AF', 'GENE', 'AF_AFR', 'AF_AMR',\n",
       "       'AF_AMR1', 'AF_AMR2', 'AF_EAS', 'AF_EUR', 'AF_SAS', 'AF_BEB', 'AF_GIH',\n",
       "       'AF_ITU', 'AF_PJL', 'AF_STU', 'FST_EUR_VS_AFR', 'FST_EUR_VS_AMR',\n",
       "       'FST_EUR_VS_AMR1', 'FST_EUR_VS_AMR2', 'FST_EUR_VS_BEB',\n",
       "       'FST_EUR_VS_EAS', 'FST_EUR_VS_GIH', 'FST_EUR_VS_ITU', 'FST_EUR_VS_PJL',\n",
       "       'FST_EUR_VS_SAS', 'FST_EUR_VS_STU', 'P_EUR_AFR', 'P_EUR_AMR',\n",
       "       'P_EUR_AMR1', 'P_EUR_AMR2', 'P_EUR_BEB', 'P_EUR_EAS', 'P_EUR_GIH',\n",
       "       'P_EUR_ITU', 'P_EUR_PJL', 'P_EUR_SAS', 'P_EUR_STU', 'db_SNP_link'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "be021558",
   "metadata": {},
   "outputs": [],
   "source": [
    "df2 = pd.read_csv(\"vep_annotated_CADD_score.tsv\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "89d55504",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Ensembl_id</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Gene</th>\n",
       "      <th>CADD_PHRED</th>\n",
       "      <th>CADD_RAW</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs112363642</td>\n",
       "      <td>chr19:1248349</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000099624</td>\n",
       "      <td>ENST00000215375</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>ATP5F1D</td>\n",
       "      <td>0.574</td>\n",
       "      <td>-0.158442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs112363642</td>\n",
       "      <td>chr19:1248349</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000167470</td>\n",
       "      <td>ENST00000300952</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MIDN</td>\n",
       "      <td>0.574</td>\n",
       "      <td>-0.158442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs112363642</td>\n",
       "      <td>chr19:1248349</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000099624</td>\n",
       "      <td>ENST00000395633</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>ATP5F1D</td>\n",
       "      <td>0.574</td>\n",
       "      <td>-0.158442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs112363642</td>\n",
       "      <td>chr19:1248349</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000099624</td>\n",
       "      <td>ENST00000588538</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>ATP5F1D</td>\n",
       "      <td>0.574</td>\n",
       "      <td>-0.158442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs112363642</td>\n",
       "      <td>chr19:1248349</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000099624</td>\n",
       "      <td>ENST00000589478</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>ATP5F1D</td>\n",
       "      <td>0.574</td>\n",
       "      <td>-0.158442</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Uploaded_variation       Location Allele       Ensembl_id          Feature  \\\n",
       "0        rs112363642  chr19:1248349      A  ENSG00000099624  ENST00000215375   \n",
       "1        rs112363642  chr19:1248349      A  ENSG00000167470  ENST00000300952   \n",
       "2        rs112363642  chr19:1248349      A  ENSG00000099624  ENST00000395633   \n",
       "3        rs112363642  chr19:1248349      A  ENSG00000099624  ENST00000588538   \n",
       "4        rs112363642  chr19:1248349      A  ENSG00000099624  ENST00000589478   \n",
       "\n",
       "               Consequence     Gene CADD_PHRED   CADD_RAW  \n",
       "0  downstream_gene_variant  ATP5F1D      0.574  -0.158442  \n",
       "1    upstream_gene_variant     MIDN      0.574  -0.158442  \n",
       "2  downstream_gene_variant  ATP5F1D      0.574  -0.158442  \n",
       "3  downstream_gene_variant  ATP5F1D      0.574  -0.158442  \n",
       "4  downstream_gene_variant  ATP5F1D      0.574  -0.158442  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "3e708dc2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Uploaded_variation', 'Location', 'Allele', 'Ensembl_id', 'Feature',\n",
       "       'Consequence', 'Gene', 'CADD_PHRED', 'CADD_RAW'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "05c97e5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "df3 = pd.read_excel(\"Final_gene_list.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "062f7602",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Gene', 'Ensembl_ID', 'chromosome', 'start', 'end', 'SRNS/SSNS',\n",
       "       'Phenotypes', 'Inheritance', ' Pediatric (Y/N)', 'Adult (Y/N)',\n",
       "       'Both (Y/N)', 'OMIM_Link', 'GWAS_Link', 'ClinPGx_Link',\n",
       "       'MalaCards_Link', 'Function', 'Pathway', 'Source Paper 1',\n",
       "       'Source Paper 2', 'Source Paper 3', 'Category', 'Variation'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df3.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "9e272ae7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Ensembl_ID</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>SRNS/SSNS</th>\n",
       "      <th>Phenotypes</th>\n",
       "      <th>Inheritance</th>\n",
       "      <th>Pediatric (Y/N)</th>\n",
       "      <th>Adult (Y/N)</th>\n",
       "      <th>...</th>\n",
       "      <th>GWAS_Link</th>\n",
       "      <th>ClinPGx_Link</th>\n",
       "      <th>MalaCards_Link</th>\n",
       "      <th>Function</th>\n",
       "      <th>Pathway</th>\n",
       "      <th>Source Paper 1</th>\n",
       "      <th>Source Paper 2</th>\n",
       "      <th>Source Paper 3</th>\n",
       "      <th>Category</th>\n",
       "      <th>Variation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NPHS1</td>\n",
       "      <td>ENSG00000161270</td>\n",
       "      <td>19</td>\n",
       "      <td>35825372.0</td>\n",
       "      <td>35869287.0</td>\n",
       "      <td>SRNS</td>\n",
       "      <td>CNS, SRNS (NPHS1)</td>\n",
       "      <td>AR</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.ebi.ac.uk/gwas/variants/rs2285450</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>https://pathcards.genecards.org/card/nephrotic...</td>\n",
       "      <td>PMC5383633</td>\n",
       "      <td>9543371,9660941 ,9915943 ,10577936 ,10652016 ,...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NPHS2</td>\n",
       "      <td>ENSG00000116218</td>\n",
       "      <td>1</td>\n",
       "      <td>179550494.0</td>\n",
       "      <td>179575952.0</td>\n",
       "      <td>SRNS</td>\n",
       "      <td>CNS, SRNS (NPHS2) &amp; Adult both</td>\n",
       "      <td>AR</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>https://pathcards.genecards.org/card/nephrotic...</td>\n",
       "      <td>PMC5383633</td>\n",
       "      <td>8589695,8606597 ,10742096 ,11729243 ,11733557 ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PLCE1</td>\n",
       "      <td>ENSG00000138193</td>\n",
       "      <td>10</td>\n",
       "      <td>93993894.0</td>\n",
       "      <td>94332823.0</td>\n",
       "      <td>SRNS</td>\n",
       "      <td>DMS, SRNS (NPHS3)</td>\n",
       "      <td>AR</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>https://pathcards.genecards.org/card/nephrotic...</td>\n",
       "      <td>PMC5383633</td>\n",
       "      <td>29127259</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CD2AP</td>\n",
       "      <td>ENSG00000198087</td>\n",
       "      <td>6</td>\n",
       "      <td>47477752.0</td>\n",
       "      <td>47627263.0</td>\n",
       "      <td>SRNS</td>\n",
       "      <td>SRNS (FSGS3)</td>\n",
       "      <td>AD/AR</td>\n",
       "      <td>yes(recessive mutation)</td>\n",
       "      <td>no</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>https://pathcards.genecards.org/card/nephrotic...</td>\n",
       "      <td>PMC5383633</td>\n",
       "      <td>19131354, 10997929, 12217865, 17459670, 146431...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FAT1</td>\n",
       "      <td>ENSG00000083857</td>\n",
       "      <td>4</td>\n",
       "      <td>186587791.0</td>\n",
       "      <td>186726722.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NS, Ciliopathy</td>\n",
       "      <td>AR</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>https://www.genecards.org/cgi-bin/carddisp.pl?...</td>\n",
       "      <td>https://pathcards.genecards.org/card/primary_f...</td>\n",
       "      <td>26905694</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene       Ensembl_ID chromosome        start          end SRNS/SSNS  \\\n",
       "0  NPHS1  ENSG00000161270         19   35825372.0   35869287.0      SRNS   \n",
       "1  NPHS2  ENSG00000116218          1  179550494.0  179575952.0      SRNS   \n",
       "2  PLCE1  ENSG00000138193         10   93993894.0   94332823.0      SRNS   \n",
       "3  CD2AP  ENSG00000198087          6   47477752.0   47627263.0      SRNS   \n",
       "4   FAT1  ENSG00000083857          4  186587791.0  186726722.0       NaN   \n",
       "\n",
       "                       Phenotypes Inheritance          Pediatric (Y/N)  \\\n",
       "0               CNS, SRNS (NPHS1)          AR                      yes   \n",
       "1  CNS, SRNS (NPHS2) & Adult both          AR                      yes   \n",
       "2               DMS, SRNS (NPHS3)          AR                      yes   \n",
       "3                    SRNS (FSGS3)       AD/AR  yes(recessive mutation)   \n",
       "4                  NS, Ciliopathy          AR                     yes    \n",
       "\n",
       "  Adult (Y/N)  ...                                      GWAS_Link  \\\n",
       "0          no  ...  https://www.ebi.ac.uk/gwas/variants/rs2285450   \n",
       "1          no  ...                                            NaN   \n",
       "2          no  ...                                            NaN   \n",
       "3          no  ...                                            NaN   \n",
       "4          no  ...                                            NaN   \n",
       "\n",
       "  ClinPGx_Link                                     MalaCards_Link  \\\n",
       "0          NaN  https://www.genecards.org/cgi-bin/carddisp.pl?...   \n",
       "1          NaN  https://www.genecards.org/cgi-bin/carddisp.pl?...   \n",
       "2          NaN  https://www.genecards.org/cgi-bin/carddisp.pl?...   \n",
       "3          NaN  https://www.genecards.org/cgi-bin/carddisp.pl?...   \n",
       "4          NaN  https://www.genecards.org/cgi-bin/carddisp.pl?...   \n",
       "\n",
       "                                            Function  \\\n",
       "0  https://www.genecards.org/cgi-bin/carddisp.pl?...   \n",
       "1  https://www.genecards.org/cgi-bin/carddisp.pl?...   \n",
       "2  https://www.genecards.org/cgi-bin/carddisp.pl?...   \n",
       "3  https://www.genecards.org/cgi-bin/carddisp.pl?...   \n",
       "4  https://www.genecards.org/cgi-bin/carddisp.pl?...   \n",
       "\n",
       "                                             Pathway Source Paper 1  \\\n",
       "0  https://pathcards.genecards.org/card/nephrotic...     PMC5383633   \n",
       "1  https://pathcards.genecards.org/card/nephrotic...     PMC5383633   \n",
       "2  https://pathcards.genecards.org/card/nephrotic...     PMC5383633   \n",
       "3  https://pathcards.genecards.org/card/nephrotic...     PMC5383633   \n",
       "4  https://pathcards.genecards.org/card/primary_f...       26905694   \n",
       "\n",
       "                                      Source Paper 2 Source Paper 3 Category  \\\n",
       "0  9543371,9660941 ,9915943 ,10577936 ,10652016 ,...            NaN      NaN   \n",
       "1  8589695,8606597 ,10742096 ,11729243 ,11733557 ...            NaN      NaN   \n",
       "2                                           29127259            NaN      NaN   \n",
       "3  19131354, 10997929, 12217865, 17459670, 146431...            NaN      NaN   \n",
       "4                                                NaN            NaN      NaN   \n",
       "\n",
       "  Variation  \n",
       "0       NaN  \n",
       "1       NaN  \n",
       "2       NaN  \n",
       "3       NaN  \n",
       "4       NaN  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "3b61d686",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading files...\n",
      "VEP file: 1048575 rows\n",
      "Final TSV file: 1048575 rows\n",
      "Gene info file: 494 rows\n",
      "\n",
      "Creating gene to Ensembl ID mapping...\n",
      "\n",
      "VEP file after removing duplicates: 107599 rows\n",
      "\n",
      "Merging files based on RS ID, Gene, Ensembl ID, and Allele...\n",
      "\n",
      "✓ Done! Output saved to Final_with_vep_annotations.tsv\n",
      "\n",
      "Summary:\n",
      "  Total rows in output: 1048575\n",
      "  Rows with VEP annotations: 63430\n",
      "  Rows without VEP annotations: 985145\n",
      "\n",
      "Top 5 most common consequences:\n",
      "Consequence\n",
      "intron_variant             54465\n",
      "missense_variant            1687\n",
      "3_prime_UTR_variant         1642\n",
      "upstream_gene_variant       1448\n",
      "downstream_gene_variant     1370\n",
      "Name: count, dtype: int64\n",
      "\n",
      "CADD_PHRED score statistics:\n",
      "count     63430\n",
      "unique    10246\n",
      "top       0.005\n",
      "freq         45\n",
      "Name: CADD_PHRED, dtype: object\n",
      "\n",
      "First 5 rows of merged data:\n",
      "             ID  GENE ALT            Consequence CADD_PHRED   CADD_RAW\n",
      "0   rs112363642  MIDN   A  upstream_gene_variant      0.574  -0.158442\n",
      "1   rs145305690  MIDN   T  upstream_gene_variant      2.348   0.206724\n",
      "2  rs1404444736  MIDN   C  upstream_gene_variant      0.332  -0.296514\n",
      "3   rs535078794  MIDN   T  upstream_gene_variant      2.588   0.233084\n",
      "4   rs371906837  MIDN   A  upstream_gene_variant      2.841   0.258988\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "# Read the three files\n",
    "print(\"Reading files...\")\n",
    "vep_df = pd.read_csv('vep_annotated_CADD_score.tsv', sep='\\t')\n",
    "final_df = pd.read_csv('Final_with_dbsnp.tsv', sep='\\t')\n",
    "gene_info_df = pd.read_excel('Final_gene_list.xlsx')\n",
    "\n",
    "print(f\"VEP file: {len(vep_df)} rows\")\n",
    "print(f\"Final TSV file: {len(final_df)} rows\")\n",
    "print(f\"Gene info file: {len(gene_info_df)} rows\")\n",
    "\n",
    "# Create a lookup dictionary from gene_info to get Ensembl_ID for each Gene\n",
    "print(\"\\nCreating gene to Ensembl ID mapping...\")\n",
    "gene_to_ensembl = dict(zip(gene_info_df['Gene'], gene_info_df['Ensembl_ID']))\n",
    "\n",
    "# Add Ensembl_ID to final_df based on GENE column\n",
    "final_df['Ensembl_ID'] = final_df['GENE'].map(gene_to_ensembl)\n",
    "\n",
    "# Clean up the VEP dataframe - keep only necessary columns and rename for clarity\n",
    "vep_clean = vep_df[['Uploaded_variation', 'Allele', 'Ensembl_id', 'Gene', \n",
    "                     'Consequence', 'CADD_PHRED', 'CADD_RAW']].copy()\n",
    "\n",
    "# Rename columns for easier merging\n",
    "vep_clean.rename(columns={\n",
    "    'Uploaded_variation': 'ID',\n",
    "    'Allele': 'ALT',\n",
    "    'Ensembl_id': 'Ensembl_ID',\n",
    "    'Gene': 'GENE'\n",
    "}, inplace=True)\n",
    "\n",
    "# Remove duplicate entries in VEP (if any) based on all matching criteria\n",
    "# Keep the first occurrence\n",
    "vep_clean = vep_clean.drop_duplicates(subset=['ID', 'ALT', 'Ensembl_ID', 'GENE'], \n",
    "                                       keep='first')\n",
    "\n",
    "print(f\"\\nVEP file after removing duplicates: {len(vep_clean)} rows\")\n",
    "\n",
    "# Merge the dataframes based on all four conditions\n",
    "print(\"\\nMerging files based on RS ID, Gene, Ensembl ID, and Allele...\")\n",
    "merged_df = final_df.merge(\n",
    "    vep_clean[['ID', 'ALT', 'GENE', 'Ensembl_ID', 'Consequence', 'CADD_PHRED', 'CADD_RAW']],\n",
    "    on=['ID', 'ALT', 'GENE', 'Ensembl_ID'],\n",
    "    how='left'\n",
    ")\n",
    "\n",
    "# Remove the temporary Ensembl_ID column if you don't want to keep it\n",
    "# merged_df = merged_df.drop('Ensembl_ID', axis=1)\n",
    "\n",
    "# Save the result\n",
    "output_file = 'Final_with_vep_annotations.tsv'\n",
    "merged_df.to_csv(output_file, sep='\\t', index=False)\n",
    "\n",
    "print(f\"\\n✓ Done! Output saved to {output_file}\")\n",
    "print(f\"\\nSummary:\")\n",
    "print(f\"  Total rows in output: {len(merged_df)}\")\n",
    "print(f\"  Rows with VEP annotations: {merged_df['Consequence'].notna().sum()}\")\n",
    "print(f\"  Rows without VEP annotations: {merged_df['Consequence'].isna().sum()}\")\n",
    "\n",
    "# Show some statistics\n",
    "if merged_df['Consequence'].notna().sum() > 0:\n",
    "    print(f\"\\nTop 5 most common consequences:\")\n",
    "    print(merged_df['Consequence'].value_counts().head())\n",
    "    \n",
    "    print(f\"\\nCADD_PHRED score statistics:\")\n",
    "    print(merged_df['CADD_PHRED'].describe())\n",
    "\n",
    "# Display first few rows of the merged data\n",
    "print(f\"\\nFirst 5 rows of merged data:\")\n",
    "print(merged_df[['ID', 'GENE', 'ALT', 'Consequence', 'CADD_PHRED', 'CADD_RAW']].head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "fb41f23a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading files for verification...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\madhu\\AppData\\Local\\Temp\\ipykernel_11320\\2067832856.py:5: DtypeWarning: Columns (43) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  merged = pd.read_csv('Final_with_vep_annotations.tsv', sep='\\t')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "======================================================================\n",
      "VERIFICATION REPORT\n",
      "======================================================================\n",
      "\n",
      "1. CHECKING VARIANT: rs112363642\n",
      "----------------------------------------------------------------------\n",
      "\n",
      "Original VEP file has 47 entries for rs112363642:\n",
      "   Uploaded_variation Allele     Gene       Ensembl_id          Feature                                   Consequence CADD_PHRED\n",
      "0         rs112363642      A  ATP5F1D  ENSG00000099624  ENST00000215375                       downstream_gene_variant      0.574\n",
      "1         rs112363642      A     MIDN  ENSG00000167470  ENST00000300952                         upstream_gene_variant      0.574\n",
      "2         rs112363642      A  ATP5F1D  ENSG00000099624  ENST00000395633                       downstream_gene_variant      0.574\n",
      "3         rs112363642      A  ATP5F1D  ENSG00000099624  ENST00000588538                       downstream_gene_variant      0.574\n",
      "4         rs112363642      A  ATP5F1D  ENSG00000099624  ENST00000589478                       downstream_gene_variant      0.574\n",
      "5         rs112363642      A  ATP5F1D  ENSG00000099624  ENST00000590265                       downstream_gene_variant      0.574\n",
      "6         rs112363642      A  ATP5F1D  ENSG00000099624  ENST00000591249                       downstream_gene_variant      0.574\n",
      "7         rs112363642      A     MIDN  ENSG00000167470  ENST00000591302                         upstream_gene_variant      0.574\n",
      "8         rs112363642      A     MIDN  ENSG00000167470  ENST00000591446                         upstream_gene_variant      0.574\n",
      "9         rs112363642      A  ATP5F1D  ENSG00000099624  ENST00000591660                       downstream_gene_variant      0.574\n",
      "10        rs112363642      A  ATP5F1D  ENSG00000099624  ENST00000592624                       downstream_gene_variant      0.574\n",
      "11        rs112363642      A  ATP5F1D  ENSG00000099624  ENST00000614837                       downstream_gene_variant      0.574\n",
      "12        rs112363642      A        -  ENSG00000288704  ENST00000681994  intron_variant,non_coding_transcript_variant      0.574\n",
      "13        rs112363642      A     MIDN  ENSG00000167470  ENST00000682408                         upstream_gene_variant      0.574\n",
      "14        rs112363642      A        -  ENSG00000288704  ENST00000757925  intron_variant,non_coding_transcript_variant      0.574\n",
      "15        rs112363642      A        -  ENSG00000288704  ENST00000757926  intron_variant,non_coding_transcript_variant      0.574\n",
      "16        rs112363642      A        -  ENSG00000288704  ENST00000757927  intron_variant,non_coding_transcript_variant      0.574\n",
      "17        rs112363642      A        -  ENSG00000288704  ENST00000757928  intron_variant,non_coding_transcript_variant      0.574\n",
      "18        rs112363642      A        -  ENSG00000288704  ENST00000757929  intron_variant,non_coding_transcript_variant      0.574\n",
      "19        rs112363642      A        -  ENSG00000288704  ENST00000757930  intron_variant,non_coding_transcript_variant      0.574\n",
      "20        rs112363642      A        -  ENSG00000288704  ENST00000757931  intron_variant,non_coding_transcript_variant      0.574\n",
      "21        rs112363642      A        -  ENSG00000288704  ENST00000757932  intron_variant,non_coding_transcript_variant      0.574\n",
      "22        rs112363642      A        -  ENSG00000288704  ENST00000757933  intron_variant,non_coding_transcript_variant      0.574\n",
      "23        rs112363642      A        -  ENSG00000288704  ENST00000757934  intron_variant,non_coding_transcript_variant      0.574\n",
      "24        rs112363642      A        -  ENSG00000288704  ENST00000757935  intron_variant,non_coding_transcript_variant      0.574\n",
      "25        rs112363642      A        -  ENSG00000288704  ENST00000757936  intron_variant,non_coding_transcript_variant      0.574\n",
      "26        rs112363642      A        -  ENSG00000288704  ENST00000757937  intron_variant,non_coding_transcript_variant      0.574\n",
      "27        rs112363642      A        -  ENSG00000288704  ENST00000757938  intron_variant,non_coding_transcript_variant      0.574\n",
      "28        rs112363642      A        -  ENSG00000288704  ENST00000757939  intron_variant,non_coding_transcript_variant      0.574\n",
      "29        rs112363642      A        -  ENSG00000288704  ENST00000757940  intron_variant,non_coding_transcript_variant      0.574\n",
      "30        rs112363642      A        -  ENSG00000288704  ENST00000757941  intron_variant,non_coding_transcript_variant      0.574\n",
      "31        rs112363642      A        -  ENSG00000288704  ENST00000757942  intron_variant,non_coding_transcript_variant      0.574\n",
      "32        rs112363642      A        -  ENSG00000288704  ENST00000757943            non_coding_transcript_exon_variant      0.574\n",
      "33        rs112363642      A        -  ENSG00000288704  ENST00000757944            non_coding_transcript_exon_variant      0.574\n",
      "34        rs112363642      A        -  ENSG00000288704  ENST00000757945            non_coding_transcript_exon_variant      0.574\n",
      "35        rs112363642      A        -  ENSG00000288704  ENST00000757946                         upstream_gene_variant      0.574\n",
      "36        rs112363642      A        -  ENSG00000298805  ENST00000758055                       downstream_gene_variant      0.574\n",
      "37        rs112363642      A        -  ENSG00000298805  ENST00000758056                       downstream_gene_variant      0.574\n",
      "38        rs112363642      A     MIDN  ENSG00000167470  ENST00000872580                         upstream_gene_variant      0.574\n",
      "39        rs112363642      A  ATP5F1D  ENSG00000099624  ENST00000906915                       downstream_gene_variant      0.574\n",
      "40        rs112363642      A  ATP5F1D  ENSG00000099624  ENST00000906916                       downstream_gene_variant      0.574\n",
      "41        rs112363642      A  ATP5F1D  ENSG00000099624  ENST00000916716                       downstream_gene_variant      0.574\n",
      "42        rs112363642      A     MIDN  ENSG00000167470  ENST00000937330                           5_prime_UTR_variant      0.574\n",
      "43        rs112363642      A     MIDN  ENSG00000167470  ENST00000937331                         upstream_gene_variant      0.574\n",
      "44        rs112363642      A     MIDN  ENSG00000167470  ENST00000937332                         upstream_gene_variant      0.574\n",
      "45        rs112363642      A     MIDN  ENSG00000167470  ENST00000937333                         upstream_gene_variant      0.574\n",
      "46        rs112363642      A     MIDN  ENSG00000167470  ENST00000937334                         upstream_gene_variant      0.574\n",
      "\n",
      "\n",
      "Merged file has 1 entries for rs112363642:\n",
      "            ID ALT  GENE       Ensembl_ID            Consequence  CADD_PHRED\n",
      "0  rs112363642   A  MIDN  ENSG00000167470  upstream_gene_variant       0.574\n",
      "\n",
      "\n",
      "2. CHECKING FOR POTENTIAL MISMATCHES\n",
      "----------------------------------------------------------------------\n",
      "\n",
      "Unique genes in VEP file: 162\n",
      "Unique genes in Excel file: 491\n",
      "Genes in merged file with annotations: 55\n",
      "\n",
      "WARNING: 106 genes in VEP are NOT in Excel file:\n",
      "Sample: ['RNU6-371P', 'SPC24', 'MARK2P21', 'Metazoa_SRP', 'KCNAB2', 'ARHGEF1', 'GSTM2', 'MPZ', 'CDC42-AS1', 'CCDC97']\n",
      "\n",
      "\n",
      "3. CADD SCORE VERIFICATION\n",
      "----------------------------------------------------------------------\n",
      "\n",
      "CADD_PHRED statistics (as numeric):\n",
      "count    63430.000000\n",
      "mean         3.756139\n",
      "std          4.173245\n",
      "min          0.001000\n",
      "25%          0.867000\n",
      "50%          2.381000\n",
      "75%          5.209000\n",
      "max         50.000000\n",
      "Name: CADD_PHRED_numeric, dtype: float64\n",
      "\n",
      "CADD_RAW statistics (as numeric):\n",
      "count    63430.000000\n",
      "mean         0.261709\n",
      "std          0.650865\n",
      "min         -3.317522\n",
      "25%         -0.054561\n",
      "50%          0.210444\n",
      "75%          0.475573\n",
      "max         10.924213\n",
      "Name: CADD_RAW_numeric, dtype: float64\n",
      "\n",
      "\n",
      "4. MATCHING STATISTICS\n",
      "----------------------------------------------------------------------\n",
      "\n",
      "Total variants in output: 1,048,575\n",
      "Variants with VEP annotations: 63,430 (6.05%)\n",
      "Variants without annotations: 985,145 (93.95%)\n",
      "\n",
      "Unmatched variants with genes in Excel: 985,027\n",
      "(These might not have VEP annotations or alleles don't match)\n",
      "\n",
      "\n",
      "5. SAMPLE MATCHED RECORDS\n",
      "----------------------------------------------------------------------\n",
      "             ID  CHROM      POS REF ALT  GENE            Consequence  CADD_PHRED  CADD_RAW\n",
      "0   rs112363642  chr19  1248349   G   A  MIDN  upstream_gene_variant       0.574 -0.158442\n",
      "1   rs145305690  chr19  1248372   C   T  MIDN  upstream_gene_variant       2.348  0.206724\n",
      "2  rs1404444736  chr19  1248399   G   C  MIDN  upstream_gene_variant       0.332 -0.296514\n",
      "3   rs535078794  chr19  1248405   C   T  MIDN  upstream_gene_variant       2.588  0.233084\n",
      "4   rs371906837  chr19  1248419   C   A  MIDN  upstream_gene_variant       2.841  0.258988\n",
      "5    rs10425167  chr19  1248572   G   C  MIDN    5_prime_UTR_variant      14.860  1.618459\n",
      "6   rs557216016  chr19  1248613   C   G  MIDN    5_prime_UTR_variant       3.481  0.318953\n",
      "7   rs190425025  chr19  1248627   C   G  MIDN    5_prime_UTR_variant       5.532  0.506645\n",
      "8   rs980730923  chr19  1248645   G   T  MIDN    5_prime_UTR_variant       8.928  0.866521\n",
      "9   rs562224607  chr19  1248673   G   C  MIDN         intron_variant      15.040  1.652609\n",
      "\n",
      "======================================================================\n",
      "END OF VERIFICATION\n",
      "======================================================================\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "print(\"Loading files for verification...\")\n",
    "vep_original = pd.read_csv('vep_annotated_CADD_score.tsv', sep='\\t')\n",
    "merged = pd.read_csv('Final_with_vep_annotations.tsv', sep='\\t')\n",
    "gene_info = pd.read_excel('Final_gene_list.xlsx')  # Replace with actual filename\n",
    "\n",
    "# Create gene to Ensembl mapping\n",
    "gene_to_ensembl = dict(zip(gene_info['Gene'], gene_info['Ensembl_ID']))\n",
    "\n",
    "print(\"\\n\" + \"=\"*70)\n",
    "print(\"VERIFICATION REPORT\")\n",
    "print(\"=\"*70)\n",
    "\n",
    "# 1. Check a specific example\n",
    "test_id = 'rs112363642'\n",
    "print(f\"\\n1. CHECKING VARIANT: {test_id}\")\n",
    "print(\"-\" * 70)\n",
    "\n",
    "# Original VEP entries\n",
    "vep_test = vep_original[vep_original['Uploaded_variation'] == test_id]\n",
    "print(f\"\\nOriginal VEP file has {len(vep_test)} entries for {test_id}:\")\n",
    "print(vep_test[['Uploaded_variation', 'Allele', 'Gene', 'Ensembl_id', \n",
    "                'Feature', 'Consequence', 'CADD_PHRED']].to_string())\n",
    "\n",
    "# Merged result\n",
    "merged_test = merged[merged['ID'] == test_id]\n",
    "print(f\"\\n\\nMerged file has {len(merged_test)} entries for {test_id}:\")\n",
    "print(merged_test[['ID', 'ALT', 'GENE', 'Ensembl_ID', \n",
    "                   'Consequence', 'CADD_PHRED']].to_string())\n",
    "\n",
    "# 2. Check if any annotations were missed due to gene name mismatch\n",
    "print(\"\\n\\n2. CHECKING FOR POTENTIAL MISMATCHES\")\n",
    "print(\"-\" * 70)\n",
    "\n",
    "# Get unique genes from VEP\n",
    "vep_genes = set(vep_original['Gene'].dropna().unique())\n",
    "excel_genes = set(gene_info['Gene'].dropna().unique())\n",
    "merged_genes_with_annot = set(merged[merged['Consequence'].notna()]['GENE'].dropna().unique())\n",
    "\n",
    "print(f\"\\nUnique genes in VEP file: {len(vep_genes)}\")\n",
    "print(f\"Unique genes in Excel file: {len(excel_genes)}\")\n",
    "print(f\"Genes in merged file with annotations: {len(merged_genes_with_annot)}\")\n",
    "\n",
    "# Genes in VEP but not in Excel (these won't get Ensembl IDs)\n",
    "vep_not_in_excel = vep_genes - excel_genes\n",
    "if vep_not_in_excel:\n",
    "    print(f\"\\nWARNING: {len(vep_not_in_excel)} genes in VEP are NOT in Excel file:\")\n",
    "    print(f\"Sample: {list(vep_not_in_excel)[:10]}\")\n",
    "else:\n",
    "    print(\"\\n✓ All VEP genes are in Excel file\")\n",
    "\n",
    "# 3. Verify CADD scores\n",
    "print(\"\\n\\n3. CADD SCORE VERIFICATION\")\n",
    "print(\"-\" * 70)\n",
    "\n",
    "# Convert CADD scores to numeric for proper statistics\n",
    "merged['CADD_PHRED_numeric'] = pd.to_numeric(merged['CADD_PHRED'], errors='coerce')\n",
    "merged['CADD_RAW_numeric'] = pd.to_numeric(merged['CADD_RAW'], errors='coerce')\n",
    "\n",
    "print(\"\\nCADD_PHRED statistics (as numeric):\")\n",
    "print(merged['CADD_PHRED_numeric'].describe())\n",
    "\n",
    "print(\"\\nCADD_RAW statistics (as numeric):\")\n",
    "print(merged['CADD_RAW_numeric'].describe())\n",
    "\n",
    "# Check for non-numeric CADD scores\n",
    "non_numeric_phred = merged[merged['Consequence'].notna() & \n",
    "                           merged['CADD_PHRED_numeric'].isna()]\n",
    "if len(non_numeric_phred) > 0:\n",
    "    print(f\"\\nWARNING: {len(non_numeric_phred)} rows have non-numeric CADD_PHRED values:\")\n",
    "    print(non_numeric_phred[['ID', 'GENE', 'CADD_PHRED', 'CADD_RAW']].head())\n",
    "\n",
    "# 4. Check matching statistics\n",
    "print(\"\\n\\n4. MATCHING STATISTICS\")\n",
    "print(\"-\" * 70)\n",
    "\n",
    "total_variants = len(merged)\n",
    "matched = merged['Consequence'].notna().sum()\n",
    "unmatched = merged['Consequence'].isna().sum()\n",
    "\n",
    "print(f\"\\nTotal variants in output: {total_variants:,}\")\n",
    "print(f\"Variants with VEP annotations: {matched:,} ({matched/total_variants*100:.2f}%)\")\n",
    "print(f\"Variants without annotations: {unmatched:,} ({unmatched/total_variants*100:.2f}%)\")\n",
    "\n",
    "# Check if unmatched variants have genes in Excel\n",
    "unmatched_df = merged[merged['Consequence'].isna()]\n",
    "unmatched_genes_in_excel = unmatched_df[unmatched_df['GENE'].isin(excel_genes)]\n",
    "print(f\"\\nUnmatched variants with genes in Excel: {len(unmatched_genes_in_excel):,}\")\n",
    "print(\"(These might not have VEP annotations or alleles don't match)\")\n",
    "\n",
    "# 5. Sample matched records\n",
    "print(\"\\n\\n5. SAMPLE MATCHED RECORDS\")\n",
    "print(\"-\" * 70)\n",
    "sample = merged[merged['Consequence'].notna()].head(10)\n",
    "print(sample[['ID', 'CHROM', 'POS', 'REF', 'ALT', 'GENE', \n",
    "              'Consequence', 'CADD_PHRED', 'CADD_RAW']].to_string())\n",
    "\n",
    "print(\"\\n\" + \"=\"*70)\n",
    "print(\"END OF VERIFICATION\")\n",
    "print(\"=\"*70)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "d8cd6bfa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analyzing why variants don't have VEP annotations...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\madhu\\AppData\\Local\\Temp\\ipykernel_11320\\3381044023.py:6: DtypeWarning: Columns (43) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  merged = pd.read_csv('Final_with_vep_annotations.tsv', sep='\\t')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "======================================================================\n",
      "MISSING ANNOTATION ANALYSIS\n",
      "======================================================================\n",
      "\n",
      "Total variants: 1,048,575\n",
      "  With VEP annotations: 63,430 (6.05%)\n",
      "  Without VEP annotations: 985,145 (93.95%)\n",
      "\n",
      "\n",
      "Breakdown of 985,145 unmatched variants:\n",
      "  Gene IS in Excel file: 985,027 (100.0%)\n",
      "  Gene NOT in Excel file: 118 (0.0%)\n",
      "\n",
      "\n",
      "Top 20 genes (in Excel) with most unmatched variants:\n",
      "----------------------------------------------------------------------\n",
      "  PTPRD          : 86,906 unmatched  (   0/86,906 matched =   0.0%)\n",
      "  MAGI2          : 41,672 unmatched  (   0/41,672 matched =   0.0%)\n",
      "  GPC5           : 41,185 unmatched  (   0/41,185 matched =   0.0%)\n",
      "  DLC1           : 20,869 unmatched  (   0/20,869 matched =   0.0%)\n",
      "  KALRN          : 17,135 unmatched  (   0/17,135 matched =   0.0%)\n",
      "  KIRREL3        : 16,419 unmatched  (   0/16,419 matched =   0.0%)\n",
      "  ELMO1          : 16,063 unmatched  (   0/16,063 matched =   0.0%)\n",
      "  DELEC1         : 15,013 unmatched  (   0/15,013 matched =   0.0%)\n",
      "  THSD7A         : 13,971 unmatched  (   0/13,971 matched =   0.0%)\n",
      "  ARHGAP24       : 13,754 unmatched  (   0/13,754 matched =   0.0%)\n",
      "  STS            : 12,744 unmatched  (   0/12,744 matched =   0.0%)\n",
      "  CLVS1          : 12,536 unmatched  (   0/12,536 matched =   0.0%)\n",
      "  TRPC6          : 12,414 unmatched  (   0/12,414 matched =   0.0%)\n",
      "  KANK1          : 11,924 unmatched  (   0/11,924 matched =   0.0%)\n",
      "  PLCG2          :  9,964 unmatched  (   0/ 9,964 matched =   0.0%)\n",
      "  CMIP           :  9,857 unmatched  (   0/ 9,857 matched =   0.0%)\n",
      "  DOCK8          :  9,775 unmatched  (   0/ 9,775 matched =   0.0%)\n",
      "  NR3C2          :  9,382 unmatched  (   0/ 9,382 matched =   0.0%)\n",
      "  LINC02182      :  9,238 unmatched  (   0/ 9,238 matched =   0.0%)\n",
      "  CUBN           :  9,046 unmatched  (   0/ 9,046 matched =   0.0%)\n",
      "\n",
      "\n",
      "Sample unmatched variants from genes in Excel file:\n",
      "----------------------------------------------------------------------\n",
      "              ID  CHROM      POS  REF                ALT  GENE\n",
      "36             .  chr19  1249688    A                 AG  MIDN\n",
      "37             .  chr19  1249688    A                 AG  MIDN\n",
      "38             .  chr19  1249688    A                AGG  MIDN\n",
      "39             .  chr19  1249688    A                AGG  MIDN\n",
      "57             .  chr19  1250110    C                  G  MIDN\n",
      "71  rs1023785579  chr19  1250628  CAG                  C  MIDN\n",
      "78             .  chr19  1250901    C                  T  MIDN\n",
      "86   rs528129526  chr19  1251000    T                 TC  MIDN\n",
      "87  rs2145485924  chr19  1251041    T  TGGAGGGGCCCATCGGG  MIDN\n",
      "93             .  chr19  1251134    G                  T  MIDN\n",
      "\n",
      "\n",
      "Detailed check for a few specific genes:\n",
      "----------------------------------------------------------------------\n",
      "\n",
      "PTPRD:\n",
      "  Total variants: 86,906\n",
      "  Matched: 0 (0.0%)\n",
      "  Unmatched: 86,906\n",
      "\n",
      "  Sample variants:\n",
      "          ID     POS REF ALT Consequence\n",
      " rs546399449 8314341   A   C         NaN\n",
      "           . 8314341   A   G         NaN\n",
      " rs145783673 8314394   C   T         NaN\n",
      "rs1369325592 8314397   G   A         NaN\n",
      "           . 8314397   G   C         NaN\n",
      "\n",
      "MAGI2:\n",
      "  Total variants: 41,672\n",
      "  Matched: 0 (0.0%)\n",
      "  Unmatched: 41,672\n",
      "\n",
      "  Sample variants:\n",
      "          ID      POS REF ALT Consequence\n",
      "rs1403062302 78017171   T   A         NaN\n",
      " rs200341869 78017316   C   T         NaN\n",
      " rs532495350 78017407   T   C         NaN\n",
      " rs182324251 78017591   A   C         NaN\n",
      " rs186051587 78017685   T   C         NaN\n",
      "\n",
      "GPC5:\n",
      "  Total variants: 41,185\n",
      "  Matched: 0 (0.0%)\n",
      "  Unmatched: 41,185\n",
      "\n",
      "  Sample variants:\n",
      "         ID      POS                       REF     ALT Consequence\n",
      "rs535736921 91398622                         T       C         NaN\n",
      "rs753928273 91398650          GGGCGGCGGAGGCGGC       G         NaN\n",
      " rs67775819 91398656 CGGAGGCGGCGGCGGCGGCGGCAGT       C         NaN\n",
      "          . 91398659                         A AGGCGGC         NaN\n",
      "          . 91398659                         A AGGCGGC         NaN\n",
      "\n",
      "\n",
      "======================================================================\n",
      "RECOMMENDATIONS\n",
      "======================================================================\n",
      "\n",
      "1. VEP Coverage Check:\n",
      "   - Your Final TSV has 876,075 unique RS IDs\n",
      "   - Your VEP file has annotations for 74,755 unique RS IDs\n",
      "   - Coverage: 8.5%\n",
      "\n",
      "2. If you need more annotations:\n",
      "   - Run VEP on the missing 801,320 variants\n",
      "   - Use the list of unmatched variants to create a new VCF for VEP\n",
      "\n",
      "3. Current merge is CORRECT for the data you have:\n",
      "   - All matching criteria are working properly\n",
      "   - You can trust the merged output\n",
      "\n",
      "4. To expand coverage:\n",
      "   - Add more genes to your Excel file, OR\n",
      "   - Accept that only variants in your 491 genes of interest will match\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "print(\"Analyzing why variants don't have VEP annotations...\")\n",
    "\n",
    "# Load files\n",
    "merged = pd.read_csv('Final_with_vep_annotations.tsv', sep='\\t')\n",
    "gene_info = pd.read_excel('Final_gene_list.xlsx')  # Replace with actual filename\n",
    "\n",
    "# Get genes from Excel\n",
    "excel_genes = set(gene_info['Gene'].dropna().unique())\n",
    "\n",
    "print(\"\\n\" + \"=\"*70)\n",
    "print(\"MISSING ANNOTATION ANALYSIS\")\n",
    "print(\"=\"*70)\n",
    "\n",
    "# Separate matched and unmatched\n",
    "matched = merged[merged['Consequence'].notna()]\n",
    "unmatched = merged[merged['Consequence'].isna()]\n",
    "\n",
    "print(f\"\\nTotal variants: {len(merged):,}\")\n",
    "print(f\"  With VEP annotations: {len(matched):,} (6.05%)\")\n",
    "print(f\"  Without VEP annotations: {len(unmatched):,} (93.95%)\")\n",
    "\n",
    "# Check unmatched variants by gene presence\n",
    "unmatched_in_excel = unmatched[unmatched['GENE'].isin(excel_genes)]\n",
    "unmatched_not_in_excel = unmatched[~unmatched['GENE'].isin(excel_genes)]\n",
    "\n",
    "print(f\"\\n\\nBreakdown of {len(unmatched):,} unmatched variants:\")\n",
    "print(f\"  Gene IS in Excel file: {len(unmatched_in_excel):,} ({len(unmatched_in_excel)/len(unmatched)*100:.1f}%)\")\n",
    "print(f\"  Gene NOT in Excel file: {len(unmatched_not_in_excel):,} ({len(unmatched_not_in_excel)/len(unmatched)*100:.1f}%)\")\n",
    "\n",
    "# Top genes without annotations\n",
    "print(\"\\n\\nTop 20 genes (in Excel) with most unmatched variants:\")\n",
    "print(\"-\" * 70)\n",
    "top_unmatched = unmatched_in_excel['GENE'].value_counts().head(20)\n",
    "for gene, count in top_unmatched.items():\n",
    "    total_for_gene = len(merged[merged['GENE'] == gene])\n",
    "    matched_for_gene = len(matched[matched['GENE'] == gene])\n",
    "    pct = (matched_for_gene / total_for_gene * 100) if total_for_gene > 0 else 0\n",
    "    print(f\"  {gene:15s}: {count:6,} unmatched  ({matched_for_gene:4,}/{total_for_gene:6,} matched = {pct:5.1f}%)\")\n",
    "\n",
    "# Sample of unmatched variants from genes in Excel\n",
    "print(\"\\n\\nSample unmatched variants from genes in Excel file:\")\n",
    "print(\"-\" * 70)\n",
    "sample = unmatched_in_excel.head(10)\n",
    "print(sample[['ID', 'CHROM', 'POS', 'REF', 'ALT', 'GENE']].to_string())\n",
    "\n",
    "# Check specific genes\n",
    "print(\"\\n\\nDetailed check for a few specific genes:\")\n",
    "print(\"-\" * 70)\n",
    "test_genes = unmatched_in_excel['GENE'].value_counts().head(3).index.tolist()\n",
    "\n",
    "for gene in test_genes:\n",
    "    gene_data = merged[merged['GENE'] == gene]\n",
    "    matched_count = len(gene_data[gene_data['Consequence'].notna()])\n",
    "    total_count = len(gene_data)\n",
    "    \n",
    "    print(f\"\\n{gene}:\")\n",
    "    print(f\"  Total variants: {total_count:,}\")\n",
    "    print(f\"  Matched: {matched_count:,} ({matched_count/total_count*100:.1f}%)\")\n",
    "    print(f\"  Unmatched: {total_count - matched_count:,}\")\n",
    "    \n",
    "    # Show sample variants\n",
    "    print(f\"\\n  Sample variants:\")\n",
    "    sample = gene_data[['ID', 'POS', 'REF', 'ALT', 'Consequence']].head(5)\n",
    "    print(sample.to_string(index=False))\n",
    "\n",
    "# Recommendation\n",
    "print(\"\\n\\n\" + \"=\"*70)\n",
    "print(\"RECOMMENDATIONS\")\n",
    "print(\"=\"*70)\n",
    "\n",
    "vep_original = pd.read_csv('vep_annotated_CADD_score.tsv', sep='\\t')\n",
    "unique_vep_ids = vep_original['Uploaded_variation'].nunique()\n",
    "unique_final_ids = merged['ID'].nunique()\n",
    "\n",
    "print(f\"\"\"\n",
    "1. VEP Coverage Check:\n",
    "   - Your Final TSV has {unique_final_ids:,} unique RS IDs\n",
    "   - Your VEP file has annotations for {unique_vep_ids:,} unique RS IDs\n",
    "   - Coverage: {unique_vep_ids/unique_final_ids*100:.1f}%\n",
    "   \n",
    "2. If you need more annotations:\n",
    "   - Run VEP on the missing {unique_final_ids - unique_vep_ids:,} variants\n",
    "   - Use the list of unmatched variants to create a new VCF for VEP\n",
    "   \n",
    "3. Current merge is CORRECT for the data you have:\n",
    "   - All matching criteria are working properly\n",
    "   - You can trust the merged output\n",
    "   \n",
    "4. To expand coverage:\n",
    "   - Add more genes to your Excel file, OR\n",
    "   - Accept that only variants in your {len(excel_genes)} genes of interest will match\n",
    "\"\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f0ff9f4e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
